---
title: "Astrocyte intron retention manuscript figures"
author: "Oliver Ziff"
date: "`r format(Sys.time(), '%d %B %Y')`"
output:
  html_document:
    fig_width: 7
    fig_height: 7
    toc_depth: 3
    theme: simplex
    dev: jpeg
editor_options:
  chunk_output_type: console
---

```{r setup, include=FALSE}
.libPaths("/camp/lab/luscomben/home/users/ziffo/.conda/envs/r4.0.3/lib/R/library")
source("/camp/home/ziffo/home/projects/astrocyte-meta-analysis/scripts/irfinder_functions.R")

# save.image("/camp/home/ziffo/home/projects/astrocyte-meta-analysis/scripts/figures_and_tables.RData")
load("/camp/home/ziffo/home/projects/astrocyte-meta-analysis/scripts/figures_and_tables.RData")

# import gene expression results generated using deseq2_analysis.Rmd script (using HTSeq counts in intersection-strict mode)
dge_res_ac_who_vcp_vs_ctrl <- readRDS("/camp/home/ziffo/home/projects/astrocyte-fractionation-bulk-rnaseq/deseq2/res_ac_who_vcp_vs_ctrl.RDS")
dge_res_ac_nuc_vcp_vs_ctrl <- readRDS("/camp/home/ziffo/home/projects/astrocyte-fractionation-bulk-rnaseq/deseq2/res_ac_nuc_vcp_vs_ctrl.RDS")
dge_res_ac_cyt_vcp_vs_ctrl <- readRDS("/camp/home/ziffo/home/projects/astrocyte-fractionation-bulk-rnaseq/deseq2/res_ac_cyt_vcp_vs_ctrl.RDS")
dge_res_ac_ctrl_cyt_vs_nuc <- readRDS("/camp/home/ziffo/home/projects/astrocyte-fractionation-bulk-rnaseq/deseq2/res_ac_ctrl_cyt_vs_nuc.RDS")
dge_res_ac_vcp_cyt_vs_nuc <- readRDS("/camp/home/ziffo/home/projects/astrocyte-fractionation-bulk-rnaseq/deseq2/res_ac_vcp_cyt_vs_nuc.RDS")
dge_res_ac_tyz_sod1_vs_ctrl <- readRDS("/camp/home/ziffo/home/downloaded-public-data/astrocyte-sod1-ipsc-tyzack-2017/expression/deseq2/res_ac_tyz_sod1_vs_ctrl.RDS")
dge_res_ac_bir_c9orf_vs_ctrl <- readRDS("/camp/home/ziffo/home/downloaded-public-data/astrocyte-c9orf72-ipsc-birger-2019/expression/deseq2/res_ac_bir_c9orf_vs_ctrl.RDS")
dge_res_ac_bar_a1_vs_a0 <- readRDS("/camp/home/ziffo/home/downloaded-public-data/reactive-astrocyte-barbar-2020/expression/deseq/res_ac_bar_a1_vs_a0.RDS")
dge_res_ac_mou_sod1_vs_ctrl <- readRDS("/camp/home/ziffo/home/downloaded-public-data/astrocyte-sod1-mouse-bacTRAP-cleveland-2015/expression/deseq/res_ac_mou_sod1_vs_ctrl.RDS")
dge_res_ac_tdp_ko_vs_ctrl <- readRDS("/camp/home/ziffo/home/downloaded-public-data/astrocyte-tdp43-mouse-peng-2020/expression/deseq/res_ac_tdp_ko_vs_ctrl.RDS")

dge_vsd_ac_who_vcp_vs_ctrl <- readRDS("/camp/home/ziffo/home/projects/astrocyte-fractionation-bulk-rnaseq/deseq2/vsd_ac_who_vcp_vs_ctrl.RDS")
dge_vsd_ac_nuc_vcp_vs_ctrl <- readRDS("/camp/home/ziffo/home/projects/astrocyte-fractionation-bulk-rnaseq/deseq2/vsd_ac_nuc_vcp_vs_ctrl.RDS")
dge_vsd_ac_cyt_vcp_vs_ctrl <- readRDS("/camp/home/ziffo/home/projects/astrocyte-fractionation-bulk-rnaseq/deseq2/vsd_ac_cyt_vcp_vs_ctrl.RDS")
dge_vsd_ac_ctrl_cyt_vs_nuc <- readRDS("/camp/home/ziffo/home/projects/astrocyte-fractionation-bulk-rnaseq/deseq2/vsd_ac_ctrl_cyt_vs_nuc.RDS")
dge_vsd_ac_vcp_cyt_vs_nuc <- readRDS("/camp/home/ziffo/home/projects/astrocyte-fractionation-bulk-rnaseq/deseq2/vsd_ac_vcp_cyt_vs_nuc.RDS")
dge_vsd_ac_tyz_sod1_vs_ctrl <- readRDS("/camp/home/ziffo/home/downloaded-public-data/astrocyte-sod1-ipsc-tyzack-2017/expression/deseq2/vsd_ac_tyz_sod1_vs_ctrl.RDS")
dge_vsd_ac_bir_c9orf_vs_ctrl <- readRDS("/camp/home/ziffo/home/downloaded-public-data/astrocyte-c9orf72-ipsc-birger-2019/expression/deseq2/vsd_ac_bir_c9orf_vs_ctrl.RDS")
dge_vsd_ac_bar_a1_vs_a0 <- readRDS("/camp/home/ziffo/home/downloaded-public-data/reactive-astrocyte-barbar-2020/expression/deseq/vsd_ac_bar_a1_vs_a0.RDS")
dge_vsd_ac_mou_sod1_vs_ctrl <- readRDS("/camp/home/ziffo/home/downloaded-public-data/astrocyte-sod1-mouse-bacTRAP-cleveland-2015/expression/deseq/vsd_ac_mou_sod1_vs_ctrl.RDS")
dge_vsd_ac_tdp_ko_vs_ctrl <- readRDS("/camp/home/ziffo/home/downloaded-public-data/astrocyte-tdp43-mouse-peng-2020/expression/deseq/vsd_ac_tdp_ko_vs_ctrl.RDS")

# Import differentially compared IRFinder output generated using IRFinder_differential script
ac_who_vcp_vs_ctrl.IRFinder <- read.table("/camp/home/ziffo/home/projects/astrocyte-meta-analysis/splicing/IRFinder/ac_who_vcp-vs-ctrl.txt", sep = '\t',header = TRUE)
ac_nuc_vcp_vs_ctrl.IRFinder <- read.table("/camp/home/ziffo/home/projects/astrocyte-meta-analysis/splicing/IRFinder/ac_nuc_vcp-vs-ctrl.txt", sep = '\t',header = TRUE)
ac_cyt_vcp_vs_ctrl.IRFinder <- read.table("/camp/home/ziffo/home/projects/astrocyte-meta-analysis/splicing/IRFinder/ac_cyt_vcp-vs-ctrl.txt", sep = '\t',header = TRUE)
ac_ctrl_cyt_vs_nuc.IRFinder <- read.table("/camp/home/ziffo/home/projects/astrocyte-meta-analysis/splicing/IRFinder/ac_ctrl_cyt-vs-nuc.txt", sep = '\t',header = TRUE)
ac_vcp_cyt_vs_nuc.IRFinder <- read.table("/camp/home/ziffo/home/projects/astrocyte-meta-analysis/splicing/IRFinder/ac_vcp_cyt-vs-nuc.txt", sep = '\t',header = TRUE)
ac_bir_c9orf_vs_ctrl.IRFinder <- read.table("/camp/home/ziffo/home/projects/astrocyte-meta-analysis/splicing/IRFinder/ac_bir_c9orf-vs-ctrl.txt", sep = '\t',header = TRUE)
ac_tyz_sod1_vs_ctrl.IRFinder <- read.table("/camp/home/ziffo/home/projects/astrocyte-meta-analysis/splicing/IRFinder/ac_tyz_sod1-vs-ctrl.txt", sep = '\t',header = TRUE)
ac_bar_a1_vs_a0.IRFinder <- read.table("/camp/home/ziffo/home/projects/astrocyte-meta-analysis/splicing/IRFinder/ac_bar_a1-vs-a0.txt", sep = '\t',header = TRUE)
ac_mou_sod1_vs_ctrl.IRFinder <- read.table("/camp/home/ziffo/home/downloaded-public-data/astrocyte-sod1-mouse-bacTRAP-cleveland-2015/splicing/IRFinder/ac_mou_sod1_vs_ctrl.txt", sep = '\t',header = TRUE)
ac_tdp_ko_vs_ctrl.IRFinder <- read.table("/camp/home/ziffo/home/downloaded-public-data/astrocyte-tdp43-mouse-peng-2020/splicing/IRFinder/ac_tdp_ko_vs_ctrl.txt", sep = '\t',header = TRUE)
# ac_tdp_het_vs_ctrl.IRFinder <- read.table("/camp/home/ziffo/home/downloaded-public-data/astrocyte-tdp43-mouse-peng-2020/splicing/IRFinder/ac_tdp_het_vs_ctrl.txt", sep = '\t',header = TRUE)
# ac_tdp_ko_vs_het.IRFinder <- read.table("/camp/home/ziffo/home/downloaded-public-data/astrocyte-tdp43-mouse-peng-2020/splicing/IRFinder/ac_tdp_ko_vs_het.txt", sep = '\t',header = TRUE)


# Merge gene expression (DESeq2), mass spec (DEP), and differential splicing (IRFinder) results
ac_who_vcp_vs_ctrl <- cleanIRFinder(IRFinder = ac_who_vcp_vs_ctrl.IRFinder, deseq_res = dge_res_ac_who_vcp_vs_ctrl)
ac_nuc_vcp_vs_ctrl <- cleanIRFinder(IRFinder = ac_nuc_vcp_vs_ctrl.IRFinder, deseq_res = dge_res_ac_nuc_vcp_vs_ctrl)
ac_cyt_vcp_vs_ctrl <- cleanIRFinder(IRFinder = ac_cyt_vcp_vs_ctrl.IRFinder, deseq_res = dge_res_ac_cyt_vcp_vs_ctrl)
ac_ctrl_cyt_vs_nuc <- cleanIRFinder(IRFinder = ac_ctrl_cyt_vs_nuc.IRFinder, deseq_res = dge_res_ac_ctrl_cyt_vs_nuc)
ac_vcp_cyt_vs_nuc <- cleanIRFinder(IRFinder = ac_vcp_cyt_vs_nuc.IRFinder, deseq_res = dge_res_ac_vcp_cyt_vs_nuc)
nuc_ir_cyt_dge <- cleanIRFinder(IRFinder = ac_nuc_vcp_vs_ctrl.IRFinder, deseq_res = dge_res_ac_cyt_vcp_vs_ctrl)
ac_bir_c9orf_vs_ctrl <- cleanIRFinder(IRFinder = ac_bir_c9orf_vs_ctrl.IRFinder, deseq_res = dge_res_ac_bir_c9orf_vs_ctrl)
ac_tyz_sod1_vs_ctrl <- cleanIRFinder(IRFinder = ac_tyz_sod1_vs_ctrl.IRFinder, deseq_res = dge_res_ac_tyz_sod1_vs_ctrl)
ac_bar_a1_vs_a0 <- cleanIRFinder(IRFinder = ac_bar_a1_vs_a0.IRFinder, deseq_res = dge_res_ac_bar_a1_vs_a0)
ac_mou_sod1_vs_ctrl <- cleanIRFinder(IRFinder = ac_mou_sod1_vs_ctrl.IRFinder, deseq_res = dge_res_ac_mou_sod1_vs_ctrl, species = "Mmusculus")
ac_tdp_ko_vs_ctrl <- cleanIRFinder(IRFinder = ac_tdp_ko_vs_ctrl.IRFinder, deseq_res = dge_res_ac_tdp_ko_vs_ctrl, species = "Mmusculus")
# ac_tdp_het_vs_ctrl <- cleanIRFinder(IRFinder = ac_tdp_het_vs_ctrl.IRFinder, species = "Mmusculus")
# ac_tdp_ko_vs_het <- cleanIRFinder(IRFinder = ac_tdp_ko_vs_het.IRFinder, species = "Mmusculus")


# Import pooled replicate quantified IRFinder Output for each condition (IRFinder-IR-dir.txt or if not stranded, then IRFinder-IR-nondir.txt) generated using IRFinder_pool_replicates script
ac_who_ctrl.IRFinder <- read.table("/camp/home/ziffo/home/projects/astrocyte-meta-analysis/splicing/IRFinder/pooled_ac_who_ctrl/IRFinder-IR-dir.txt", sep = '\t',header = TRUE)
ac_who_vcp.IRFinder <- read.table("/camp/home/ziffo/home/projects/astrocyte-meta-analysis/splicing/IRFinder/pooled_ac_who_vcp/IRFinder-IR-dir.txt", sep = '\t',header = TRUE)
ac_nuc_ctrl.IRFinder <- read.table("/camp/home/ziffo/home/projects/astrocyte-meta-analysis/splicing/IRFinder/pooled_ac_nuc_ctrl/IRFinder-IR-dir.txt", sep = '\t',header = TRUE)
ac_nuc_vcp.IRFinder <- read.table("/camp/home/ziffo/home/projects/astrocyte-meta-analysis/splicing/IRFinder/pooled_ac_nuc_vcp/IRFinder-IR-dir.txt", sep = '\t',header = TRUE)
ac_cyt_ctrl.IRFinder <- read.table("/camp/home/ziffo/home/projects/astrocyte-meta-analysis/splicing/IRFinder/pooled_ac_cyt_ctrl/IRFinder-IR-dir.txt", sep = '\t',header = TRUE)
ac_cyt_vcp.IRFinder <- read.table("/camp/home/ziffo/home/projects/astrocyte-meta-analysis/splicing/IRFinder/pooled_ac_cyt_vcp/IRFinder-IR-dir.txt", sep = '\t',header = TRUE)
ac_bir_ctrl.IRFinder <- read.table("/camp/home/ziffo/home/projects/astrocyte-meta-analysis/splicing/IRFinder/pooled_ac_bir_ctrl/IRFinder-IR-dir.txt", sep = '\t',header = TRUE)
ac_bir_c9orf.IRFinder <- read.table("/camp/home/ziffo/home/projects/astrocyte-meta-analysis/splicing/IRFinder/pooled_ac_bir_c9orf/IRFinder-IR-dir.txt", sep = '\t',header = TRUE)
ac_tyz_ctrl.IRFinder <- read.table("/camp/home/ziffo/home/projects/astrocyte-meta-analysis/splicing/IRFinder/pooled_ac_tyz_ctrl/IRFinder-IR-dir.txt", sep = '\t',header = TRUE)
ac_tyz_sod1.IRFinder <- read.table("/camp/home/ziffo/home/projects/astrocyte-meta-analysis/splicing/IRFinder/pooled_ac_tyz_sod1/IRFinder-IR-dir.txt", sep = '\t',header = TRUE)
ac_bar_a0.IRFinder <- read.table("/camp/home/ziffo/home/projects/astrocyte-meta-analysis/splicing/IRFinder/pooled_ac_bar_a0/IRFinder-IR-nondir.txt", sep = '\t',header = TRUE)
ac_bar_a1.IRFinder <- read.table("/camp/home/ziffo/home/projects/astrocyte-meta-analysis/splicing/IRFinder/pooled_ac_bar_a1/IRFinder-IR-nondir.txt", sep = '\t',header = TRUE)
ac_tdp_ctrl.IRFinder <- read.table("/camp/home/ziffo/home/downloaded-public-data/astrocyte-tdp43-mouse-peng-2020/splicing/IRFinder/pooled_ac_tdp_ctrl/IRFinder-IR-dir.txt", sep = '\t',header = TRUE)
# ac_tdp_het.IRFinder <- read.table("/camp/home/ziffo/home/downloaded-public-data/astrocyte-tdp43-mouse-peng-2020/splicing/IRFinder/pooled_ac_tdp_het/IRFinder-IR-dir.txt", sep = '\t',header = TRUE)
ac_tdp_ko.IRFinder <- read.table("/camp/home/ziffo/home/downloaded-public-data/astrocyte-tdp43-mouse-peng-2020/splicing/IRFinder/pooled_ac_tdp_ko/IRFinder-IR-dir.txt", sep = '\t',header = TRUE)

ac_who_ctrl.IR <- cleanIRFinderQuant(ac_who_ctrl.IRFinder)
ac_who_vcp.IR <- cleanIRFinderQuant(ac_who_vcp.IRFinder)
ac_nuc_ctrl.IR <- cleanIRFinderQuant(ac_nuc_ctrl.IRFinder)
ac_nuc_vcp.IR <- cleanIRFinderQuant(ac_nuc_vcp.IRFinder)
ac_cyt_ctrl.IR <- cleanIRFinderQuant(ac_cyt_ctrl.IRFinder)
ac_cyt_vcp.IR <- cleanIRFinderQuant(ac_cyt_vcp.IRFinder)
ac_bir_ctrl.IR <- cleanIRFinderQuant(ac_bir_ctrl.IRFinder)
ac_bir_c9orf.IR <- cleanIRFinderQuant(ac_bir_c9orf.IRFinder)
ac_tyz_ctrl.IR <- cleanIRFinderQuant(ac_tyz_ctrl.IRFinder)
ac_tyz_sod1.IR <- cleanIRFinderQuant(ac_tyz_sod1.IRFinder)
ac_bar_a0.IR <- cleanIRFinderQuant(ac_bar_a0.IRFinder)
ac_bar_a1.IR <- cleanIRFinderQuant(ac_bar_a1.IRFinder)
ac_tdp_ctrl.IR <- cleanIRFinderQuant(ac_tdp_ctrl.IRFinder, species = "Mmusculus")
# ac_tdp_het.IR <- cleanIRFinderQuant(ac_tdp_het.IRFinder, species = "Mmusculus")
ac_tdp_ko.IR <- cleanIRFinderQuant(ac_tdp_ko.IRFinder, species = "Mmusculus")

# Merge gene expression (DESeq2), mass spec (DEP), and absolute splicing (IRFinder) results
ac_who_vcp_vs_ctrl.absoluteIR <- cleanIRFinderQuantCompared(mutant = ac_who_vcp.IR, control = ac_who_ctrl.IR, deseq_res = dge_res_ac_who_vcp_vs_ctrl)
ac_nuc_vcp_vs_ctrl.absoluteIR <- cleanIRFinderQuantCompared(mutant = ac_nuc_vcp.IR, control = ac_nuc_ctrl.IR, deseq_res = dge_res_ac_nuc_vcp_vs_ctrl)
ac_cyt_vcp_vs_ctrl.absoluteIR <- cleanIRFinderQuantCompared(mutant = ac_cyt_vcp.IR, control = ac_cyt_ctrl.IR, deseq_res = dge_res_ac_cyt_vcp_vs_ctrl)
ac_ctrl_cyt_vs_nuc.absoluteIR <- cleanIRFinderQuantCompared(mutant = ac_cyt_ctrl.IR, control = ac_nuc_ctrl.IR, deseq_res = dge_res_ac_ctrl_cyt_vs_nuc)
ac_vcp_cyt_vs_nuc.absoluteIR <- cleanIRFinderQuantCompared(mutant = ac_cyt_vcp.IR, control = ac_nuc_vcp.IR, deseq_res = dge_res_ac_vcp_cyt_vs_nuc)
ac_nucIR_cytDGE.absoluteIR <- cleanIRFinderQuantCompared(mutant = ac_nuc_vcp.IR, control = ac_nuc_ctrl.IR, deseq_res = dge_res_ac_cyt_vcp_vs_ctrl)
ac_bir_c9orf_vs_ctrl.absoluteIR <- cleanIRFinderQuantCompared(mutant = ac_bir_c9orf.IR, control = ac_bir_ctrl.IR, deseq_res = dge_res_ac_bir_c9orf_vs_ctrl)
ac_tyz_sod1_vs_ctrl.absoluteIR <- cleanIRFinderQuantCompared(mutant = ac_tyz_sod1.IR, control = ac_tyz_ctrl.IR, deseq_res = dge_res_ac_tyz_sod1_vs_ctrl)
ac_bar_a1_vs_a0.absoluteIR <- cleanIRFinderQuantCompared(mutant = ac_bar_a1.IR, control = ac_bar_a0.IR, deseq_res = dge_res_ac_bar_a1_vs_a0)
ac_tdp_ko_vs_ctrl.absoluteIR <- cleanIRFinderQuantCompared(mutant = ac_tdp_ko.IR, control = ac_tdp_ctrl.IR, deseq_res = dge_res_ac_tdp_ko_vs_ctrl, differentialIR = ac_tdp_ko_vs_ctrl)
# ac_tdp_het_vs_ctrl.absoluteIR <- cleanIRFinderQuantCompared(mutant = ac_tdp_het.IR, control = ac_tdp_ctrl.IR, differentialIR = ac_tdp_het_vs_ctrl)
# ac_tdp_ko_vs_het.absoluteIR <- cleanIRFinderQuantCompared(mutant = ac_tdp_ko.IR, control = ac_tdp_het.IR, differentialIR = ac_tdp_ko_vs_het)
```

# Figure S1: Sample clustering

A biorender schematic - add in illustrator

```{r}
## Individual dataset PCA
htseqdir <- "/camp/home/ziffo/home/projects/astrocyte-meta-analysis/alignment/htseq"

# VCP
patani <- read.csv("/camp/home/ziffo/home/projects/astrocyte-fractionation-bulk-rnaseq/sample-details/sample_table.csv")
patani <- patani[!patani$cellline == "ctrl2",] # remove CTRL2 samples due to batch effects
patani$cellline <- patani$cellline %>% gsub("ctrl5", "ctrl2", .) # for the paper we rename ctrl5 as ctrl2 to avoid confusion as to why we have ctrl1 & 5 (but not 2,3 and 4)
patani <- patani %>% distinct(fraction_cellline, .keep_all = TRUE) # remove rows for technical repeats
patani <- patani %>% mutate(mutation = as.factor(vcp), fraction = factor(fraction, levels = c("whole", "nuclear", "cytoplasmic")))
patani$condition <- gsub("vcp", "als", patani$mutation)
patani$condition <- factor(patani$condition, levels = c("ctrl", "als")) # set the reference control first to ensure comparison is ALS vs CTRL (rather than CTRL vs ALS)
patani$sample <- patani$fraction_cellline %>% gsub("whole", "who", .) %>% gsub("nuclear", "nuc", .) %>% gsub("cytoplasmic", "cyt", .) #%>% gsub("cb1e", "R191Q", .) %>% gsub("glia", "R155C", .)
patani$sample <- paste("ac_",patani$sample,sep="")
patani$patient <- patani$cellline %>% gsub("cb1e", "R191Q", .) %>% gsub("glia", "R155C", .)
patani <- patani %>% dplyr::filter(fraction == "whole") # astrocyte whole-cell samples for VCP only
patani <- patani %>% dplyr::select(sampleName=fraction_cellline, condition, fraction, mutation, cellline, patient, sample) %>% mutate(fileName = paste(sample, ".tab", sep = ""), patient_fraction = paste(patient, fraction, sep = "_")) %>% dplyr::select(sampleName, fileName, condition, everything())
htseqdir <- "/camp/home/ziffo/home/projects/astrocyte-meta-analysis/alignment/htseq"
dds.vcp <- DESeqDataSetFromHTSeqCount(sampleTable = patani, directory = htseqdir, design = ~ condition) # create DESeqDataSet with this design - Import htseq .tab tables
dds.vcp <- DESeq(dds.vcp) # run DESeq2
vsd.vcp <- vst(dds.vcp, blind=FALSE) # VST transformation
colnames(vsd.vcp) <- colnames(vsd.vcp) %>% gsub("whole_", "", .) %>% gsub("5", "2", .) %>% gsub("cb1e", "R191Q", .) %>% gsub("glia", "R155C", .) # Labels for mutants are based on VCP mutation. R155C mutant = GLIA. R191Q mutant = CB1E.
pcaData.vcp <- plotPCA(vsd.vcp, intgroup=c("condition", "mutation", "patient"), returnData=TRUE)
percentVar.vcp <- round(100 * attr(pcaData.vcp, "percentVar"))
pca.vcp <- ggplot(pcaData.vcp, aes(PC1, PC2, color=mutation)) + geom_text_repel(aes(label=patient), size = 3) + 
  scale_colour_manual( values = c("dodgerblue", "firebrick2")) +  theme_bw() +  
  theme(panel.grid = element_blank(), legend.title = element_blank(), legend.text=element_text(size=8), legend.position = "top", legend.box = "horizontal", legend.margin=margin(0,0,0,0),
        legend.box.margin=margin(-10,-10,-10,-10)) + geom_point(size=1) +   xlab(paste0("PC1: ",percentVar.vcp[1],"% variance")) +  ylab(paste0("PC2: ",percentVar.vcp[2],"% variance")) + coord_fixed()
pca.vcp

# SOD1
tyzack <- read_excel("/camp/home/ziffo/home/downloaded-public-data/astrocyte-sod1-ipsc-tyzack-2017/sample-details/metadata.xlsx")
tyzack <- tyzack %>% distinct(sample, .keep_all = TRUE)
tyzack <- tyzack %>% mutate(mutation = factor(condition, levels = c("ctrl", "sod1")),
                            filename = sample,
                            fraction = "whole")
tyzack$sample <- paste("ac_tyz_",tyzack$sample,sep="")
tyzack$condition <- gsub("sod1", "als", tyzack$mutation)
tyzack$condition <- factor(tyzack$condition, levels = c("ctrl", "als")) # set the reference control first to ensure comparison is ALS vs CTRL (rather than CTRL vs ALS)
tyzack$cellline <- c("ctrl1","ctrl2","sod1")
tyzack$patient <- c("ctrl1","ctrl2","sod1")
tyzack <- tyzack %>% dplyr::select(sampleName=sample, condition, mutation, patient) %>% mutate(dataset = "SOD1", fileName = paste(sampleName, ".tab", sep = "")) %>% dplyr::select(sampleName, fileName, condition, everything()) %>% as.data.frame
dds.sod1 <- DESeqDataSetFromHTSeqCount(sampleTable = tyzack, directory = htseqdir, design = ~ condition) # create DESeqDataSet with this design - Import htseq .tab tables
dds.sod1 <- DESeq(dds.sod1) # run DESeq2
vsd.sod1 <- vst(dds.sod1, blind=FALSE) # VST transformation
pcaData.sod1 <- plotPCA(vsd.sod1, intgroup=c("condition", "mutation", "patient"), returnData=TRUE)
percentVar.sod1 <- round(100 * attr(pcaData.sod1, "percentVar"))
pca.sod1 <- ggplot(pcaData.sod1, aes(PC1, PC2, color=mutation)) + geom_text_repel(aes(label=patient), size = 3) + 
  scale_colour_manual( values = c("dodgerblue", "firebrick2")) +  theme_bw() +  
  theme(panel.grid = element_blank(), legend.title = element_blank(), legend.text=element_text(size=8), legend.position = "top", legend.box = "horizontal", legend.spacing = unit(0.02, 'cm'), legend.margin=margin(0,0,0,0), legend.box.margin=margin(-10,-10,-10,-10)) + geom_point(size=1) +   xlab(paste0("PC1: ",percentVar.sod1[1],"% variance")) +  ylab(paste0("PC2: ",percentVar.sod1[2],"% variance")) + coord_fixed()
pca.sod1

# C9orf72
birger <- read_excel("/camp/home/ziffo/home/downloaded-public-data/astrocyte-c9orf72-ipsc-birger-2019/sample-details/metadata.xlsx")
birger$condition <- birger$condition %>% gsub("control", "ctrl", .)
birger <- birger %>% mutate(sample = factor(sample, levels = c("control_L3", "control_L9", "C9orf72_L5", "C9orf72_L8")), 
                            mutation = factor(condition, levels = c("ctrl", "C9orf72")),
                            filename = sample,
                            fraction = "whole")
birger$patient <- birger$sample %>% gsub("control_L", "ctrl_L", .)
birger$sample <- birger$sample %>% gsub("control_L", "ctrl", .) %>% gsub("C9orf72_L", "C9orf", .) %>% paste("ac_bir_", .,sep="")
birger$condition <- gsub("C9orf72", "als", birger$mutation)
birger$condition <- factor(birger$condition, levels = c("ctrl", "als")) # set the reference control first to ensure comparison is ALS vs CTRL (rather than CTRL vs ALS)
birger <- birger %>% dplyr::select(sampleName = sample, patient, condition, mutation) %>% mutate(dataset = "C9ORF72", fileName = paste(sampleName, ".tab", sep = "")) %>% dplyr::select(sampleName, fileName, condition, dataset, everything()) %>% as.data.frame
dds.c9orf <- DESeqDataSetFromHTSeqCount(sampleTable = birger, directory = htseqdir, design = ~ condition) # create DESeqDataSet with this design - Import htseq .tab tables
dds.c9orf <- DESeq(dds.c9orf) # run DESeq2
vsd.c9orf <- vst(dds.c9orf, blind=FALSE) # VST transformation
pcaData.c9orf <- plotPCA(vsd.c9orf, intgroup=c("condition", "mutation", "patient"), returnData=TRUE)
percentVar.c9orf <- round(100 * attr(pcaData.c9orf, "percentVar"))
pca.c9orf <- ggplot(pcaData.c9orf, aes(PC1, PC2, color=mutation)) + geom_text_repel(aes(label=patient), size = 3) + 
  scale_colour_manual( values = c("dodgerblue", "firebrick2")) +  theme_bw() +  
  theme(panel.grid = element_blank(), legend.title = element_blank(), legend.text=element_text(size=8), legend.position = "top", legend.box = "horizontal", legend.spacing = unit(0.02, 'cm'), legend.margin=margin(0,0,0,0), legend.box.margin=margin(-10,-10,-10,-10)) + geom_point(size=1) +   xlab(paste0("PC1: ",percentVar.c9orf[1],"% variance")) +  ylab(paste0("PC2: ",percentVar.c9orf[2],"% variance")) + coord_fixed()
pca.c9orf


# Stimulated Barbar
barbar <- read_excel("/camp/home/ziffo/home/downloaded-public-data/reactive-astrocyte-barbar-2020/sample-details/sample_table.xls")
barbar <- barbar %>% mutate(sampleName = sample, # set the reference control first
                            fileName = paste(sampleName, ".tab", sep = ""),
                            condition = factor(reactivity, levels = c("A0", "A1")),
                            patient = gsub("ac_bar_", "", sample), patient = gsub("ac_bar_", "", patient), patient = gsub("m", "_M", patient), patient = gsub("f", "_F", patient), patient = gsub("u", "_U", patient), 
                            patient = gsub("a1", "stimulated", patient), patient = gsub("a0", "untreated", patient), 
                            dataset = "Cytokine-stimulated", mutation = condition,
                            ) %>% 
  dplyr::select(sampleName, fileName, condition, patient, dataset, mutation) %>% as.data.frame
dds.a1 <- DESeqDataSetFromHTSeqCount(sampleTable = barbar, directory = htseqdir, design = ~ condition) # create DESeqDataSet with this design - Import htseq .tab tables
dds.a1 <- DESeq(dds.a1) # run DESeq2
vsd.a1 <- vst(dds.a1, blind=FALSE) # VST transformation
pcaData.a1 <- plotPCA(vsd.a1, intgroup=c("condition", "mutation", "patient"), returnData=TRUE)
percentVar.a1 <- round(100 * attr(pcaData.a1, "percentVar"))
pca.a1 <- ggplot(pcaData.a1, aes(PC1, PC2, color=mutation)) + geom_text_repel(aes(label=patient), size = 3) + 
  scale_colour_manual( values = c("dodgerblue", "firebrick2")) +  theme_bw() +  
  theme(panel.grid = element_blank(), legend.title = element_blank(), legend.text=element_text(size=8), legend.position = "top", legend.box = "horizontal", legend.spacing = unit(0.02, 'cm'), legend.margin=margin(0,0,0,0), legend.box.margin=margin(-10,-10,-10,-10)) + geom_point(size=1) +   xlab(paste0("PC1: ",percentVar.a1[1],"% variance")) +  ylab(paste0("PC2: ",percentVar.a1[2],"% variance")) + coord_fixed()
pca.a1

# TDP43 knockout
peng <- read_excel("/camp/home/ziffo/home/downloaded-public-data/astrocyte-tdp43-mouse-peng-2020/sample-details/metadata.xlsx") %>%
  filter(group != "ac_tdp_het")
peng$group <- peng$group %>% gsub("ac_", "", .)
peng$sampleName <- peng$sample %>% gsub("ac_", "", .)
peng <- peng %>% mutate(sampleName = sampleName, 
                        fileName = paste(sample, ".tab", sep = ""),
                        condition = factor(group, levels = c("tdp_ctrl", "tdp_ko"))) %>%  # set the reference control first
  dplyr::select(sampleName, fileName, condition) %>% as.data.frame
dds.tdp43 <- DESeqDataSetFromHTSeqCount(sampleTable = peng, directory = htseqdir, design = ~ condition) # create DESeqDataSet with this design - Import htseq .tab tables
dds.tdp43 <- DESeq(dds.tdp43) # run DESeq2
vsd.tdp43 <- vst(dds.tdp43, blind=FALSE) # VST transformation
pcaData.tdp43 <- plotPCA(vsd.tdp43, intgroup=c("condition"), returnData=TRUE)
percentVar.tdp43 <- round(100 * attr(pcaData.tdp43, "percentVar"))
pca.tdp43 <- ggplot(pcaData.tdp43, aes(PC1, PC2, color=condition)) + geom_text_repel(aes(label=rownames(pcaData.tdp43)), size = 3) + 
  scale_colour_manual( values = c("dodgerblue", "firebrick2")) +  theme_bw() +  
  theme(panel.grid = element_blank(), legend.title = element_blank(), legend.text=element_text(size=8), legend.position = "top", legend.box = "horizontal", legend.spacing = unit(0.02, 'cm'), legend.margin=margin(0,0,0,0), legend.box.margin=margin(-10,-10,-10,-10)) + geom_point(size=1) +   xlab(paste0("PC1: ",percentVar.tdp43[1],"% variance")) +  ylab(paste0("PC2: ",percentVar.tdp43[2],"% variance")) + coord_fixed()
pca.tdp43

# SOD1 mouse TRAP-seq
cleveland <- read_excel("/camp/home/ziffo/home/downloaded-public-data/astrocyte-sod1-mouse-bacTRAP-cleveland-2015/sample-details/metadata.xlsx")
cleveland <- cleveland %>% mutate(sampleName = cell_line, 
                            fileName = paste(sample, ".tab", sep = ""),
                            condition = factor(condition, levels = c("ctrl", "sod1"))) %>%  # set the reference control first
  dplyr::select(sampleName, fileName, condition) %>% as.data.frame
dds.TRAPseq <- DESeqDataSetFromHTSeqCount(sampleTable = cleveland, 
                                            directory = htseqdir, 
                                            design = ~ condition) # create DESeqDataSet with this design - Import htseq .tab tables
dds.TRAPseq <- DESeq(dds.TRAPseq) # run DESeq2
vsd.TRAPseq <- vst(dds.TRAPseq, blind=FALSE) # VST transformation
pcaData.TRAPseq <- plotPCA(vsd.TRAPseq, intgroup=c("condition"), returnData=TRUE)
percentVar.TRAPseq <- round(100 * attr(pcaData.TRAPseq, "percentVar"))
pca.TRAPseq <- ggplot(pcaData.TRAPseq, aes(PC1, PC2, color=condition)) + geom_text_repel(aes(label=rownames(pcaData.TRAPseq)), size = 3) + 
  scale_colour_manual( values = c("dodgerblue", "firebrick2")) +  theme_bw() +  
  theme(panel.grid = element_blank(), legend.title = element_blank(), legend.text=element_text(size=8), legend.position = "top", legend.box = "horizontal", legend.spacing = unit(0.02, 'cm'), legend.margin=margin(0,0,0,0), legend.box.margin=margin(-10,-10,-10,-10)) + geom_point(size=1) +   xlab(paste0("PC1: ",percentVar.TRAPseq[1],"% variance")) +  ylab(paste0("PC2: ",percentVar.TRAPseq[2],"% variance")) + coord_fixed()
pca.TRAPseq

pca_individual_datasets_separated <- ggarrange(ggarrange(pca.vcp, pca.sod1, pca.c9orf, nrow = 1, ncol = 3, widths = c(1,1,1)), 
          ggarrange(pca.a1, pca.tdp43, pca.TRAPseq, nrow = 1, ncol = 3, widths = c(1.5,1,1)),
          nrow = 2)
pca_individual_datasets_separated

### Dendogram of all datasets together
# merge dataset dfs
### Dendrogram all human datasets
patani <- patani %>% mutate(sampleName = sample, fileName = sample) %>% mutate(fileName = paste(fileName, ".tab", sep = ""))
patani$sampleName <- gsub("ac_who_", "hiPSC_VCP_", patani$sampleName) %>% gsub("ctrl5", "ctrl2", .) %>% gsub("cb1e", "R191Q", .) %>% gsub("glia", "R155C", .) 
tyzack$sampleName <- gsub("ac_tyz_", "hiPSC_SOD1_", tyzack$sampleName)
birger$sampleName <- gsub("ac_bir_", "hiPSC_C9_", birger$sampleName)
barbar$sampleName <- gsub("ac_bar_a0", "hiPSC_unstim_", barbar$sampleName) %>% gsub("ac_bar_a1", "hiPSC_cytokine_", .)
peng$sampleName <- gsub("tdp_", "tdp43_", peng$sampleName)
cleveland$sampleName <- gsub("ctrl", "TRAPseq_ctrl", cleveland$sampleName) %>% gsub("g37r", "TRAPseq_g37r", .)

sampleTable <- bind_rows(patani, tyzack, birger, barbar) %>% #, peng, cleveland) %>% # cannot run DESeq if different GTFs used (mouse vs human - need to filter HTSeq.tab to only matching gene names)
  select(sampleName, fileName) %>% as.data.frame
# run DESeq2
htseqdir="/camp/home/ziffo/home/projects/astrocyte-meta-analysis/alignment/htseq"
dds <- DESeqDataSetFromHTSeqCount(sampleTable = sampleTable, 
                                            directory = htseqdir, 
                                            design = ~ 1) # create DESeqDataSet with this design - Import htseq .tab tables
dds <- DESeq(dds) # run DESeq2
vsd <- vst(dds, blind=FALSE) # VST transformation
# create dendrogram
sampleDists <- dist(t(assay(vsd)))
hc <- hclust(sampleDists, method = "ward.D2")
a <- c("dodgerblue", "dodgerblue", "firebrick2","firebrick2","dodgerblue", "dodgerblue", "firebrick2","firebrick2","firebrick2","dodgerblue", "dodgerblue","firebrick2","firebrick2","firebrick2","dodgerblue", "dodgerblue","dodgerblue")
ac_human_dendrogram <- ggdendrogram(hc, rotate = TRUE, theme_dendro = FALSE) +  theme_classic()  + theme(axis.line=element_blank(), axis.title.x=element_blank(), axis.title.y=element_blank(), axis.text.x=element_blank(), axis.ticks.x=element_blank(), axis.text.y= element_text(colour = a), strip.background = element_rect(colour="black", fill="white"), panel.grid = element_blank()) 
ac_human_dendrogram

# Mouse datasets dendogram
sampleTable <- bind_rows(peng, cleveland) %>% select(sampleName, fileName) %>% as.data.frame
htseqdir="/camp/home/ziffo/home/projects/astrocyte-meta-analysis/alignment/htseq"
dds <- DESeqDataSetFromHTSeqCount(sampleTable = sampleTable, 
                                            directory = htseqdir, 
                                            design = ~ 1) # create DESeqDataSet with this design - Import htseq .tab tables
dds <- DESeq(dds) # run DESeq2
vsd <- vst(dds, blind=FALSE) # VST transformation
# create dendrogram
sampleDists <- dist(t(assay(vsd)))
hc <- hclust(sampleDists, method = "ward.D2")
a <- c("dodgerblue", "dodgerblue", "dodgerblue", "dodgerblue","dodgerblue", "dodgerblue",
       "firebrick2","firebrick2", "firebrick2","firebrick2",
       "firebrick2","firebrick2", "firebrick2","firebrick2",
       "dodgerblue", "dodgerblue", "dodgerblue", "dodgerblue")
ac_mouse_dendrogram <- ggdendrogram(hc, rotate = TRUE, theme_dendro = FALSE) +  theme_classic()  + theme(axis.line=element_blank(), axis.title.x=element_blank(), axis.title.y=element_blank(), axis.text.x=element_blank(), axis.ticks.x=element_blank(), axis.text.y= element_text(colour = a), strip.background = element_rect(colour="black", fill="white"), panel.grid = element_blank()) 
ac_mouse_dendrogram


### Gene Expression GO plot VCP vs CTRL
patani <- read.csv("/camp/home/ziffo/home/projects/astrocyte-fractionation-bulk-rnaseq/sample-details/sample_table.csv")
patani <- patani[!patani$cellline == "ctrl2",] # remove CTRL2 samples due to batch effects
patani$cellline <- patani$cellline %>% gsub("ctrl5", "ctrl2", .) # for the paper we rename ctrl5 as ctrl2 to avoid confusion as to why we have ctrl1 & 5 (but not 2,3 and 4)
patani <- patani %>% distinct(fraction_cellline, .keep_all = TRUE) # remove rows for technical repeats
patani <- patani %>% mutate(mutation = as.factor(vcp), fraction = factor(fraction, levels = c("whole", "nuclear", "cytoplasmic")))
patani$condition <- gsub("vcp", "als", patani$mutation)
patani$condition <- factor(patani$condition, levels = c("ctrl", "als")) # set the reference control first to ensure comparison is ALS vs CTRL (rather than CTRL vs ALS)
patani$sample <- patani$fraction_cellline %>% gsub("whole", "who", .) %>% gsub("nuclear", "nuc", .) %>% gsub("cytoplasmic", "cyt", .) #%>% gsub("cb1e", "R191Q", .) %>% gsub("glia", "R155C", .)
patani$sample <- paste("ac_",patani$sample,sep="")
patani$patient <- patani$cellline %>% gsub("cb1e", "R191Q", .) %>% gsub("glia", "R155C", .)
patani <- patani %>% dplyr::filter(fraction == "whole") # astrocyte whole-cell samples for VCP only
sampleTable <- patani %>% dplyr::select(sampleName=fraction_cellline, condition, fraction, mutation, cellline, patient, sample) %>% mutate(fileName = paste(sample, ".tab", sep = ""), patient_fraction = paste(patient, fraction, sep = "_")) %>% dplyr::select(sampleName, fileName, condition, everything())
htseqdir <- "/camp/home/ziffo/home/projects/astrocyte-meta-analysis/alignment/htseq"
dds.vcp <- DESeqDataSetFromHTSeqCount(sampleTable = sampleTable, directory = htseqdir, design = ~ condition) # create DESeqDataSet with this design - Import htseq .tab tables
dds.vcp <- DESeq(dds.vcp) # run DESeq2
res <- DESeq2::results(dds.vcp, name = "condition_als_vs_ctrl") # Astrocyte VCP vs CTRL
gene2ens_filtered <-  filter(gene2ens, gene_id %in% rownames(res)) %>% unique # filter gene to ensID to only ENS names in the results tibble
dge_res_ac_who_vcp_vs_ctrl <- res %>% as_tibble(rownames = "ensID") %>% left_join(gene2ens_filtered, by=c("ensID"="gene_id")) %>% arrange(padj) # join gene gene_nameS to res tibble & order by padj
dge_res_ac_who_vcp_vs_ctrl.FDR0.05 <- dge_res_ac_who_vcp_vs_ctrl %>% filter(padj < 0.05) # 170
dge_genes <- dge_res_ac_who_vcp_vs_ctrl.FDR0.05 %>% mutate(direction = case_when(log2FoldChange > 0 ~ "Up in VCP", TRUE ~ "Down in VCP")) %>% split(.$direction) %>%  map("ensID") # create list of IR up/down & Expression up/down genes
dge_genes <- gost(dge_genes) # run gprofiler2 on all groups - runs on all KEGG, GO, REAC,TF, WP pathways simultaneously
dge_genes$result
dge_gprofiles_filt <- dge_genes$result %>% filter(str_detect(source, "KEGG|GO:BP|GO:MF|GO:CC|REAC|CORUM")) %>% arrange( -log10(p_value) ) %>% mutate( term_name = as.factor(term_name))
dge_gprofiles_filt
dge_gprofiles_filt$term_name <- gsub("extracellular matrix structural constituent conferring compression resistance", "extracellular matrix", dge_gprofiles_filt$term_name)
dge_gprofiles_filt$term_name <- gsub("central nervous system", "", dge_gprofiles_filt$term_name)
dge_gprofiles_filt$query <- factor(dge_gprofiles_filt$query, levels = c("Up in VCP", "Down in VCP"))
ac_who_vcp_vs_ctrl.DGE_only.go_curated <- curated_profiles(gprofiles = dge_gprofiles_filt)
ac_who_vcp_vs_ctrl.DGE_only.go_curated
```

## Merge

Astrocyte differentiation Biorender figure merged in illustrator

```{r}
figS1.merged <- ggarrange(
  pca_individual_datasets_separated,
  ggarrange(ac_human_dendrogram, ac_mouse_dendrogram, ncol = 2, widths = c(1,1), labels = c("C", "D")),
  ggarrange(NULL, ac_who_vcp_vs_ctrl.DGE_only.go_curated, ncol = 2, widths = c(0.1, 1)),
  nrow = 3, heights = c(3.5, 2.7, 2), labels = c("B", "", "E"))
figS1.merged
ggsave(figS1.merged, filename = "/camp/home/ziffo/home/projects/astrocyte-meta-analysis/splicing/IRFinder/irlib/figures/figS1.merged.png", height = 10.5, width = 9)
```


# Figure S2: VASTTOOLS & SGSeq

```{r}
# D VAST TOOLS splicing modes
source("/camp/home/ziffo/home/projects/astrocyte-meta-analysis/scripts/vast_tools_functions.r")
ac_who_vcp_vs_ctrl_compare <- read.table("/camp/home/ziffo/home/projects/astrocyte-meta-analysis/splicing/vast-tools/vast_out/whole/compare.tab", sep = '\t',header = TRUE)
ac_who_vcp_vs_ctrl_compare_all_events <- read.table("/camp/home/ziffo/home/projects/astrocyte-meta-analysis/splicing/vast-tools/vast_out/whole/compare_all_events.tab", sep = '\t',header = TRUE)
ac_who_vcp_vs_ctrl_diff <- read.table("/camp/home/ziffo/home/projects/astrocyte-meta-analysis/splicing/vast-tools/vast_out/whole/INCLUSION_table.DIFF.txt", sep = '\t',header = TRUE)
ac_who_vcp_vs_ctrl.VASTTOOLS <- cleanVASTTOOLS(diff = ac_who_vcp_vs_ctrl_diff, comp = ac_who_vcp_vs_ctrl_compare)
ac_who_vcp_vs_ctrl.VASTTOOLS_all_events <- cleanVASTTOOLS(diff = ac_who_vcp_vs_ctrl_diff, comp = ac_who_vcp_vs_ctrl_compare_all_events)

vast_tools.splicingtype_test <- filter(ac_who_vcp_vs_ctrl.VASTTOOLS, MV0.1 == "significant") %>% dplyr::select(E.dPsi., type) %>% group_by(type) %>% t_test(E.dPsi. ~ 1, mu = 0) %>% adjust_pvalue() %>%  add_significance() %>% mutate(y.position = 1)
vast_tools.splicingtype_test
vast_tools.all_type_order <- filter(ac_who_vcp_vs_ctrl.VASTTOOLS, MV0.1 == "significant") %>% group_by(type) %>% tally %>% arrange(n) %>%  pull(type)
vast_tools.all_splice_ordered <- filter(ac_who_vcp_vs_ctrl.VASTTOOLS, MV0.1 == "significant") %>% mutate(type = factor(type, levels = vast_tools.all_type_order))
vast_tools.sinaplot_splice_types <- ggplot(vast_tools.all_splice_ordered, aes(x = type, y= E.dPsi., color = type)) + geom_violin() + geom_sina(size = 0.5) + 
  geom_boxplot(data = vast_tools.all_splice_ordered, aes(fill = type), colour = "black", width=0.1, outlier.shape = NA) +
  # scale_colour_manual(values = rev(get_palette("npg",4))) + scale_fill_manual(values = rev(get_palette("npg",4))) +
  scale_colour_manual(values = get_palette("npg",4)) + scale_fill_manual(values = get_palette("npg",4)) +
  theme_classic() + theme(legend.position = "none", axis.text.y=element_text(size=10)) +
  geom_hline(yintercept =0, linetype = 2) + coord_flip() +# ylim(c(-1,1)) +
  geom_segment(aes(x = 0.6, xend = 0.6, y= 0.7, yend= 1), arrow=arrow(length=unit(0.4,"cm")), color = "dodgerblue3") +
  geom_segment(aes(x = 0.6, xend = 0.6, y= -0.7, yend= -1),arrow=arrow(length=unit(0.4,"cm")), color = "firebrick2") +
  annotate("text", x = 0.7, y = 0.5, label = "Up in VCP", color = "dodgerblue3") +   annotate("text", x = 0.7, y = -0.5, label = "Down in VCP", color = "firebrick2") +
  stat_pvalue_manual(vast_tools.splicingtype_test, label = "p.adj.signif", xmin = "type", xmax = NULL, hide.ns = TRUE) + # add manually p values from binomial test
  ylab(label = expression(VAST~TOOLS~Delta~PSI~VCP-CTRL) ) + xlab("")
vast_tools.sinaplot_splice_types

# E SGSeq splicing modes
res.events <- readRDS("/camp/home/ziffo/home/projects/astrocyte-meta-analysis/splicing/sgseq/annotated/ac_who_vcp_vs_ctrl/res.events.RDS")
res.events$type_translation <- gsub("retained intron", "intron retention", res.events$type_translation)
res.events$type_translation <- gsub("cassette exon", "exon skipping", res.events$type_translation)
type_order <- res.events %>% filter( status_strict != "none_significant" ) %>% group_by(type_translation) %>% tally %>% arrange(n) %>%  pull(type_translation)
type_counts <- res.events %>% filter(status_strict != "none_significant") %>% mutate(type_translation = factor(type_translation, levels = type_order)) %>%
  mutate(status_strict = factor(status_strict, levels = c("significant", "none_significant"))) %>% group_by( status_strict, type_translation) %>% tally(name = "typecount") %>% ungroup()
splicingtype_test <- filter(res.events, type_translation %in% c("alternative start", "intron retention", "alternate first exon", "alternate end", " alternate last exon", "alternate 3'", "alternate 5'", "exon skipping")) %>% dplyr::select(lfc, type_translation) %>% group_by(type_translation) %>% t_test(lfc ~ 1, mu = 0) %>% adjust_pvalue() %>%  add_significance() %>% mutate(y.position = 1)
splicingtype_test
all_type_order <- filter(res.events, type_translation %in% c("alternative start", "intron retention", "alternate first exon", "alternate end", " alternate last exon", "alternate 3'", "alternate 5'", "exon skipping")) %>% group_by(type_translation) %>% tally %>% arrange(n) %>%  pull(type_translation)
all_splice_ordered <- filter(res.events, type_translation %in% c("alternative start", "intron retention", "alternate first exon", "alternate end", " alternate last exon", "alternate 3'", "alternate 5'", "exon skipping")) %>% mutate(type = factor(type_translation, levels = all_type_order))
sgseq.sinaplot_splice_types <- ggplot(all_splice_ordered, aes(x = type_translation, y= lfc, color = type_translation)) + geom_violin() + geom_sina(size = 0.5) + 
  geom_boxplot(data = all_splice_ordered, aes(fill = type_translation), colour = "black", width=0.2, outlier.shape = NA) +
  scale_colour_manual(values = get_palette("npg",7)) + scale_fill_manual(values = get_palette("npg",7)) +
  theme_classic() + theme(legend.position = "none", axis.text.y=element_text(size=10)) +
  geom_hline(yintercept =0, linetype = 2) + coord_flip() + ylim(c(-0.5,0.5)) +
  geom_segment(aes(x = 0.6, xend = 0.6, y= 0.4, yend= 0.5), arrow=arrow(length=unit(0.4,"cm")), color = "dodgerblue3") +
  geom_segment(aes(x = 0.6, xend = 0.6, y= -0.4, yend= -0.5),arrow=arrow(length=unit(0.4,"cm")), color = "firebrick2") +
  annotate("text", x = 0.7, y = 0.3, label = "Up in VCP", color = "dodgerblue3") +   annotate("text", x = 0.7, y = -0.3, label = "Down in VCP", color = "firebrick2") +
  stat_pvalue_manual(splicingtype_test, label = "p.adj.signif", xmin = "type_translation", xmax = NULL, hide.ns = TRUE, y.position = 0.5) + # add manually p values from binomial test
  ylab(label = expression(SGSeq~Log[2]~foldchange~VCP:CTRL) ) + xlab("")
sgseq.sinaplot_splice_types
```

## Merge

Astrocyte differentiation Biorender figure merged in illustrator

```{r}
figS2.merged <- ggarrange(vast_tools.sinaplot_splice_types,
  sgseq.sinaplot_splice_types,
  nrow = 2, heights = c(1, 2), labels = c("A", "B"))
figS2.merged
ggsave(figS2.merged, filename = "/camp/home/ziffo/home/projects/astrocyte-meta-analysis/splicing/IRFinder/irlib/figures/figS2.merged.png", height = 8, width = 8)
```


# Figure 1: Decreased IR in ALS mutants

Panels A & B are from Biorender so merged with illustrator

```{r}
ac_who_vcp_vs_ctrl.absoluteIR %>% dplyr::filter(A.reliable_retained == "reliable-retained" | B.reliable_retained == "reliable-retained") %>% nrow # 16,317 reliable unique retained introns 
ac_who_vcp_vs_ctrl.absoluteIR %>% dplyr::filter(A.reliable_retained == "reliable-retained") %>% nrow  # 12,169 reliable unique retained introns in VCP
ac_who_vcp_vs_ctrl.absoluteIR %>% dplyr::filter(B.reliable_retained == "reliable-retained") %>% nrow  # 14,162 reliable unique retained introns in control
(14162-12169)/14162 * 100 # 14.1% fewer retained introns in VCP compared with control 
ac_who_vcp_vs_ctrl %>% dplyr::filter(p0.05_reliable == "significant") %>% nrow # 2,309 reliable introns retained at significantly different levels
ac_who_vcp_vs_ctrl %>% dplyr::filter(p0.05_reliable == "significant") %>% distinct(gene_name) %>% nrow # from 1,699 unique genes
ac_who_vcp_vs_ctrl %>% dplyr::filter(p0.05_reliable == "significant", IRratio.diff < 0) %>% nrow # 1,885 (81.6%) displaying decreased IR levels in VCP
(2309-(2309-1885))/2309 * 100 # 81.6% decreased in VCP

# sinaplot of IR events in VCP, C9orf72, SOD1 & reactive astrocytes
vcp <- ac_who_vcp_vs_ctrl %>% dplyr::filter(p0.05_reliable == "significant") %>% dplyr::select(Intron.GeneName.GeneID, Chr, Start, End, gene_name, ensID, p.diff, IRratio.diff, p0.05_reliable_IRdirection, log2FoldChange) %>% mutate(mutation = "VCP - CTRL")
sod1 <-ac_tyz_sod1_vs_ctrl %>% dplyr::filter(p0.05_reliable == "significant") %>% dplyr::select(Intron.GeneName.GeneID, Chr, Start, End, gene_name, ensID, p.diff, IRratio.diff, p0.05_reliable_IRdirection, log2FoldChange) %>% mutate(mutation = "SOD1 - CTRL")
c9orf <- ac_bir_c9orf_vs_ctrl %>% dplyr::filter(p0.05_reliable == "significant") %>% dplyr::select(Intron.GeneName.GeneID, Chr, Start, End, gene_name, ensID, p.diff, IRratio.diff, p0.05_reliable_IRdirection, log2FoldChange) %>% mutate(mutation = "C9orf72 - CTRL")
vcp_sod1_c9orf <- bind_rows(vcp, sod1, c9orf) %>% mutate(mutation = factor(mutation, levels = c("VCP - CTRL", "C9orf72 - CTRL", "SOD1 - CTRL")))

vcp_sod1_c9orf_sina_plot <- sinaplot.one.sample(data = vcp_sod1_c9orf, groups = "mutation", continuous = "IRratio.diff", labels = "gene_name", label_number = 0, stat.y.position = 0.3, width = 0.15, flip = "no", colours = 4) + ylab(label = expression(Delta~IR~ratio) ) + xlab("") + ylim(c(-0.3,0.3)) +
  geom_segment(aes(x = 3.5, xend = 3.5, y= 0.05, yend= 0.25), arrow=arrow(length=unit(0.4,"cm")), color = "dodgerblue") +
  geom_segment(aes(x = 3.5, xend = 3.5, y= -0.05, yend= -0.25),arrow=arrow(length=unit(0.4,"cm")), color = "firebrick2") +
  annotate("text", x = 3.3, y = 0.27, label = "IR Up", colour = "dodgerblue", size = 3) +   annotate("text", x = 3.3, y = -0.27, label = "IR Down", colour = "firebrick2", size = 3)
vcp_sod1_c9orf_sina_plot

#  "Stats test used =  t_test"
# # A tibble: 3 x 10
#   group          .y.   group1 group2         n statistic    df        p p.signif y.position
#   <fct>          <chr> <chr>  <chr>      <int>     <dbl> <dbl>    <dbl> <chr>         <dbl>
# 1 VCP - CTRL     lfc   1      null model  2309    -18.1   2308 2.39e-68 ****            0.3
# 2 C9orf72 - CTRL lfc   1      null model   448    -16.7    447 8.76e-49 ****            0.3
# 3 SOD1 - CTRL    lfc   1      null model   139     -4.95   138 2.13e- 6 ****            0.3

ac_who_vcp_vs_ctrl %>% dplyr::filter(p0.05_reliable == "significant", IRratio.diff < 0, padj < 0.05) %>% distinct(gene_name) 
# Of genes with significantly decreased IR, 4 genes (ERAP2, PTPRN, IL20RB, and KCNE4) were also significantly increased in gene expression in VCP versus control (FDR < 0.05) 
# 1     ERAP2
# 2     PTPRN
# 3    IL20RB
# 4     KCNE4

ac_bir_c9orf_vs_ctrl.absoluteIR %>% dplyr::filter(A.reliable_retained == "reliable-retained" | B.reliable_retained == "reliable-retained") %>% nrow # 16,316 reliable unique retained introns 
ac_bir_c9orf_vs_ctrl.absoluteIR %>% dplyr::filter(A.reliable_retained == "reliable-retained") %>% nrow  # 7,925 reliable unique retained introns in C9orf72
ac_bir_c9orf_vs_ctrl.absoluteIR %>% dplyr::filter(B.reliable_retained == "reliable-retained") %>% nrow  # 14,937 reliable unique retained introns in control
(14938-7925)/14938 * 100 # 47% fewer retained introns in c9orf compared with control 

ac_tyz_sod1_vs_ctrl.absoluteIR %>% dplyr::filter(A.reliable_retained == "reliable-retained" | B.reliable_retained == "reliable-retained") %>% nrow # 8610 reliable unique retained introns 
ac_tyz_sod1_vs_ctrl.absoluteIR %>% dplyr::filter(A.reliable_retained == "reliable-retained") %>% nrow  # 3602 reliable unique retained introns in sod1
ac_tyz_sod1_vs_ctrl.absoluteIR %>% dplyr::filter(B.reliable_retained == "reliable-retained") %>% nrow  # 7842 reliable unique retained introns in control
(7842-3602)/7842 * 100 # 54% fewer retained introns in sod1 compared with control 

ac_bir_c9orf_vs_ctrl %>% dplyr::filter(p0.05_reliable == "significant") %>% nrow # 448 reliable introns retained at significantly different levels
ac_bir_c9orf_vs_ctrl %>% dplyr::filter(p0.05_reliable == "significant") %>% distinct(gene_name) %>% nrow # from 376 unique genes
ac_bir_c9orf_vs_ctrl %>% dplyr::filter(p0.05_reliable == "significant", IRratio.diff < 0) %>% nrow # 416 displaying decreased IR levels in c9orf
(448-(448-416))/448 * 100 # 93% decreased in c9orf

ac_tyz_sod1_vs_ctrl %>% dplyr::filter(p0.05_reliable == "significant") %>% nrow # 139 reliable introns retained at significantly different levels
ac_tyz_sod1_vs_ctrl %>% dplyr::filter(p0.05_reliable == "significant") %>% distinct(gene_name) %>% nrow # from 131 unique genes
ac_tyz_sod1_vs_ctrl %>% dplyr::filter(p0.05_reliable == "significant", IRratio.diff < 0) %>% nrow # 82 displaying decreased IR levels in sod1
(139-(139-82))/139 * 100 # 59% decreased in sod1

# Overlap IR events between VCP, C9orf72 and SOD1
vcp.filt <- ac_who_vcp_vs_ctrl %>% dplyr::filter(p0.05_reliable == "significant") %>% dplyr::select(gene_name, ensID, IR_vcp = p0.05_reliable_IRdirection) # Intron.GeneName.GeneID, Chr, Start, End, 
sod1.filt <-ac_tyz_sod1_vs_ctrl %>% dplyr::filter(p0.05_reliable == "significant") %>% dplyr::select(gene_name, ensID, IR_sod1 = p0.05_reliable_IRdirection) # Intron.GeneName.GeneID, Chr, Start, End, 
c9orf.filt <- ac_bir_c9orf_vs_ctrl %>% dplyr::filter(p0.05_reliable == "significant") %>% dplyr::select(gene_name, ensID, IR_c9orf = p0.05_reliable_IRdirection) # Intron.GeneName.GeneID, Chr, Start, End, 
vcp_sod1_c9orf.overlap <- vcp.filt %>% full_join(sod1.filt, c("gene_name", "ensID")) %>% # "Intron.GeneName.GeneID", "Chr", "Start", "End", 
  full_join(c9orf.filt, c("gene_name", "ensID")) %>% # "Intron.GeneName.GeneID", "Chr", "Start", "End",
  distinct(.keep_all = TRUE) # keep unique gene names
vcp_sod1_c9orf.overlap <- vcp_sod1_c9orf.overlap %>% mutate(overlap.down = case_when(IR_vcp == "IR down" & IR_sod1 == "IR down" & IR_c9orf == "IR down" ~ "overlapping",
                                                                              IR_vcp == "IR down" & IR_sod1 == "IR down" ~ "VCP & SOD1", 
                                                                              IR_vcp == "IR down" & IR_c9orf == "IR down" ~ "VCP & C9orf72", 
                                                                              IR_c9orf == "IR down" ~ "C9orf72 specific", 
                                                                              IR_sod1 == "IR down" ~ "SOD1 specific", 
                                                                              IR_vcp == "IR down" ~ "VCP specific",
                                                                              TRUE ~ "none"))
vcp_sod1_c9orf.overlap.filt <- filter(vcp_sod1_c9orf.overlap, overlap.down != "none")

table(vcp_sod1_c9orf.overlap.filt$overlap.down)
# C9orf72 specific      overlapping    SOD1 specific    VCP & C9orf72       VCP & SOD1     VCP specific 
#              230               10               41              119               21             1247 

# venn overlap of IR events
venn_vcp_sod1_c9orf_overlap.down <- as.ggplot(
  ~draw.triple.venn(
    area1 = nrow( filter(vcp_sod1_c9orf.overlap.filt, overlap.down %in% c("VCP specific", "VCP & SOD1", "VCP & C9orf72", "overlapping" ) ) ), # VCP events
    area2 = nrow( filter(vcp_sod1_c9orf.overlap.filt, overlap.down %in% c("SOD1 specific", "VCP & SOD1", "C9orf72 & SOD1", "overlapping" ) ) ), # SOD1 events
    area3 = nrow( filter(vcp_sod1_c9orf.overlap.filt, overlap.down %in% c("C9orf72 specific", "C9orf72 & SOD1", "VCP & C9orf72", "overlapping" ) ) ), # C9orf72 events
    n12 = nrow( filter(vcp_sod1_c9orf.overlap.filt, overlap.down %in% c("VCP & SOD1", "overlapping" ) ) ), # VCP & SOD1
    n23 = nrow( filter(vcp_sod1_c9orf.overlap.filt, overlap.down %in% c("C9orf72 & SOD1", "overlapping" ) ) ), # SOD1 & C9orf72
    n13 = nrow( filter(vcp_sod1_c9orf.overlap.filt, overlap.down %in% c("VCP & C9orf72", "overlapping" ) ) ), # VCP & SOD1
    n123 = nrow(filter(vcp_sod1_c9orf.overlap.filt, overlap.down == "overlapping")),
    category = c("VCP", "SOD1", "C9orf72"),
    alpha = 0.5,  fill = c("firebrick2", "forestgreen", "dodgerblue3"), 
    cat.col = c("firebrick2", "forestgreen", "dodgerblue3"), cat.fontfamily = "sans", 
    cat.pos = c(330,30,0), cat.dist = 0.05) ) # cat.cex = c(1,1,1),
venn_vcp_sod1_c9orf_overlap.down


# hypergeometric testing
IRdown_VCP <- ac_who_vcp_vs_ctrl %>% filter(p0.05_reliable == "significant" & IRdirection == "down") %>% pull(gene_name) %>% unique
IRdown_SOD1 <- ac_tyz_sod1_vs_ctrl %>% filter(p0.05_reliable == "significant" & IRdirection == "down") %>% pull(gene_name) %>% unique
IRdown_C9ORF72 <- ac_bir_c9orf_vs_ctrl %>% filter(p0.05_reliable == "significant" & IRdirection == "down") %>% pull(gene_name) %>% unique

IRdown_SOD1[IRdown_SOD1 %in% IRdown_VCP] # 31 / 76
go.obj <- newGeneOverlap(IRdown_VCP,
                         IRdown_SOD1,
                         genome.size=nrow(distinct(gtf, gene_name)))
testGeneOverlap(go.obj)
# GeneOverlap object:
# listA size=1396
# listB size=76
# Intersection size=31
# Overlapping p-value=9e-31
# Jaccard Index=0.0

IRdown_C9ORF72[IRdown_C9ORF72 %in% IRdown_VCP] # 129 / 347
go.obj <- newGeneOverlap(IRdown_VCP,
                         IRdown_C9ORF72,
                         genome.size=nrow(distinct(gtf, gene_name)))
testGeneOverlap(go.obj)
# GeneOverlap object:
# listA size=1396
# listB size=347
# Intersection size=129
# Overlapping p-value=2e-118
# Jaccard Index=0.1

# 3-way overlap https://github.com/singlecoated/htSeqTools/blob/master/R/listOverlap.R
# test overlap of SOD1 VCP & C9orf72 3 gene lists:
list1 <- unique(IRdown_VCP[!is.na(IRdown_VCP)])
list2 <- unique(IRdown_SOD1[!is.na(IRdown_SOD1)])
list3 <- unique(IRdown_C9ORF72[!is.na(IRdown_C9ORF72)])
univ <- unique(gtf$gene_name[!is.na(gtf$gene_name)])
xtab <- data.frame(Univ=univ, List1=as.integer(univ %in% list1), List2=as.integer(univ %in% list2), List3=as.integer(univ %in% list3))
ftable <- as.data.frame(table(xtab[,c('List1','List2','List3')]))
glm1 <- glm(Freq ~ List1 * List2 * List3, family='poisson',data=ftable)
glm2 <- glm(Freq ~ List1*List2 + List1*List3 + List2*List3, family='poisson',data=ftable)
anovatest <- anova(glm1,glm2,test='Chisq')
pvalue <- anovatest[['Pr(>Chi)']][2] # 0.004703483
ans <- list(xtab,ftable,glm1,glm2,pvalue)
names(ans) <- c('xtab','ftable','glm1','glm2','pvalue')

# $ftable
#   List1 List2 List3  Freq
# 1     0     0     0 58892
# 2     1     0     0  1246
# 3     0     1     0    40
# 4     1     1     0    21
# 5     0     0     1   213
# 6     1     0     1   119
# 7     0     1     1     5
# 8     1     1     1    10
# 
# $glm1
# Call:  glm(formula = Freq ~ List1 * List2 * List3, family = "poisson",  data = ftable)
# Coefficients:
#          (Intercept)                List11                List21                List31         List11:List21         List11:List31         List21:List31      List11:List21:List31
#               10.983                -3.856                -7.295                -5.622                 3.211                 3.274                 3.543      -1.936 
# Degrees of Freedom: 7 Total (i.e. Null);  0 Residual
# Null Deviance:	    233800 
# Residual Deviance: -9.319e-12 	AIC: 69.66
# 
# $glm2
# Call:  glm(formula = Freq ~ List1 * List2 + List1 * List3 + List2 * List3, family = "poisson", data = ftable)
# Coefficients:
#   (Intercept)         List11         List21         List31  List11:List21  List11:List31  List21:List31  
#        10.983         -3.853         -7.208         -5.605          2.931          3.223          2.141  
# Degrees of Freedom: 7 Total (i.e. Null);  1 Residual
# Null Deviance:	    233800 
# Residual Deviance: 7.99 	AIC: 75.65
# 
# $pvalue
# [1] 0.004703483

IRdown_SOD1_VCP <- IRdown_SOD1[IRdown_SOD1 %in% IRdown_VCP]
IRdown_C9ORF72_VCP <- IRdown_C9ORF72[IRdown_C9ORF72 %in% IRdown_VCP]
IRdown_SOD1_VCP_C9ORF72 <- IRdown_SOD1_VCP[IRdown_SOD1_VCP %in% IRdown_C9ORF72] # Ten genes with decreased IR were common to all three mutations "FLNA"    "PLXND1"  "THBS3"   "ATP13A1" "PLOD3"   "MRC2"    "ACTN4"   "GPS2"    "CELSR2"  "MCAM"  
IRdown_SOD1_VCP_C9ORF72[IRdown_SOD1_VCP_C9ORF72 %in% all.reactivity.go] # 6 are relevant to reactivity "FLNA"  "PLOD3" "MRC2"  "ACTN4" "GPS2"  "MCAM" 


### Differential IR characteristics IR up vs IR down
VCP <- ac_who_vcp_vs_ctrl %>% dplyr::select(Intron.GeneName.GeneID.Coords, coords, ensID, gene_name, Chr, Start, End, Direction, gc_content, intron_length, p0.05_reliable, VCP.p0.05_reliable_IRdirection = p0.05_reliable_IRdirection) %>% filter(p0.05_reliable == "significant")
SOD1 <- ac_tyz_sod1_vs_ctrl %>% dplyr::select(Intron.GeneName.GeneID.Coords, coords, ensID, gene_name, Chr, Start, End, Direction,p0.05_reliable, SOD1.p0.05_reliable_IRdirection = p0.05_reliable_IRdirection, gc_content, intron_length)  %>% filter(p0.05_reliable == "significant")
C9ORF72 <- ac_bir_c9orf_vs_ctrl %>% dplyr::select(Intron.GeneName.GeneID.Coords, coords, ensID, gene_name, Chr, Start, End, Direction,p0.05_reliable, C9ORF72.p0.05_reliable_IRdirection = p0.05_reliable_IRdirection, gc_content, intron_length) %>% filter(p0.05_reliable == "significant")
differential.3datasets.IR <- VCP %>% full_join(SOD1, by = c("Intron.GeneName.GeneID.Coords", "coords", "ensID", "gene_name", "Chr", "Start", "End", "Direction", "gc_content", "intron_length")) %>%
  full_join(C9ORF72, by = c("Intron.GeneName.GeneID.Coords", "coords", "ensID", "gene_name", "Chr", "Start", "End", "Direction", "gc_content", "intron_length")) %>%
  mutate(p0.05_reliable_IRdirection = case_when(VCP.p0.05_reliable_IRdirection == "IR down" | SOD1.p0.05_reliable_IRdirection == "IR down" | C9ORF72.p0.05_reliable_IRdirection == "IR down" ~ "IR down", TRUE ~ "IR up"))

gc.differential.sina_plot <- sinaplot(data = differential.3datasets.IR, groups = "p0.05_reliable_IRdirection", continuous = "gc_content", dashed.line = 0.5) + ylim(c(0,1)) + ylab(label = expression(GC~content) ) + xlab("") + theme(axis.text.x = element_text(angle = 45, hjust = 1))
gc.differential.sina_plot

length.differential.sina_plot <- sinaplot(data = differential.3datasets.IR, groups = "p0.05_reliable_IRdirection", continuous = "intron_length", dashed.line = "NA", stat.y.position = 4.5, stats.test = "wilcox_test") + #ylim(c(0,1)) + 
  ylab(label = expression(log[10]~intron~length) ) + xlab("") + scale_y_log10() + theme(axis.text.y = element_text(size=6, angle = 45), axis.text.x = element_text(angle = 45, hjust = 1))
length.differential.sina_plot

gr_introns <- as.GenomicRange(differential.3datasets.IR$Intron.GeneName.GeneID.Coords)
seqlevelsStyle(gr_introns) <- "UCSC"
differential.3datasets.IR$phast_cons <- gscores(phast, gr_introns)$default # takes ~1min - too slow to add to function
PhastCons.differential.sina_plot <- sinaplot(data = differential.3datasets.IR, groups = "p0.05_reliable_IRdirection", continuous = "phast_cons", dashed.line = "NA", stat.y.position = 0.01, stats.test = "t_test") + ylab(label = expression(log[10]~PhastCons~score) ) + xlab("") + scale_y_log10() + theme(axis.text.y = element_text(size=6, angle = 45), axis.text.x = element_text(angle = 45, hjust = 1))
PhastCons.differential.sina_plot

# clip_hepg2 <- list.files("/camp/home/ziffo/home/genomes/clip-bedgraphs/HepG2/")
# clip_hepg2 <- clip_hepg2[!grepl("3nt_peaks",clip_hepg2)] # only assess raw crosslink files without any peak calling for enrichment
# grange_introns <- as.GenomicRange(differential.3datasets.IR$Intron.GeneName.GeneID.Coords) # create GRanges for each intron (both retained and non retained)
# intron_coords <- tibble(intron = 1:length(differential.3datasets.IR$coords), coords = differential.3datasets.IR$coords) # intron coordinates for to enable matching back to original coords in IRFinder dataframe
# coverage_list_who <- list()
# for (i in seq_along(clip_hepg2)) {
#   print(clip_hepg2[[i]])
#   rbp <- str_remove(clip_hepg2[[i]], ".merged.bed.gz")
#   bed <- import(paste("/camp/home/ziffo/home/genomes/clip-bedgraphs/HepG2/", clip_hepg2[[i]],sep=""))
#   seqlevelsStyle(bed) <- "NCBI" # remove "chr"
#   grange_overlap <- findOverlaps(query = grange_introns, subject=bed, ignore.strand=FALSE) # find overlaps between RBP binding sites & retained introns
#   range <- bed[ queryHits(grange_overlap) ] %>% as.data.frame() # subset iCLIP xlinks that overlap with the introns - 50,287 rows each with a xlinks score. There are 8739 introns - many introns have multiple scores.
#   coverage_list_who[[i]] <- range %>%  mutate( intron = queryHits(grange_overlap)) %>%  group_by(intron) %>% summarise(iclip_score = sum(score) ) # sum the scores for each intron - 4266 introns with scores, remaining introns have 0 scores
#   coverage_list_who[[i]]$iclip_score[is.na(coverage_list_who[[i]]$iclip_score)]<-0 # set NA scores to 0
#   coverage_list_who[[i]] <- coverage_list_who[[i]] %>% left_join(intron_coords, by = "intron") # add intron coordinates to match back to IRFinder results
#   names(coverage_list_who)[i] <- rbp
# }
# saveRDS(coverage_list_who, "/camp/home/ziffo/home/projects/astrocyte-meta-analysis/splicing/IRFinder/coverage_list_who.absolute_IR.differential.3datasets.IR.CLIPscores.RDS")
# coverage_list_who <- readRDS("/camp/home/ziffo/home/projects/astrocyte-meta-analysis/splicing/IRFinder/coverage_list_who.absolute_IR.differential.3datasets.IR.CLIPscores.RDS")
# coverage_list_who_df <- bind_rows(coverage_list_who, .id = "intron") # collapse list of dataframes into a single df
# iclip_scores <- coverage_list_who_df %>% distinct() %>% group_by(intron, coords) %>% # each column becomes an RBP. Each row is an intron - there are 310,450 in IRFinder but only 155,313 have xlink scores.
#   summarise(iclip_score.sum = sum(iclip_score)) %>% # https://stackoverflow.com/questions/63365516/tidyr-pivot-wider-error-cant-convert-double-to-list
#   tidyr::pivot_wider(names_from = intron, values_from = iclip_score.sum, values_fill = 0)
# # iclip_scores <- pivot_wider(coverage_list_who_df, names_from = intron, values_from = iclip_score, values_fill = 0)
# iclip_scores <- differential.3datasets.IR %>% dplyr::select(coords) %>% full_join(iclip_scores, by = "coords") # add back the introns without xlink scores
# iclip_scores[is.na(iclip_scores)] <- 0  # set NA scores to 0
# iclip_scores <- na.omit(iclip_scores)
# iclip_scores <- iclip_scores %>% left_join(select(differential.3datasets.IR, coords, intron_length), by = c("coords"))
# # iclip_scores$intron_length <- differential.3datasets.IR$intron_length[differential.3datasets.IR$coords %in% iclip_scores$coords] # add intron_length to df
# iclip_scores_normalised <- iclip_scores %>% mutate_if(is.numeric, ~ (./intron_length)) %>% dplyr::select(-intron_length) # normalise all scores for intron_length
# iclip_scores_normalised$sum <- iclip_scores_normalised %>% select_if(is.numeric) %>% rowSums # sum all normalised RBP scores for each intron
# saveRDS(iclip_scores_normalised, "/camp/home/ziffo/home/projects/astrocyte-meta-analysis/splicing/IRFinder/absolute_IR.differential.3datasets.IR.CLIPscores.RDS")
iclip_scores_normalised <- readRDS("/camp/home/ziffo/home/projects/astrocyte-meta-analysis/splicing/IRFinder/absolute_IR.differential.3datasets.IR.CLIPscores.RDS")
differential.3datasets.IR <- differential.3datasets.IR %>% left_join(dplyr::select(iclip_scores_normalised, sum, coords), by = "coords")
clip.differential.sina_plot <- sinaplot(data = differential.3datasets.IR, groups = "p0.05_reliable_IRdirection", continuous = "sum", dashed.line = "NA", stat.y.position = 2, stats.test = "t_test") + 
  ylab(label = expression(log[10]~RBP~iCLIP~xlinks) ) + xlab("") + scale_y_log10() + theme(axis.text.y = element_text(size=6, angle = 45), axis.text.x = element_text(angle = 45, hjust = 1))
clip.differential.sina_plot

### MaxEntScan https://github.com/monahton/GencoDymo https://rdrr.io/github/monahton/GencoDymo/f/vignettes/Vignette.Rmd https://bioconductor.org/packages/devel/bioc/manuals/VarCon/man/VarCon.pdf
# differential.3datasets.IR.coords <- differential.3datasets.IR %>% dplyr::select(coords)
# hg38 <- BSgenome.Hsapiens.UCSC.hg38::BSgenome.Hsapiens.UCSC.hg38
# # download.file(url="ftp://ftp.ebi.ac.uk/pub/databases/gencode/Gencode_human/release_36/gencode.v36.basic.annotation.gtf.gz",
# #               destfile = "/camp/home/ziffo/home/genomes/gencode/GRCh38.p13.v36.basic.annotation.gtf.gz", method = "wget")
# gen36 <- load_gtf("/camp/home/ziffo/home/genomes/gencode/GRCh38.p13.v36.basic.annotation.gtf.gz")
# gen36_introns <- extract_introns(gen36) # Extract introns from the gtf file
# gen36_introns_unique <- eliminate_redundant_introns(gen36_introns) # Remove redundant introns
# # filter gen36_introns gtf to IRFinder differentially retained introns
# differential.3datasets.IR.coords <- differential.3datasets.IR %>% dplyr::select("seqnames" = Chr, "intron_start" = Start, "intron_end" = End, "strand" = Direction) %>%
#   mutate(seqnames = paste("chr", seqnames, sep = ""),  # add chr to match gencode annotation style
#          intron_start = 1 + intron_start) # add 1 to start position of intron to avoid overlapping the exon
# gen36_IRFinder_introns <- differential.3datasets.IR.coords %>% left_join(gen36_introns_unique, by = c("seqnames", "intron_start", "intron_end", "strand"))
# setwd("/camp/home/ziffo/home/projects/astrocyte-meta-analysis/splicing/IRFinder/MaxEntScan") # set director to save splice site fasta files for MaxEntScan
# ss5_motif_df <- extract_5ss_motif(gen36_IRFinder_introns, BSgenome.Hsapiens.UCSC.hg38) # Produce a fasta file with 5'ss 9-mers:
# ss3_motif_df <- extract_3ss_motif(gen36_IRFinder_introns, BSgenome.Hsapiens.UCSC.hg38) # Produce a fasta file with 3'ss 23-mers:
# ss_motif <- ss5_motif_df %>% left_join(ss3_motif_df, by = c("seqnames", "strand", "transcript_id", "intron_number")) %>%
#   dplyr::rename("motif_start.5" = motif_start.x, "motif_start.3" = motif_start.y, "motif_end.5" = motif_end.x, "motif_end.3" = motif_end.y) %>%
#   mutate(MaxEntScore_5ss = calculateMaxEntScanScore(donor_ss_motif,5), MaxEntScore_3ss = calculateMaxEntScanScore(acceptor_ss_motif,3)) %>% # Calculate MaxEntScores at both splice sites
#   left_join(dplyr::select(gen36_introns_unique, seqnames, intron_start, intron_end, strand, transcript_id, intron_number), by = c("seqnames", "strand", "transcript_id", "intron_number")) %>% # intron start end
#   mutate(seqnames = gsub("chr", "", seqnames), intron_start = intron_start - 1) %>% rename("Chr" = seqnames, "Direction" = strand, "Start" = intron_start, "End" = intron_end)
# saveRDS(ss_motif, "/camp/lab/luscomben/home/users/ziffo/projects/astrocyte-meta-analysis/splicing/IRFinder/MaxEntScan/MaxEntScores.absolute_IR.differential.3datasets.RDS")
ss_motif <- readRDS("/camp/lab/luscomben/home/users/ziffo/projects/astrocyte-meta-analysis/splicing/IRFinder/MaxEntScan/MaxEntScores.absolute_IR.differential.3datasets.RDS")
differential.3datasets.IR <- differential.3datasets.IR %>% left_join(dplyr::select(ss_motif, Chr, Start, End, Direction, MaxEntScore_5ss, MaxEntScore_3ss), by = c("Chr", "Start", "End", "Direction")) %>%
  mutate(MaxEntScore_5ss = as.numeric(MaxEntScore_5ss), MaxEntScore_3ss = as.numeric(MaxEntScore_3ss)) # add MaxEntScan scores back to IRFinder results
maxent5.differential.sina_plot <- sinaplot(data = differential.3datasets.IR, groups = "p0.05_reliable_IRdirection", continuous = "MaxEntScore_5ss", dashed.line = "NA", stat.y.position = 13) + 
  ylim(c(0,13)) + # negative scores are likely spurious https://www.cell.com/molecular-cell/pdfExtended/S1097-2765(18)30832-3
  ylab(label = "5' MaxEntScore" ) + xlab("")  + theme(axis.text.x = element_text(angle = 45, hjust = 1))
maxent5.differential.sina_plot
maxent3.differential.sina_plot <- sinaplot(data = differential.3datasets.IR, groups = "p0.05_reliable_IRdirection", continuous = "MaxEntScore_3ss", dashed.line = "NA", stat.y.position = 15) + 
  ylim(c(0,15)) +
  ylab(label = "3' MaxEntScore" ) + xlab("") + theme(axis.text.x = element_text(angle = 45, hjust = 1))
maxent3.differential.sina_plot
maxent.differential.sina_plot <- ggarrange(maxent5.differential.sina_plot, maxent3.differential.sina_plot, nrow = 1)
maxent.differential.sina_plot

### notNMD https://www.bioconductor.org/packages/devel/bioc/vignettes/GeneStructureTools/inst/doc/Vignette.html https://github.com/betsig/notNMD  http://htmlpreview.github.io/?https://github.com/betsig/notNMD/blob/master/vignette.html
# Dr Beth Signal wrote this script. NB add 1 to the start of the IRFinder introns for this to work.
# Also GeneStructureTools functions clash with other packages so need a fresh R environment (dont source irfinder_functions.R)
# .libPaths("/camp/lab/luscomben/home/users/ziffo/.conda/envs/r4.0.3/lib/R/library")
# load("/camp/home/ziffo/home/projects/astrocyte-meta-analysis/scripts/figures_and_tables.RData")
# library(GenomicRanges)
# library(stringr)
# library(BSgenome.Hsapiens.UCSC.hg38)
# library(Gviz)
# library(rtracklayer)
# library(irlib)
# library(tidyverse)
# library(GeneStructureTools)
# VCP <- ac_who_vcp_vs_ctrl %>% dplyr::select(Intron.GeneName.GeneID.Coords, coords, ensID, gene_name, Chr, Start, End, Direction, gc_content, intron_length, p0.05_reliable, VCP.p0.05_reliable_IRdirection = p0.05_reliable_IRdirection) %>% filter(p0.05_reliable == "significant")
# SOD1 <- ac_tyz_sod1_vs_ctrl %>% dplyr::select(Intron.GeneName.GeneID.Coords, coords, ensID, gene_name, Chr, Start, End, Direction,p0.05_reliable, SOD1.p0.05_reliable_IRdirection = p0.05_reliable_IRdirection, gc_content, intron_length)  %>% filter(p0.05_reliable == "significant")
# C9ORF72 <- ac_bir_c9orf_vs_ctrl %>% dplyr::select(Intron.GeneName.GeneID.Coords, coords, ensID, gene_name, Chr, Start, End, Direction,p0.05_reliable, C9ORF72.p0.05_reliable_IRdirection = p0.05_reliable_IRdirection, gc_content, intron_length) %>% filter(p0.05_reliable == "significant")
# differential.3datasets.IR <- VCP %>% full_join(SOD1, by = c("Intron.GeneName.GeneID.Coords", "coords", "ensID", "gene_name", "Chr", "Start", "End", "Direction", "gc_content", "intron_length")) %>%
#   full_join(C9ORF72, by = c("Intron.GeneName.GeneID.Coords", "coords", "ensID", "gene_name", "Chr", "Start", "End", "Direction", "gc_content", "intron_length")) %>%
#   mutate(p0.05_reliable_IRdirection = case_when(VCP.p0.05_reliable_IRdirection == "IR down" | SOD1.p0.05_reliable_IRdirection == "IR down" | C9ORF72.p0.05_reliable_IRdirection == "IR down" ~ "IR down", TRUE ~ "IR up"))
# gr_IRFinder <- as.GenomicRange(differential.3datasets.IR$Intron.GeneName.GeneID.Coords) # GRanges of IRFinder 8,739 retained introns
# start(gr_IRFinder) = start(gr_IRFinder) +1
# seqlevelsStyle(gr_IRFinder) <- "UCSC"
# gr_IRFinder$id = paste0(seqnames(gr_IRFinder), ":", start(gr_IRFinder), "-", end(gr_IRFinder)) # make an id for each intron
# Homo_sapiens.GRCh38.99.exons <- readRDS("/camp/home/ziffo/home/genomes/ensembl/Homo_sapiens.GRCh38.99.exons.RDS")
# seqlevelsStyle(Homo_sapiens.GRCh38.99.exons) <- "UCSC"
# colnames(mcols(Homo_sapiens.GRCh38.99.exons)) = gsub("biotype","type", colnames(mcols(Homo_sapiens.GRCh38.99.exons))) # changes biotype to type in column names (required for GeneStrutureTools)
# exons.intronRetention <- findIntronContainingTranscripts(gr_IRFinder, Homo_sapiens.GRCh38.99.exons )
# IntronRetentionTranscripts <- addIntronInTranscript(exons.intronRetention, Homo_sapiens.GRCh38.99.exons)
# intron_ids = unique(IntronRetentionTranscripts$whippet_id)
## get ORF details for each set of transcripts
# g <- BSgenome.Hsapiens.UCSC.hg38::BSgenome.Hsapiens.UCSC.hg38
# # reference transcripts
# normalTranscripts <- Homo_sapiens.GRCh38.99.exons[Homo_sapiens.GRCh38.99.exons$transcript_id %in% unlist(lapply(stringr::str_split(IntronRetentionTranscripts$transcript_id, "[+]"), "[[", 1))]
# orfs_normal <- getOrfs(normalTranscripts, BSgenome = g, returnLongestOnly = FALSE, allFrames = TRUE,uORFs = TRUE)
# orfs_skipped <- getOrfs(IntronRetentionTranscripts[IntronRetentionTranscripts$set == "spliced_intron"],BSgenome = g,returnLongestOnly = FALSE, allFrames = TRUE,uORFs = TRUE)
# orfs_retained <- getOrfs(IntronRetentionTranscripts[IntronRetentionTranscripts$set == "retained_intron"],BSgenome = g,returnLongestOnly = FALSE, allFrames = TRUE,uORFs = TRUE)
# # differences comparing ORFs
# orfChange <- orfDiff(orfsX = orfs_skipped, 
#                      orfsY = orfs_retained, 
#                      filterNMD = FALSE,
#                      compareBy="gene",
#                      geneSimilarity = TRUE,
#                      compareUTR=TRUE,
#                      allORFs = orfs_normal)
# # add NMD calculations
# orfs_normal$nmd_prob <- notNMD::predictNMD(orfs_normal, "prob")
# orfs_normal$nmd_class <- notNMD::predictNMD(orfs_normal)
# orfs_skipped$nmd_prob <- notNMD::predictNMD(orfs_skipped, "prob")
# orfs_skipped$nmd_class <- notNMD::predictNMD(orfs_skipped)
# orfs_retained$nmd_prob <- notNMD::predictNMD(orfs_retained, "prob")
# orfs_retained$nmd_class <- notNMD::predictNMD(orfs_retained)
# orfs_normal <- orfs_normal[which(!is.na(orfs_normal$orf_length)),]
# orfs_skipped <- orfs_skipped[which(!is.na(orfs_skipped$orf_length)),]
# orfs_retained <- orfs_retained[which(!is.na(orfs_retained$orf_length)),]
# # compare normal and retained isoforms
# # this time setting filterNMD to TRUE, which removes NMD targeted frames/isoforms where possible
# orfChange_filter <- orfDiff(orfsX = orfs_skipped, # normal transcripts
#                      orfsY = orfs_retained, # alternative transcripts
#                      filterNMD = TRUE, 
#                      geneSimilarity = TRUE,
#                      compareUTR=TRUE,
#                      allORFs = orfs_normal)
# # test changes in NMD between retained and skipped introns.
# nmdChange <- attrChangeAltSpliced(orfs_skipped, orfs_retained,
#                                   attribute="nmd_prob",
#                                   compareBy="gene",
#                                   useMax=FALSE) # want useMax=F, as we want the minimum NMD probablity (if NMD prob is low, means that at least 1 isoform not targeted)
# m <- match(orfChange_filter$id, nmdChange$id)
# orfChange_filter <- cbind(orfChange_filter, nmdChange[m,-1])
# write.table(orfChange_filter, file="/camp/home/ziffo/home/projects/astrocyte-meta-analysis/splicing/IRFinder/notNMD/differential.3datasets.IR_orf_nmd_changes.txt", sep='\t', row.names = F)
# save.image("/camp/home/ziffo/home/projects/astrocyte-meta-analysis/splicing/IRFinder/notNMD/differential.3datasets.IR_notNMD_results.RData")
# load(file = "/camp/home/ziffo/home/projects/astrocyte-meta-analysis/splicing/IRFinder/notNMD/differential.3datasets.IR_notNMD_results.RData")
# notNMDprob <- orfChange_filter %>% select("coords" = id, "nmd_prob_retained" = nmd_prob_bygroup_y, "nmd_prob_spliced" = nmd_prob_bygroup_x) # 
# notNMDprob$coords <- gsub("chr", "", notNMDprob$coords)
# notNMDprob$Chr <- str_split_fixed(notNMDprob$coords, ":", 2)[,1]
# notNMDprob$End <- as.numeric(str_split_fixed(notNMDprob$coords, "-", 2)[,2])
# notNMDprob$Start <- str_split_fixed(notNMDprob$coords, "-", 2)[,1]
# notNMDprob$Start <- as.numeric(str_split_fixed(notNMDprob$Start, ":", 2)[,2])
# notNMDprob <- notNMDprob %>% mutate("Start" = Start - 1) %>% select(-coords) # remove 1 from Start to revert back to original IRFinder coords
# saveRDS(notNMDprob, "/camp/home/ziffo/home/projects/astrocyte-meta-analysis/splicing/IRFinder/notNMD/notNMDprob.differential.3datasets.IR.RDS")
notNMDprob <- readRDS("/camp/home/ziffo/home/projects/astrocyte-meta-analysis/splicing/IRFinder/notNMD/notNMDprob.differential.3datasets.IR.RDS")
differential.3datasets.IR <-  differential.3datasets.IR %>% left_join(notNMDprob, by = c("Chr", "Start", "End")) # left join results to IRFinder
differential.3datasets.IR <-  differential.3datasets.IR %>% mutate(nmd_prob = nmd_prob_retained, #case_when(retained == "retained" ~ nmd_prob_retained, TRUE ~ nmd_prob_spliced),
                                 nmd_outcome = case_when(nmd_prob > 0.5 ~ "NMD", TRUE ~ "not NMD"))
# Differential Sina plot
notNMD.differential.sina_plot <- sinaplot(data = differential.3datasets.IR, groups = "p0.05_reliable_IRdirection", continuous = "nmd_prob", stat.y.position = 1, stats.test = "wilcox_test", width = 0.2, dashed.line = 0.5) + ylab(label = expression(NMD~probability) ) + xlab("") + theme(axis.text.x = element_text(angle = 45, hjust = 1))
notNMD.differential.sina_plot
```

## Merge

```{r}
fig1.merged <- ggarrange(ggarrange(vcp_sod1_c9orf_sina_plot, venn_vcp_sod1_c9orf_overlap.down, nrow = 1, ncol = 2, widths = c(1.3,1), labels = c("C", "D")),
  ggarrange(gc.differential.sina_plot, length.differential.sina_plot, PhastCons.differential.sina_plot, clip.differential.sina_plot, maxent5.differential.sina_plot, maxent3.differential.sina_plot, notNMD.differential.sina_plot, nrow = 1, ncol = 7), # notNMD.differential.sina_plot
  nrow = 2, heights = c(1,1), labels = c("", "E"))
fig1.merged
ggsave(fig1.merged, filename = "/camp/home/ziffo/home/projects/astrocyte-meta-analysis/splicing/IRFinder/irlib/figures/fig1.merged.png", height = 7, width = 12)
```

## Coverage plots IR down

Top IR events irrespective of gene expression

```{r}
# Coverage plots for MCAM, RECK, UBN1, LOXL3, TGFB1I1, COL7A1
txdb <- makeTxDbFromGFF("/camp/home/ziffo/home/genomes/ensembl/Homo_sapiens.GRCh38.99.chr_patch_hapl_scaff.gtf", format="gtf") # make txdb from GTF - turn on only when needed
seqlevels(txdb, pruning.mode="coarse") <- c("1", "2", "3", "4", "5", "6", "7", "8", "9", "10", "11", "12", "13", "14", "15", "16", "17", "18", "19", "20", "21", "22", "X", "Y") # removes all the other chromsome annotations
ac_who_ctrl.bam <- "/camp/home/ziffo/home/projects/astrocyte-meta-analysis/alignment/STAR/merged/ac_who_ctrl.bam"
ac_who_vcp.bam <- "/camp/home/ziffo/home/projects/astrocyte-meta-analysis/alignment/STAR/merged/ac_who_vcp.bam"

# Top 4 reliable delta IR events
ac_who_vcp_vs_ctrl_top10events <- ac_who_vcp_vs_ctrl %>% filter(reliable == "reliable")  %>% dplyr::select(gene_name, ensemblID = ensID, Chr, Start, End, Strand = Direction, length = intron_length, IRratio.diff, coverage = IR.coverage, p.value = p.diff, IR.DGE.direction = direction) %>% arrange(IRratio.diff) %>% top_n(wt = -IRratio.diff, n = 10)
ac_who_vcp_vs_ctrl_top10events

track_viewer_tx_top_IRratio_event(gene_test = "RND3", txt = ac_who_vcp_vs_ctrl, mut.bam = ac_who_vcp.bam, ctrl.bam = ac_who_ctrl.bam)
track_viewer_tx_top_IRratio_event(gene_test = "SLC26A6", txt = ac_who_vcp_vs_ctrl, mut.bam = ac_who_vcp.bam, ctrl.bam = ac_who_ctrl.bam)
track_viewer_tx_top_IRratio_event(gene_test = "PIMREG", txt = ac_who_vcp_vs_ctrl, mut.bam = ac_who_vcp.bam, ctrl.bam = ac_who_ctrl.bam)
track_viewer_tx_top_IRratio_event(gene_test = "HSF4", txt = ac_who_vcp_vs_ctrl, mut.bam = ac_who_vcp.bam, ctrl.bam = ac_who_ctrl.bam)
```



# Figure S3: MA plots & reliability histograms

```{r}
# VCP reliable events
ac_who_vcp_ctrl.reliability_histogram <- histogram_plot_irfinder(ctrl = ac_who_ctrl.IR, mutant = ac_who_vcp.IR, ctrl.name = "CTRL", mutant.name = "VCP")
ac_who_vcp_ctrl.reliability_histogram

# C9ORF72 reliable events
ac_bir_c9orf_ctrl.reliability_histogram <- histogram_plot_irfinder(ctrl = ac_bir_ctrl.IR, mutant = ac_bir_c9orf.IR, ctrl.name = "CTRL", mutant.name = "C9ORF72", legend = FALSE)
ac_bir_c9orf_ctrl.reliability_histogram

# SOD1 reliable events
SOD1.reliability_histogram <- histogram_plot_irfinder(ctrl = ac_tyz_ctrl.IR, mutant = ac_tyz_sod1.IR, ctrl.name = "CTRL", mutant.name = "SOD1", legend = FALSE)
SOD1.reliability_histogram

# Cytokine stimulated reliable events
ac_bar_a1_a0.reliability_histogram <- histogram_plot_irfinder(ctrl = ac_bar_a0.IR, mutant = ac_bar_a1.IR, ctrl.name = "Unstimulated CTRL", mutant.name = "Cytokine stimulated", legend = FALSE)
ac_bar_a1_a0.reliability_histogram

# TDP43 deletion reliable events
ac_tdp_ko_ctrl.reliability_histogram <- histogram_plot_irfinder(ctrl = ac_tdp_ctrl.IR, mutant = ac_tdp_ko.IR, ctrl.name = "CTRL", mutant.name = "TDP43 deleted", legend = FALSE)
ac_tdp_ko_ctrl.reliability_histogram

# MA plots of IR events in VCP
ac_who_vcp_vs_ctrl.labels <- c("VIM", "COL1A1", "MYL6", "FN1", "FLNA", "PDLIM7", "TNC", "ILK", " MCAM", "HLA-B", "TGFB1L1", "OSMR", "RND3", "PIMREG", "LOXL3", "RECK", "COL7A1", "UBN1", "TTBK1", "TSC1", "ADAM19")
ac_who_ma <- ma_plot_irfinder(data = ac_who_vcp_vs_ctrl, labels_list = ac_who_vcp_vs_ctrl.labels) + ylab(expression(Delta~IR~ratio~VCP)) + xlim(1,5.2) + ylim(-0.36,0.36) +
  geom_segment(aes(x = 5.0, xend = 5.0, y= 0.1, yend= 0.3), arrow=arrow(length=unit(0.4,"cm")), color = "dodgerblue") +
  geom_segment(aes(x = 5.0, xend = 5.0, y= -0.1, yend= -0.3),arrow=arrow(length=unit(0.4,"cm")), color = "firebrick2") +
  annotate("text", x = 4.6, y = 0.3, label = "IR Up", colour = "dodgerblue", size = 3) +   annotate("text", x = 4.6, y = -0.3, label = "IR Down", colour = "firebrick2", size = 3)
ac_who_ma

# MA plots of IR events in C9orf72
C9ORF72.ma <- ma_plot_irfinder(data = ac_bir_c9orf_vs_ctrl, labels_list = astrocyte.reactivity.labels) + ylab(expression(Delta~IR~ratio~C9orf72)) + xlim(0.6,4.5) + ylim(-0.4,0.4) +
  geom_segment(aes(x = 4.3, xend = 4.3, y= 0.1, yend= 0.3), arrow=arrow(length=unit(0.4,"cm")), color = "dodgerblue") +
  geom_segment(aes(x = 4.3, xend = 4.3, y= -0.1, yend= -0.3),arrow=arrow(length=unit(0.4,"cm")), color = "firebrick2") +
  annotate("text", x = 3.7, y = 0.3, label = "IR Up", colour = "dodgerblue", size = 3) +   annotate("text", x = 3.7, y = -0.3, label = "IR Down", colour = "firebrick2", size = 3)
C9ORF72.ma

# MA plots of IR events in SOD1
SOD1.ma <- ma_plot_irfinder(data = ac_tyz_sod1_vs_ctrl, labels_list = astrocyte.reactivity.labels) + ylab(expression(Delta~IR~ratio~SOD1)) + xlim(0.6,4) + ylim(-0.5,0.5) +
  geom_segment(aes(x = 3.8, xend = 3.8, y= 0.1, yend= 0.3), arrow=arrow(length=unit(0.4,"cm")), color = "dodgerblue") +
  geom_segment(aes(x = 3.8, xend = 3.8, y= -0.1, yend= -0.3),arrow=arrow(length=unit(0.4,"cm")), color = "firebrick2") +
  annotate("text", x = 3.6, y = 0.4, label = "IR Up", colour = "dodgerblue", size = 3) +   annotate("text", x = 3.6, y = -0.4, label = "IR Down", colour = "firebrick2", size = 3)
SOD1.ma

# MA plot of IR events in Stimulated astrocytes
ac_bar_a1_ma <- ma_plot_irfinder(data = ac_bar_a1_vs_a0, labels_list = astrocyte.reactivity.labels.ac_a1) + ylab(expression(Delta~IR~ratio~Stimulated)) + xlim(1,4.5) + ylim(-0.3,0.3)  +
  geom_segment(aes(x = 4.3, xend = 4.3, y= 0.1, yend= 0.3), arrow=arrow(length=unit(0.4,"cm")), color = "dodgerblue") +
  geom_segment(aes(x = 4.3, xend = 4.3, y= -0.1, yend= -0.3),arrow=arrow(length=unit(0.4,"cm")), color = "firebrick2") +
  annotate("text", x = 3.9, y = 0.25, label = "IR Up", colour = "dodgerblue", size = 3) +   annotate("text", x = 3.9, y = -0.25, label = "IR Down", colour = "firebrick2", size = 3)
ac_bar_a1_ma

# MA plot of IR events in TDP43 knockout
ac_tdp_ko_ma <- ma_plot_irfinder(data = ac_tdp_ko_vs_ctrl, labels_list = astrocyte.reactivity.labels.ac_tdp43) + ylab(expression(Delta~IR~ratio~TDP43~KO)) + xlim(1,4) + ylim(-0.3,0.3)  +
  geom_segment(aes(x = 4.0, xend = 4.0, y= 0.1, yend= 0.3), arrow=arrow(length=unit(0.4,"cm")), color = "dodgerblue") +
  geom_segment(aes(x = 4.0, xend = 4.0, y= -0.1, yend= -0.3),arrow=arrow(length=unit(0.4,"cm")), color = "firebrick2") +
  annotate("text", x = 3.6, y = 0.25, label = "IR Up", colour = "dodgerblue", size = 3) +   annotate("text", x = 3.6, y = -0.25, label = "IR Down", colour = "firebrick2", size = 3)
ac_tdp_ko_ma
```

## Merge

```{r}
figS3.merged <- ggarrange(
  ggarrange(ac_who_ma, ac_who_vcp_ctrl.reliability_histogram, ncol = 2, widths = c(1,1.4), labels = c("A", "F")),
  ggarrange(C9ORF72.ma, ac_bir_c9orf_ctrl.reliability_histogram, ncol = 2, widths = c(1,1.4), labels = c("B", "G")),
  ggarrange(SOD1.ma, SOD1.reliability_histogram, ncol = 2, widths = c(1,1.4), labels = c("C", "H")),
  ggarrange(ac_bar_a1_ma, ac_bar_a1_a0.reliability_histogram, ncol = 2, widths = c(1,1.4), labels = c("D", "I")),
  ggarrange(ac_tdp_ko_ma, ac_tdp_ko_ctrl.reliability_histogram, ncol = 2, widths = c(1,1.4), labels = c("E", "J")),
  nrow = 5, heights = c(1.1,1,1,1,1))
figS3.merged
ggsave(figS3.merged, filename = "/camp/home/ziffo/home/projects/astrocyte-meta-analysis/splicing/IRFinder/irlib/figures/figS3.merged.png", height = 11.75, width = 11)
```


# Figure S4: IR characteristics 

## Retained vs spliced

```{r}
#### Absolute IR characteristics in ALS ####
VCP.ctrl.filt <- ac_who_ctrl.IR %>% dplyr::select(Name.Coords, coords, ensID, gene_name, Chr, Start, End, Strand, gc_content, intron_length, IRratio, reliable, VCP.ctrl.retained = retained) %>% filter(reliable == "reliable")
VCP.mutant.filt <- ac_who_vcp.IR %>% dplyr::select(Name.Coords, coords, ensID, gene_name, Chr, Start, End, Strand, IRratio, reliable, VCP.mutant.retained = retained, gc_content, intron_length) %>% filter(reliable == "reliable") 
SOD1.ctrl.filt <- ac_tyz_ctrl.IR %>% dplyr::select(Name.Coords, coords, ensID, gene_name, Chr, Start, End, Strand,IRratio, reliable, SOD1.ctrl.retained = retained, gc_content, intron_length)  %>% filter(reliable == "reliable")
SOD1.mutant.filt <- ac_tyz_sod1.IR %>% dplyr::select(Name.Coords, coords, ensID, gene_name, Chr, Start, End, Strand, IRratio, reliable, SOD1.mutant.retained = retained, gc_content, intron_length) %>% filter(reliable == "reliable") 
C9ORF72.ctrl.filt <- ac_bir_ctrl.IR %>% dplyr::select(Name.Coords, coords, ensID, gene_name, Chr, Start, End, Strand,IRratio, reliable, C9ORF72.ctrl.retained = retained, gc_content, intron_length) %>% filter(reliable == "reliable")
C9ORF72.mutant.filt <- ac_bir_c9orf.IR %>% dplyr::select(Name.Coords, coords, ensID, gene_name, Chr, Start, End, Strand, IRratio, reliable, C9ORF72.mutant.retained = retained, gc_content, intron_length) %>% filter(reliable == "reliable") 
ctrl.mutant.3datasets.IR <- VCP.ctrl.filt %>% 
  full_join(VCP.mutant.filt, by = c("Name.Coords", "coords", "ensID", "gene_name", "Chr", "Start", "End", "Strand", "gc_content", "intron_length")) %>%
  full_join(SOD1.ctrl.filt, by = c("Name.Coords", "coords", "ensID", "gene_name", "Chr", "Start", "End", "Strand", "gc_content", "intron_length")) %>%
  full_join(SOD1.mutant.filt, by = c("Name.Coords", "coords", "ensID", "gene_name", "Chr", "Start", "End", "Strand", "gc_content", "intron_length")) %>%
  full_join(C9ORF72.ctrl.filt, by = c("Name.Coords", "coords", "ensID", "gene_name", "Chr", "Start", "End", "Strand", "gc_content", "intron_length")) %>%
  full_join(C9ORF72.mutant.filt, by = c("Name.Coords", "coords", "ensID", "gene_name", "Chr", "Start", "End", "Strand", "gc_content", "intron_length")) %>%
  dplyr::mutate(any_retained = case_when(VCP.ctrl.retained == "retained" | VCP.mutant.retained == "retained" | SOD1.ctrl.retained == "retained" | SOD1.mutant.retained == "retained" | C9ORF72.ctrl.retained == "retained" | C9ORF72.mutant.retained == "retained" ~ "retained", TRUE ~ "spliced"))

## GC content
gc.retained.sina_plot <- sinaplot(data = ctrl.mutant.3datasets.IR, groups = "any_retained", continuous = "gc_content", dashed.line = 0.5, stats.test = "wilcox_test") + ylim(c(0,1)) + ylab(label = expression(GC~content) ) + xlab("")  + theme(axis.text.x = element_text(angle = 45, hjust = 1))
gc.retained.sina_plot

## Intron length
length.retained.sina_plot <- sinaplot(data = ctrl.mutant.3datasets.IR, groups = "any_retained", continuous = "intron_length", dashed.line = "NA", stats.test = "wilcox_test", stat.y.position = 5.5) + ylim(c(0,5)) +  ylab(label = expression(log[10]~intron~length) ) + xlab("") +  scale_y_log10() + theme(axis.text.x = element_text(angle = 45, hjust = 1))
length.retained.sina_plot

## PhastCons 
gr_introns <- as.GenomicRange(ctrl.mutant.3datasets.IR$Name.Coords)
seqlevelsStyle(gr_introns) <- "UCSC"
ctrl.mutant.3datasets.IR$phast_cons <- gscores(phast, gr_introns)$default # takes ~1min - too slow to add to function
PhastCons.retained.sina_plot <- sinaplot(data = ctrl.mutant.3datasets.IR, groups = "any_retained", continuous = "phast_cons", dashed.line = "NA", stats.test = "t_test", stat.y.position = 0.01) + 
  ylab(label = expression(log[10]~PhastCons~score) ) + xlab("") + scale_y_log10()   + theme(axis.text.x = element_text(angle = 45, hjust = 1))
PhastCons.retained.sina_plot

## iCLIP
# sum the iclip xlink scores for each RBP in HepG2 in a for loop
# clip_hepg2 <- list.files("/camp/home/ziffo/home/genomes/clip-bedgraphs/HepG2/")
# clip_hepg2 <- clip_hepg2[!grepl("3nt_peaks",clip_hepg2)] # only assess raw crosslink files without any peak calling for enrichment
# grange_introns <- as.GenomicRange(ctrl.mutant.3datasets.IR$Name.Coords) # create GRanges for each intron (both retained and non retained)
# intron_coords <- tibble(intron = 1:length(ctrl.mutant.3datasets.IR$coords), coords = ctrl.mutant.3datasets.IR$coords) # intron coordinates for to enable matching back to original coords in IRFinder dataframe
# coverage_list_who <- list()
# for (i in seq_along(clip_hepg2)) {
#   print(clip_hepg2[[i]])
#   rbp <- str_remove(clip_hepg2[[i]], ".merged.bed.gz")
#   bed <- import(paste("/camp/home/ziffo/home/genomes/clip-bedgraphs/HepG2/", clip_hepg2[[i]],sep=""))
#   seqlevelsStyle(bed) <- "NCBI" # remove "chr"
#   grange_overlap <- findOverlaps(query = grange_introns, subject=bed, ignore.strand=FALSE) # find overlaps between RBP binding sites & retained introns
#   range <- bed[ queryHits(grange_overlap) ] %>% as.data.frame() # subset iCLIP xlinks that overlap with the introns - 50,287 rows each with a xlinks score. There are 8739 introns - many introns have multiple scores.
#   coverage_list_who[[i]] <- range %>%  mutate( intron = queryHits(grange_overlap)) %>%  group_by(intron) %>% summarise(iclip_score = sum(score) ) # sum the scores for each intron - 4266 introns with scores, remaining introns have 0 scores
#   coverage_list_who[[i]]$iclip_score[is.na(coverage_list_who[[i]]$iclip_score)]<-0 # set NA scores to 0
#   coverage_list_who[[i]] <- coverage_list_who[[i]] %>% left_join(intron_coords, by = "intron") # add intron coordinates to match back to IRFinder results
#   names(coverage_list_who)[i] <- rbp
# }
# saveRDS(coverage_list_who, "/camp/home/ziffo/home/projects/astrocyte-meta-analysis/splicing/IRFinder/coverage_list_who.absolute_IR.ctrl.mutant.3datasets.IR.CLIPscores.RDS")
# 
# coverage_list_who_df <- bind_rows(coverage_list_who, .id = "intron") # collapse list of dataframes into a single df
# iclip_scores <- coverage_list_who_df %>% distinct() %>% group_by(intron, coords) %>% # each column becomes an RBP. Each row is an intron - there are 310,450 in IRFinder but only 155,313 have xlink scores.
#   summarise(iclip_score.sum = sum(iclip_score)) %>% # https://stackoverflow.com/questions/63365516/tidyr-pivot-wider-error-cant-convert-double-to-list
#   tidyr::pivot_wider(names_from = intron, values_from = iclip_score.sum, values_fill = 0)
# # iclip_scores <- pivot_wider(coverage_list_who_df, names_from = intron, values_from = iclip_score, values_fill = 0)
# iclip_scores <- ctrl.mutant.3datasets.IR %>% dplyr::select(coords) %>% full_join(iclip_scores, by = "coords") # add back the introns without xlink scores
# iclip_scores[is.na(iclip_scores)] <- 0  # set NA scores to 0
# iclip_scores$intron_length <- ctrl.mutant.3datasets.IR$intron_length[iclip_scores$coords %in% ctrl.mutant.3datasets.IR$coords] # add intron_length to df
# iclip_scores_normalised <- iclip_scores %>% mutate_if(is.numeric, ~ (./intron_length)) %>% dplyr::select(-intron_length) # normalise all scores for intron_length
# iclip_scores_normalised$sum <- iclip_scores_normalised %>% select_if(is.numeric) %>% rowSums # sum all normalised RBP scores for each intron
# saveRDS(iclip_scores_normalised, "/camp/home/ziffo/home/projects/astrocyte-meta-analysis/splicing/IRFinder/absolute_IR.ctrl.mutant.3datasets.IR.CLIPscores.RDS")
absolute_IR.ctrl.mutant.3datasets.iclip_scores_normalised <- readRDS("/camp/home/ziffo/home/projects/astrocyte-meta-analysis/splicing/IRFinder/absolute_IR.ctrl.mutant.3datasets.IR.CLIPscores.RDS")
ctrl.mutant.3datasets.IR$iclip_enrichment <- absolute_IR.ctrl.mutant.3datasets.iclip_scores_normalised$sum[absolute_IR.ctrl.mutant.3datasets.iclip_scores_normalised$coords %in% ctrl.mutant.3datasets.IR$coords] 
clip.retained.sina_plot <- sinaplot(data = ctrl.mutant.3datasets.IR, groups = "any_retained", continuous = "iclip_enrichment", dashed.line = "NA", stats.test = "t_test", stat.y.position = 2.5) + #ylim(c(0,1)) + 
  ylab(label = expression(log[10]~iCLIP~xlinks~score) ) + xlab("") +  scale_y_log10()  + theme(axis.text.x = element_text(angle = 45, hjust = 1))
clip.retained.sina_plot

### MaxEntScan https://github.com/monahton/GencoDymo https://rdrr.io/github/monahton/GencoDymo/f/vignettes/Vignette.Rmd https://bioconductor.org/packages/devel/bioc/manuals/VarCon/man/VarCon.pdf
# ctrl.mutant.3datasets.IR.coords <- ctrl.mutant.3datasets.IR %>% dplyr::select(coords)
# hg38 <- BSgenome.Hsapiens.UCSC.hg38::BSgenome.Hsapiens.UCSC.hg38
# # download.file(url="ftp://ftp.ebi.ac.uk/pub/databases/gencode/Gencode_human/release_36/gencode.v36.basic.annotation.gtf.gz",
# #               destfile = "/camp/home/ziffo/home/genomes/gencode/GRCh38.p13.v36.basic.annotation.gtf.gz", method = "wget")
# gen36 <- load_gtf("/camp/home/ziffo/home/genomes/gencode/GRCh38.p13.v36.basic.annotation.gtf.gz")
# gen36_introns <- extract_introns(gen36) # Extract introns from the gtf file
# gen36_introns_unique <- eliminate_redundant_introns(gen36_introns) # Remove redundant introns
# # filter gen36_introns gtf to IRFinder differentially retained introns
# ctrl.mutant.3datasets.IR.coords <- ctrl.mutant.3datasets.IR %>% dplyr::select("seqnames" = Chr, "intron_start" = Start, "intron_end" = End, "strand" = Strand) %>%
#   mutate(seqnames = paste("chr", seqnames, sep = ""),  # add chr to match gencode annotation style
#          intron_start = 1 + intron_start) # add 1 to start position of intron to avoid overlapping the exon
# gen36_IRFinder_introns <- ctrl.mutant.3datasets.IR.coords %>% left_join(gen36_introns_unique, by = c("seqnames", "intron_start", "intron_end", "strand"))
# setwd("/camp/home/ziffo/home/projects/astrocyte-meta-analysis/splicing/IRFinder/MaxEntScan") # set director to save splice site fasta files for MaxEntScan
# ss5_motif_df <- extract_5ss_motif(gen36_IRFinder_introns, BSgenome.Hsapiens.UCSC.hg38) # Produce a fasta file with 5'ss 9-mers:
# ss3_motif_df <- extract_3ss_motif(gen36_IRFinder_introns, BSgenome.Hsapiens.UCSC.hg38) # Produce a fasta file with 3'ss 23-mers:
# ss_motif <- ss5_motif_df %>% left_join(ss3_motif_df, by = c("seqnames", "strand", "transcript_id", "intron_number")) %>%
#   dplyr::rename("motif_start.5" = motif_start.x, "motif_start.3" = motif_start.y, "motif_end.5" = motif_end.x, "motif_end.3" = motif_end.y) %>%
#   mutate(MaxEntScore_5ss = calculateMaxEntScanScore(donor_ss_motif,5), MaxEntScore_3ss = calculateMaxEntScanScore(acceptor_ss_motif,3)) %>% # Calculate MaxEntScores at both splice sites
#   left_join(dplyr::select(gen36_introns_unique, seqnames, intron_start, intron_end, strand, transcript_id, intron_number), by = c("seqnames", "strand", "transcript_id", "intron_number")) %>% # intron start end
#   mutate(seqnames = gsub("chr", "", seqnames), intron_start = intron_start - 1) %>% rename("Chr" = seqnames, "Strand" = strand, "Start" = intron_start, "End" = intron_end)
# saveRDS(ss_motif, "/camp/lab/luscomben/home/users/ziffo/projects/astrocyte-meta-analysis/splicing/IRFinder/MaxEntScan/MaxEntScores.absolute_ctrl.mutant.3datasets.IR.RDS")
```

Run above in separate script with:

```{bash}
cd /camp/lab/luscomben/home/users/ziffo/projects/astrocyte-meta-analysis/splicing/IRFinder/MaxEntScan

sbatch -N 1 -c 10 --mem=0 -t 48:00:00 --wrap="Rscript /camp/lab/luscomben/home/users/ziffo/projects/astrocyte-meta-analysis/scripts/MaxEntScan_absolute_retained_vs_spliced.R" --mail-type=ALL,ARRAY_TASKS --mail-user=oliver.ziff@crick.ac.uk --job-name=MaxEntScan
```

```{r}
ss_motif <- readRDS("/camp/lab/luscomben/home/users/ziffo/projects/astrocyte-meta-analysis/splicing/IRFinder/MaxEntScan/MaxEntScores.absolute_ctrl.mutant.3datasets.IR.RDS")
ctrl.mutant.3datasets.IR <- ctrl.mutant.3datasets.IR %>% left_join(dplyr::select(ss_motif, Chr, Start, End, Strand, MaxEntScore_5ss, MaxEntScore_3ss), by = c("Chr", "Start", "End", "Strand")) %>%
  mutate(MaxEntScore_5ss = as.numeric(MaxEntScore_5ss), MaxEntScore_3ss = as.numeric(MaxEntScore_3ss)) # add MaxEntScan scores back to IRFinder results
maxent5.retained.sina_plot <- sinaplot(data = ctrl.mutant.3datasets.IR, groups = "any_retained", continuous = "MaxEntScore_5ss", dashed.line = "NA", stat.y.position = 13) + 
  ylim(c(0,13)) + # negative scores are likely spurious https://www.cell.com/molecular-cell/pdfExtended/S1097-2765(18)30832-3
  ylab(label = "5' MaxEntScore" ) + xlab("")  + theme(axis.text.x = element_text(angle = 45, hjust = 1))
maxent5.retained.sina_plot
maxent3.retained.sina_plot <- sinaplot(data = ctrl.mutant.3datasets.IR, groups = "any_retained", continuous = "MaxEntScore_3ss", dashed.line = "NA", stat.y.position = 15) + 
  ylim(c(0,15)) +
  ylab(label = "3' MaxEntScore" ) + xlab("") + theme(axis.text.x = element_text(angle = 45, hjust = 1))
maxent3.retained.sina_plot
maxent.retained.sina_plot <- ggarrange(maxent5.retained.sina_plot, maxent3.retained.sina_plot, nrow = 1)
maxent.retained.sina_plot

### notNMD https://www.bioconductor.org/packages/devel/bioc/vignettes/GeneStructureTools/inst/doc/Vignette.html https://github.com/betsig/notNMD  http://htmlpreview.github.io/?https://github.com/betsig/notNMD/blob/master/vignette.html
# Dr Beth Signal wrote this script. NB add 1 to the start of the IRFinder introns for this to work.
# Also GeneStructureTools functions clash with other packages so need a fresh R environment (dont source irfinder_functions.R)
# .libPaths("/camp/lab/luscomben/home/users/ziffo/.conda/envs/r4.0.3/lib/R/library")
# load("/camp/home/ziffo/home/projects/astrocyte-meta-analysis/scripts/figures_and_tables.RData")
# library(GenomicRanges)
# library(stringr)
# library(BSgenome.Hsapiens.UCSC.hg38)
# library(Gviz)
# library(rtracklayer)
# library(irlib)
# library(tidyverse)
# library(GeneStructureTools)
# VCP.ctrl.filt <- ac_who_ctrl.IR %>% dplyr::select(Name.Coords, coords, ensID, gene_name, Chr, Start, End, Strand, gc_content, intron_length, IRratio, reliable, VCP.ctrl.retained = retained) %>% filter(reliable == "reliable")
# VCP.mutant.filt <- ac_who_vcp.IR %>% dplyr::select(Name.Coords, coords, ensID, gene_name, Chr, Start, End, Strand, IRratio, reliable, VCP.mutant.retained = retained, gc_content, intron_length) %>% filter(reliable == "reliable") 
# SOD1.ctrl.filt <- ac_tyz_ctrl.IR %>% dplyr::select(Name.Coords, coords, ensID, gene_name, Chr, Start, End, Strand,IRratio, reliable, SOD1.ctrl.retained = retained, gc_content, intron_length)  %>% filter(reliable == "reliable")
# SOD1.mutant.filt <- ac_tyz_sod1.IR %>% dplyr::select(Name.Coords, coords, ensID, gene_name, Chr, Start, End, Strand, IRratio, reliable, SOD1.mutant.retained = retained, gc_content, intron_length) %>% filter(reliable == "reliable") 
# C9ORF72.ctrl.filt <- ac_bir_ctrl.IR %>% dplyr::select(Name.Coords, coords, ensID, gene_name, Chr, Start, End, Strand,IRratio, reliable, C9ORF72.ctrl.retained = retained, gc_content, intron_length) %>% filter(reliable == "reliable")
# C9ORF72.mutant.filt <- ac_bir_c9orf.IR %>% dplyr::select(Name.Coords, coords, ensID, gene_name, Chr, Start, End, Strand, IRratio, reliable, C9ORF72.mutant.retained = retained, gc_content, intron_length) %>% filter(reliable == "reliable") 
# ctrl.mutant.3datasets.IR <- VCP.ctrl.filt %>% 
#   full_join(VCP.mutant.filt, by = c("Name.Coords", "coords", "ensID", "gene_name", "Chr", "Start", "End", "Strand", "gc_content", "intron_length")) %>%
#   full_join(SOD1.ctrl.filt, by = c("Name.Coords", "coords", "ensID", "gene_name", "Chr", "Start", "End", "Strand", "gc_content", "intron_length")) %>%
#   full_join(SOD1.mutant.filt, by = c("Name.Coords", "coords", "ensID", "gene_name", "Chr", "Start", "End", "Strand", "gc_content", "intron_length")) %>%
#   full_join(C9ORF72.ctrl.filt, by = c("Name.Coords", "coords", "ensID", "gene_name", "Chr", "Start", "End", "Strand", "gc_content", "intron_length")) %>%
#   full_join(C9ORF72.mutant.filt, by = c("Name.Coords", "coords", "ensID", "gene_name", "Chr", "Start", "End", "Strand", "gc_content", "intron_length")) %>%
#   dplyr::mutate(any_retained = case_when(VCP.ctrl.retained == "retained" | VCP.mutant.retained == "retained" | SOD1.ctrl.retained == "retained" | SOD1.mutant.retained == "retained" | C9ORF72.ctrl.retained == "retained" | C9ORF72.mutant.retained == "retained" ~ "retained", TRUE ~ "spliced"))
# 
# gr_IRFinder <- as.GenomicRange(ctrl.mutant.3datasets.IR$Name.Coords) # GRanges of IRFinder 8,739 retained introns
# start(gr_IRFinder) = start(gr_IRFinder) +1
# seqlevelsStyle(gr_IRFinder) <- "UCSC"
# gr_IRFinder$id = paste0(seqnames(gr_IRFinder), ":", start(gr_IRFinder), "-", end(gr_IRFinder)) # make an id for each intron
# Homo_sapiens.GRCh38.99.exons <- readRDS("/camp/home/ziffo/home/genomes/ensembl/Homo_sapiens.GRCh38.99.exons.RDS")
# seqlevelsStyle(Homo_sapiens.GRCh38.99.exons) <- "UCSC"
# colnames(mcols(Homo_sapiens.GRCh38.99.exons)) = gsub("biotype","type", colnames(mcols(Homo_sapiens.GRCh38.99.exons))) # changes biotype to type in column names (required for GeneStrutureTools)
# exons.intronRetention <- findIntronContainingTranscripts(gr_IRFinder, Homo_sapiens.GRCh38.99.exons )
# IntronRetentionTranscripts <- addIntronInTranscript(exons.intronRetention, Homo_sapiens.GRCh38.99.exons)
# intron_ids = unique(IntronRetentionTranscripts$whippet_id)
# # get ORF details for each set of transcripts
# g <- BSgenome.Hsapiens.UCSC.hg38::BSgenome.Hsapiens.UCSC.hg38
# 
# # reference transcripts
# normalTranscripts <- Homo_sapiens.GRCh38.99.exons[Homo_sapiens.GRCh38.99.exons$transcript_id %in% unlist(lapply(stringr::str_split(IntronRetentionTranscripts$transcript_id, "[+]"), "[[", 1))]
# orfs_normal <- getOrfs(normalTranscripts, BSgenome = g, returnLongestOnly = FALSE, allFrames = TRUE,uORFs = TRUE)
# orfs_skipped <- getOrfs(IntronRetentionTranscripts[IntronRetentionTranscripts$set == "spliced_intron"],BSgenome = g,returnLongestOnly = FALSE, allFrames = TRUE,uORFs = TRUE)
# orfs_retained <- getOrfs(IntronRetentionTranscripts[IntronRetentionTranscripts$set == "retained_intron"],BSgenome = g,returnLongestOnly = FALSE, allFrames = TRUE,uORFs = TRUE)
# # differences comparing ORFs
# orfChange <- orfDiff(orfsX = orfs_skipped,
#                      orfsY = orfs_retained,
#                      filterNMD = FALSE,
#                      compareBy="gene",
#                      geneSimilarity = TRUE,
#                      compareUTR=TRUE,
#                      allORFs = orfs_normal)
# # add NMD calculations
# orfs_normal$nmd_prob <- notNMD::predictNMD(orfs_normal, "prob")
# orfs_normal$nmd_class <- notNMD::predictNMD(orfs_normal)
# orfs_skipped$nmd_prob <- notNMD::predictNMD(orfs_skipped, "prob")
# orfs_skipped$nmd_class <- notNMD::predictNMD(orfs_skipped)
# orfs_retained$nmd_prob <- notNMD::predictNMD(orfs_retained, "prob")
# orfs_retained$nmd_class <- notNMD::predictNMD(orfs_retained)
# orfs_normal <- orfs_normal[which(!is.na(orfs_normal$orf_length)),]
# orfs_skipped <- orfs_skipped[which(!is.na(orfs_skipped$orf_length)),]
# orfs_retained <- orfs_retained[which(!is.na(orfs_retained$orf_length)),]
# # compare normal and retained isoforms
# # this time setting filterNMD to TRUE, which removes NMD targeted frames/isoforms where possible
# orfChange_filter <- orfDiff(orfsX = orfs_skipped, # normal transcripts
#                      orfsY = orfs_retained, # alternative transcripts
#                      filterNMD = TRUE,
#                      geneSimilarity = TRUE,
#                      compareUTR=TRUE,
#                      allORFs = orfs_normal)
# # test changes in NMD between retained and skipped introns.
# nmdChange <- attrChangeAltSpliced(orfs_skipped, orfs_retained,
#                                   attribute="nmd_prob",
#                                   compareBy="gene",
#                                   useMax=FALSE) # want useMax=F, as we want the minimum NMD probablity (if NMD prob is low, means that at least 1 isoform not targeted)
# m <- match(orfChange_filter$id, nmdChange$id)
# orfChange_filter <- cbind(orfChange_filter, nmdChange[m,-1])
# write.table(orfChange_filter, file="/camp/home/ziffo/home/projects/astrocyte-meta-analysis/splicing/IRFinder/notNMD/ctrl.mutant.3datasets.IR_orf_nmd_changes.txt", sep='\t', row.names = F)
# save.image("/camp/home/ziffo/home/projects/astrocyte-meta-analysis/splicing/IRFinder/notNMD/ctrl.mutant.3datasets.IR_notNMD_results.RData")
# 
# load(file = "/camp/home/ziffo/home/projects/astrocyte-meta-analysis/splicing/IRFinder/notNMD/ctrl.mutant.3datasets.IR_notNMD_results.RData")
# notNMDprob <- orfChange_filter %>% select("coords" = id, "nmd_prob_retained" = nmd_prob_bygroup_y, "nmd_prob_spliced" = nmd_prob_bygroup_x) #
# notNMDprob$coords <- gsub("chr", "", notNMDprob$coords)
# notNMDprob$Chr <- str_split_fixed(notNMDprob$coords, ":", 2)[,1]
# notNMDprob$End <- as.numeric(str_split_fixed(notNMDprob$coords, "-", 2)[,2])
# notNMDprob$Start <- str_split_fixed(notNMDprob$coords, "-", 2)[,1]
# notNMDprob$Start <- as.numeric(str_split_fixed(notNMDprob$Start, ":", 2)[,2])
# notNMDprob <- notNMDprob %>% mutate("Start" = Start - 1) %>% select(-coords) # remove 1 from Start to revert back to original IRFinder coords
# saveRDS(notNMDprob, "/camp/home/ziffo/home/projects/astrocyte-meta-analysis/splicing/IRFinder/notNMD/notNMDprob.ctrl.mutant.3datasets.IR.RDS")
```

Run above in a separate script with:

```{bash}
cd /camp/lab/luscomben/home/users/ziffo/projects/astrocyte-meta-analysis/splicing/IRFinder/notNMD

sbatch --cpus-per-task=10 --mem=1500G -N 1 --partition=hmem -t 48:00:00 --wrap="Rscript /camp/lab/luscomben/home/users/ziffo/projects/astrocyte-meta-analysis/scripts/notNMD_absolute_retained_vs_spliced.R" --mail-type=ALL,ARRAY_TASKS --mail-user=oliver.ziff@crick.ac.uk  --job-name=notNMD
```

Error even with hmem node:
system call failed: Cannot allocate memory

```{r}
notNMDprob <- readRDS("/camp/home/ziffo/home/projects/astrocyte-meta-analysis/splicing/IRFinder/notNMD/notNMDprob.ctrl.mutant.3datasets.IR.RDS")
ctrl.mutant.3datasets.IR <-  ctrl.mutant.3datasets.IR %>% left_join(notNMDprob, by = c("Chr", "Start", "End")) # left join results to IRFinder
ctrl.mutant.3datasets.IR <-  ctrl.mutant.3datasets.IR %>% mutate(nmd_prob = nmd_prob_retained, #case_when(retained == "retained" ~ nmd_prob_retained, TRUE ~ nmd_prob_spliced),
                                 nmd_outcome = case_when(nmd_prob > 0.5 ~ "NMD", TRUE ~ "not NMD"))
# retained Sina plot
notNMD.retained.sina_plot <- sinaplot(data = ctrl.mutant.3datasets.IR, groups = "p0.05_reliable_IRdirection", continuous = "nmd_prob", stat.y.position = 1, stats.test = "wilcox_test", width = 0.2, dashed.line = 0.5) + ylab(label = expression(NMD~probability) ) + xlab("") + theme(axis.text.x = element_text(angle = 45, hjust = 1))
notNMD.retained.sina_plot
```

## VCP vs C9orf72 vs SOD1

Reviewer question: Are there sequence-specific factors that associated with reduced IR in affected genes in VCP, C9orf72, or SOD1 mutant astrocytes?

```{r}
# IR down events in all 3 datasets
vcp <- ac_who_vcp_vs_ctrl %>% dplyr::filter(p0.05_reliable == "significant") %>% dplyr::select(Intron.GeneName.GeneID.Coords, coords, Chr, Start, End, Direction, gene_name, ensID, p.diff, IRratio.diff, p0.05_reliable_IRdirection, gc_content, intron_length) %>% mutate(mutation = "VCP - CTRL")
sod1 <-ac_tyz_sod1_vs_ctrl %>% dplyr::filter(p0.05_reliable == "significant") %>% dplyr::select(Intron.GeneName.GeneID.Coords, coords, Chr, Start, End, Direction, gene_name, ensID, p.diff, IRratio.diff, p0.05_reliable_IRdirection, gc_content, intron_length) %>% mutate(mutation = "SOD1 - CTRL")
c9orf <- ac_bir_c9orf_vs_ctrl %>% dplyr::filter(p0.05_reliable == "significant") %>% dplyr::select(Intron.GeneName.GeneID.Coords, coords, Chr, Start, End, Direction, gene_name, ensID, p.diff, IRratio.diff, p0.05_reliable_IRdirection, gc_content, intron_length) %>% mutate(mutation = "C9orf72 - CTRL")
vcp_sod1_c9orf <- bind_rows(vcp, sod1, c9orf) %>% mutate(mutation = factor(mutation, levels = c("VCP - CTRL", "C9orf72 - CTRL", "SOD1 - CTRL")))

## GC content vs IR up/down in VCP, C9orf72, SOD1
vcp_sod1_c9orf_gc_content_sina_plot <- ggplot(data = vcp_sod1_c9orf, aes(x = p0.05_reliable_IRdirection, y= gc_content, colour = p0.05_reliable_IRdirection)) + 
          geom_violin() + geom_sina(size = 0.5) + geom_boxplot(data = vcp_sod1_c9orf, aes(fill = p0.05_reliable_IRdirection), colour = "black", width=0.2, outlier.shape = NA) +
          scale_colour_manual(values = c("dodgerblue3", "firebrick2", "forestgreen")) + scale_fill_manual(values = c("dodgerblue3", "firebrick2", "forestgreen")) +
          theme_classic() + theme(legend.position = "none", axis.text=element_text(size=10)) +
          facet_grid(. ~ mutation) +
          stat_compare_means(comparisons = list(c("IR down", "IR up")), label = "p.signif", label.y = 1, tip.length = 0.01)  +
  ylab(label = expression(GC~content) ) + xlab("")
vcp_sod1_c9orf_gc_content_sina_plot

## Intron length vs IR up/down in VCP, C9orf72, SOD1
vcp_sod1_c9orf_intron_length_sina_plot <- ggplot(data = vcp_sod1_c9orf, aes(x = p0.05_reliable_IRdirection, y= intron_length, colour = p0.05_reliable_IRdirection)) + 
          geom_violin() + geom_sina(size = 0.5) + geom_boxplot(data = vcp_sod1_c9orf, aes(fill = p0.05_reliable_IRdirection), colour = "black", width=0.2, outlier.shape = NA) +
          scale_colour_manual(values = c("dodgerblue3", "firebrick2", "forestgreen")) + scale_fill_manual(values = c("dodgerblue3", "firebrick2", "forestgreen")) +
          theme_classic() + theme(legend.position = "none", axis.text=element_text(size=10)) +
          facet_grid(. ~ mutation) +
          stat_compare_means(comparisons = list(c("IR down", "IR up")), label = "p.signif", label.y = 4.3, tip.length = 0.01)  + scale_y_log10() + 
  ylab(label = expression(log[10]~intron~length) ) + xlab("")
vcp_sod1_c9orf_intron_length_sina_plot

## PhastCons 
gr_introns <- as.GenomicRange(vcp_sod1_c9orf$Intron.GeneName.GeneID.Coords)
seqlevelsStyle(gr_introns) <- "UCSC"
vcp_sod1_c9orf$phast_cons <- gscores(phast, gr_introns)$default # takes ~1min - too slow to add to function
vcp_sod1_c9orf_PhastCons_sina_plot <- ggplot(data = vcp_sod1_c9orf, aes(x = p0.05_reliable_IRdirection, y= phast_cons, colour = p0.05_reliable_IRdirection)) + 
          geom_violin() + geom_sina(size = 0.5) + geom_boxplot(data = vcp_sod1_c9orf, aes(fill = p0.05_reliable_IRdirection), colour = "black", width=0.2, outlier.shape = NA) +
          scale_colour_manual(values = c("dodgerblue3", "firebrick2", "forestgreen")) + scale_fill_manual(values = c("dodgerblue3", "firebrick2", "forestgreen")) +
          theme_classic() + theme(legend.position = "none", axis.text=element_text(size=10)) +
          facet_grid(. ~ mutation) +
          stat_compare_means(comparisons = list(c("IR down", "IR up")), label = "p.signif", label.y = 0.1, tip.length = 0.01)  + scale_y_log10() + 
  ylab(label = expression(log[10]~PhastCons~score) ) + xlab("")
vcp_sod1_c9orf_PhastCons_sina_plot

## iCLIP
# sum the iclip xlink scores for each RBP in HepG2 in a for loop
# clip_hepg2 <- list.files("/camp/home/ziffo/home/genomes/clip-bedgraphs/HepG2/")
# clip_hepg2 <- clip_hepg2[!grepl("3nt_peaks",clip_hepg2)] # only assess raw crosslink files without any peak calling for enrichment
# grange_introns <- as.GenomicRange(vcp_sod1_c9orf$Intron.GeneName.GeneID.Coords) # create GRanges for each intron (both retained and non retained)
# intron_coords <- tibble(intron = 1:length(vcp_sod1_c9orf$coords), coords = vcp_sod1_c9orf$coords) # intron coordinates for to enable matching back to original coords in IRFinder dataframe
# coverage_list_who <- list()
# for (i in seq_along(clip_hepg2)) {
#   print(clip_hepg2[[i]])
#   rbp <- str_remove(clip_hepg2[[i]], ".merged.bed.gz")
#   bed <- import(paste("/camp/home/ziffo/home/genomes/clip-bedgraphs/HepG2/", clip_hepg2[[i]],sep=""))
#   seqlevelsStyle(bed) <- "NCBI" # remove "chr"
#   grange_overlap <- findOverlaps(query = grange_introns, subject=bed, ignore.strand=FALSE) # find overlaps between RBP binding sites & retained introns
#   range <- bed[ queryHits(grange_overlap) ] %>% as.data.frame() # subset iCLIP xlinks that overlap with the introns - 50,287 rows each with a xlinks score. There are 8739 introns - many introns have multiple scores.
#   coverage_list_who[[i]] <- range %>%  mutate( intron = queryHits(grange_overlap)) %>%  group_by(intron) %>% summarise(iclip_score = sum(score) ) # sum the scores for each intron - 4266 introns with scores, remaining introns have 0 scores
#   coverage_list_who[[i]]$iclip_score[is.na(coverage_list_who[[i]]$iclip_score)]<-0 # set NA scores to 0
#   coverage_list_who[[i]] <- coverage_list_who[[i]] %>% left_join(intron_coords, by = "intron") # add intron coordinates to match back to IRFinder results
#   names(coverage_list_who)[i] <- rbp
# }
# saveRDS(coverage_list_who, "/camp/home/ziffo/home/projects/astrocyte-meta-analysis/splicing/IRFinder/coverage_list.absolute_IR.vcp_sod1_c9orf.CLIPscores.RDS")
# 
# coverage_list_who_df <- bind_rows(coverage_list_who, .id = "intron") # collapse list of dataframes into a single df
# iclip_scores <- coverage_list_who_df %>% distinct() %>% group_by(intron, coords) %>% # each column becomes an RBP. Each row is an intron - there are 310,450 in IRFinder but only 155,313 have xlink scores.
#   summarise(iclip_score.sum = sum(iclip_score)) %>% # https://stackoverflow.com/questions/63365516/tidyr-pivot-wider-error-cant-convert-double-to-list
#   tidyr::pivot_wider(names_from = intron, values_from = iclip_score.sum, values_fill = 0)
# # iclip_scores <- pivot_wider(coverage_list_who_df, names_from = intron, values_from = iclip_score, values_fill = 0)
# iclip_scores <- vcp_sod1_c9orf %>% dplyr::select(coords) %>% full_join(iclip_scores, by = "coords") # add back the introns without xlink scores
# iclip_scores[is.na(iclip_scores)] <- 0  # set NA scores to 0
# iclip_scores$intron_length <- vcp_sod1_c9orf$intron_length[iclip_scores$coords %in% vcp_sod1_c9orf$coords] # add intron_length to df
# iclip_scores_normalised <- iclip_scores %>% mutate_if(is.numeric, ~ (./intron_length)) %>% dplyr::select(-intron_length) # normalise all scores for intron_length
# iclip_scores_normalised$sum <- iclip_scores_normalised %>% select_if(is.numeric) %>% rowSums # sum all normalised RBP scores for each intron
# saveRDS(iclip_scores_normalised, "/camp/home/ziffo/home/projects/astrocyte-meta-analysis/splicing/IRFinder/absolute_IR.vcp_sod1_c9orf.CLIPscores.RDS")
iclip_scores_normalised <- readRDS("/camp/home/ziffo/home/projects/astrocyte-meta-analysis/splicing/IRFinder/absolute_IR.vcp_sod1_c9orf.CLIPscores.RDS")
vcp_sod1_c9orf <- vcp_sod1_c9orf %>% left_join(dplyr::select(iclip_scores_normalised, sum, coords), by = "coords") # add normalised sum of sums of RBP xlinks to IRFinder results
vcp_sod1_c9orf_iclip_sina_plot <- ggplot(data = vcp_sod1_c9orf, aes(x = p0.05_reliable_IRdirection, y= sum, colour = p0.05_reliable_IRdirection)) + 
          geom_violin() + geom_sina(size = 0.5) + geom_boxplot(data = vcp_sod1_c9orf, aes(fill = p0.05_reliable_IRdirection), colour = "black", width=0.2, outlier.shape = NA) +
          scale_colour_manual(values = c("dodgerblue3", "firebrick2", "forestgreen")) + scale_fill_manual(values = c("dodgerblue3", "firebrick2", "forestgreen")) +
          theme_classic() + theme(legend.position = "none", axis.text=element_text(size=10)) +
          facet_grid(. ~ mutation) +
          stat_compare_means(comparisons = list(c("IR down", "IR up")), label = "p.signif", label.y = 2, tip.length = 0.01)  + scale_y_log10() + 
  ylab(label = expression(log[10]~RBP~iCLIP~xlinks) ) + xlab("")
vcp_sod1_c9orf_iclip_sina_plot


### MaxEntScan https://github.com/monahton/GencoDymo https://rdrr.io/github/monahton/GencoDymo/f/vignettes/Vignette.Rmd https://bioconductor.org/packages/devel/bioc/manuals/VarCon/man/VarCon.pdf
# vcp_sod1_c9orf.coords <- vcp_sod1_c9orf %>% dplyr::select(coords)
# hg38 <- BSgenome.Hsapiens.UCSC.hg38::BSgenome.Hsapiens.UCSC.hg38
# # download.file(url="ftp://ftp.ebi.ac.uk/pub/databases/gencode/Gencode_human/release_36/gencode.v36.basic.annotation.gtf.gz",
# #               destfile = "/camp/home/ziffo/home/genomes/gencode/GRCh38.p13.v36.basic.annotation.gtf.gz", method = "wget")
# gen36 <- load_gtf("/camp/home/ziffo/home/genomes/gencode/GRCh38.p13.v36.basic.annotation.gtf.gz")
# gen36_introns <- extract_introns(gen36) # Extract introns from the gtf file
# gen36_introns_unique <- eliminate_redundant_introns(gen36_introns) # Remove redundant introns
# # filter gen36_introns gtf to IRFinder differentially retained introns
# vcp_sod1_c9orf.coords <- vcp_sod1_c9orf %>% dplyr::select("seqnames" = Chr, "intron_start" = Start, "intron_end" = End, "strand" = Direction) %>%
#   mutate(seqnames = paste("chr", seqnames, sep = ""),  # add chr to match gencode annotation style
#          intron_start = 1 + intron_start) # add 1 to start position of intron to avoid overlapping the exon
# gen36_IRFinder_introns <- vcp_sod1_c9orf.coords %>% left_join(gen36_introns_unique, by = c("seqnames", "intron_start", "intron_end", "strand"))
# setwd("/camp/home/ziffo/home/projects/astrocyte-meta-analysis/splicing/IRFinder/MaxEntScan") # set director to save splice site fasta files for MaxEntScan
# ss5_motif_df <- extract_5ss_motif(gen36_IRFinder_introns, BSgenome.Hsapiens.UCSC.hg38) # Produce a fasta file with 5'ss 9-mers:
# ss3_motif_df <- extract_3ss_motif(gen36_IRFinder_introns, BSgenome.Hsapiens.UCSC.hg38) # Produce a fasta file with 3'ss 23-mers:
# ss_motif <- ss5_motif_df %>% left_join(ss3_motif_df, by = c("seqnames", "strand", "transcript_id", "intron_number")) %>%
#   dplyr::rename("motif_start.5" = motif_start.x, "motif_start.3" = motif_start.y, "motif_end.5" = motif_end.x, "motif_end.3" = motif_end.y) %>%
#   mutate(MaxEntScore_5ss = calculateMaxEntScanScore(donor_ss_motif,5), MaxEntScore_3ss = calculateMaxEntScanScore(acceptor_ss_motif,3)) %>% # Calculate MaxEntScores at both splice sites
#   left_join(dplyr::select(gen36_introns_unique, seqnames, intron_start, intron_end, strand, transcript_id, intron_number), by = c("seqnames", "strand", "transcript_id", "intron_number")) %>% # intron start end
#   mutate(seqnames = gsub("chr", "", seqnames), intron_start = intron_start - 1) %>% rename("Chr" = seqnames, "Direction" = strand, "Start" = intron_start, "End" = intron_end)
# saveRDS(ss_motif, "/camp/lab/luscomben/home/users/ziffo/projects/astrocyte-meta-analysis/splicing/IRFinder/MaxEntScan/MaxEntScores.differential_IR.vcp_sod1_c9orf.RDS")
ss_motif <- readRDS("/camp/lab/luscomben/home/users/ziffo/projects/astrocyte-meta-analysis/splicing/IRFinder/MaxEntScan/MaxEntScores.differential_IR.vcp_sod1_c9orf.RDS")
vcp_sod1_c9orf.MaxEntScan <- vcp_sod1_c9orf %>% left_join(dplyr::select(ss_motif, Chr, Start, End, "Direction" = Strand, MaxEntScore_5ss, MaxEntScore_3ss), by = c("Chr", "Start", "End", "Direction")) %>%
  mutate(MaxEntScore_5ss = as.numeric(MaxEntScore_5ss), MaxEntScore_3ss = as.numeric(MaxEntScore_3ss)) # add MaxEntScan scores back to IRFinder results

vcp_sod1_c9orf_maxent5_sina_plot <- ggplot(data = vcp_sod1_c9orf.MaxEntScan, aes(x = p0.05_reliable_IRdirection, y= MaxEntScore_5ss, colour = p0.05_reliable_IRdirection)) + 
          geom_violin() + geom_sina(size = 0.5) + geom_boxplot(data = vcp_sod1_c9orf.MaxEntScan, aes(fill = p0.05_reliable_IRdirection), colour = "black", width=0.2, outlier.shape = NA) +
          scale_colour_manual(values = c("dodgerblue3", "firebrick2", "forestgreen")) + scale_fill_manual(values = c("dodgerblue3", "firebrick2", "forestgreen")) +
          theme_classic() + theme(legend.position = "none", axis.text=element_text(size=10)) + facet_grid(. ~ mutation) + ylim(c(0,13)) +
          stat_compare_means(comparisons = list(c("IR down", "IR up")), label = "p.signif", label.y = 13, tip.length = 0.01) + ylab(label = "5' MaxEntScore" ) + xlab("")
vcp_sod1_c9orf_maxent5_sina_plot

vcp_sod1_c9orf_maxent3_sina_plot <- ggplot(data = vcp_sod1_c9orf.MaxEntScan, aes(x = p0.05_reliable_IRdirection, y= MaxEntScore_3ss, colour = p0.05_reliable_IRdirection)) + 
          geom_violin() + geom_sina(size = 0.5) + geom_boxplot(data = vcp_sod1_c9orf.MaxEntScan, aes(fill = p0.05_reliable_IRdirection), colour = "black", width=0.2, outlier.shape = NA) +
          scale_colour_manual(values = c("dodgerblue3", "firebrick2", "forestgreen")) + scale_fill_manual(values = c("dodgerblue3", "firebrick2", "forestgreen")) +
          theme_classic() + theme(legend.position = "none", axis.text=element_text(size=10)) + facet_grid(. ~ mutation) + ylim(c(0,15)) +
          stat_compare_means(comparisons = list(c("IR down", "IR up")), label = "p.signif", label.y = 15, tip.length = 0.01) + ylab(label = "3' MaxEntScore" ) + xlab("")
vcp_sod1_c9orf_maxent3_sina_plot

### notNMD https://www.bioconductor.org/packages/devel/bioc/vignettes/GeneStructureTools/inst/doc/Vignette.html https://github.com/betsig/notNMD  http://htmlpreview.github.io/?https://github.com/betsig/notNMD/blob/master/vignette.html
# Dr Beth Signal wrote this script. NB add 1 to the start of the IRFinder introns for this to work.
# Also GeneStructureTools functions clash with other packages so need a fresh R environment (dont source irfinder_functions.R)
# .libPaths("/camp/lab/luscomben/home/users/ziffo/.conda/envs/r4.0.3/lib/R/library")
# load("/camp/home/ziffo/home/projects/astrocyte-meta-analysis/scripts/figures_and_tables.RData")
# library(GenomicRanges)
# library(stringr)
# library(BSgenome.Hsapiens.UCSC.hg38)
# library(Gviz)
# library(rtracklayer)
# library(irlib)
# library(tidyverse)
# library(GeneStructureTools)
# vcp <- ac_who_vcp_vs_ctrl %>% dplyr::filter(p0.05_reliable == "significant") %>% dplyr::select(Intron.GeneName.GeneID.Coords, coords, Chr, Start, End, Direction, gene_name, ensID, p.diff, IRratio.diff, p0.05_reliable_IRdirection, gc_content, intron_length) %>% mutate(mutation = "VCP - CTRL")
# sod1 <-ac_tyz_sod1_vs_ctrl %>% dplyr::filter(p0.05_reliable == "significant") %>% dplyr::select(Intron.GeneName.GeneID.Coords, coords, Chr, Start, End, Direction, gene_name, ensID, p.diff, IRratio.diff, p0.05_reliable_IRdirection, gc_content, intron_length) %>% mutate(mutation = "SOD1 - CTRL")
# c9orf <- ac_bir_c9orf_vs_ctrl %>% dplyr::filter(p0.05_reliable == "significant") %>% dplyr::select(Intron.GeneName.GeneID.Coords, coords, Chr, Start, End, Direction, gene_name, ensID, p.diff, IRratio.diff, p0.05_reliable_IRdirection, gc_content, intron_length) %>% mutate(mutation = "C9orf72 - CTRL")
# vcp_sod1_c9orf <- bind_rows(vcp, sod1, c9orf) %>% mutate(mutation = factor(mutation, levels = c("VCP - CTRL", "C9orf72 - CTRL", "SOD1 - CTRL")))
# gr_IRFinder <- as.GenomicRange(vcp_sod1_c9orf$Intron.GeneName.GeneID.Coords) # GRanges of IRFinder 8,739 retained introns
# start(gr_IRFinder) = start(gr_IRFinder) +1
# seqlevelsStyle(gr_IRFinder) <- "UCSC"
# gr_IRFinder$id = paste0(seqnames(gr_IRFinder), ":", start(gr_IRFinder), "-", end(gr_IRFinder)) # make an id for each intron
# Homo_sapiens.GRCh38.99.exons <- readRDS("/camp/home/ziffo/home/genomes/ensembl/Homo_sapiens.GRCh38.99.exons.RDS")
# seqlevelsStyle(Homo_sapiens.GRCh38.99.exons) <- "UCSC"
# colnames(mcols(Homo_sapiens.GRCh38.99.exons)) = gsub("biotype","type", colnames(mcols(Homo_sapiens.GRCh38.99.exons))) # changes biotype to type in column names (required for GeneStrutureTools)
# exons.intronRetention <- findIntronContainingTranscripts(gr_IRFinder, Homo_sapiens.GRCh38.99.exons )
# IntronRetentionTranscripts <- addIntronInTranscript(exons.intronRetention, Homo_sapiens.GRCh38.99.exons)
# intron_ids = unique(IntronRetentionTranscripts$whippet_id)
# # get ORF details for each set of transcripts
# g <- BSgenome.Hsapiens.UCSC.hg38::BSgenome.Hsapiens.UCSC.hg38
# # reference transcripts
# normalTranscripts <- Homo_sapiens.GRCh38.99.exons[Homo_sapiens.GRCh38.99.exons$transcript_id %in% unlist(lapply(stringr::str_split(IntronRetentionTranscripts$transcript_id, "[+]"), "[[", 1))]
# orfs_normal <- getOrfs(normalTranscripts, BSgenome = g, returnLongestOnly = FALSE, allFrames = TRUE,uORFs = TRUE)
# orfs_skipped <- getOrfs(IntronRetentionTranscripts[IntronRetentionTranscripts$set == "spliced_intron"],BSgenome = g,returnLongestOnly = FALSE, allFrames = TRUE,uORFs = TRUE)
# orfs_retained <- getOrfs(IntronRetentionTranscripts[IntronRetentionTranscripts$set == "retained_intron"],BSgenome = g,returnLongestOnly = FALSE, allFrames = TRUE,uORFs = TRUE)
# # differences comparing ORFs
# orfChange <- orfDiff(orfsX = orfs_skipped,
#                      orfsY = orfs_retained,
#                      filterNMD = FALSE,
#                      compareBy="gene",
#                      geneSimilarity = TRUE,
#                      compareUTR=TRUE,
#                      allORFs = orfs_normal)
# # add NMD calculations
# orfs_normal$nmd_prob <- notNMD::predictNMD(orfs_normal, "prob")
# orfs_normal$nmd_class <- notNMD::predictNMD(orfs_normal)
# orfs_skipped$nmd_prob <- notNMD::predictNMD(orfs_skipped, "prob")
# orfs_skipped$nmd_class <- notNMD::predictNMD(orfs_skipped)
# orfs_retained$nmd_prob <- notNMD::predictNMD(orfs_retained, "prob")
# orfs_retained$nmd_class <- notNMD::predictNMD(orfs_retained)
# orfs_normal <- orfs_normal[which(!is.na(orfs_normal$orf_length)),]
# orfs_skipped <- orfs_skipped[which(!is.na(orfs_skipped$orf_length)),]
# orfs_retained <- orfs_retained[which(!is.na(orfs_retained$orf_length)),]
# # compare normal and retained isoforms
# # this time setting filterNMD to TRUE, which removes NMD targeted frames/isoforms where possible
# orfChange_filter <- orfDiff(orfsX = orfs_skipped, # normal transcripts
#                      orfsY = orfs_retained, # alternative transcripts
#                      filterNMD = TRUE,
#                      geneSimilarity = TRUE,
#                      compareUTR=TRUE,
#                      allORFs = orfs_normal)
# # test changes in NMD between retained and skipped introns.
# nmdChange <- attrChangeAltSpliced(orfs_skipped, orfs_retained,
#                                   attribute="nmd_prob",
#                                   compareBy="gene",
#                                   useMax=FALSE) # want useMax=F, as we want the minimum NMD probablity (if NMD prob is low, means that at least 1 isoform not targeted)
# m <- match(orfChange_filter$id, nmdChange$id)
# orfChange_filter <- cbind(orfChange_filter, nmdChange[m,-1])
# write.table(orfChange_filter, file="/camp/home/ziffo/home/projects/astrocyte-meta-analysis/splicing/IRFinder/notNMD/vcp_sod1_c9orf_orf_nmd_changes.txt", sep='\t', row.names = F)
# save.image("/camp/home/ziffo/home/projects/astrocyte-meta-analysis/splicing/IRFinder/notNMD/vcp_sod1_c9orf_notNMD_results.RData")
# load(file = "/camp/home/ziffo/home/projects/astrocyte-meta-analysis/splicing/IRFinder/notNMD/vcp_sod1_c9orf_notNMD_results.RData")
# notNMDprob <- orfChange_filter %>% select("coords" = id, "nmd_prob_retained" = nmd_prob_bygroup_y, "nmd_prob_spliced" = nmd_prob_bygroup_x) #
# notNMDprob$coords <- gsub("chr", "", notNMDprob$coords)
# notNMDprob$Chr <- str_split_fixed(notNMDprob$coords, ":", 2)[,1]
# notNMDprob$End <- as.numeric(str_split_fixed(notNMDprob$coords, "-", 2)[,2])
# notNMDprob$Start <- str_split_fixed(notNMDprob$coords, "-", 2)[,1]
# notNMDprob$Start <- as.numeric(str_split_fixed(notNMDprob$Start, ":", 2)[,2])
# notNMDprob <- notNMDprob %>% mutate("Start" = Start - 1) %>% select(-coords) # remove 1 from Start to revert back to original IRFinder coords
# saveRDS(notNMDprob, "/camp/home/ziffo/home/projects/astrocyte-meta-analysis/splicing/IRFinder/notNMD/notNMDprob.vcp_sod1_c9orf.RDS")
notNMDprob <- readRDS("/camp/home/ziffo/home/projects/astrocyte-meta-analysis/splicing/IRFinder/notNMD/notNMDprob.vcp_sod1_c9orf.RDS")
vcp_sod1_c9orf.notNMD <-  vcp_sod1_c9orf %>% left_join(notNMDprob, by = c("Chr", "Start", "End")) # left join results to IRFinder
vcp_sod1_c9orf.notNMD <-  vcp_sod1_c9orf.notNMD %>% mutate(nmd_prob = nmd_prob_retained, #case_when(retained == "retained" ~ nmd_prob_retained, TRUE ~ nmd_prob_spliced),
                                 nmd_outcome = case_when(nmd_prob > 0.5 ~ "NMD", TRUE ~ "not NMD"))
vcp_sod1_c9orf_notNMD_sina_plot <- ggplot(data = vcp_sod1_c9orf.notNMD, aes(x = p0.05_reliable_IRdirection, y= nmd_prob, colour = p0.05_reliable_IRdirection)) + 
          geom_violin() + geom_sina(size = 0.5) + geom_boxplot(data = vcp_sod1_c9orf.notNMD, aes(fill = p0.05_reliable_IRdirection), colour = "black", width=0.2, outlier.shape = NA) +
          scale_colour_manual(values = c("dodgerblue3", "firebrick2", "forestgreen")) + scale_fill_manual(values = c("dodgerblue3", "firebrick2", "forestgreen")) +
          theme_classic() + theme(legend.position = "none", axis.text=element_text(size=10)) + facet_grid(. ~ mutation) + 
          geom_hline(yintercept =0.5, linetype = 2) +
          stat_compare_means(comparisons = list(c("IR down", "IR up")), label = "p.signif", label.y = 1, tip.length = 0.01, method = "wilcox.test") + ylab(label = expression(NMD~probability) ) + xlab("")
vcp_sod1_c9orf_notNMD_sina_plot
```

## Merge

```{r}
figS4.merged <- ggarrange(
  ggarrange(gc.retained.sina_plot, length.retained.sina_plot, PhastCons.retained.sina_plot, clip.retained.sina_plot, maxent5.retained.sina_plot, maxent3.retained.sina_plot,nrow = 1, ncol = 6, widths = c(1,1,1,1,1,1)), # notNMD.retained.sina_plot
  ggarrange( vcp_sod1_c9orf_gc_content_sina_plot, vcp_sod1_c9orf_intron_length_sina_plot, nrow = 1, ncol = 2, widths = c(1,1)),
  ggarrange(vcp_sod1_c9orf_PhastCons_sina_plot, vcp_sod1_c9orf_iclip_sina_plot, nrow = 1, ncol = 2, widths = c(1,1)),
  ggarrange(vcp_sod1_c9orf_maxent5_sina_plot, vcp_sod1_c9orf_maxent3_sina_plot, nrow = 1, ncol = 2, widths = c(1,1)), # vcp_sod1_c9orf_notNMD_sina_plot
  nrow = 4, heights = c(1,1,1,1), labels = c("A", "B", "", ""))
figS4.merged
ggsave(figS4.merged, filename = "/camp/home/ziffo/home/projects/astrocyte-meta-analysis/splicing/IRFinder/irlib/figures/figS4.merged.png", height = 12.5, width = 9)
```


# Figure 2: IR down & GE up in ALS mutants

```{r}
# 2A Sina plot DGE vs IR up/down in VCP, C9orf72, SOD1, reactivity
vcp <- ac_who_vcp_vs_ctrl %>% dplyr::filter(p0.05_reliable == "significant") %>% dplyr::select(Intron.GeneName.GeneID, Chr, Start, End, gene_name, ensID, p.diff, IRratio.diff, p0.05_reliable_IRdirection, log2FoldChange) %>% mutate(mutation = "VCP - CTRL")
sod1 <-ac_tyz_sod1_vs_ctrl %>% dplyr::filter(p0.05_reliable == "significant") %>% dplyr::select(Intron.GeneName.GeneID, Chr, Start, End, gene_name, ensID, p.diff, IRratio.diff, p0.05_reliable_IRdirection, log2FoldChange) %>% mutate(mutation = "SOD1 - CTRL")
c9orf <- ac_bir_c9orf_vs_ctrl %>% dplyr::filter(p0.05_reliable == "significant") %>% dplyr::select(Intron.GeneName.GeneID, Chr, Start, End, gene_name, ensID, p.diff, IRratio.diff, p0.05_reliable_IRdirection, log2FoldChange) %>% mutate(mutation = "C9orf72 - CTRL")
vcp_sod1_c9orf <- bind_rows(vcp, sod1, c9orf) %>% mutate(mutation = factor(mutation, levels = c("VCP - CTRL", "C9orf72 - CTRL", "SOD1 - CTRL")))

vcp_sod1_c9orf_dge_ir_sina_plot <- ggplot(data = vcp_sod1_c9orf, aes(x = p0.05_reliable_IRdirection, y= log2FoldChange, colour = p0.05_reliable_IRdirection)) + 
          geom_violin() + geom_sina(size = 0.5) + geom_boxplot(data = vcp_sod1_c9orf, aes(fill = p0.05_reliable_IRdirection), colour = "black", width=0.2, outlier.shape = NA) +
          scale_colour_manual(values = c("dodgerblue3", "firebrick2", "forestgreen")) + scale_fill_manual(values = c("dodgerblue3", "firebrick2", "forestgreen")) +
          theme_classic() + theme(legend.position = "none", axis.text=element_text(size=10)) +
          geom_hline(yintercept = 0, linetype = 2) + facet_grid(. ~ mutation) +
          stat_compare_means(comparisons = list(c("IR down", "IR up")), label = "p.signif", label.y = 1.7, tip.length = 0.01)  + ylim(c(-2,2)) + ylab(label = expression(Gene~exp~LFC) ) + xlab("")
vcp_sod1_c9orf_dge_ir_sina_plot

# show t-test p-values on plot
ggplot(data = vcp_sod1_c9orf, aes(x = p0.05_reliable_IRdirection, y= log2FoldChange, colour = p0.05_reliable_IRdirection)) + 
          geom_violin() + geom_sina(size = 0.5) + geom_boxplot(data = vcp_sod1_c9orf, aes(fill = p0.05_reliable_IRdirection), colour = "black", width=0.2, outlier.shape = NA) +
          scale_colour_manual(values = c("dodgerblue3", "firebrick2", "forestgreen")) + scale_fill_manual(values = c("dodgerblue3", "firebrick2", "forestgreen")) +
          theme_classic() + theme(legend.position = "none", axis.text=element_text(size=10)) +
          geom_hline(yintercept = 0, linetype = 2) + facet_grid(. ~ mutation) +
          stat_compare_means(comparisons = list(c("IR down", "IR up")), label = "p.label", label.y = 1.7, tip.length = 0.01)  + ylim(c(-2,2)) + ylab(label = expression(Gene~exp~LFC) ) + xlab("")
# VCP 2.6e-11
# C9orf72 4.5e-8
# SOD1 1.5e-4

### VCP scatter plots: Delta IR vs differential GE
ac_who_vcp_vs_ctrl.labels.DGE.IR <- c("COL1A1", "FN1", "FLNA", "PDLIM7", "TNC", "ILK", " MCAM", "HLA-B", "HLA-C", "ADAMTSL4", "TGFB1I1", "LAMB2", "ANKZF1", "OSMR", "LOXL3", "RECK", "COL7A1", "UBN1",  "ADAM19")
labels_lenient <- filter(ac_who_vcp_vs_ctrl, p0.05_reliable == "significant") %>% filter(gene_name %in% ac_who_vcp_vs_ctrl.labels.DGE.IR) %>% arrange( -(abs(IRratio.diff)) ) %>% distinct(gene_name, .keep_all = TRUE)
ac_who_dge_ir_scatter <- ggplot(filter(ac_who_vcp_vs_ctrl, p0.05_reliable == "significant"), aes( x = IRratio.diff, y = log2FoldChange,  label = gene_name, colour = log10(baseMean))) + 
  geom_point(alpha = 0.5, size = 0.5) +  theme_bw() +  xlab( expression( Delta~IR~ratio~VCP-CTRL)  ) +  ylab( expression(Gene~exp~VCP:CTRL) ) +
  scale_colour_gradient(low = "dodgerblue3", high = "firebrick2") +
  geom_text_repel(data = labels_lenient, aes(x = IRratio.diff, y = log2FoldChange, label = gene_name), fontface = "italic", colour = "black", min.segment.length = 0, box.padding = 0.5, point.padding = 0.5, size = 2.5, max.overlaps = Inf) +
  theme(strip.background = element_rect(colour="black", fill="white"), panel.grid = element_blank(), legend.position = "none") + 
  geom_hline(yintercept = 0, linetype = 3, colour = "darkgrey") + geom_vline(xintercept = 0, linetype = 3, colour = "darkgrey") + 
  geom_smooth(data = filter(ac_who_vcp_vs_ctrl, p0.05_reliable == "significant"), aes(x = IRratio.diff, y = log2FoldChange), method = "lm", se = FALSE, show.legend = TRUE, colour = "black", linetype = 2) + 
  xlim(-0.36,0.36) +  ylim(-2,2.3) +
  annotate(geom="text", x=.25, y=-2, label="IR up & expression down", color="darkgrey", size = 2.7)  + annotate(geom="text", x=.25, y=2, label="IR up & expression up", color="darkgrey", size = 2.7)  + 
  annotate(geom="text", x=-.25, y=2, label="IR down & expression up", color="#E64B35FF", size = 2.7) +   annotate(geom="text", x=-.25, y=-2, label="IR down & expression down", color="darkgrey", size = 2.7)
ac_who_dge_ir_scatter

ac_who_vcp_vs_ctrl_filt <- ac_who_vcp_vs_ctrl %>% dplyr::filter(p0.05_reliable == "significant") %>%
  drop_na(IRratio.diff, log2FoldChange) %>% filter(IRratio.diff != Inf) %>% filter(IRratio.diff != -Inf) %>% filter(log2FoldChange != Inf) %>% filter(log2FoldChange != -Inf)
cor.test(x = ac_who_vcp_vs_ctrl_filt$IRratio.diff, y = ac_who_vcp_vs_ctrl_filt$log2FoldChange) # R = -.20
summary( lm( IRratio.diff ~ log2FoldChange, data = ac_who_vcp_vs_ctrl_filt))

ac_who_vcp_vs_ctrl %>% dplyr::filter(p0.05_reliable == "significant", log2FoldChange >0) %>% distinct(gene_name) %>% nrow # 817 genes with increased expression and exhibiting a reliable IR event
ac_who_vcp_vs_ctrl %>% dplyr::filter(p0.05_reliable == "significant", IRratio.diff < 0, log2FoldChange >0) %>% distinct(gene_name) %>% nrow # 702 exhibited decreased IR in VCP astrocytes
(817-(817-702))/817 * 100 # 85.8% decreased in VCP

# 2F GO terms IR down & DGE up in VCP
vcp_dge_ir_genes <- ac_who_vcp_vs_ctrl %>% filter(p0.05_reliable == "significant") %>% split(.$direction) %>%  map("ensID") # create list of IR up/down & Expression up/down genes
vcp_dge_ir_genes <- gost(vcp_dge_ir_genes) # run gprofiler2 on all groups - runs on all KEGG, GO, REAC,TF, WP pathways simultaneously
vcp_dge_ir_gprofiles_filt <- vcp_dge_ir_genes$result %>% filter(str_detect(source, "KEGG|GO:BP|GO:MF|GO:CC|REAC|CORUM")) %>% arrange( -log10(p_value) ) %>% mutate( term_name = as.factor(term_name)) 
vcp_dge_ir_gprofiles_curated_irdown_expup <- vcp_dge_ir_gprofiles_filt %>% filter(query == "IR down & expression up")
paste('"',unique(vcp_dge_ir_gprofiles_curated_irdown_expup$term_name),'",', sep = "", collapse = '\n') %>% cat()
irdown_expup_curated_list <- c("NIK-->noncanonical NF-kB signaling",
                               #"Antigen processing-Cross presentation",
                  "regulation of cell morphogenesis",
                  "Class I MHC mediated antigen processing & presentation",
                  # "ECM-receptor interaction",
                  "Wnt signaling pathway",
                  # "RNA metabolic process",
                  "Extracellular matrix organization",
                  # "ribonucleoprotein complex biogenesis",
                  "RNA processing",
                  "cellular response to stress",
                  "Collagen formation",
                  # "metabolic process",
                  "RNA binding",
                  "cellular metabolic process",
                  "focal adhesion") #,  "cellular component organization","nucleus", "cytoplasm")#,"protein binding")
vcp_dge_ir_gprofiles_curated_irdown_expup <- vcp_dge_ir_gprofiles_curated_irdown_expup %>% filter(term_name %in% irdown_expup_curated_list) %>% mutate(query = "")
vcp_dge_ir_gprofiles_curated_irdown_expup$term_name <- gsub("Class I MHC mediated antigen processing & presentation", "Class I MHC antigen processing", vcp_dge_ir_gprofiles_curated_irdown_expup$term_name)
# vcp_dge_ir_gprofiles_curated_irdown_expup$term_name <- strtrim(vcp_dge_ir_gprofiles_curated_irdown_expup$term_name, 39) # limit term_name string length
ac_who_vcp_vs_ctrl.go_curated <- curated_profiles(gprofiles = vcp_dge_ir_gprofiles_curated_irdown_expup, colours = "#E64B35FF") + ggtitle("VCP")
ac_who_vcp_vs_ctrl.go_curated


# Liddelow heatmaps DGE & IR
# Annotation for all Liddelow heatmaps
annot <- data.frame(row.names = astrocyte.reactivity.markers$gene_name, group = factor(astrocyte.reactivity.markers$group))
annot_cols <- c("black", "firebrick2", "dodgerblue3")
names(annot_cols) <- unique(astrocyte.reactivity.markers$group)
annot_cols <- list(group = annot_cols)
breaksList = seq(-2, 2, by = 0.1) 
vcp_dge_ir_filt <- ac_who_vcp_vs_ctrl.absoluteIR %>% filter(gene_name %in% astrocyte.reactive.genes) %>% arrange(IR.lfc) %>% distinct(gene_name, .keep_all = TRUE) %>% 
  dplyr::select(gene_name, "VCP IR" = IR.lfc, "VCP GE" = log2FoldChange)
vcp_dge_ir_filt$"VCP IR" <- replace_na(vcp_dge_ir_filt$"VCP IR", 0)
vcp_dge_ir_filt$"VCP GE" <- replace_na(vcp_dge_ir_filt$"VCP GE", 0)
vcp_dge_ir_filt <- vcp_dge_ir_filt %>% dplyr::arrange(factor(gene_name, levels = rownames(annot))) # order the matrix by the annotation dataframe
vcp_dge_ir_filt.mat <- make_matrix(select(vcp_dge_ir_filt,-gene_name),pull(vcp_dge_ir_filt,gene_name))
vcp_dge_ir_filt.mat[vcp_dge_ir_filt.mat == -Inf] <- -2.0
vcp_dge_ir_filt.mat[vcp_dge_ir_filt.mat == Inf] <- 2.0
vcp_dge_ir_filt.mat[vcp_dge_ir_filt.mat == NA] <- 0
vcp_dge_ir_filt.mat[vcp_dge_ir_filt.mat > 2.0] <- 2.0
vcp_dge_ir_filt.mat[vcp_dge_ir_filt.mat < -2.0] <- -2.0
ac_who_vcp_vs_ctrl.all <- cleanIRFinder(IRFinder = ac_who_vcp_vs_ctrl.IRFinder) %>% full_join(dge_res_ac_who_vcp_vs_ctrl, by = c("ensID", "gene_name")) # full join DESeq to avoid loosing DGE values where no IR values exist
vcp_dge_res_padj <- ac_who_vcp_vs_ctrl.all %>% filter(gene_name %in% astrocyte.reactive.genes) %>% arrange(IR.lfc) %>% distinct(gene_name, .keep_all = TRUE) %>%
  mutate(IR.sig = case_when(p0.05_reliable == "significant" ~ "\u2217", TRUE ~ ""), DGE.sig = case_when(padj <= 0.05 ~  "\u2217", TRUE ~ "")) %>%
  dplyr::select(gene_name, IR.sig, DGE.sig) %>%
  arrange(factor(gene_name, levels = rownames(annot))) %>% dplyr::select(IR.sig,DGE.sig) %>%  as.matrix  # order the matrix by the annotation dataframe
ac_who_liddelow_heatmap <- pheatmap(t(vcp_dge_ir_filt.mat), cluster_rows=F, show_rownames=T, show_colnames = T, cluster_cols=F, fontsize = 8, gaps_col = c(13,29), gaps_row = 1,
                  annotation_col = annot, annotation_names_col = F, annotation_legend = FALSE, annotation_colors = annot_cols, main = "", display_numbers = t(vcp_dge_res_padj), 
                  color = colorRampPalette(rev(greenred(75)))(length(breaksList)),breaks = breaksList, 
                  cellheight = 9, cellwidth = 14,
                  border_color=NA, legend = TRUE)

# We contrasted genes with reduced IR and increased expression with astrocyte reactivity genes identified by Liddelow et al (25) and found 27% (11 / 41, bivariate gene set enrichment p = 0.04) exhibit decreased IR and increased expression
LFC.IR_DGE <- ac_who_vcp_vs_ctrl.absoluteIR %>% group_by(gene_name) %>% dplyr::summarise(IR = mean(IRratio.diff, na.rm = TRUE), GE = median(log2FoldChange))
LFC.IR_DGE %>% filter(gene_name %in% astrocyte.reactivity.markers$gene_name, IR < 0, GE > 0) # 11 / 41 liddelow genes have decreased IR & increased GE
11/41 * 100 # 27%
LFC.IR_DGE[is.na(LFC.IR_DGE)] <- 0
LFC.IR_DGE <- LFC.IR_DGE %>% data.frame(row.names = "gene_name")
LFC.IR_DGE <- scale(LFC.IR_DGE) # rescale both IR & GE to ensure they have the same variance
LFC.IR_DGE <- LFC.IR_DGE %>% as_tibble(rownames = "gene_name") %>% mutate("IR_DGE" = (GE + (IR*-1)/2)) # take mean GE & IR (inverse) 
geneList = LFC.IR_DGE$IR_DGE
names(geneList) <- as.character(LFC.IR_DGE$gene_name)
geneList = sort(geneList, decreasing = TRUE)
liddelow_genes.list <- list("Astrocyte_reactivity" = astrocyte.reactivity.markers$gene_name[astrocyte.reactivity.markers$gene_name %in% ac_who_vcp_vs_ctrl.absoluteIR$gene_name]) # liddelow genes

fgseaRes <- fgsea(pathways = liddelow_genes.list, stats = geneList)
fgseaRes # NES = 0.87 p = 0.66


#  Scatter delta IR ratio vs DGE in C9ORF72
labels_lenient <- filter(ac_bir_c9orf_vs_ctrl, p0.05_reliable == "significant") %>% filter(gene_name %in% astrocyte.reactivity.labels.ac_c9orf72) %>% arrange( -(abs(IRratio.diff)) ) %>% distinct(gene_name, .keep_all = TRUE)
ac_bir_c9orf_scatter <- ggplot(filter(ac_bir_c9orf_vs_ctrl, p0.05_reliable == "significant"), aes( x = IRratio.diff, y = log2FoldChange,  label = gene_name, colour = log10(baseMean))) + 
  geom_point(alpha = 0.5, size = 0.5) +  theme_bw() +  xlab( expression( Delta~IR~ratio~C9orf72-CTRL)  ) +  ylab( expression(Gene~exp~C9orf72:CTRL) ) +
  scale_colour_gradient(low = "dodgerblue3", high = "firebrick2") +
  geom_text_repel(data = labels_lenient, aes(x = IRratio.diff, y = log2FoldChange, label = gene_name), fontface = "italic", colour = "black", min.segment.length = 0, box.padding = 0.5, point.padding = 0.5, size = 2.5, max.overlaps = Inf) +
  theme(strip.background = element_rect(colour="black", fill="white"), panel.grid = element_blank(), legend.position = "none", legend.box.margin=margin(-10,-10,-10,-10), legend.spacing.x = unit(1.0, 'cm'), axis.text=element_text(size=10)) + labs(colour = expression(log[10]~expression)) +
  geom_hline(yintercept = 0, linetype = 3, colour = "darkgrey") + geom_vline(xintercept = 0, linetype = 3, colour = "darkgrey") + 
  geom_smooth(data = filter(ac_bir_c9orf_vs_ctrl, p0.05_reliable == "significant"), aes(x = IRratio.diff, y = log2FoldChange), method = "lm", se = FALSE, show.legend = TRUE, colour = "black", linetype = 2) + 
  xlim(-0.4,0.4) +  ylim(-2,2.3) +
  annotate(geom="text", x=.25, y=-2, label="IR up & expression down", color="darkgrey", size = 2.7)  + annotate(geom="text", x=.25, y=2.2, label="IR up & expression up", color="darkgrey", size = 2.7)  + 
  annotate(geom="text", x=-.25, y=2.2, label="IR down & expression up", color="#4DBBD5FF", size = 2.7) +   annotate(geom="text", x=-.25, y=-2, label="IR down & expression down", color="darkgrey", size = 2.7)
ac_bir_c9orf_scatter

ac_bir_c9orf_vs_ctrl_filt <- ac_bir_c9orf_vs_ctrl %>% dplyr::filter(p0.05_reliable == "significant") %>%
  drop_na(IRratio.diff, log2FoldChange) %>% filter(IRratio.diff != Inf) %>% filter(IRratio.diff != -Inf) %>% filter(log2FoldChange != Inf) %>% filter(log2FoldChange != -Inf)
cor.test(x = ac_bir_c9orf_vs_ctrl_filt$IRratio.diff, y = ac_bir_c9orf_vs_ctrl_filt$log2FoldChange) # R = -0.05
summary( lm( IRratio.diff ~ log2FoldChange, data = ac_bir_c9orf_vs_ctrl_filt)) # beta coefficient under Coefficients: Estimate under log2FoldChange row


# Scatter delta IR ratio vs DGE in SOD1
labels_lenient <- filter(ac_tyz_sod1_vs_ctrl, p0.05_reliable == "significant") %>% filter(gene_name %in% astrocyte.reactivity.labels.ac_sod1) %>% arrange( -(abs(IRratio.diff)) ) %>% distinct(gene_name, .keep_all = TRUE)
ac_tyz_sod1_scatter <- ggplot(filter(ac_tyz_sod1_vs_ctrl, p0.05_reliable == "significant"), aes( x = IRratio.diff, y = log2FoldChange,  label = gene_name, colour = log10(baseMean))) + 
  geom_point(alpha = 0.5, size = 0.5) +  theme_bw() +  xlab( expression( Delta~IR~ratio~SOD1-CTRL)  ) +  ylab( expression(Gene~exp~SOD1:CTRL) ) +
  scale_colour_gradient(low = "dodgerblue3", high = "firebrick2") +
  geom_text_repel(data = labels_lenient, aes(x = IRratio.diff, y = log2FoldChange, label = gene_name), fontface = "italic", colour = "black", min.segment.length = 0, box.padding = 0.5, point.padding = 0.5, size = 2.5, max.overlaps = Inf) +
  theme(strip.background = element_rect(colour="black", fill="white"), panel.grid = element_blank(), legend.position = "none", legend.box.margin=margin(-10,-10,-10,-10), legend.spacing.x = unit(1.0, 'cm'), axis.text=element_text(size=10)) + labs(colour = expression(log[10]~expression)) +
  geom_hline(yintercept = 0, linetype = 3, colour = "darkgrey") + geom_vline(xintercept = 0, linetype = 3, colour = "darkgrey") + 
  geom_smooth(data = filter(ac_tyz_sod1_vs_ctrl, p0.05_reliable == "significant"), aes(x = IRratio.diff, y = log2FoldChange), method = "lm", se = FALSE, show.legend = TRUE, colour = "black", linetype = 2) + 
  xlim(-0.5,0.5) +  ylim(-2,2.3) +
  annotate(geom="text", x=.3, y=-2, label="IR up & expression down", color="darkgrey", size = 2.7)  + annotate(geom="text", x=.3, y=2.2, label="IR up & expression up", color="darkgrey", size = 2.7)  + 
  annotate(geom="text", x=-.3, y=2.2, label="IR down & expression up", color="#00A087FF", size = 2.7) +   annotate(geom="text", x=-.3, y=-2, label="IR down & expression down", color="darkgrey", size = 2.7)
ac_tyz_sod1_scatter

ac_tyz_sod1_vs_ctrl_filt <- ac_tyz_sod1_vs_ctrl %>% dplyr::filter(p0.05_reliable == "significant") %>%
  drop_na(IRratio.diff, log2FoldChange) %>% filter(IRratio.diff != Inf) %>% filter(IRratio.diff != -Inf) %>% filter(log2FoldChange != Inf) %>% filter(log2FoldChange != -Inf)
cor.test(x = ac_tyz_sod1_vs_ctrl_filt$IRratio.diff, y = ac_tyz_sod1_vs_ctrl_filt$log2FoldChange) # R = -0.09
summary( lm( IRratio.diff ~ log2FoldChange, data = ac_tyz_sod1_vs_ctrl_filt))

# 2G GO terms IR down & DGE up in C9ORF72
bir_c9orf_dge_ir_genes <- filter(ac_bir_c9orf_vs_ctrl, p0.05_reliable == "significant") %>%  split(.$direction) %>%  map("ensID") # create list of IR up/down & Expression up/down genes
bir_c9orf_dge_ir_genes <- gost(bir_c9orf_dge_ir_genes) # run gprofiler2 on all groups - runs on all KEGG, GO, REAC,TF, WP pathways simultaneously
bir_c9orf_dge_ir_gprofiles_filt <- bir_c9orf_dge_ir_genes$result %>% filter(str_detect(source, "KEGG|GO:BP|GO:MF|GO:CC|REAC|CORUM")) %>% arrange( -log10(p_value) ) %>% mutate( term_name = as.factor(term_name)) 
bir_c9orf_dge_ir_gprofiles_curated_irdown_expup <- bir_c9orf_dge_ir_gprofiles_filt %>% filter(query == "IR down & expression up")
irdown_expup_curated_list <- c("polysomal ribosome",
"actin binding",
# "extracellular matrix",
"cell morphogenesis involved in differentiation",
"translation",
"secretory granule",
# "Collagen formation",
"response to wounding",
# "cadherin binding",
"collagen-containing extracellular matrix",
"cell migration",
"cell adhesion",
"RNA binding",
"response to endoplasmic reticulum stress",
# "platelet degranulation",
"Extracellular matrix organization",
"cellular localization",
"integrin binding",
"anchoring junction",
# "extracellular vesicle",
# "cell-substrate junction",
"focal adhesion")    
bir_c9orf_dge_ir_gprofiles_curated_irdown_expup <- bir_c9orf_dge_ir_gprofiles_curated_irdown_expup %>% filter(term_name %in% irdown_expup_curated_list) %>% mutate(query = "")
bir_c9orf_dge_ir_gprofiles_curated_irdown_expup$term_name <- gsub("response to endoplasmic reticulum stress", "endoplasmic reticulum stress", bir_c9orf_dge_ir_gprofiles_curated_irdown_expup$term_name)
bir_c9orf_dge_ir_gprofiles_curated_irdown_expup$term_name <- gsub("cell morphogenesis involved in differentiation", "cell morphogenesis", bir_c9orf_dge_ir_gprofiles_curated_irdown_expup$term_name)
bir_c9orf_dge_ir_gprofiles_curated_irdown_expup$term_name <- gsub("collagen-containing extracellular matrix", "collagen extracellular matrix", bir_c9orf_dge_ir_gprofiles_curated_irdown_expup$term_name)
ac_bir_c9orf_go_curated <- curated_profiles(gprofiles = bir_c9orf_dge_ir_gprofiles_curated_irdown_expup, colours = "#4DBBD5FF") + ggtitle("C9ORF72")
ac_bir_c9orf_go_curated

# 2H GO terms IR down & DGE up in SOD1
tyz_sod1_dge_ir_genes <- filter(ac_tyz_sod1_vs_ctrl,p0.05_reliable == "significant") %>%  split(.$direction) %>%  map("ensID") # create list of IR up/down & Expression up/down genes
tyz_sod1_dge_ir_genes <- gost(tyz_sod1_dge_ir_genes) # run gprofiler2 on all groups - runs on all KEGG, GO, REAC,TF, WP pathways simultaneously
tyz_sod1_dge_ir_gprofiles_filt <- tyz_sod1_dge_ir_genes$result %>% filter(str_detect(source, "KEGG|GO:BP|GO:MF|GO:CC|REAC|CORUM")) %>% arrange( -log10(p_value) ) %>% mutate( term_name = as.factor(term_name)) 
tyz_sod1_dge_ir_gprofiles_curated_irdown_expup <- tyz_sod1_dge_ir_gprofiles_filt %>% filter(query == "IR down & expression up")
irdown_expup_curated_list <- c("NCAM1 interactions",
"HSF1 activation",
"integrin binding",
"Laminin interactions",
"extracellular vesicle",
"anchoring junction",
"Signaling by PDGF",
"viral transcription",
"Influenza Infection",
"Focal adhesion",
"Collagen formation",
"Nonsense-Mediated Decay (NMD)",
"Viral mRNA Translation",
"Extracellular matrix organization",
"cytosolic ribosome",
"Peptide chain elongation")       
tyz_sod1_dge_ir_gprofiles_curated_irdown_expup <- tyz_sod1_dge_ir_gprofiles_curated_irdown_expup %>% filter(term_name %in% irdown_expup_curated_list) %>% mutate(query = "")
ac_tyz_sod1_go_curated <- curated_profiles(gprofiles = tyz_sod1_dge_ir_gprofiles_curated_irdown_expup, colours = "#00A087FF") + ggtitle("SOD1")
ac_tyz_sod1_go_curated


# Of the Liddelow et al (25) astrocyte reactivity genes, we observed decreased IR and increased expression in 29.3% (12/41, enrichment p = 0.05) for C9orf72 mutants and 46.3% (19/41, enrichment p = 0.02) for SOD1 mutants (Fig. 2I-J)
ac_bir_c9orf72_vs_ctrl.IR_GE <- ac_bir_c9orf_vs_ctrl.absoluteIR %>% group_by(gene_name)  %>% dplyr::summarise(IR = mean(IRratio.diff, na.rm = TRUE), GE = median(log2FoldChange))
ac_bir_c9orf72_vs_ctrl.IR_GE %>% filter(gene_name %in% astrocyte.reactivity.markers$gene_name, IR < 0, GE > 0) # 12 / 41 liddelow genes have decreased IR & increased GE
12/41 * 100 # 29%
ac_bir_c9orf72_vs_ctrl.IR_GE[is.na(ac_bir_c9orf72_vs_ctrl.IR_GE)] <- 0
ac_bir_c9orf72_vs_ctrl.IR_GE <- ac_bir_c9orf72_vs_ctrl.IR_GE %>% data.frame(row.names = "gene_name")
ac_bir_c9orf72_vs_ctrl.IR_GE <- scale(ac_bir_c9orf72_vs_ctrl.IR_GE) # rescale both IR & GE to ensure they have the same variance
ac_bir_c9orf72_vs_ctrl.IR_GE <- ac_bir_c9orf72_vs_ctrl.IR_GE %>% as_tibble(rownames = "gene_name") %>% mutate("IR_DGE" = (GE + (IR*-1)/2))
ac_bir_c9orf72.geneList = ac_bir_c9orf72_vs_ctrl.IR_GE$IR_DGE
names(ac_bir_c9orf72.geneList) <- as.character(ac_bir_c9orf72_vs_ctrl.IR_GE$gene_name) # name vector with gene_name
ac_bir_c9orf72.geneList = sort(ac_bir_c9orf72.geneList, decreasing = TRUE)
liddelow_genes.list <- list("Astrocyte_reactivity" = astrocyte.reactivity.markers$gene_name[astrocyte.reactivity.markers$gene_name %in% ac_bir_c9orf_vs_ctrl.absoluteIR$gene_name]) # astrocyte reactivity genes
c9orf72.fgseaRes <- fgsea(pathways = liddelow_genes.list, stats = ac_bir_c9orf72.geneList)
c9orf72.fgseaRes # NES -1.40 p = 0.05


ac_tyz_sod1_vs_ctrl.IR_GE <- ac_tyz_sod1_vs_ctrl.absoluteIR %>% group_by(gene_name)  %>% dplyr::summarise(IR = mean(IRratio.diff), GE = median(log2FoldChange))
ac_tyz_sod1_vs_ctrl.IR_GE %>% filter(gene_name %in% astrocyte.reactivity.markers$gene_name, IR < 0, GE > 0) # 13 / 41 liddelow genes have decreased IR & increased GE
13/41 * 100 # 31.7%
ac_tyz_sod1_vs_ctrl.IR_GE[is.na(ac_tyz_sod1_vs_ctrl.IR_GE)] <- 0
ac_tyz_sod1_vs_ctrl.IR_GE <- ac_tyz_sod1_vs_ctrl.IR_GE %>% data.frame(row.names = "gene_name")
ac_tyz_sod1_vs_ctrl.IR_GE <- scale(ac_tyz_sod1_vs_ctrl.IR_GE) # rescale both IR & GE to ensure they have the same variance
ac_tyz_sod1_vs_ctrl.IR_GE <- ac_tyz_sod1_vs_ctrl.IR_GE %>% as_tibble(rownames = "gene_name") %>% mutate("IR_DGE" = (GE + (IR*-1)/2))
geneList = ac_tyz_sod1_vs_ctrl.IR_GE$IR_DGE
names(geneList) <- as.character(ac_tyz_sod1_vs_ctrl.IR_GE$gene_name)
geneList = sort(geneList, decreasing = TRUE)
liddelow_genes.list <- list("Astrocyte_reactivity" = astrocyte.reactivity.markers$gene_name[astrocyte.reactivity.markers$gene_name %in% ac_tyz_sod1_vs_ctrl.absoluteIR$gene_name]) # astrocyte reactivity genes
sod1.fgseaRes <- fgsea(pathways = liddelow_genes.list, 
                  stats    = geneList,
                  nperm = 10000)
sod1.fgseaRes # NES 1.51 p = 0.019


# 2K Liddelow heatmap DGE & IR in C9ORF72
c9orf_dge_ir_filt <- ac_bir_c9orf_vs_ctrl.absoluteIR %>% filter(gene_name %in% astrocyte.reactive.genes) %>% arrange(IR.lfc) %>% distinct(gene_name, .keep_all = TRUE) %>%
  dplyr::select(gene_name, "C9ORF72 IR" = IR.lfc, "C9ORF72 GE" = log2FoldChange)
c9orf_dge_ir_filt$"C9ORF72 IR" <- replace_na(c9orf_dge_ir_filt$"C9ORF72 IR", 0)
c9orf_dge_ir_filt$"C9ORF72 GE" <- replace_na(c9orf_dge_ir_filt$"C9ORF72 GE", 0)
c9orf_dge_ir_filt <- c9orf_dge_ir_filt %>% arrange(factor(gene_name, levels = rownames(annot))) # order the matrix by the annotation dataframe
c9orf_dge_ir_filt.mat <- make_matrix(select(c9orf_dge_ir_filt,-gene_name),pull(c9orf_dge_ir_filt,gene_name))
c9orf_dge_ir_filt.mat[c9orf_dge_ir_filt.mat == -Inf] <- -2.0
c9orf_dge_ir_filt.mat[c9orf_dge_ir_filt.mat == Inf] <- 2.0
c9orf_dge_ir_filt.mat[c9orf_dge_ir_filt.mat == NA] <- 0
c9orf_dge_ir_filt.mat[c9orf_dge_ir_filt.mat > 2.0] <- 2.0
c9orf_dge_ir_filt.mat[c9orf_dge_ir_filt.mat < -2.0] <- -2.0
ac_bir_c9orf_vs_ctrl.all <- cleanIRFinder(IRFinder = ac_bir_c9orf_vs_ctrl.IRFinder) %>% full_join(dge_res_ac_bir_c9orf_vs_ctrl, by = c("ensID", "gene_name")) # FULL join deseq2 to avoid loosing DGE values with no IR values
bir_dge_res_padj <- ac_bir_c9orf_vs_ctrl.all %>% filter(gene_name %in% astrocyte.reactive.genes) %>% arrange(IR.lfc) %>% distinct(gene_name, .keep_all = TRUE) %>%
  mutate(IR.sig = case_when(p0.05_reliable == "significant" ~ "\u2217", TRUE ~ ""), DGE.sig = case_when(padj <= 0.05 ~  "\u2217", TRUE ~ "")) %>%
  dplyr::select(gene_name, IR.sig, DGE.sig) %>%
  arrange(factor(gene_name, levels = rownames(annot))) %>% dplyr::select(IR.sig,DGE.sig) %>%  as.matrix  # order the matrix by the annotation dataframe
ac_c9orf_liddelow_heatmap <- pheatmap(t(c9orf_dge_ir_filt.mat), cluster_rows=F, show_rownames=T, show_colnames = T, cluster_cols=F, fontsize = 8, gaps_col = c(13,29), gaps_row = 1,
                  annotation_col = annot, annotation_names_col = F, annotation_legend = FALSE, annotation_colors = annot_cols, main = "", display_numbers = t(bir_dge_res_padj), 
                  color = colorRampPalette(rev(greenred(75)))(length(breaksList)),breaks = breaksList, 
                  cellheight = 9, cellwidth = 15,
                  border_color=NA, legend = FALSE)

# 2L Liddelow heatmap DGE & IR in SOD1
sod1_dge_ir_filt <- ac_tyz_sod1_vs_ctrl.absoluteIR %>% filter(gene_name %in% astrocyte.reactive.genes) %>% arrange(IR.lfc) %>% distinct(gene_name, .keep_all = TRUE) %>%
  dplyr::select(gene_name, "SOD1 IR" = IR.lfc, "SOD1 GE" = log2FoldChange)
sod1_dge_ir_filt$"SOD1 IR" <- replace_na(sod1_dge_ir_filt$"SOD1 IR", 0)
sod1_dge_ir_filt$"SOD1 GE" <- replace_na(sod1_dge_ir_filt$"SOD1 GE", 0)
sod1_dge_ir_filt <- sod1_dge_ir_filt %>% arrange(factor(gene_name, levels = rownames(annot))) # order the matrix by the annotation dataframe
sod1_dge_ir_filt.mat <- make_matrix(select(sod1_dge_ir_filt,-gene_name),pull(sod1_dge_ir_filt,gene_name))
sod1_dge_ir_filt.mat[sod1_dge_ir_filt.mat == -Inf] <- -2.0
sod1_dge_ir_filt.mat[sod1_dge_ir_filt.mat == Inf] <- 2.0
sod1_dge_ir_filt.mat[sod1_dge_ir_filt.mat == NA] <- 0
sod1_dge_ir_filt.mat[sod1_dge_ir_filt.mat > 2.0] <- 2.0
sod1_dge_ir_filt.mat[sod1_dge_ir_filt.mat < -2.0] <- -2.0
ac_tyz_sod1_vs_ctrl.all <- cleanIRFinder(IRFinder = ac_tyz_sod1_vs_ctrl.IRFinder) %>% full_join(dge_res_ac_tyz_sod1_vs_ctrl, by = c("ensID", "gene_name")) # FULL join deseq2 to avoid loosing DGE values with no IR values
tyz_dge_res_padj <- ac_tyz_sod1_vs_ctrl.all %>% filter(gene_name %in% astrocyte.reactive.genes) %>% arrange(IR.lfc) %>% distinct(gene_name, .keep_all = TRUE) %>%
  mutate(IR.sig = case_when(p0.05_reliable == "significant" ~ "\u2217", TRUE ~ ""), DGE.sig = case_when(padj <= 0.05 ~  "\u2217", TRUE ~ "")) %>%
  dplyr::select(gene_name, IR.sig, DGE.sig) %>%
  arrange(factor(gene_name, levels = rownames(annot))) %>% dplyr::select(IR.sig,DGE.sig) %>%  as.matrix  # order the matrix by the annotation dataframe
ac_sod1_liddelow_heatmap <- pheatmap(t(sod1_dge_ir_filt.mat), cluster_rows=F, show_rownames=T, show_colnames = T, cluster_cols=F, fontsize = 8,
                  gaps_col = c(13,29), gaps_row = 1,
                  annotation_col = annot, annotation_names_col = F, annotation_legend = FALSE, annotation_colors = annot_cols, main = "", display_numbers = t(tyz_dge_res_padj), 
                  color = colorRampPalette(rev(greenred(75)))(length(breaksList)),breaks = breaksList,
                  cellheight = 9, cellwidth = 15,
                  border_color=NA, legend = FALSE)
```

## Merge

```{r}
fig2.merged <- ggarrange(
  ggarrange(vcp_sod1_c9orf_dge_ir_sina_plot, ac_who_dge_ir_scatter, ncol = 2, labels = c("A", "B"), widths = c(1,1)),
  ggarrange(ac_bir_c9orf_scatter, ac_tyz_sod1_scatter, ncol = 2, labels = c("C", "D"), widths = c(1,1)),
  ggarrange(NULL, ac_who_vcp_vs_ctrl.go_curated, ac_bir_c9orf_go_curated, ac_tyz_sod1_go_curated, ncol = 4, labels = c("", "E", "F", "G"), widths = c(0.05,1,1,1)),
  NULL,
  ggarrange(
    annotate_figure(ac_who_liddelow_heatmap[[4]],  bottom = text_grob("Pan-reactive                                                  A1                                                             A2", color = "black", hjust = 0.6, vjust = -10.1, x = 0.5, face = "italic", size = 10)),
    annotate_figure(ac_c9orf_liddelow_heatmap[[4]],  bottom = text_grob("Pan-reactive                                                   A1                                                             A2", color = "black", hjust = 0.6, vjust = -10.1, x = 0.5, face = "italic", size = 10)),
    annotate_figure(ac_sod1_liddelow_heatmap[[4]],  bottom = text_grob("Pan-reactive                                                   A1                                                             A2", color = "black", hjust = 0.6, vjust = -10.1, x = 0.5, face = "italic", size = 10)), 
    labels = c("H", "I", "J"), nrow = 3, heights = c(1,1,1)),
  nrow = 5, heights = c(1.2,1.2,1,0.02,1.2)) # rows 5-7
fig2.merged
ggsave(fig2.merged, filename = "/camp/home/ziffo/home/projects/astrocyte-meta-analysis/splicing/IRFinder/irlib/figures/fig2.merged.annot.png", height = 12.5, width = 10)
```

## Coverage tracks IR down expression up

Insert these to the bottom of the figure in Illustrator.

```{r}
# Coverage plots for MCAM, RECK, UBN1, LOXL3, TGFB1I1, COL7A1
txdb <- makeTxDbFromGFF("/camp/home/ziffo/home/genomes/ensembl/Homo_sapiens.GRCh38.99.chr_patch_hapl_scaff.gtf", format="gtf") # make txdb from GTF - turn on only when needed
seqlevels(txdb, pruning.mode="coarse") <- c("1", "2", "3", "4", "5", "6", "7", "8", "9", "10", "11", "12", "13", "14", "15", "16", "17", "18", "19", "20", "21", "22", "X", "Y") # removes all the other chromsome annotations
ac_who_ctrl.bam <- "/camp/home/ziffo/home/projects/astrocyte-meta-analysis/alignment/STAR/merged/ac_who_ctrl.bam"
ac_who_vcp.bam <- "/camp/home/ziffo/home/projects/astrocyte-meta-analysis/alignment/STAR/merged/ac_who_vcp.bam"

# Top 4 reliable delta IR events
ac_who_vcp_vs_ctrl_top10events <- ac_who_vcp_vs_ctrl %>% filter(reliable == "reliable", direction == "IR down & expression up", pvalue < 0.05)  %>% dplyr::select(gene_name, ensemblID = ensID, Chr, Start, End, Strand = Direction, length = intron_length, IRratio.diff, coverage = IR.coverage, p.value = p.diff, IR.DGE.direction = direction, log2FoldChange, padj, pvalue) %>% arrange(IRratio.diff) %>% top_n(wt = abs(IRratio.diff), n = 10)
ac_who_vcp_vs_ctrl_top10events

# ERAP2, PTPRN, IL20RB, and KCNE4
track_viewer_tx_top_IRratio_event(gene_test = "ERAP2", txt = ac_who_vcp_vs_ctrl, mut.bam = ac_who_vcp.bam, ctrl.bam = ac_who_ctrl.bam)
track_viewer_tx_top_IRratio_event(gene_test = "PTPRN", txt = ac_who_vcp_vs_ctrl, mut.bam = ac_who_vcp.bam, ctrl.bam = ac_who_ctrl.bam)
# track_viewer_tx_top_IRratio_event(gene_test = "IL20RB", txt = ac_who_vcp_vs_ctrl, mut.bam = ac_who_vcp.bam, ctrl.bam = ac_who_ctrl.bam)
# track_viewer_tx_top_IRratio_event(gene_test = "KCNE4", txt = ac_who_vcp_vs_ctrl, mut.bam = ac_who_vcp.bam, ctrl.bam = ac_who_ctrl.bam)

# supp
# track_viewer_tx_top_IRratio_event(gene_test = "CUTALP", txt = ac_who_vcp_vs_ctrl, mut.bam = ac_who_vcp.bam, ctrl.bam = ac_who_ctrl.bam)
# track_viewer_tx_top_IRratio_event(gene_test = "SRPK3", txt = ac_who_vcp_vs_ctrl, mut.bam = ac_who_vcp.bam, ctrl.bam = ac_who_ctrl.bam)
# track_viewer_tx_top_IRratio_event(gene_test = "DNHD1", txt = ac_who_vcp_vs_ctrl, mut.bam = ac_who_vcp.bam, ctrl.bam = ac_who_ctrl.bam)
# track_viewer_tx_top_IRratio_event(gene_test = "TFAP2E", txt = ac_who_vcp_vs_ctrl, mut.bam = ac_who_vcp.bam, ctrl.bam = ac_who_ctrl.bam)
# track_viewer_tx_top_IRratio_event(gene_test = "RND3", txt = ac_who_vcp_vs_ctrl, mut.bam = ac_who_vcp.bam, ctrl.bam = ac_who_ctrl.bam)
# track_viewer_tx_top_IRratio_event(gene_test = "PIMREG", txt = ac_who_vcp_vs_ctrl, mut.bam = ac_who_vcp.bam, ctrl.bam = ac_who_ctrl.bam)
# track_viewer_tx_top_IRratio_event(gene_test = "CCDC57", txt = ac_who_vcp_vs_ctrl, mut.bam = ac_who_vcp.bam, ctrl.bam = ac_who_ctrl.bam)
# track_viewer_tx_top_IRratio_event(gene_test = "PER1", txt = ac_who_vcp_vs_ctrl, mut.bam = ac_who_vcp.bam, ctrl.bam = ac_who_ctrl.bam)
# track_viewer_tx_top_IRratio_event(gene_test = "REC8", txt = ac_who_vcp_vs_ctrl, mut.bam = ac_who_vcp.bam, ctrl.bam = ac_who_ctrl.bam)
# track_viewer_tx_top_IRratio_event(gene_test = "MFSD8", txt = ac_who_vcp_vs_ctrl, mut.bam = ac_who_vcp.bam, ctrl.bam = ac_who_ctrl.bam)
# track_viewer_tx(gene_test = "RECK", txt = ac_who_vcp_vs_ctrl, mut.bam = ac_who_vcp.bam, ctrl.bam = ac_who_ctrl.bam)
# track_viewer_tx(gene_test = "LOXL3", txt = ac_who_vcp_vs_ctrl, mut.bam = ac_who_vcp.bam, ctrl.bam = ac_who_ctrl.bam)
# track_viewer_tx(gene_test = "STX4", txt = ac_who_vcp_vs_ctrl, mut.bam = ac_who_vcp.bam, ctrl.bam = ac_who_ctrl.bam)
# track_viewer_tx(gene_test = "COL7A1", txt = ac_who_vcp_vs_ctrl, mut.bam = ac_who_vcp.bam, ctrl.bam = ac_who_ctrl.bam)
# track_viewer_tx(gene_test = "TGFB1I1", txt = ac_who_vcp_vs_ctrl, mut.bam = ac_who_vcp.bam, ctrl.bam = ac_who_ctrl.bam)
# track_viewer_tx(gene_test = "TEP1", txt = ac_who_vcp_vs_ctrl, mut.bam = ac_who_vcp.bam, ctrl.bam = ac_who_ctrl.bam)
# track_viewer_tx(gene_test = "NFKB2", txt = ac_who_vcp_vs_ctrl, mut.bam = ac_who_vcp.bam, ctrl.bam = ac_who_ctrl.bam)
# track_viewer_tx(gene_test = "FBXL8", txt = ac_who_vcp_vs_ctrl, mut.bam = ac_who_vcp.bam, ctrl.bam = ac_who_ctrl.bam)
```


# Figure 3: Cytokine stimulated & TDP43 deletion

BioRender schematics inserted in illustrator at top of figure

## Cytokine stimulated astrocytes

```{r}
ac_bar_a1_vs_a0.absoluteIR %>% dplyr::filter(A.reliable_retained == "reliable-retained" | B.reliable_retained == "reliable-retained") %>% nrow # 3164 reliable unique retained introns 
ac_bar_a1_vs_a0.absoluteIR %>% dplyr::filter(A.reliable_retained == "reliable-retained") %>% nrow  # 2143 reliable unique retained introns in VCP
ac_bar_a1_vs_a0.absoluteIR %>% dplyr::filter(B.reliable_retained == "reliable-retained") %>% nrow  # 2402 reliable unique retained introns in control
(2402-2143)/2402 * 100 # 10.8% fewer retained introns in VCP compared with control 
ac_bar_a1_vs_a0 %>% dplyr::filter(p0.05_reliable == "significant") %>% nrow # 578 reliable introns retained at significantly different levels
ac_bar_a1_vs_a0 %>% dplyr::filter(p0.05_reliable == "significant") %>% distinct(gene_name) %>% nrow # from 430 unique genes
ac_bar_a1_vs_a0 %>% dplyr::filter(p0.05_reliable == "significant", IRratio.diff < 0) %>% nrow # 304 displaying decreased IR levels in VCP
(578-(578-304))/578 * 100 # 52.6% decreased in VCP

# Of the 240 genes with decreased IR in stimulated reactive astrocytes, 86 (35.8%) also exhibited decreased IR in the ALS astrocytes (overlap p = 3.8x10-71)
IRdown_VCP <- ac_who_vcp_vs_ctrl %>% filter(p0.05_reliable == "significant" & IRdirection == "down") %>% pull(gene_name) %>% unique
IRdown_SOD1 <- ac_tyz_sod1_vs_ctrl %>% filter(p0.05_reliable == "significant" & IRdirection == "down") %>% pull(gene_name) %>% unique
IRdown_C9ORF72 <- ac_bir_c9orf_vs_ctrl %>% filter(p0.05_reliable == "significant" & IRdirection == "down") %>% pull(gene_name) %>% unique
IRdown_ALS <- unique(c(IRdown_VCP,IRdown_SOD1, IRdown_C9ORF72)) # 834 genes
IRdown_A1 <- ac_bar_a1_vs_a0 %>% filter(p0.05_reliable == "significant" & IRdirection == "down") %>% pull(gene_name) %>% unique # 189 genes
IRdown_A1[IRdown_A1 %in% IRdown_ALS] # 86 / 240 (35.8%)

IRdown_A1_ALS <- IRdown_A1[IRdown_A1 %in% IRdown_ALS] 
IRdown_A1_ALS[IRdown_A1_ALS %in% all.reactivity.go] # "HSPG2"   "PSMD11"  "NFKBIZ"  "DNAJC2"  "ERF"     "TINF2"   "OSMR"    "TNIK"    "ILK"     "NUP188"  "STIP1"   "TBK1"    "MCAM"    "ARHGEF2" "NT5C3A"  "FN1"     "POLR3C" 

go.obj <- newGeneOverlap(IRdown_ALS,
                         IRdown_A1,
                         genome.size=nrow(distinct(gtf, gene_name)))
testGeneOverlap(go.obj)
# GeneOverlap object:
# listA size=1654
# listB size=240
# Intersection size=86
# Overlapping p-value=3.8e-71
# Jaccard Index=0.0


a1 <- ac_bar_a1_vs_a0 %>% dplyr::filter(p0.05_reliable == "significant") %>% dplyr::select(Intron.GeneName.GeneID, Chr, Start, End, gene_name, ensID, p.diff, IRratio.diff, p0.05_reliable_IRdirection, log2FoldChange) %>% mutate(mutation = "Stimulated - CTRL")
tdp43 <- ac_tdp_ko_vs_ctrl %>% dplyr::filter(p0.05_reliable == "significant") %>% dplyr::select(Intron.GeneName.GeneID, Chr, Start, End, gene_name, ensID, p.diff, IRratio.diff, p0.05_reliable_IRdirection, log2FoldChange) %>% mutate(mutation = "TDP43 KO - CTRL")
a1_tdp43 <- bind_rows(a1, tdp43) %>% mutate(mutation = factor(mutation, levels = c("Stimulated - CTRL", "TDP43 KO - CTRL")))

# Sina plot
a1_tdp43_sina_plot <- sinaplot.one.sample(data = a1_tdp43, groups = "mutation", continuous = "IRratio.diff", labels = "gene_name", label_number = 0, stat.y.position = 0.2, width = 0.15, flip = "no", colours = 2) + ylab(label = expression(Delta~IR~ratio) ) + xlab("") + ylim(c(-0.2,0.2)) +
  geom_segment(aes(x = 2.5, xend = 2.5, y= 0.05, yend= 0.2), arrow=arrow(length=unit(0.4,"cm")), color = "dodgerblue") +
  geom_segment(aes(x = 2.5, xend = 2.5, y= -0.05, yend= -0.2),arrow=arrow(length=unit(0.4,"cm")), color = "firebrick2") +
  annotate("text", x = 2.3, y = 0.17, label = "IR Up", colour = "dodgerblue", size = 3) +   annotate("text", x = 2.3, y = -0.17, label = "IR Down", colour = "firebrick2", size = 3)
a1_tdp43_sina_plot

# [1] "Stats test used =  t_test"
# # A tibble: 2 x 10
#   group             .y.   group1 group2         n statistic    df        p p.signif y.position
#   <fct>             <chr> <chr>  <chr>      <int>     <dbl> <dbl>    <dbl> <chr>         <dbl>
# 1 Stimulated - CTRL lfc   1      null model   578      3.27   577 1.14e- 3 **              0.2
# 2 TDP43 KO - CTRL   lfc   1      null model   700     -9.63   699 1.10e-20 ****            0.2

# Sina violin plot: LFC gene expression for IR up & down genes
a1_tdp43_dge_ir_sina_plot <- ggplot(data = a1_tdp43, aes(x = p0.05_reliable_IRdirection, y= log2FoldChange, colour = p0.05_reliable_IRdirection)) + 
          geom_violin() + geom_sina(size = 0.5) + geom_boxplot(data = a1_tdp43, aes(fill = p0.05_reliable_IRdirection), colour = "black", width=0.2, outlier.shape = NA) +
          scale_colour_manual(values = c("dodgerblue3", "firebrick2", "forestgreen")) + scale_fill_manual(values = c("dodgerblue3", "firebrick2", "forestgreen")) +
          theme_classic() + theme(legend.position = "none", axis.text=element_text(size=10)) +
          geom_hline(yintercept = 0, linetype = 2) + facet_grid(. ~ mutation) +
          stat_compare_means(comparisons = list(c("IR down", "IR up")), method = "t.test", label = "p.signif", label.y = 1.9, tip.length = 0.01)  + ylim(c(-2,2)) + ylab(label = expression(Gene~exp~log[2]~fold~change) ) + xlab("")
a1_tdp43_dge_ir_sina_plot

# show p-value labels
 ggplot(data = a1_tdp43, aes(x = p0.05_reliable_IRdirection, y= log2FoldChange, colour = p0.05_reliable_IRdirection)) + 
          geom_violin() + geom_sina(size = 0.5) + geom_boxplot(data = a1_tdp43, aes(fill = p0.05_reliable_IRdirection), colour = "black", width=0.2, outlier.shape = NA) +
          scale_colour_manual(values = c("dodgerblue3", "firebrick2", "forestgreen")) + scale_fill_manual(values = c("dodgerblue3", "firebrick2", "forestgreen")) +
          theme_classic() + theme(legend.position = "none", axis.text=element_text(size=10)) +
          geom_hline(yintercept = 0, linetype = 2) + facet_grid(. ~ mutation) +
          stat_compare_means(comparisons = list(c("IR down", "IR up")), method = "t.test", label = "p.label", label.y = 1.9, tip.length = 0.01)  + ylim(c(-2,2)) + ylab(label = expression(Gene~exp~log[2]~fold~change) ) + xlab("")
 # cytokine stimulated < 2.2e-16
 # TDP43 ko 0.012

# GO terms IR down & DGE up in Stimulated
bar_a1_dge_ir_genes <- filter(ac_bar_a1_vs_a0, p0.05_reliable == "significant") %>%  split(.$direction) %>%  map("ensID") # create list of IR up/down & Expression up/down genes
bar_a1_dge_ir_genes <- gost(bar_a1_dge_ir_genes) # run gprofiler2 on all groups - runs on all KEGG, GO, REAC,TF, WP pathways simultaneously
bar_a1_dge_ir_gprofiles_filt <- bar_a1_dge_ir_genes$result %>% filter(str_detect(source, "KEGG|GO:BP|GO:MF|GO:CC|REAC|CORUM")) %>% arrange( -log10(p_value) ) %>% mutate( term_name = as.factor(term_name)) 
bar_a1_dge_ir_gprofiles_curated_irdown_expup <- bar_a1_dge_ir_gprofiles_filt %>% filter(query == "IR down & expression up")
irdown_expup_curated_list <- c("TANGO1L-cTAGE5-SEC12 complex",
"cellular metabolic process",
"protein-containing complex",
"protein phosphorylation",
"macromolecule modification",
"regulation of phosphorylation",
"nucleus",
"Metabolism of RNA",
"nuclear lumen",
"cytoplasm", "nucleoplasm")
bar_a1_dge_ir_gprofiles_curated_irdown_expup <- bar_a1_dge_ir_gprofiles_filt %>% filter(query == "IR down & expression up") %>% filter(term_name %in% irdown_expup_curated_list) %>% mutate(query = "")
bar_a1_dge_ir_gprofiles_curated_irdown_expup$term_name <- gsub("TANGO1L-cTAGE5-SEC12", "collagen cargo", bar_a1_dge_ir_gprofiles_curated_irdown_expup$term_name)
ac_bar_a1_go_curated <- curated_profiles(gprofiles = bar_a1_dge_ir_gprofiles_curated_irdown_expup, colours = "dodgerblue3") + labs(subtitle = "Cytokine stimulated")
ac_bar_a1_go_curated
 
# Cytokine stimulated Scatter plot: Delta IR vs DGE LFC
labels_lenient <- filter(ac_bar_a1_vs_a0, p0.05_reliable == "significant") %>% filter(gene_name %in% astrocyte.reactivity.labels.ac_a1) %>% group_by(gene_name) %>% slice_max(abs(IRratio.diff), n = 1, with_ties = FALSE)
ac_bar_a1_scatter <- ggplot(filter(ac_bar_a1_vs_a0, p0.05_reliable == "significant"), aes( x = IRratio.diff, y = log2FoldChange,  label = gene_name, colour = log10(baseMean))) + 
  geom_point(alpha = 0.5, size = 0.5) +  theme_bw() +  xlab( expression( Delta~IR~ratio~Stimulated-CTRL)  ) +  ylab( expression(Gene~exp~Stimulated:CTRL) ) +
  scale_colour_gradient(low = "dodgerblue3", high = "firebrick2") +
  geom_text_repel(data = labels_lenient, aes(x = IRratio.diff, y = log2FoldChange, label = gene_name), fontface = "italic", colour = "black", min.segment.length = 0, box.padding = 0.5, point.padding = 0.5, size = 2.5) +
  theme(strip.background = element_rect(colour="black", fill="white"), panel.grid = element_blank(), legend.position = "none", legend.box.margin=margin(-10,-10,-10,-10),legend.spacing.x = unit(1.0, 'cm'), axis.text=element_text(size=10)) + labs(colour = expression(log[10]~expression)) +
  geom_hline(yintercept = 0, linetype = 3, colour = "darkgrey") + geom_vline(xintercept = 0, linetype = 3, colour = "darkgrey") + 
    geom_smooth(data = filter(ac_bar_a1_vs_a0, p0.05_reliable == "significant"), aes(x = IRratio.diff, y = log2FoldChange), method = "lm", se = FALSE, show.legend = TRUE, colour = "black", linetype = 2) + 
  xlim(-0.36,0.36) +  ylim(-2.5,6.0) +
  annotate(geom="text", x=.2, y=-2, label="IR up & expression down", color="darkgrey", size = 2.7)  + annotate(geom="text", x=.2, y=5.7, label="IR up & expression up", color="darkgrey", size = 2.7)  + 
  annotate(geom="text", x=-.2, y=5.7, label="IR down & expression up", color="#3C5488FF", size = 2.7) +   annotate(geom="text", x=-.2, y=-2, label="IR down & expression down", color="darkgrey", size = 2.7)
ac_bar_a1_scatter

ac_bar_a1_vs_a0_filt <- ac_bar_a1_vs_a0 %>% dplyr::filter(p0.05_reliable == "significant") %>%
  drop_na(IRratio.diff, log2FoldChange) %>% filter(IRratio.diff != Inf) %>% filter(IRratio.diff != -Inf) %>% filter(log2FoldChange != Inf) %>% filter(log2FoldChange != -Inf)
cor.test(x = ac_bar_a1_vs_a0_filt$IRratio.diff, y = ac_bar_a1_vs_a0_filt$log2FoldChange) # R = -0.34
summary( lm( IRratio.diff ~ log2FoldChange, data = ac_bar_a1_vs_a0_filt))


# Annotation for all Liddelow heatmaps
annot <- data.frame(row.names = astrocyte.reactivity.markers$gene_name, group = factor(astrocyte.reactivity.markers$group))
annot_cols <- c("black", "firebrick2", "dodgerblue3")
names(annot_cols) <- unique(astrocyte.reactivity.markers$group)
annot_cols <- list(group = annot_cols)
breaksList = seq(-2, 2, by = 0.1) 

# Liddelow heatmap DGE & IR in Cytokine stimulated
a1_dge_ir_filt <- ac_bar_a1_vs_a0.absoluteIR %>% filter(gene_name %in% astrocyte.reactive.genes) %>% arrange(IR.lfc) %>% distinct(gene_name, .keep_all = TRUE) %>%
  dplyr::select(gene_name, "Stimulated IR" = IR.lfc, "Stimulated GE" = log2FoldChange)
a1_dge_ir_filt$"Stimulated IR" <- replace_na(a1_dge_ir_filt$"Stimulated IR", 0)
a1_dge_ir_filt$"Stimulated GE" <- replace_na(a1_dge_ir_filt$"Stimulated GE", 0)
a1_dge_ir_filt <- a1_dge_ir_filt %>% arrange(factor(gene_name, levels = rownames(annot))) # order the matrix by the annotation dataframe
a1_dge_ir_filt.mat <- make_matrix(select(a1_dge_ir_filt,-gene_name),pull(a1_dge_ir_filt,gene_name))
a1_dge_ir_filt.mat[a1_dge_ir_filt.mat == -Inf] <- -2.0
a1_dge_ir_filt.mat[a1_dge_ir_filt.mat == Inf] <- 2.0
a1_dge_ir_filt.mat[a1_dge_ir_filt.mat == NA] <- 0
a1_dge_ir_filt.mat[a1_dge_ir_filt.mat > 2.0] <- 2.0
a1_dge_ir_filt.mat[a1_dge_ir_filt.mat < -2.0] <- -2.0
ac_bar_a1_vs_a0.all <- cleanIRFinder(IRFinder = ac_bar_a1_vs_a0.IRFinder) %>% full_join(dge_res_ac_bar_a1_vs_a0, by = c("ensID", "gene_name")) # FULL join deseq2 to avoid loosing DGE values with no IR values
dge_res_padj <- ac_bar_a1_vs_a0.all %>% filter(gene_name %in% astrocyte.reactive.genes) %>% arrange(IR.lfc) %>% distinct(gene_name, .keep_all = TRUE) %>%
  mutate(IR.sig = case_when(p0.05_reliable == "significant" ~ "\u2217", TRUE ~ ""), DGE.sig = case_when(padj <= 0.05 ~  "\u2217", TRUE ~ "")) %>%
  dplyr::select(gene_name, IR.sig, DGE.sig) %>%
  arrange(factor(gene_name, levels = rownames(annot))) %>% dplyr::select(IR.sig,DGE.sig) %>%  as.matrix 
ac_a1_liddelow_heatmap <- pheatmap(t(a1_dge_ir_filt.mat), cluster_rows=F, show_rownames=T, show_colnames = T, cluster_cols=F, fontsize = 8,
                  gaps_col = c(13,29), gaps_row = 1, 
                  annotation_col = annot, annotation_names_col = F, annotation_legend = FALSE, annotation_colors = annot_cols, main = "", display_numbers = t(dge_res_padj), 
                  color = colorRampPalette(rev(greenred(75)))(length(breaksList)),breaks = breaksList,
                  cellheight = 9, cellwidth = 14, border_color=NA, legend = TRUE)

#  Liddelow reactivity genes were strongly upregulated in expression across pan-reactive, A1, and A2 subgroups and were decreased in IR predominantly in A1 and A2 markers (enrichment p = 0.0002)
ac_bar_a1_vs_a0.IR_GE <- ac_bar_a1_vs_a0.absoluteIR %>% group_by(gene_name) %>% dplyr::summarise(IR.mean = mean(IRratio.diff, na.rm = TRUE), GE.lfc = median(log2FoldChange))
ac_bar_a1_vs_a0.IR_GE[is.na(ac_bar_a1_vs_a0.IR_GE)] <- 0
ac_bar_a1_vs_a0.IR_GE <- ac_bar_a1_vs_a0.IR_GE %>% data.frame(row.names = "gene_name")
ac_bar_a1_vs_a0.IR_GE <- scale(ac_bar_a1_vs_a0.IR_GE) # rescale both IR & GE to ensure they have the same variance
ac_bar_a1_vs_a0.IR_GE <- ac_bar_a1_vs_a0.IR_GE %>% as_tibble(rownames = "gene_name") %>% mutate("IR_DGE" = (GE.lfc + (IR.mean*-1)/2))
geneList = ac_bar_a1_vs_a0.IR_GE$IR_DGE
names(geneList) <- as.character(ac_bar_a1_vs_a0.IR_GE$gene_name)
geneList = sort(geneList, decreasing = TRUE)
liddelow_genes.list <- list("Astrocyte_reactivity" = astrocyte.reactivity.markers$gene_name[astrocyte.reactivity.markers$gene_name %in% ac_bar_a1_vs_a0.absoluteIR$gene_name]) # astrocyte reactivity gene set vector
a1.fgseaRes <- fgsea(pathways = liddelow_genes.list, stats    = geneList)
a1.fgseaRes # 9.52e-6

# Comparing genes with decreased IR and increased expression between the A1 and ALS astrocytes revealed an overlap of 45 / 189 (23.8%), of which 17 are directly relevant to astrocyte reactivity
IRdown_DGEup_VCP <- ac_who_vcp_vs_ctrl %>% filter(p0.05_reliable == "significant" & direction == "IR down & expression up") %>% pull(gene_name) %>% unique
IRdown_DGEup_SOD1 <- ac_tyz_sod1_vs_ctrl %>% filter(p0.05_reliable == "significant" & direction == "IR down & expression up") %>% pull(gene_name) %>% unique
IRdown_DGEup_C9ORF72 <- ac_bir_c9orf_vs_ctrl %>% filter(p0.05_reliable == "significant" & direction == "IR down & expression up") %>% pull(gene_name) %>% unique
IRdown_DGEup_ALS <- unique(c(IRdown_DGEup_VCP,IRdown_DGEup_SOD1, IRdown_DGEup_C9ORF72)) # 834 genes
IRdown_DGEup_A1 <- ac_bar_a1_vs_a0 %>% filter(p0.05_reliable == "significant" & direction == "IR down & expression up") %>% pull(gene_name) %>% unique # 189 genes
IRdown_DGEup_A1[IRdown_DGEup_A1 %in% IRdown_DGEup_ALS] # 45 / 189 (23.8%)
IRdown_DGEup_A1_ALS <- IRdown_DGEup_A1[IRdown_DGEup_A1 %in% IRdown_DGEup_ALS] 
IRdown_DGEup_A1_ALS[IRdown_DGEup_A1_ALS %in% all.reactivity.go] 
# [1] "ITPR3"   "HSPG2"   "PSMD11"  "ERF"     "PKD1"    "TINF2"   "DGKQ"    "OSMR"    "ILK"     "NUP188" "IL11"    "CCNF"    "TBK1"    "MCAM"    "ARHGEF2" "FN1"     "SIK3"   
```

## TDP43 deleted astrocytes

```{r}
 # TARDBP iCLIP
clip_k562 <- list.files("/camp/home/ziffo/home/genomes/clip-bedgraphs/K562/")
clip_k562 <- clip_k562[!grepl("3nt_peaks",clip_k562)] # only assess raw crosslink files without any peak calling for enrichment
VCP <- ac_who_vcp_vs_ctrl %>% dplyr::select(Intron.GeneName.GeneID.Coords, coords, ensID, gene_name, Chr, Start, End, Direction, gc_content, intron_length, p0.05_reliable, VCP.p0.05_reliable_IRdirection = p0.05_reliable_IRdirection) %>% filter(p0.05_reliable == "significant")
SOD1 <- ac_tyz_sod1_vs_ctrl %>% dplyr::select(Intron.GeneName.GeneID.Coords, coords, ensID, gene_name, Chr, Start, End, Direction,p0.05_reliable, SOD1.p0.05_reliable_IRdirection = p0.05_reliable_IRdirection, gc_content, intron_length)  %>% filter(p0.05_reliable == "significant")
C9ORF72 <- ac_bir_c9orf_vs_ctrl %>% dplyr::select(Intron.GeneName.GeneID.Coords, coords, ensID, gene_name, Chr, Start, End, Direction,p0.05_reliable, C9ORF72.p0.05_reliable_IRdirection = p0.05_reliable_IRdirection, gc_content, intron_length) %>% filter(p0.05_reliable == "significant")
differential.3datasets.IR <- VCP %>% full_join(SOD1, by = c("Intron.GeneName.GeneID.Coords", "coords", "ensID", "gene_name", "Chr", "Start", "End", "Direction", "gc_content", "intron_length")) %>%
  full_join(C9ORF72, by = c("Intron.GeneName.GeneID.Coords", "coords", "ensID", "gene_name", "Chr", "Start", "End", "Direction", "gc_content", "intron_length")) %>%
  mutate(p0.05_reliable_IRdirection = case_when(VCP.p0.05_reliable_IRdirection == "IR down" | SOD1.p0.05_reliable_IRdirection == "IR down" | C9ORF72.p0.05_reliable_IRdirection == "IR down" ~ "IR down", TRUE ~ "IR up"))
grange_introns <- as.GenomicRange(differential.3datasets.IR$Intron.GeneName.GeneID.Coords) # create GRanges for each intron (both retained and non retained)
intron_coords <- tibble(intron = 1:length(differential.3datasets.IR$coords), coords = differential.3datasets.IR$coords) # intron coordinates for to enable matching back to original coords in IRFinder dataframe
rbp <- "TARDBP"
bed <- import("/camp/home/ziffo/home/genomes/clip-bedgraphs/K562/TARDBP.merged.bed.gz")
seqlevelsStyle(bed) <- "NCBI" # remove "chr"
grange_overlap <- findOverlaps(query = grange_introns, subject=bed, ignore.strand=FALSE) # find overlaps between RBP binding sites & retained introns
range <- bed[ queryHits(grange_overlap) ] %>% as.data.frame() # subset TDP43 xlinks that overlap with the introns.
TARDBP_ac_who_k562 <- range %>%  mutate( intron = queryHits(grange_overlap)) %>%  group_by(intron) %>% summarise(iclip_score = sum(score) ) %>% # sum the scores for each intron - 1834 introns with scores
  left_join(intron_coords, by = "intron") %>% dplyr::select(-intron) # add intron coordinates to match back to IRFinder results
TARDBP_iclip_scores <- differential.3datasets.IR %>% dplyr::select(coords, intron_length) %>% full_join(TARDBP_ac_who_k562, by = "coords") %>% # add back the introns without xlink scores
  drop_na() # NAs indicate no xlink score for the intron. Can drop NAs or set NA scores to 0
# TARDBP_iclip_scores[is.na(TARDBP_iclip_scores)] <- 0  # set NA scores to 0
TARDBP_iclip_scores <- TARDBP_iclip_scores %>% mutate(iclip_score.normalised = iclip_score/intron_length) # normalise all scores for intron_length
differential.3datasets.IR <- differential.3datasets.IR %>% left_join(select(TARDBP_iclip_scores, iclip_score.normalised, coords), by = "coords") #%>% # add normalised scores to IRFinder results
TARDBP.clip.differential.sina_plot <- sinaplot(data = differential.3datasets.IR, groups = "p0.05_reliable_IRdirection", continuous = "iclip_score.normalised", dashed.line = "NA", stat.y.position = 0.11, stats.test = "t_test") +  ylab(label = expression(log[10]~TARDBP~iCLIP~binding) ) + xlab("ALS vs CTRL astrocytes") + scale_y_log10() + theme(axis.text=element_text(size=10))
TARDBP.clip.differential.sina_plot

# [1] "Stats test used =  t_test"
# # A tibble: 1 x 11
#   .y.   group1  group2    n1    n2 statistic    df         p     p.adj p.adj.signif y.position
#   <chr> <chr>   <chr>  <int> <int>     <dbl> <dbl>     <dbl>     <dbl> <chr>             <dbl>
# 1 y     IR down IR up   2339   490     -4.00  1832 0.0000657 0.0000657 ****               0.11


# 11% less retained introns in TDP-43 deleted astrocytes compared with control (12,716 vs 14,338 respectively; Fig. S3J). Differential IR analysis revealed 700 introns from 643 genes that were retained at significantly different levels (p < 0.05, Fig. S3E), with 581 (83%) being downregulated in reactive astrocytes (p = 1.18x10-45)
ac_tdp_ko_vs_ctrl.absoluteIR %>% dplyr::filter(A.reliable_retained == "reliable-retained" | B.reliable_retained == "reliable-retained") %>% nrow # 16136 reliable unique retained introns 
ac_tdp_ko_vs_ctrl.absoluteIR %>% dplyr::filter(A.reliable_retained == "reliable-retained") %>% nrow  # 12716 reliable unique retained introns in VCP
ac_tdp_ko_vs_ctrl.absoluteIR %>% dplyr::filter(B.reliable_retained == "reliable-retained") %>% nrow  # 14338 reliable unique retained introns in control
(14338-12716)/14338 * 100 # 11.3% fewer retained introns in VCP compared with control 
ac_tdp_ko_vs_ctrl %>% dplyr::filter(p0.05_reliable == "significant") %>% nrow # 700 reliable introns retained at significantly different levels
ac_tdp_ko_vs_ctrl %>% dplyr::filter(p0.05_reliable == "significant") %>% distinct(gene_name) %>% nrow # from 643 unique genes
ac_tdp_ko_vs_ctrl %>% dplyr::filter(p0.05_reliable == "significant", IRratio.diff < 0) %>% nrow # 581 displaying decreased IR levels in VCP
(700-(700-581))/700 * 100 # 83% decreased in VCP

# Comparing genes with decreased IR between the TARDBP knockout astrocytes and the ALS astrocytes revealed an overlap of 139 / 525 (26.5%, p = 1.2x10-92; Table S4). These results indicate that TDP-43 is required to maintain the prevalent IR levels in astrocytes and its loss may trigger the ALS decreased IR signature.
IRdown_VCP <- ac_who_vcp_vs_ctrl %>% filter(p0.05_reliable == "significant" & IRdirection == "down") %>% pull(gene_name) %>% unique
IRdown_SOD1 <- ac_tyz_sod1_vs_ctrl %>% filter(p0.05_reliable == "significant" & IRdirection == "down") %>% pull(gene_name) %>% unique
IRdown_C9ORF72 <- ac_bir_c9orf_vs_ctrl %>% filter(p0.05_reliable == "significant" & IRdirection == "down") %>% pull(gene_name) %>% unique
IRdown_ALS <- unique(c(IRdown_VCP,IRdown_SOD1, IRdown_C9ORF72)) # 834 genes
IRdown_TDP43 <- ac_tdp_ko_vs_ctrl %>% filter(p0.05_reliable == "significant" & IRdirection == "down") %>% pull(gene_name) %>% unique # 189 genes
IRdown_TDP43[IRdown_TDP43 %in% IRdown_ALS] # 139 / 540 (25.7%)
IRdown_TDP43_ALS <- IRdown_TDP43[IRdown_TDP43 %in% IRdown_ALS] 
IRdown_TDP43_ALS[IRdown_TDP43_ALS %in% all.reactivity.go] # 58

go.obj <- newGeneOverlap(IRdown_ALS,
                         IRdown_TDP43,
                         genome.size=nrow(distinct(gtf, gene_name)))
testGeneOverlap(go.obj)
# GeneOverlap object:
# listA size=1654
# listB size=540
# Intersection size=139
# Overlapping p-value=1.2e-92
# Jaccard Index=0.1

# TDP43 deletion Scatter plot: Delta IR vs DGE LFC
labels_lenient <- filter(ac_tdp_ko_vs_ctrl, p0.05_reliable == "significant", IRratio.diff < 0, log2FoldChange > 0) %>% filter(gene_name %in% all.reactivity.go) %>% group_by(gene_name) %>% slice_max(abs(IRratio.diff), n = 1, with_ties = FALSE)
ac_tdp_ko_scatter <- ggplot(filter(ac_tdp_ko_vs_ctrl, p0.05_reliable == "significant"), aes( x = IRratio.diff, y = log2FoldChange,  label = gene_name, colour = log10(baseMean))) + 
  geom_point(alpha = 0.5, size = 0.5) +  theme_bw() +  xlab( expression( Delta~IR~ratio~TDP43~KO-CTRL)  ) +  ylab( expression(Gene~exp~TDP43~KO:CTRL) ) +
  scale_colour_gradient(low = "dodgerblue3", high = "firebrick2") +
  geom_text_repel(data = labels_lenient, aes(x = IRratio.diff, y = log2FoldChange, label = gene_name), fontface = "italic", colour = "black", min.segment.length = 0, box.padding = 0.5, point.padding = 0.5, size = 2.5) +
  theme(strip.background = element_rect(colour="black", fill="white"), panel.grid = element_blank(), legend.position = "none", axis.text=element_text(size=10)) +
  geom_hline(yintercept = 0, linetype = 3, colour = "darkgrey") + geom_vline(xintercept = 0, linetype = 3, colour = "darkgrey") + 
    geom_smooth(data = filter(ac_tdp_ko_vs_ctrl, p0.05_reliable == "significant"), aes(x = IRratio.diff, y = log2FoldChange), method = "lm", se = FALSE, show.legend = TRUE, colour = "black", linetype = 2) + 
  xlim(-0.31,0.31) +  ylim(-0.7,1.3) +
  annotate(geom="text", x=.17, y=-0.6, label="IR up & expression down", color="darkgrey", size = 2.7)  + annotate(geom="text", x=.17, y=1.2, label="IR up & expression up", color="darkgrey", size = 2.7)  + 
  annotate(geom="text", x=-.17, y=1.2, label="IR down & expression up", color="firebrick2", size = 2.7) +   annotate(geom="text", x=-.17, y=-0.6, label="IR down & expression down", color="darkgrey", size = 2.7)
ac_tdp_ko_scatter

ac_tdp_ko_vs_ctrl_filt <- ac_tdp_ko_vs_ctrl %>% dplyr::filter(p0.05_reliable == "significant") %>%
  drop_na(IRratio.diff, log2FoldChange) %>% filter(IRratio.diff != Inf) %>% filter(IRratio.diff != -Inf) %>% filter(log2FoldChange != Inf) %>% filter(log2FoldChange != -Inf)
cor.test(x = ac_tdp_ko_vs_ctrl_filt$IRratio.diff, y = ac_tdp_ko_vs_ctrl_filt$log2FoldChange) # R = -0.07
summary( lm( IRratio.diff ~ log2FoldChange, data = ac_tdp_ko_vs_ctrl_filt)) 

# GO terms IR down & DGE up in TDP-43 deleted
ac_tdp_dge_ir_genes <- filter(ac_tdp_ko_vs_ctrl, p0.05_reliable == "significant") %>%  split(.$direction) %>%  map("ensID") # create list of IR up/down & Expression up/down genes
ac_tdp_dge_ir_genes <- gost(query = ac_tdp_dge_ir_genes, organism = "mmusculus") # run gprofiler2 on all groups - runs on all KEGG, GO, REAC,TF, WP pathways simultaneously
ac_tdp_dge_ir_gprofiles_filt <- ac_tdp_dge_ir_genes$result %>% filter(str_detect(source, "KEGG|GO:BP|GO:MF|GO:CC|REAC|CORUM")) %>% arrange( -log10(p_value) ) %>% mutate( term_name = as.factor(term_name)) 
ac_tdp_dge_ir_gprofiles_curated_irdown_expup <- ac_tdp_dge_ir_gprofiles_filt %>% filter(query == "IR down & expression up")
# paste('"',unique(ac_tdp_dge_ir_gprofiles_curated_irdown_expup$term_name),'",', sep = "", collapse = '\n') %>% cat()
irdown_expup_curated_list <- c(#"nuclear body",
# "catalytic activity",
# "macromolecule metabolic process",
# "cellular macromolecule metabolic process",
"cellular component organization",
# "intrinsic component of endoplasmic reticulum membrane",
# "cellular component organization or biogenesis",
# "integral component of endoplasmic reticulum membrane",
"protein modification process",
# "cellular protein modification process",
"macromolecule modification",
"RNA binding",
"mitochondrion",
# "GTPase binding",
# "cellular protein metabolic process",
# "nitrogen compound metabolic process",
"protein metabolic process",
# "primary metabolic process",
"protein binding",
# "metabolic process",
"cellular metabolic process",
"enzyme binding",
"intracellular",
"cytoplasm")    
ac_tdp_dge_ir_gprofiles_curated_irdown_expup <- ac_tdp_dge_ir_gprofiles_filt %>% filter(query == "IR down & expression up") %>% filter(term_name %in% irdown_expup_curated_list) %>% mutate(query = "")
ac_tdp_go_curated <- curated_profiles(gprofiles = ac_tdp_dge_ir_gprofiles_curated_irdown_expup, colours = "firebrick2") + labs(subtitle = "TDP43 deleted")
ac_tdp_go_curated

# Liddelow heatmap DGE & IR in TDP43 depleted
tdp_dge_ir_filt <- ac_tdp_ko_vs_ctrl.absoluteIR %>% filter(gene_name %in% astrocyte.reactive.genes) %>% arrange(IR.lfc) %>% distinct(gene_name, .keep_all = TRUE) %>% # Use absolute IR results to avoid loosing DGE values where no IR values exist
  dplyr::select(gene_name, "TDP43 KO IR" = IR.lfc, "TDP43 KO GE" = log2FoldChange)
tdp_dge_ir_filt$"TDP43 KO IR" <- replace_na(tdp_dge_ir_filt$"TDP43 KO IR", 0)
tdp_dge_ir_filt$"TDP43 KO GE" <- replace_na(tdp_dge_ir_filt$"TDP43 KO GE", 0)
tdp_dge_ir_filt <- tdp_dge_ir_filt %>% arrange(factor(gene_name, levels = rownames(annot))) # order the matrix by the annotation dataframe
tdp_dge_ir_filt.mat <- make_matrix(select(tdp_dge_ir_filt,-gene_name),pull(tdp_dge_ir_filt,gene_name))
tdp_dge_ir_filt.mat[tdp_dge_ir_filt.mat == -Inf] <- -2.0
tdp_dge_ir_filt.mat[tdp_dge_ir_filt.mat == Inf] <- 2.0
# tdp_dge_ir_filt.mat[tdp_dge_ir_filt.mat == NA] <- 0
tdp_dge_ir_filt.mat[tdp_dge_ir_filt.mat > 2.0] <- 2.0
tdp_dge_ir_filt.mat[tdp_dge_ir_filt.mat < -2.0] <- -2.0
ac_tdp_ko_vs_ctrl.all <- cleanIRFinder(IRFinder = ac_tdp_ko_vs_ctrl.IRFinder, species = "Mmusculus") %>% full_join(dge_res_ac_tdp_ko_vs_ctrl, by = c("ensID", "gene_name")) # full join DESeq to avoid loosing DGE values where no IR values exist
tdp_dge_res_padj <- ac_tdp_ko_vs_ctrl.all %>% filter(gene_name %in% astrocyte.reactive.genes) %>% arrange(IR.lfc) %>% distinct(gene_name, .keep_all = TRUE) %>%
  mutate(IR.sig = case_when(p0.05_reliable == "significant" ~ "\u2217", TRUE ~ ""), DGE.sig = case_when(padj <= 0.05 ~  "\u2217", TRUE ~ "")) %>%
  dplyr::select(gene_name, IR.sig, DGE.sig) %>%
  arrange(factor(gene_name, levels = rownames(annot))) %>% dplyr::select(IR.sig,DGE.sig) %>%  as.matrix  # order the matrix by the annotation dataframe
ac_tdp_ko_vs_ctrl_liddelow_heatmap <- pheatmap(t(tdp_dge_ir_filt.mat), cluster_rows=F, show_rownames=T, show_colnames = T, cluster_cols=F, fontsize = 8, gaps_col = c(12,25), gaps_row = 1,
                  annotation_col = annot, annotation_names_col = F, annotation_legend = FALSE, annotation_colors = annot_cols, main = "", display_numbers = t(tdp_dge_res_padj), 
                  color = colorRampPalette(rev(greenred(75)))(length(breaksList)), breaks = breaksList, cellheight = 9, cellwidth = 15, border_color=NA, legend = FALSE)

# fgsea
ac_tdp_ko_vs_ctrl.IR_GE <- ac_tdp_ko_vs_ctrl.absoluteIR %>% group_by(gene_name) %>% dplyr::summarise(IR.mean = mean(IRratio.diff, na.rm = TRUE), GE.lfc = median(log2FoldChange))
ac_tdp_ko_vs_ctrl.IR_GE[is.na(ac_tdp_ko_vs_ctrl.IR_GE)] <- 0
ac_tdp_ko_vs_ctrl.IR_GE <- ac_tdp_ko_vs_ctrl.IR_GE %>% data.frame(row.names = "gene_name")
ac_tdp_ko_vs_ctrl.IR_GE <- scale(ac_tdp_ko_vs_ctrl.IR_GE) # rescale both IR & GE to ensure they have the same variance
ac_tdp_ko_vs_ctrl.IR_GE <- ac_tdp_ko_vs_ctrl.IR_GE %>% as_tibble(rownames = "gene_name") %>% mutate("IR_DGE" = (GE.lfc + (IR.mean*-1)/2))
geneList = ac_tdp_ko_vs_ctrl.IR_GE$IR_DGE
names(geneList) <- as.character(ac_tdp_ko_vs_ctrl.IR_GE$gene_name)
geneList = sort(geneList, decreasing = TRUE)
liddelow_genes.list <- list("Astrocyte_reactivity" = astrocyte.reactivity.markers$gene_name[astrocyte.reactivity.markers$gene_name %in% ac_tdp_ko_vs_ctrl.absoluteIR$gene_name]) # astrocyte reactivity genes
tdp.fgseaRes <- fgsea(pathways = liddelow_genes.list, stats = geneList)
tdp.fgseaRes # 0.21

# Comparing genes with decreased IR and increased expression between TARDBP knockout  and ALS astrocytes revealed an overlap of 46 / 269 (17.1%, p = 4.6x10-34), 19 are relevant to astrocyte reactivity 
IRdown_DGEup_VCP <- ac_who_vcp_vs_ctrl %>% filter(p0.05_reliable == "significant" & direction == "IR down & expression up") %>% pull(gene_name) %>% unique
IRdown_DGEup_SOD1 <- ac_tyz_sod1_vs_ctrl %>% filter(p0.05_reliable == "significant" & direction == "IR down & expression up") %>% pull(gene_name) %>% unique
IRdown_DGEup_C9ORF72 <- ac_bir_c9orf_vs_ctrl %>% filter(p0.05_reliable == "significant" & direction == "IR down & expression up") %>% pull(gene_name) %>% unique
IRdown_DGEup_ALS <- unique(c(IRdown_DGEup_VCP,IRdown_DGEup_SOD1, IRdown_DGEup_C9ORF72)) # 834 genes
IRdown_DGEup_TDP43 <- ac_tdp_ko_vs_ctrl %>% filter(p0.05_reliable == "significant" & direction == "IR down & expression up", gene_name %in% gtf$gene_name) %>% pull(gene_name) %>% unique # 269 genes
IRdown_DGEup_TDP43[IRdown_DGEup_TDP43 %in% IRdown_DGEup_ALS] # 46 / 269 (17.1%)
IRdown_DGEup_TDP43_ALS <- IRdown_DGEup_TDP43[IRdown_DGEup_TDP43 %in% IRdown_DGEup_ALS] 
IRdown_DGEup_TDP43_ALS[IRdown_DGEup_TDP43_ALS %in% all.reactivity.go] 
#  [1] "INPPL1"  "ELN"     "SIPA1"   "FZR1"    "GPS2"    "PLOD3"   "TSTA3"   "ADAR"    "NAPRT"   "FARP1"  
# [11] "CD81"    "RIC8A"   "PKD2"    "MYO1C"   "AEN"     "CNTNAP1" "PSMA1"   "PSMD4"   "DGKQ" 

go.obj <- newGeneOverlap(IRdown_DGEup_ALS,
                         IRdown_DGEup_TDP43,
                         genome.size=nrow(distinct(gtf, gene_name)))
testGeneOverlap(go.obj)
# GeneOverlap object:
# listA size=916
# listB size=269
# Intersection size=46
# Overlapping p-value=4.6e-34
# Jaccard Index=0.0
```

## SOD1 TRAP-seq astrocytes

```{r}
# SOD1 TRAP-Seq GO terms upregulated
DGEevents.ensID <- dge_res_ac_mou_sod1_vs_ctrl %>% filter(padj < 0.05) %>% mutate(direction = case_when(log2FoldChange > 0 ~ "translation up", log2FoldChange < 0 ~ "translation down", TRUE ~ "no_change")) %>% split(.$direction) %>%  map("ensID")
DGEevents.ensID$no_change <- NULL
DGEevents.gprofiles <- gost(query = DGEevents.ensID, organism = "mmusculus") # run gprofiler2 on all groups - runs on all KEGG, GO, REAC,TF, WP pathways simultaneously
DGEevents.gprofiles_filt <- DGEevents.gprofiles$result %>% filter(str_detect(source, "KEGG|GO:BP|GO:MF|GO:CC|REAC|CORUM")) %>% arrange( -log10(p_value) ) %>% mutate( term_name = as.factor(term_name)) 
DGEevents.gprofiles_filt.up <- DGEevents.gprofiles_filt %>% filter(query == "translation up")
expup_curated_list <- c("response to interferon-beta",
"antigen processing and presentation",
# "gliogenesis",
"cell adhesion",
"tumor necrosis factor superfamily cytokine production",
"interleukin-6 production",
"response to interferon-gamma",
# "macrophage activation",
"collagen-containing extracellular matrix",
"actin cytoskeleton organization",
# "anatomical structure morphogenesis",
# "immune effector process",
"positive regulation of cell motility",
# "immune response",
"cytokine production",
# "apoptotic process",
"response to cytokine",
"inflammatory response",
"immune system process",
"response to stress")
DGEevents.gprofiles_curated_expup <- DGEevents.gprofiles_filt %>% filter(query == "translation up")  %>% filter(term_name %in% expup_curated_list) %>% mutate(query = "")
DGEevents.gprofiles_curated_expup$term_name <- gsub("tumor necrosis factor", "TNF", DGEevents.gprofiles_curated_expup$term_name)
DGEevents.gprofiles_curated_expup$term_name <- gsub("collagen-containing extracellular matrix", "collagen extracellular matrix", DGEevents.gprofiles_curated_expup$term_name)
TRAP.translationUP.go_curated <- curated_profiles(gprofiles = DGEevents.gprofiles_curated_expup, colours = "forestgreen") + labs(subtitle = "SOD1 TRAPseq")
TRAP.translationUP.go_curated

# Comparing genes with significantly increased translation in Sun et al. (FDR < 0.05) with those with decreased IR in ALS astrocytes, revealed 69 / 765 (9%, p = 0.03) overlapping genes, of which 27 are directly relevant to astrocyte reactivity, including OSMR, FBLN5, VIM, and PTX3
# Overlap IR down in ALS & DGE up in TRAP with astrocyte reactivity genes
DGEup_TRAP <- dge_res_ac_mou_sod1_vs_ctrl %>% filter(padj < 0.05) %>% pull(gene_name) %>% toupper %>% unique # 902 genes
DGEup_TRAP <- unique(DGEup_TRAP[DGEup_TRAP %in% gtf$gene_name]) # 765 / 902 have a human gene ID
IRdown_VCP <- ac_who_vcp_vs_ctrl %>% filter(p0.05_reliable == "significant" & IRratio.diff < 0) %>% pull(gene_name) %>% unique # 1312 genes
IRdown_SOD1 <- ac_tyz_sod1_vs_ctrl %>% filter(p0.05_reliable == "significant" & IRratio.diff < 0) %>% pull(gene_name) %>% unique # 115 genes
IRdown_C9ORF72 <- ac_bir_c9orf_vs_ctrl %>% filter(p0.05_reliable == "significant" & IRratio.diff < 0) %>% pull(gene_name) %>% unique # 220 genes
IRdown_ALS <- unique(c(IRdown_VCP, IRdown_C9ORF72, IRdown_SOD1)) # 1532 genes
DGEup_TRAP[DGEup_TRAP %in% IRdown_ALS] # 69 / 765 (8.5%)
IRdownALS_DGEupTRAP <- DGEup_TRAP[DGEup_TRAP %in% IRdown_ALS] 
IRdownALS_DGEupTRAP[IRdownALS_DGEupTRAP %in% astrocyte.reactive.genes] # "VIM"   "OSMR"  "FBLN5" "PTX3"
IRdownALS_DGEupTRAP[IRdownALS_DGEupTRAP %in% all.reactivity.go] 
go.obj <- newGeneOverlap(IRdown_ALS,
                         DGEup_TRAP,
                         genome.size=nrow(distinct(gtf, gene_name)))
testGeneOverlap(go.obj)
# GeneOverlap object:
# listA size=1654
# listB size=765
# Intersection size=69
# Overlapping p-value=8.6e-18
# Jaccard Index=0.0

# Of the genes exhibiting decreased IR in stimulated reactive astrocytes, 8 / 223 (p = 0/01) displayed increased translation in SOD1G37R mouse astrocytes, of which all 8 were decreased in IR in ALS astrocytes. 
go.obj <- newGeneOverlap(IRdown_A1,
                         DGEup_TRAP,
                         genome.size=nrow(distinct(gtf, gene_name)))
testGeneOverlap(go.obj)
# GeneOverlap object:
# listA size=240
# listB size=765
# Intersection size=8
# Overlapping p-value=0.012

# in the TARDBP knockout mouse, 26 genes with decreased IR were increased in translation in SOD1G37R astrocytes, of which 10 were decreased in IR in ALS mutants. 
go.obj <- newGeneOverlap(IRdown_TDP43,
                         DGEup_TRAP,
                         genome.size=nrow(distinct(gtf, gene_name)))
testGeneOverlap(go.obj)
# GeneOverlap object:
# listA size=540
# listB size=765
# Intersection size=26
# Overlapping p-value=9.8e-09

IRdownALS_DGEupTRAP[IRdownALS_DGEupTRAP %in% IRdown_TDP43] # "SGK1"     "MEST"     "NAPRT"    "HDAC5"    "MCAM"     "SIPA1"    "ADAMTSL4" "FLOT1"    "INSIG1"  "BRD2"  

# SOD1 TRAP Seq liddelow heatmap
# plot LFC as a single row
sod1TRAP.dge_res <- dge_res_ac_mou_sod1_vs_ctrl %>% mutate(gene_name = toupper(gene_name)) %>% filter(gene_name %in% astrocyte.reactive.genes) %>% dplyr::select(gene_name, "SOD1 Translation" = log2FoldChange) %>%  arrange(factor(gene_name, levels = rownames(annot))) %>% drop_na # order the matrix by the annotation dataframe
sod1TRAP.dge_res.mat <- make_matrix(select(sod1TRAP.dge_res,-gene_name),pull(sod1TRAP.dge_res,gene_name))
sod1TRAP.dge_res.mat[sod1TRAP.dge_res.mat > 1.5] <- 1.5
sod1TRAP.dge_res.mat[sod1TRAP.dge_res.mat < -1.5] <- -1.5
sod1TRAP.dge_res_padj <- dge_res_ac_mou_sod1_vs_ctrl %>% mutate(gene_name = toupper(gene_name)) %>% filter(gene_name %in% astrocyte.reactive.genes) %>% filter(log2FoldChange != "NA") %>% mutate(sig = case_when(padj <= 0.05 ~ "\u2217", TRUE ~ "")) %>% arrange(factor(gene_name, levels = rownames(annot))) %>% dplyr::select(sig) %>%  as.matrix  # order the matrix by the annotation dataframe
ac_mou_sod1_liddelow_heatmap <- pheatmap(t(sod1TRAP.dge_res.mat), cluster_rows=F, show_rownames=T, show_colnames = T, cluster_cols=F, fontsize = 8, gaps_col = c(12,24),
                  annotation_col = annot, annotation_names_col = F, annotation_legend = FALSE, annotation_colors = annot_cols, main = "", display_numbers = t(sod1TRAP.dge_res_padj), 
                  color = colorRampPalette(rev(greenred(75)))(length(breaksList)),breaks = breaksList, cellheight = 20, cellwidth = 15, border_color=NA, legend = FALSE)

# upregulation of inflammatory pathways (Fig. 3K, S5G) as well as established astrocyte reactivity genes (25) (enrichment p = 7.3x10-10
geneList = dge_res_ac_mou_sod1_vs_ctrl$log2FoldChange
names(geneList) <- as.character(dge_res_ac_mou_sod1_vs_ctrl$gene_name)
geneList = sort(geneList, decreasing = TRUE)
liddelow_genes.list <- list("Astrocyte_reactivity" = astrocyte.reactivity.markers$gene_name[astrocyte.reactivity.markers$gene_name %in% dge_res_ac_mou_sod1_vs_ctrl$gene_name]) 
TRAP.fgseaRes <- fgsea(pathways = liddelow_genes.list, stats = geneList)
TRAP.fgseaRes 
#                 pathway         pval         padj   log2err        ES      NES size
# 1: Astrocyte_reactivity 1.361612e-09 1.361612e-09 0.7881868 0.8129225 2.301581   36
```

## Merge

```{r}
fig3.merged.new <- ggarrange(
  ggarrange(a1_tdp43_sina_plot, a1_tdp43_dge_ir_sina_plot, TARDBP.clip.differential.sina_plot, nrow = 1, ncol = 3, widths = c(0.8,1.2, 0.8), labels = c("D", "E", "F")), # IR sina & IR GE sina
  ggarrange(ac_bar_a1_scatter, ac_tdp_ko_scatter, nrow = 1, ncol = 2, widths = c(1,1), labels = c("G", "H")), # IR & GE scatter
  ggarrange(ac_bar_a1_go_curated, ac_tdp_go_curated, TRAP.translationUP.go_curated, nrow = 1, ncol = 3, widths = c(1,1,1), labels = c("I", "J", "K")), # GO terms
  NULL,
  ggarrange(
    annotate_figure(ac_a1_liddelow_heatmap[[4]],  bottom = text_grob("Pan-reactive                                                 A1                                                           A2", color = "black", hjust = 0.6, vjust = -10.9, x = 0.5, face = "italic", size = 10)), 
    annotate_figure(ac_tdp_ko_vs_ctrl_liddelow_heatmap[[4]],  bottom = text_grob("Pan-reactive                                                 A1                                                           A2", color = "black", hjust = 0.6, vjust = -10.9, x = 0.5, face = "italic", size = 10)), 
    annotate_figure(ac_mou_sod1_liddelow_heatmap[[4]],  bottom = text_grob("Pan-reactive                                                 A1                                                           A2", color = "black", hjust = 0.6, vjust = -10.6, x = 0.5, face = "italic", size = 10)), 
    nrow = 3, ncol = 1, heights = c(1,1,0.9), labels = c("L", "M", "N")), # liddelow heatmap
  nrow = 5, heights = c(1,1,0.9,0.02,1.1))
fig3.merged.new
ggsave(fig3.merged.new, filename = "/camp/home/ziffo/home/projects/astrocyte-meta-analysis/splicing/IRFinder/irlib/figures/fig3.merged.new.png", height = 13, width = 11.5)
```


# Table S2: Differentially expressed genes VCP vs CTRL

Differentially expressed genes in VCP mutant astrocytes

```{r}
# dge_res_ac_who_vcp_vs_ctrl.FDR0.05 created in Figure S1A
readr::write_csv(select(dge_res_ac_who_vcp_vs_ctrl.FDR0.05, gene_name, everything()), path=paste("/camp/home/ziffo/home/projects/astrocyte-fractionation-bulk-rnaseq/deseq2/ac_who_vcp_vs_ctrl_deseq2_results.FDR0.05.csv", sep = ""))
```

# Table S3: Dataset QC summary

Sequencing depth, read types, read lengths taken from nfcore/rnaseq pipeline MultiQC output. 


# Table S4: Overlap decreased IR across datasets

Overlap IR between datasets

```{r}
IRdown_VCP <- ac_who_vcp_vs_ctrl %>% filter(p0.05_reliable == "significant" & IRdirection == "down") %>% pull(gene_name) %>% unique
IRdown_SOD1 <- ac_tyz_sod1_vs_ctrl %>% filter(p0.05_reliable == "significant" & IRdirection == "down") %>% pull(gene_name) %>% unique
IRdown_C9ORF72 <- ac_bir_c9orf_vs_ctrl %>% filter(p0.05_reliable == "significant" & IRdirection == "down") %>% pull(gene_name) %>% unique
unique(c(IRdown_VCP, IRdown_C9ORF72)) # 834 genes
IRdown_ALS <- unique(c(IRdown_VCP,IRdown_SOD1, IRdown_C9ORF72)) # 834 genes
IRdown_A1 <- ac_bar_a1_vs_a0 %>% filter(p0.05_reliable == "significant" & IRdirection == "down") %>% pull(gene_name) %>% unique # 189 genes
IRdown_VCP_C9ORF72 <- IRdown_C9ORF72[IRdown_C9ORF72 %in% IRdown_VCP]
IRdown_VCP_C9ORF72_SOD1 <- IRdown_VCP_C9ORF72[IRdown_VCP_C9ORF72 %in% IRdown_SOD1]
DGEup_TRAP <- dge_res_ac_mou_sod1_vs_ctrl %>% filter(padj < 0.05) %>% pull(gene_name) %>% toupper %>% unique # 902 genes
DGEup_TRAP <- unique(DGEup_TRAP[DGEup_TRAP %in% gtf$gene_name]) # 765 / 902 have a human gene ID
IRdown_TDP43 <- ac_tdp_ko_vs_ctrl %>% filter(p0.05_reliable == "significant" & IRratio.diff < 0) %>% dplyr::pull(gene_name) %>% unique # 540 genes
IRdown_TDP43 <- unique(IRdown_TDP43[IRdown_TDP43 %in% gtf$gene_name]) # 525 / 540 have a human gene ID
IRdown_TDP43[IRdown_TDP43 %in% IRdown_ALS] # 139 / 525 (26.5%)


SOD1G37R.TRAP_TARDBP.KO

DGEup_TRAP[DGEup_TRAP %in% IRdown_TDP43][DGEup_TRAP[DGEup_TRAP %in% IRdown_TDP43] %in% IRdown_ALS]

IRdown_A1_ALS <- IRdown_A1[IRdown_A1 %in% IRdown_ALS] 
IRdown_overlap <- cbindX(arrange(as_tibble_col(IRdown_SOD1[IRdown_SOD1 %in% IRdown_VCP], column_name = "SOD1_VCP"), SOD1_VCP), 
                         arrange(as_tibble_col(IRdown_C9ORF72[IRdown_C9ORF72 %in% IRdown_VCP], column_name = "C9ORF72_VCP"), C9ORF72_VCP),  
                         arrange(as_tibble_col(IRdown_VCP_C9ORF72[IRdown_VCP_C9ORF72 %in% IRdown_SOD1], column_name = "SOD1_C9ORF72_VCP"), SOD1_C9ORF72_VCP),  
                         arrange(as_tibble_col(IRdown_A1[IRdown_A1 %in% IRdown_ALS], column_name = "Reactive_any.ALS"), Reactive_any.ALS),  
                         arrange(as_tibble_col(IRdown_A1[IRdown_A1 %in% IRdown_VCP_C9ORF72_SOD1], column_name = "Reactive_all.ALS"), Reactive_all.ALS),
                         arrange(as_tibble_col(IRdown_TDP43[IRdown_TDP43 %in% IRdown_ALS], column_name = "TARDBP.KO_any.ALS"), TARDBP.KO_any.ALS),
                         arrange(as_tibble_col(IRdown_TDP43[IRdown_TDP43 %in% IRdown_VCP_C9ORF72_SOD1], column_name = "TARDBP.KO_all.ALS"), TARDBP.KO_all.ALS),
                         arrange(as_tibble_col(IRdown_TDP43[IRdown_TDP43 %in% IRdown_A1], column_name = "TARDBP.KO_Reactive"), TARDBP.KO_Reactive),
                         arrange(as_tibble_col(DGEup_TRAP[DGEup_TRAP %in% IRdown_ALS], column_name = "SOD1G37R.TRAP_ALS"), SOD1G37R.TRAP_ALS), 
                         arrange(as_tibble_col(DGEup_TRAP[DGEup_TRAP %in% IRdown_A1], column_name = "SOD1G37R.TRAP_Reactive"), SOD1G37R.TRAP_Reactive),
                         arrange(as_tibble_col(DGEup_TRAP[DGEup_TRAP %in% IRdown_TDP43], column_name = "SOD1G37R.TRAP_TARDBP.KO"), SOD1G37R.TRAP_TARDBP.KO))
IRdown_overlap[is.na(IRdown_overlap)] <- ""
write.csv(IRdown_overlap, "/camp/home/ziffo/home/projects/astrocyte-meta-analysis/splicing/IRFinder/irlib/results/IR_down.overlap.csv")
```


# Figure S5: GO terms in 3 categories (IR & GE)

```{r}
# 3 cats GO VCP IR & DGE
ac_who_dge_ir_genes <- ac_who_vcp_vs_ctrl %>% filter(p0.05_reliable == "significant") %>% split(.$direction) %>%  map("ensID") # create list of IR up/down & Expression up/down genes
ac_who_dge_ir_genes <- gost(ac_who_dge_ir_genes) # run gprofiler2 on all groups - runs on all KEGG, GO, REAC,TF, WP pathways simultaneously
ac_who_dge_ir_gprofiles_filt <- ac_who_dge_ir_genes$result %>% filter(str_detect(source, "KEGG|GO:BP|GO:MF|GO:CC|REAC|CORUM")) %>% arrange( -log10(p_value) ) %>% mutate( term_name = as.factor(term_name)) 
ac_who_dge_ir_gprofiles_filt$query <- gsub("expression", "GE", ac_who_dge_ir_gprofiles_filt$query)
ac_who_dge_ir_gprofiles_curated_irdown_expdown <- ac_who_dge_ir_gprofiles_filt %>% filter(query == "IR down & GE down" & (grepl("RNA", term_name) | grepl("nucle", term_name) | grepl("cyt", term_name)))
paste('"',unique(ac_who_dge_ir_gprofiles_curated_irdown_expdown$term_name),'",', sep = "", collapse = '\n') %>% cat()
irdown_expdown_curated_list <- c("microtubule cytoskeleton",
"ribonucleoprotein complex",
"mRNA export from nucleus",
"RNA transport",
"RNA localization",
"transcription by RNA polymerase II",
"nucleocytoplasmic transport",
"ribonucleoprotein granule",
"nuclear speck",
"Processing of Capped Intron-Containing Pre-mRNA",
"RNA splicing",
"mRNA processing",
"mRNA metabolic process",
# "cytoplasm",
"RNA binding") #"nucleus")
ac_who_dge_ir_gprofiles_curated_irdown_expdown <- ac_who_dge_ir_gprofiles_curated_irdown_expdown %>% filter(term_name %in% irdown_expdown_curated_list)
ac_who_dge_ir_gprofiles_curated_irup_expdown <- ac_who_dge_ir_gprofiles_filt %>% filter(query == "IR up & GE down")
paste('"',unique(ac_who_dge_ir_gprofiles_curated_irup_expdown$term_name),'",', sep = "", collapse = '\n') %>% cat()
irup_expdown_curated_list <- c("small GTPase binding",
"microtubule cytoskeleton",
"protein binding",
"cell projection organization",
"cellular component organization or biogenesis") #"plasma membrane bounded cell projection organization")
ac_who_dge_ir_gprofiles_curated_irup_expdown <- ac_who_dge_ir_gprofiles_curated_irup_expdown %>% filter(term_name %in% irup_expdown_curated_list)
ac_who_dge_ir_gprofiles_curated_irup_expup <- ac_who_dge_ir_gprofiles_filt %>% filter(query == "IR up & GE up")
paste('"',unique(ac_who_dge_ir_gprofiles_curated_irup_expup$term_name),'",', sep = "", collapse = '\n') %>% cat()
irup_expup_curated_list <- c("enzyme activator activity",
"transferase activity") #"plasma membrane bounded cell projection organization")
ac_who_dge_ir_gprofiles_curated_irup_expup <- ac_who_dge_ir_gprofiles_curated_irup_expup %>% filter(term_name %in% irup_expup_curated_list)
ac_who_vcp_vs_ctrl.go_curated_3cats <- curated_profiles_colours(gprofiles = bind_rows(ac_who_dge_ir_gprofiles_curated_irdown_expdown, ac_who_dge_ir_gprofiles_curated_irup_expdown, ac_who_dge_ir_gprofiles_curated_irup_expup))  +
  ggtitle("VCP")
ac_who_vcp_vs_ctrl.go_curated_3cats

# 3 cats GO C9ORF
bir_c9orf_dge_ir_genes <- filter(ac_bir_c9orf_vs_ctrl, p0.05_reliable == "significant") %>%  split(.$direction) %>%  map("ensID") # create list of IR up/down & Expression up/down genes
bir_c9orf_dge_ir_genes <- gost(bir_c9orf_dge_ir_genes) # run gprofiler2 on all groups - runs on all KEGG, GO, REAC,TF, WP pathways simultaneously
bir_c9orf_dge_ir_gprofiles_filt <- bir_c9orf_dge_ir_genes$result %>% filter(str_detect(source, "KEGG|GO:BP|GO:MF|GO:CC|REAC|CORUM")) %>% arrange( -log10(p_value) ) %>% mutate( term_name = as.factor(term_name)) 
bir_c9orf_dge_ir_gprofiles_filt$query <- gsub("expression", "GE", bir_c9orf_dge_ir_gprofiles_filt$query)
bir_c9orf_dge_ir_gprofiles_curated_irdown_expdown <- bir_c9orf_dge_ir_gprofiles_filt %>% filter(query == "IR down & GE down") 
irdown_expdown_curated_list <- c("regulation of lipid biosynthetic process",
"steroid metabolic process",
"Activation of gene expression by SREBF (SREBP)",
"cholesterol metabolic process")
bir_c9orf_dge_ir_gprofiles_curated_irdown_expdown <- bir_c9orf_dge_ir_gprofiles_curated_irdown_expdown %>% filter(term_name %in% irdown_expdown_curated_list)
bir_c9orf_dge_ir_gprofiles_curated_irdown_expdown$term_name <- gsub("Activation of gene expression by", "Activation by", bir_c9orf_dge_ir_gprofiles_curated_irdown_expdown$term_name)
bir_c9orf_dge_ir_gprofiles_curated_irup_expup <- bir_c9orf_dge_ir_gprofiles_filt %>% filter(query == "IR up & GE up") 
bir_c9orf_dge_ir_gprofiles_curated_irup_expdown <- bir_c9orf_dge_ir_gprofiles_filt %>% filter(query == "IR up & GE down") 
irup_expdown_curated_list <- c("cellular response to DNA damage stimulus",
"double-strand break repair",
"helicase activity",
"DNA conformation change")
bir_c9orf_dge_ir_gprofiles_curated_irup_expdown <- bir_c9orf_dge_ir_gprofiles_curated_irup_expdown %>% filter(term_name %in% irup_expdown_curated_list)
bir_c9orf_dge_ir_gprofiles_curated_irup_expdown$term_name <- gsub("cellular response to DNA damage stimulus", "response to DNA damage", bir_c9orf_dge_ir_gprofiles_curated_irup_expdown$term_name)
C9orf72.go_curated_3cats <- curated_profiles_colours(gprofiles = bind_rows(bir_c9orf_dge_ir_gprofiles_curated_irup_expup, bir_c9orf_dge_ir_gprofiles_curated_irdown_expdown, bir_c9orf_dge_ir_gprofiles_curated_irup_expdown)) +
  ggtitle("C9ORF72")
C9orf72.go_curated_3cats

# 3 cats GO SOD1
tyz_sod1_dge_ir_genes <- filter(ac_tyz_sod1_vs_ctrl,p0.05_reliable == "significant") %>%  split(.$direction) %>%  map("ensID") # create list of IR up/down & Expression up/down genes
tyz_sod1_dge_ir_genes <- gost(tyz_sod1_dge_ir_genes) # run gprofiler2 on all groups - runs on all KEGG, GO, REAC,TF, WP pathways simultaneously
tyz_sod1_dge_ir_gprofiles_filt <- tyz_sod1_dge_ir_genes$result %>% filter(str_detect(source, "KEGG|GO:BP|GO:MF|GO:CC|REAC|CORUM")) %>% arrange( -log10(p_value) ) %>% mutate( term_name = as.factor(term_name)) 
tyz_sod1_dge_ir_gprofiles_filt$query <- gsub("expression", "GE", tyz_sod1_dge_ir_gprofiles_filt$query)
tyz_sod1_dge_ir_gprofiles_filt$term_name <- gsub("endoplasmic reticulum to Golgi vesicle-mediated transport", "endoplasmic reticulum - golgi", tyz_sod1_dge_ir_gprofiles_filt$term_name)
tyz_sod1_dge_ir_gprofiles_curated_irdown_expdown <- tyz_sod1_dge_ir_gprofiles_filt %>% filter(query == "IR down & GE down")
tyz_sod1_dge_ir_gprofiles_curated_irup_expdown <- tyz_sod1_dge_ir_gprofiles_filt %>% filter(query == "IR up & GE down")
tyz_sod1_dge_ir_gprofiles_curated_irup_expup <- tyz_sod1_dge_ir_gprofiles_filt %>% filter(query == "IR up & GE up")
SOD1.go_curated_3cats <- curated_profiles_colours(gprofiles = bind_rows(tyz_sod1_dge_ir_gprofiles_curated_irdown_expdown, tyz_sod1_dge_ir_gprofiles_curated_irup_expdown, tyz_sod1_dge_ir_gprofiles_curated_irup_expup))  +
  ggtitle("SOD1")
SOD1.go_curated_3cats

# Cytokine stimulated GO 3 cats has no relevent terms - retinoic acid & citrate dehydrogenase
bar_a1_dge_ir_genes <- filter(ac_bar_a1_vs_a0, p0.05_reliable == "significant") %>%  split(.$direction) %>%  map("ensID") # create list of IR up/down & Expression up/down genes
bar_a1_dge_ir_genes <- gost(bar_a1_dge_ir_genes) # run gprofiler2 on all groups - runs on all KEGG, GO, REAC,TF, WP pathways simultaneously
bar_a1_dge_ir_gprofiles_filt <- bar_a1_dge_ir_genes$result %>% filter(str_detect(source, "KEGG|GO:BP|GO:MF|GO:CC|REAC|CORUM")) %>% arrange( -log10(p_value) ) %>% mutate( term_name = as.factor(term_name)) 
bar_a1_dge_ir_gprofiles_filt$query <- gsub("expression", "GE", bar_a1_dge_ir_gprofiles_filt$query)
bar_a1_dge_ir_gprofiles_curated_irdown_expdown <- bar_a1_dge_ir_gprofiles_filt %>% filter(query == "IR down & GE down") 
bar_a1_dge_ir_gprofiles_curated_irup_expup <- bar_a1_dge_ir_gprofiles_filt %>% filter(query == "IR up & GE up") 
bar_a1_dge_ir_gprofiles_curated_irup_expdown <- bar_a1_dge_ir_gprofiles_filt %>% filter(query == "IR up & GE down") 
bar_a1.go_curated_3cats <- curated_profiles_colours(gprofiles = bind_rows(bar_a1_dge_ir_gprofiles_curated_irup_expup, bar_a1_dge_ir_gprofiles_curated_irdown_expdown, bar_a1_dge_ir_gprofiles_curated_irup_expdown)) +
  ggtitle("Cytokine stimulated")
bar_a1.go_curated_3cats

# TDP43 deleted GO 3 cats
tdp43_ko_dge_ir_genes <- filter(ac_tdp_ko_vs_ctrl, p0.05_reliable == "significant") %>%  split(.$direction) %>%  map("ensID") # create list of IR up/down & Expression up/down genes
tdp43_ko_dge_ir_genes <- gost(tdp43_ko_dge_ir_genes, organism = "mmusculus") # run gprofiler2 on all groups - runs on all KEGG, GO, REAC,TF, WP pathways simultaneously
tdp43_ko_dge_ir_gprofiles_filt <- tdp43_ko_dge_ir_genes$result %>% filter(str_detect(source, "KEGG|GO:BP|GO:MF|GO:CC|REAC|CORUM")) %>% arrange( -log10(p_value) ) %>% mutate( term_name = as.factor(term_name)) 
tdp43_ko_dge_ir_gprofiles_filt$query <- gsub("expression", "GE", tdp43_ko_dge_ir_gprofiles_filt$query)
tdp43_ko_dge_ir_gprofiles_curated_irdown_expdown <- tdp43_ko_dge_ir_gprofiles_filt %>% filter(query == "IR down & GE down")
paste('"',unique(tdp43_ko_dge_ir_gprofiles_curated_irdown_expdown$term_name),'",', sep = "", collapse = '\n') %>% cat()
tdp43_ko_IRdown_GEdown_curated_list <- c("Dynein-dynactin complex",
# "small molecule binding",
"presynapse",
"mRNA metabolic process",
"protein modification process",
"inclusion body",
"nucleoplasm",
# "nuclear body",
"nuclear speck",
"intracellular transport",
"protein localization",
"catalytic activity, acting on RNA",
# "endomembrane system",
# "protein-containing complex",
# "macromolecule localization",
"nitrogen compound transport",
"cytoplasmic vesicle",
# "endosome",
"cellular metabolic process",
"protein binding",
# "vesicle",
# "cellular component organization",
# "enzyme binding",
"cytoplasm")    
tdp43_ko_dge_ir_gprofiles_curated_irdown_expdown <- tdp43_ko_dge_ir_gprofiles_filt %>% filter(query == "IR down & GE down") %>% filter(term_name %in% tdp43_ko_IRdown_GEdown_curated_list)
tdp43_ko_dge_ir_gprofiles_curated_irup_expup <- tdp43_ko_dge_ir_gprofiles_filt %>% filter(query == "IR up & GE up") 
paste('"',unique(tdp43_ko_dge_ir_gprofiles_curated_irup_expup$term_name),'",', sep = "", collapse = '\n') %>% cat()
tdp43_ko_IRup_GEup_curated_list <- c(#"Apaf1 homo-oligomer complex",
# "Hepatitis C",
# "p75 NTR receptor-mediated signalling",
"Apoptosis",
"G alpha (12/13) signalling events",
# "Cell death signalling via NRAGE, NRIF and NADE",
"Death Receptor Signalling")#,
# "NRAGE signals death through JNK")
tdp43_ko_dge_ir_gprofiles_curated_irup_expup <- tdp43_ko_dge_ir_gprofiles_filt %>% filter(query == "IR up & GE up") %>% filter(term_name %in% tdp43_ko_IRup_GEup_curated_list)
tdp43_ko_dge_ir_gprofiles_curated_irup_expdown <- tdp43_ko_dge_ir_gprofiles_filt %>% filter(query == "IR up & GE down") 
paste('"',unique(tdp43_ko_dge_ir_gprofiles_curated_irup_expdown$term_name),'",', sep = "", collapse = '\n') %>% cat()
tdp43_ko_IRup_GEdown_curated_list <- c("regulation of DNA recombination",
"MMXD complex",
"DNA recombination")
tdp43_ko_dge_ir_gprofiles_curated_irup_expdown <- tdp43_ko_dge_ir_gprofiles_filt %>% filter(query == "IR up & GE down") %>% filter(term_name %in% tdp43_ko_IRup_GEdown_curated_list)
tdp43_ko.go_curated_3cats <- curated_profiles_colours(gprofiles = bind_rows(tdp43_ko_dge_ir_gprofiles_curated_irup_expup, tdp43_ko_dge_ir_gprofiles_curated_irdown_expdown, tdp43_ko_dge_ir_gprofiles_curated_irup_expdown)) +
  ggtitle("TDP43 deleted")
tdp43_ko.go_curated_3cats

# TRAPSeq SOD1 mouse downregulated genes
DGEevents.ensID <- dge_res_ac_mou_sod1_vs_ctrl %>% filter(padj < 0.05) %>% mutate(direction = case_when(log2FoldChange > 0 ~ "translation up", log2FoldChange < 0 ~ "translation down", TRUE ~ "no_change")) %>% split(.$direction) %>%  map("ensID")
DGEevents.ensID$no_change <- NULL
DGEevents.gprofiles <- gost(query = DGEevents.ensID, organism = "mmusculus") # run gprofiler2 on all groups - runs on all KEGG, GO, REAC,TF, WP pathways simultaneously
DGEevents.gprofiles_filt <- DGEevents.gprofiles$result %>% filter(str_detect(source, "KEGG|GO:BP|GO:MF|GO:CC|REAC|CORUM")) %>% arrange( -log10(p_value) ) %>% mutate( term_name = as.factor(term_name)) 
DGEevents.gprofiles_filt.down <- DGEevents.gprofiles_filt %>% filter(query == "translation down")
paste('"',unique(DGEevents.gprofiles_filt.down$term_name),'",', sep = "", collapse = '\n') %>% cat()
expdown_curated_list <- c("cytoskeletal protein binding",
"central nervous system development",
"beta-catenin binding",
"negative regulation of canonical Wnt signaling pathway",
"negative regulation of cellular metabolic process",
"negative regulation of gene expression",
"negative regulation of transcription by RNA polymerase II",
"cell morphogenesis",
"RNA metabolic process",
"nervous system development",
"developmental process",
"regulation of metabolic process",
"RNA biosynthetic process",
"regulation of cellular metabolic process",
"regulation of RNA metabolic process",
"DNA-binding transcription factor activity")
DGEevents.gprofiles_curated_expdown <- DGEevents.gprofiles_filt %>% filter(query == "translation down") %>% filter(term_name %in% expdown_curated_list) #%>% mutate(query = "")
DGEevents.gprofiles_curated_expdown$term_name <- gsub("negative regulation of transcription by RNA polymerase II", "negative regulation of transcription by RNA pol2", DGEevents.gprofiles_curated_expdown$term_name)
DGEevents.gprofiles_curated_expdown$term_name <- gsub("negative regulation of canonical Wnt signaling pathway", "negative regulation of canonical Wnt signaling", DGEevents.gprofiles_curated_expdown$term_name)
DGEevents.gprofiles_curated_expdown$term_name <- gsub("negative regulation of cellular metabolic process", "negative regulation of metabolic process", DGEevents.gprofiles_curated_expdown$term_name)
DGEevents.gprofiles_curated_expdown$term_name <- gsub("negative regulation", "downregulation", DGEevents.gprofiles_curated_expdown$term_name)
TRAP.translationDOWN.go_curated <- curated_profiles(gprofiles = DGEevents.gprofiles_curated_expdown, colours = "dodgerblue3") + labs(subtitle = "SOD1 TRAPSeq")
TRAP.translationDOWN.go_curated

# Venn overlap IR down & GE up
vcp.filt <- ac_who_vcp_vs_ctrl %>% dplyr::filter(p0.05_reliable == "significant") %>% dplyr::select(gene_name, ensID, IR_vcp = direction) # Intron.GeneName.GeneID, Chr, Start, End, 
sod1.filt <-ac_tyz_sod1_vs_ctrl %>% dplyr::filter(p0.05_reliable == "significant") %>% dplyr::select(gene_name, ensID, IR_sod1 = direction) # Intron.GeneName.GeneID, Chr, Start, End, 
c9orf.filt <- ac_bir_c9orf_vs_ctrl %>% dplyr::filter(p0.05_reliable == "significant") %>% dplyr::select(gene_name, ensID, IR_c9orf = direction) # Intron.GeneName.GeneID, Chr, Start, End, 
vcp_sod1_c9orf.overlap <- vcp.filt %>% full_join(sod1.filt, c("gene_name", "ensID")) %>% # "Intron.GeneName.GeneID", "Chr", "Start", "End", 
  full_join(c9orf.filt, c("gene_name", "ensID")) %>% # "Intron.GeneName.GeneID", "Chr", "Start", "End",
  distinct(.keep_all = TRUE) # keep unique gene names
vcp_sod1_c9orf.overlap <- vcp_sod1_c9orf.overlap %>% mutate(overlap.IRdown.GEup = case_when(IR_vcp == "IR down & expression up" & IR_sod1 == "IR down & expression up" & IR_c9orf == "IR down & expression up" ~ "overlapping",
                                                                              IR_vcp == "IR down & expression up" & IR_sod1 == "IR down & expression up" ~ "VCP & SOD1", 
                                                                              IR_vcp == "IR down & expression up" & IR_c9orf == "IR down & expression up" ~ "VCP & C9orf72", 
                                                                              IR_c9orf == "IR down & expression up" ~ "C9orf72 specific", 
                                                                              IR_sod1 == "IR down & expression up" ~ "SOD1 specific", 
                                                                              IR_vcp == "IR down & expression up" ~ "VCP specific",
                                                                              TRUE ~ "none"))
vcp_sod1_c9orf.overlap.filt <- filter(vcp_sod1_c9orf.overlap, overlap.IRdown.GEup != "none")
venn_vcp_sod1_c9orf_overlap.IRdown.GEup <- as.ggplot(
  ~draw.triple.venn(
    area1 = nrow( filter(vcp_sod1_c9orf.overlap.filt, overlap.IRdown.GEup %in% c("VCP specific", "VCP & SOD1", "VCP & C9orf72", "overlapping" ) ) ), # VCP events
    area2 = nrow( filter(vcp_sod1_c9orf.overlap.filt, overlap.IRdown.GEup %in% c("SOD1 specific", "VCP & SOD1", "C9orf72 & SOD1", "overlapping" ) ) ), # SOD1 events
    area3 = nrow( filter(vcp_sod1_c9orf.overlap.filt, overlap.IRdown.GEup %in% c("C9orf72 specific", "C9orf72 & SOD1", "VCP & C9orf72", "overlapping" ) ) ), # C9orf72 events
    n12 = nrow( filter(vcp_sod1_c9orf.overlap.filt, overlap.IRdown.GEup %in% c("VCP & SOD1", "overlapping" ) ) ), # VCP & SOD1
    n23 = nrow( filter(vcp_sod1_c9orf.overlap.filt, overlap.IRdown.GEup %in% c("C9orf72 & SOD1", "overlapping" ) ) ), # SOD1 & C9orf72
    n13 = nrow( filter(vcp_sod1_c9orf.overlap.filt, overlap.IRdown.GEup %in% c("VCP & C9orf72", "overlapping" ) ) ), # VCP & SOD1
    n123 = nrow(filter(vcp_sod1_c9orf.overlap.filt, overlap.IRdown.GEup == "overlapping")),
    category = c("VCP", "SOD1", "C9orf72"),
    alpha = 0.5,  fill = c("firebrick2", "forestgreen", "dodgerblue3"), 
    cat.col = c("firebrick2", "forestgreen", "dodgerblue3"), cat.fontfamily = "sans", 
    cat.pos = c(330,30,0), cat.dist = 0.07) ) # cat.cex = c(1,1,1),
venn_vcp_sod1_c9orf_overlap.IRdown.GEup

# hypergeometric testing
IRdown_DGEup_VCP <- ac_who_vcp_vs_ctrl %>% filter(p0.05_reliable == "significant" & direction == "IR down & expression up") %>% pull(gene_name) %>% unique
IRdown_DGEup_SOD1 <- ac_tyz_sod1_vs_ctrl %>% filter(p0.05_reliable == "significant" & direction == "IR down & expression up") %>% pull(gene_name) %>% unique
IRdown_DGEup_C9ORF72 <- ac_bir_c9orf_vs_ctrl %>% filter(p0.05_reliable == "significant" & direction == "IR down & expression up") %>% pull(gene_name) %>% unique

# Comparing genes exhibiting decreased IR and increased expression in the SOD1 dataset with that of VCP revealed a significant overlap of 18 / 55 (33%, p = 2.1x10-14, hypergeometric test; Fig. S5D). 
IRdown_DGEup_SOD1[IRdown_DGEup_SOD1 %in% IRdown_DGEup_VCP] # 18 / 55
go.obj <- newGeneOverlap(IRdown_DGEup_VCP,
                         IRdown_DGEup_SOD1,
                         genome.size=nrow(distinct(gtf, gene_name)))
testGeneOverlap(go.obj)
# GeneOverlap object:
# listA size=702
# listB size=55
# Intersection size=18
# Overlapping p-value=1.1e-21
# Jaccard Index=0.0

# Of the 251 genes exhibiting decreased IR and increased expression in C9orf72, 70 (28%) were also found in VCP mutants (overlap p = 3.3x10-47;  Table S5). 
IRdown_DGEup_C9ORF72[IRdown_DGEup_C9ORF72 %in% IRdown_DGEup_VCP]# 70 / 251
go.obj <- newGeneOverlap(IRdown_DGEup_VCP,
                         IRdown_DGEup_C9ORF72,
                         genome.size=nrow(distinct(gtf, gene_name)))
testGeneOverlap(go.obj)
# GeneOverlap object:
# listA size=702
# listB size=251
# Intersection size=70
# Overlapping p-value=2.7e-75
# Jaccard Index=0.1

# Six genes with decreased IR and increased expression were common to all three mutations (chi-squared anova p = 0.008), of which 4 are directly relevant to reactivity: FLNA, PLOD3, MRC2 and ACTN4.
list1 <- unique(IRdown_DGEup_VCP[!is.na(IRdown_DGEup_VCP)])
list2 <- unique(IRdown_DGEup_SOD1[!is.na(IRdown_DGEup_SOD1)])
list3 <- unique(IRdown_DGEup_C9ORF72[!is.na(IRdown_DGEup_C9ORF72)])
univ <- unique(gtf$gene_name[!is.na(gtf$gene_name)])
xtab <- data.frame(Univ=univ, List1=as.integer(univ %in% list1), List2=as.integer(univ %in% list2), List3=as.integer(univ %in% list3))
ftable <- as.data.frame(table(xtab[,c('List1','List2','List3')]))
glm1 <- glm(Freq ~ List1 * List2 * List3, family='poisson',data=ftable)
glm2 <- glm(Freq ~ List1*List2 + List1*List3 + List2*List3, family='poisson',data=ftable)
anovatest <- anova(glm1,glm2,test='Chisq')
pvalue <- anovatest[['Pr(>Chi)']][2] # 0.004703483
ans <- list(xtab,ftable,glm1,glm2,pvalue)
names(ans) <- c('xtab','ftable','glm1','glm2','pvalue')
# $ftable
#   List1 List2 List3  Freq
# 1     0     0     0 59630
# 2     1     0     0   620
# 3     0     1     0    33
# 4     1     1     0    12
# 5     0     0     1   177
# 6     1     0     1    64
# 7     0     1     1     4
# 8     1     1     1     6
# 
# $glm1
# Call:  glm(formula = Freq ~ List1 * List2 * List3, family = "poisson", data = ftable)
# 
# Coefficients:
#          (Intercept)                List11                List21                List31         List11:List21  
#               10.996                -4.566                -7.499                -5.820                 3.555  
#        List11:List31         List21:List31  List11:List21:List31  
#                3.549                 3.710                -2.132  
# Degrees of Freedom: 7 Total (i.e. Null);  0 Residual
# Null Deviance:	    240500 
# Residual Deviance: -9.184e-12 	AIC: 66.72
# 
# $glm2
# Call:  glm(formula = Freq ~ List1 * List2 + List1 * List3 + List2 * List3, family = "poisson", data = ftable)
# Coefficients:
#   (Intercept)         List11         List21         List31  List11:List21  List11:List31  List21:List31  
#        10.996         -4.561         -7.415         -5.803          3.187          3.481          2.303  
# 
# Degrees of Freedom: 7 Total (i.e. Null);  1 Residual
# Null Deviance:	    240500 
# Residual Deviance: 6.979 	AIC: 71.69
# 
# $pvalue
# [1] 0.008244918

IRdown_DGEup_SOD1_VCP <- IRdown_DGEup_SOD1[IRdown_DGEup_SOD1 %in% IRdown_DGEup_VCP]
IRdown_DGEup_C9ORF72_VCP <- IRdown_DGEup_C9ORF72[IRdown_DGEup_C9ORF72 %in% IRdown_DGEup_VCP]
IRdown_DGEup_SOD1_VCP_C9ORF72 <- IRdown_DGEup_SOD1_VCP[IRdown_DGEup_SOD1_VCP %in% IRdown_DGEup_C9ORF72]
IRdown_DGEup_SOD1_VCP_C9ORF72[IRdown_DGEup_SOD1_VCP_C9ORF72 %in% all.reactivity.go] # 4 are relevent to reactivity "FLNA"  "PLOD3" "MRC2"  "ACTN4"
```

## Merge

```{r}
figS5.merged <- ggarrange(
  ac_who_vcp_vs_ctrl.go_curated_3cats,
  ggarrange(C9orf72.go_curated_3cats, SOD1.go_curated_3cats, ncol = 2, widths = c(1,1), labels = c("B", "C")),
  ggarrange(venn_vcp_sod1_c9orf_overlap.IRdown.GEup, bar_a1.go_curated_3cats, ncol = 2, widths = c(1, 1), labels = c("D", "E")),
  tdp43_ko.go_curated_3cats,
  ggarrange(NULL, TRAP.translationDOWN.go_curated, ncol = 2, widths = c(0.02, 1)),
  nrow = 5, heights = c(2.0, 1.3, 1.3, 2.1, 1.8), labels = c("A", "", "", "F", "G"))
figS5.merged
ggsave(figS5.merged, filename = "/camp/home/ziffo/home/projects/astrocyte-meta-analysis/splicing/IRFinder/irlib/figures/figS5.merged.png", height = 12.5, width = 8.25)
```

# Table S5: Overlap IR down & GE up 

Overlap IR and gene expression between datasets

```{r}
IRdown_DGEup_VCP <- ac_who_vcp_vs_ctrl %>% filter(p0.05_reliable == "significant" & direction == "IR down & expression up") %>% pull(gene_name) %>% unique
IRdown_DGEup_SOD1 <- ac_tyz_sod1_vs_ctrl %>% filter(p0.05_reliable == "significant" & direction == "IR down & expression up") %>% pull(gene_name) %>% unique
IRdown_DGEup_C9ORF72 <- ac_bir_c9orf_vs_ctrl %>% filter(p0.05_reliable == "significant" & direction == "IR down & expression up") %>% pull(gene_name) %>% unique
IRdown_DGEup_ALS <- unique(c(IRdown_DGEup_VCP,IRdown_DGEup_SOD1, IRdown_DGEup_C9ORF72))
IRdown_DGEup_A1 <- ac_bar_a1_vs_a0 %>% filter(p0.05_reliable == "significant" & direction == "IR down & expression up") %>% pull(gene_name) %>% unique
IRdown_DGEup_TDP43 <- ac_tdp_ko_vs_ctrl %>% filter(p0.05_reliable == "significant" & direction == "IR down & expression up") %>% pull(gene_name) %>% unique
IRdown_DGEup_TDP43 <- unique(IRdown_DGEup_TDP43[IRdown_DGEup_TDP43 %in% gtf$gene_name]) # 269 / 275 have a human gene ID
DGEup_TRAP <- dge_res_ac_mou_sod1_vs_ctrl %>% filter(padj < 0.05) %>% pull(gene_name) %>% toupper %>% unique # 902 genes
DGEup_TRAP <- unique(DGEup_TRAP[DGEup_TRAP %in% gtf$gene_name]) # 765 / 902 have a human gene ID

IRdown_DGEup_SOD1_VCP <- IRdown_DGEup_SOD1[IRdown_DGEup_SOD1 %in% IRdown_DGEup_VCP]
IRdown_DGEup_C9ORF72_VCP <- IRdown_DGEup_C9ORF72[IRdown_DGEup_C9ORF72 %in% IRdown_DGEup_VCP]
IRdown_DGEup_SOD1_VCP_C9ORF72 <- IRdown_DGEup_SOD1_VCP[IRdown_DGEup_SOD1_VCP %in% IRdown_DGEup_C9ORF72]

IRdown_DGEup_TDP43[IRdown_DGEup_TDP43 %in% IRdown_DGEup_ALS][IRdown_DGEup_TDP43[IRdown_DGEup_TDP43 %in% IRdown_DGEup_ALS] %in% all.reactivity.go] # 19 / 46 overlap relevent to reactivity

IRdown_DGEup_overlap <- cbindX(arrange(as_tibble_col(IRdown_DGEup_SOD1[IRdown_DGEup_SOD1 %in% IRdown_DGEup_VCP], column_name = "SOD1_VCP"), SOD1_VCP), 
                         arrange(as_tibble_col(IRdown_DGEup_C9ORF72[IRdown_DGEup_C9ORF72 %in% IRdown_DGEup_VCP], column_name = "C9ORF72_VCP"), C9ORF72_VCP),  
                         arrange(as_tibble_col(IRdown_DGEup_C9ORF72_VCP[IRdown_DGEup_C9ORF72_VCP %in% IRdown_DGEup_SOD1], column_name = "SOD1_C9ORF72_VCP"), SOD1_C9ORF72_VCP),  
                         arrange(as_tibble_col(IRdown_DGEup_A1[IRdown_DGEup_A1 %in% IRdown_DGEup_ALS], column_name = "Reactive_any.ALS"), Reactive_any.ALS),  
                         # arrange(as_tibble_col(IRdown_DGEup_A1[IRdown_DGEup_A1 %in% IRdown_DGEup_SOD1_VCP_C9ORF72], column_name = "Reactive_all.ALS"), Reactive_all.ALS), # 0
                         arrange(as_tibble_col(IRdown_DGEup_TDP43[IRdown_DGEup_TDP43 %in% IRdown_DGEup_ALS], column_name = "TARDBP.KO_any.ALS"), TARDBP.KO_any.ALS),  
                         # arrange(as_tibble_col(IRdown_DGEup_TDP43[IRdown_DGEup_TDP43 %in% IRdown_DGEup_SOD1_VCP_C9ORF72], column_name = "TARDBP.KO_all.ALS"), TARDBP.KO_all.ALS), # PLOD1 only
                         arrange(as_tibble_col(IRdown_DGEup_TDP43[IRdown_DGEup_TDP43 %in% IRdown_DGEup_A1], column_name = "TARDBP.KO_Reactive"), TARDBP.KO_Reactive),  
                         arrange(as_tibble_col(DGEup_TRAP[DGEup_TRAP %in% IRdown_DGEup_ALS], column_name = "SOD1G37R.TRAP_ALS"), SOD1G37R.TRAP_ALS), 
                         arrange(as_tibble_col(DGEup_TRAP[DGEup_TRAP %in% IRdown_DGEup_A1], column_name = "SOD1G37R.TRAP_Reactive"), SOD1G37R.TRAP_Reactive),
                         arrange(as_tibble_col(DGEup_TRAP[DGEup_TRAP %in% IRdown_DGEup_TDP43], column_name = "SOD1G37R.TRAP_TARDBP.KO"), SOD1G37R.TRAP_TARDBP.KO))
IRdown_DGEup_overlap[is.na(IRdown_DGEup_overlap)] <- ""
write.csv(IRdown_DGEup_overlap, "/camp/home/ziffo/home/projects/astrocyte-meta-analysis/splicing/IRFinder/irlib/results/IRdown_DGEup.overlap.csv")
```

# Figure S6: QC of fractionation

```{r}
# A RNAseq of fractionation
ctrl <- dge_res_ac_ctrl_cyt_vs_nuc %>%  dplyr::select(ensID, log2FoldChange_ctrl = log2FoldChange, padj_ctrl = padj, gene_name, baseMean_ctrl = baseMean)
vcp <- dge_res_ac_vcp_cyt_vs_nuc %>% dplyr::select(ensID, log2FoldChange_vcp = log2FoldChange, padj_vcp = padj, gene_name, baseMean_vcp = baseMean)
ctrl_vcp <- full_join(ctrl, vcp, by = c("gene_name", "ensID"))
cor.test(x = ctrl_vcp$log2FoldChange_ctrl, y = ctrl_vcp$log2FoldChange_vcp)
t.test( x = ctrl_vcp$log2FoldChange_ctrl, y = ctrl_vcp$log2FoldChange_vcp, paired = TRUE )
fractionation_labels = c("MALAT1", "GAPDH", "NEAT", "MALAT1", "GFAP", "TUBB", "ASH2L", "PAF49", "CENPA", "CHOP", "EIF6", "ERS1", "H2AX", "H2AZ1", "H4F3", "H3F3B", "LSD1", "LMNA", "LMNC", "MYC", "NOP2", "NUP98", "SIR1", "SRSF2")
fractionation_labels_filt = c("MLXIPL", "MALAT1", "GAPDH", "NEAT1", "GLUL", "ACTB", "ASS1", " SLC2A2", "APOE", "PCK1", "MALAT1", "GFAP", "TUBB", "PAF49", "CHOP", "ERS1", "H4F3", "LSD1", "HIST", "LMNC", "NOP2", "NUP98", "SIR1", "HIST4H4", "HIST1H2APS3", "HIST1H2APS5", "HIST2H2BD", "HIST2H2BF", "HIST2H2BK", "HIST2H2AK", "HIST1H2PS3") #NLRP6 GCGR
cyt_vs_nuc_ctrl.vcp.scatter <- ggplot(ctrl_vcp, aes( x = log2FoldChange_ctrl, y = log2FoldChange_vcp,  label = gene_name, colour = log10(baseMean_ctrl))) + 
  scale_colour_gradient(low = "dodgerblue1", high = "firebrick1") + 
  geom_point(alpha = 0.5, size = 0.5) +  theme_bw() +  xlab( expression( CTRL~mRNA~LFC~Cytoplasm:Nuclear ) ) +  ylab( expression( VCP~mRNA~LFC~Cytoplasm:Nuclear) ) +
  geom_text_repel(data = filter(ctrl_vcp, gene_name %in%  fractionation_labels_filt), aes(x = log2FoldChange_ctrl, y = log2FoldChange_vcp, label = gene_name), fontface = "italic", colour = "black", min.segment.length = 0, box.padding = 0.5, point.padding = 0.5, size = 3) +
  theme(strip.background = element_rect(colour="black", fill="white"), panel.grid = element_blank(), legend.position = "none") +
  geom_hline(yintercept = 0, linetype = 3, colour = "darkgrey") + geom_vline(xintercept = 0, linetype = 3, colour = "darkgrey") + 
  geom_smooth(data = ctrl_vcp, aes(x = log2FoldChange_ctrl, y = log2FoldChange_vcp), method = "lm", se = FALSE, show.legend = TRUE, colour = "black", linetype = 2) + #geom_abline(slope = 1, intercept = 0, linetype = 2) +
  xlim(-8,8) + ylim(-8,8) +
  annotate(geom="text", x=-4.5, y=8, label=expression("p < 2.2e-16; R = 0.69"), color="black", size = 2.8)  +  
  annotate(geom="text", x=4.5, y=-8, label="nuclear in VCP but not CTRL", color="darkgrey", size = 2.8)  + annotate(geom="text", x=4.5, y=8, label="cytoplasmic in both VCP & CTRL", color="dodgerblue3", size = 2.8)  + 
  annotate(geom="text", x=-4.5, y=6, label="nuclear in CTRL but not VCP", color="darkgrey", size = 2.8) +   annotate(geom="text", x=-4.5, y=-8, label="nuclear in both VCP & CTRL", color="firebrick2", size = 2.8)
cyt_vs_nuc_ctrl.vcp.scatter

# B Mass spec fractionation
ctrl_mass_spec_results <- readRDS("/camp/home/ziffo/home/projects/astrocyte-meta-analysis/expression/mass-spectrometry/ctrl_mass_spec_results.RDS")
vcp_mass_spec_results <- readRDS("/camp/home/ziffo/home/projects/astrocyte-meta-analysis/expression/mass-spectrometry/vcp_mass_spec_results.RDS")
ctrl <- ctrl_mass_spec_results %>% dplyr::select(name, ID, ctrl.lfc = cyt_vs_nuc_ratio) %>% mutate(condition = "ctrl")
vcp <- vcp_mass_spec_results %>% dplyr::select(name, ID, vcp.lfc = cyt_vs_nuc_ratio) %>% mutate(condition = "vcp")
ctrl_vcp <- full_join(ctrl, vcp, by = c("name", "ID"))
cor.test(x = ctrl_vcp$ctrl.lfc, y = ctrl_vcp$vcp.lfc)
t.test( x = ctrl_vcp$ctrl.lfc, y = ctrl_vcp$vcp.lfc, paired = TRUE )
bahar_halpern = c("LMNA", "GFAP", "GAPDH", "TUBB", "GCGR", "MALAT1", "GCK", "NEAT1", "MLXIPL", "INS2", "ALDOA", "INS1", "GLUL", "MLRP6", "APOE", "ASS1", "SLC2A2", "KRT19", "NOS2",
                  "HIST1H2BK", "HIST1H2BM", "HIST1H4A", "HIST2H2AA3", "HIST2H3A", "HIST1H2AC", "HIST1H1E", "HIST1H1C", 
                  "PAF49", "CENPA", "CHOP", "EIF6", "HIF1A", "LSD1", "MYC", "NOP2", "NUP98", "SIRT1")
labels_lenient <- ctrl_vcp %>% filter(name %in% bahar_halpern) %>% arrange( -(abs(ctrl.lfc)) ) %>% distinct(name, .keep_all = TRUE)
mass.spec_cyt_vs_nuc.scatter <- ggplot(ctrl_vcp, aes( x = ctrl.lfc, y = vcp.lfc,  label = name)) + 
  geom_point(colour = "#B768A2", alpha = 0.5, size = 0.5) +  theme_bw() +  xlab( expression( CTRL~protein~LFC~Cytoplasm:Nuclear)  ) +  ylab( expression( VCP~protein~LFC~Cytoplasm:Nuclear) ) +
  geom_text_repel(data = labels_lenient, aes(x = ctrl.lfc, y = vcp.lfc, label = name), fontface = "italic", colour = "black", min.segment.length = 0, box.padding = 0.5, point.padding = 0.5, size = 3) +
  theme(strip.background = element_rect(colour="black", fill="white"), panel.grid = element_blank()) +
  geom_hline(yintercept = 0, linetype = 3, colour = "darkgrey") + geom_vline(xintercept = 0, linetype = 3, colour = "darkgrey") + 
  geom_smooth(data = ctrl_vcp, aes(x = ctrl.lfc, y = vcp.lfc), method = "lm", se = FALSE, show.legend = TRUE, colour = "black", linetype = 2) + 
  xlim(-11,11) + ylim(-13,10) +
  annotate(geom="text", x=-6, y=10, label=expression("P < 2.2e-16; R = 0.89"), color="black", size = 2.8)  +  
  annotate(geom="text", x=6, y=-13, label="nuclear in VCP but not CTRL", color="darkgrey", size = 2.8)  + annotate(geom="text", x=6, y=10, label="cytoplasmic in both VCP & CTRL", color="dodgerblue3", size = 2.8)  + 
  annotate(geom="text", x=-6, y=8, label="nuclear in CTRL but not VCP", color="darkgrey", size = 2.8) + annotate(geom="text", x=-6, y=-13, label="nuclear in both VCP & CTRL", color="firebrick2", size = 2.8)
mass.spec_cyt_vs_nuc.scatter


# PCA fractionation
vsd_ac_nuc_cyt_vcp_vs_ctrl <- readRDS("/camp/home/ziffo/home/projects/astrocyte-fractionation-bulk-rnaseq/deseq2/vsd_ac_nuc_cyt_vcp_vs_ctrl.RDS")
colnames(vsd_ac_nuc_cyt_vcp_vs_ctrl) <- colData(vsd_ac_nuc_cyt_vcp_vs_ctrl)$patient_fraction
pcaData_astrocyte <- plotPCA(vsd_ac_nuc_cyt_vcp_vs_ctrl, intgroup=c("sample", "condition", "mutation", "cellline","fraction", "patient_fraction", "patient"), returnData=TRUE)
percentVar_astrocyte <- round(100 * attr(pcaData_astrocyte, "percentVar"))
ac_fractions_pca <- ggplot(pcaData_astrocyte, aes(PC1, PC2, color=mutation, shape = fraction)) + geom_text_repel(aes(label=patient), size = 4) + 
  scale_shape_manual(values=c(15, 17)) + scale_colour_manual( values = c("dodgerblue", "firebrick2") ) +  theme_bw() +  
  theme(panel.grid = element_blank(), legend.title = element_blank(), legend.text=element_text(size=12), legend.position = "right", legend.spacing = unit(0.2, 'cm')) + geom_point(size=3) + 
  xlab(paste0("PC1: ",percentVar_astrocyte[1],"% variance")) +  ylab(paste0("PC2: ",percentVar_astrocyte[2],"% variance")) + coord_fixed()
ac_fractions_pca

# Dendrogram
sampleDists <- dist(t(assay(vsd_ac_nuc_cyt_vcp_vs_ctrl)))
hc <- hclust(sampleDists, method = "ward.D2")
a <- c("dodgerblue", "dodgerblue", "firebrick2","firebrick2", "dodgerblue", "dodgerblue", "firebrick2","firebrick2")
ac_fractions_dendrogram <- ggdendrogram(hc, rotate = TRUE, theme_dendro = FALSE) +  theme_classic()  + 
  theme(axis.line=element_blank(), axis.title.x=element_blank(), axis.title.y=element_blank(), axis.text.x=element_blank(), axis.ticks=element_blank(), axis.text.y= element_text(size = 12, colour = a), strip.background = element_rect(colour="black", fill="white"), panel.grid = element_blank()) 
ac_fractions_dendrogram
```

## Merge

```{r}
figS6.merged <- ggarrange(
  ggarrange(cyt_vs_nuc_ctrl.vcp.scatter, mass.spec_cyt_vs_nuc.scatter, ncol = 2, labels = c("A", "B")),
  ggarrange(ac_fractions_pca, ac_fractions_dendrogram, ncol = 2, widths = c(2,1), labels = c("C", "D")),
  nrow = 2, heights = c(1, 1.0), labels = c("", ""))
figS6.merged
ggsave(figS6.merged, filename = "/camp/home/ziffo/home/projects/astrocyte-meta-analysis/splicing/IRFinder/irlib/figures/figS6.merged.png", height = 8.25, width = 8.25)
```

# Figure S7: Nuclear & Cytoplasmic, IR & GE

```{r}
# Reliability histogram Nuclear
ac_nuc.reliability_histogram <- histogram_plot_irfinder(ctrl = ac_nuc_ctrl.IR, mutant = ac_nuc_vcp.IR, ctrl.name = "Nuclear CTRL", mutant.name = "Nuclear VCP")
ac_nuc.reliability_histogram

# Reliability histogram Cytoplasmic
ac_cyt.reliability_histogram <- histogram_plot_irfinder(ctrl = ac_cyt_ctrl.IR, mutant = ac_cyt_vcp.IR, ctrl.name = "Cytoplasmic CTRL", mutant.name = "Cytoplasmic VCP", legend = FALSE)
ac_cyt.reliability_histogram

# MA plot nuclear
astrocyte.reactivity.labels.ac_nuc <- c("FLNA", "COL1A1", "MYL6", "HLA-B", "HLA-C", "CSRP1", "MCAM", "ADAM19", "ILK", "RND3", "PLSCR1", "IRF7", "FLOT1", "OSMR", "HLA-A", "PDLIM7")
nuc_ma <- ma_plot_irfinder(data = ac_nuc_vcp_vs_ctrl, labels_list = astrocyte.reactivity.labels.ac_nuc) + ylab(expression( Nuclear~Delta~IR~ratio )) + xlim(0.6,5.2) + ylim(-0.4,0.4) +
  geom_segment(aes(x = 5.0, xend = 5.0, y= 0.1, yend= 0.39), arrow=arrow(length=unit(0.4,"cm")), color = "dodgerblue") +
  geom_segment(aes(x = 5.0, xend = 5.0, y= -0.1, yend= -0.39),arrow=arrow(length=unit(0.4,"cm")), color = "firebrick2") +
  annotate("text", x = 4.2, y = 0.3, label = "VCP IR Up", colour = "dodgerblue", size = 3) +   annotate("text", x = 4.2, y = -0.4, label = "VCP IR Down", colour = "firebrick2", size = 3)
nuc_ma

# MA plot cyto
astrocyte.reactivity.labels.ac_cyt.filt <- c("SFPQ", "DDX39B", "ACADVL", "LENG8", "FADS3", "VIM", "COL1A1", "FN1", "FLNA", "OSMR", "HLA-A", "HLA-B", "HLA-C", "RND3", "SLC26AA6","PIMREG", "MX1", "EMP1",
  "NARPT", "CSRP1", "AGER", "GBP2", "GTBP2", "TGFB1I1", "COL1A1A", "TNC", "NUP199", "IDH1", "TP53", "ILK" ,    
  "MCAM", "FBLN5", "CAPN2", "CDKN1A", "ADAM19", "SERPINE1", "ITGA5", "PDLIM7", "DIXDC1",  "BAG3", "LCN2", "STEAP4", "S1PR3", "TIMP1", "HSPB1", "CXCL10", "CD44", "CP", "GFAP", "SERPINA3",
  "ASPG", "C3", "HLA-E", "MICA", "H2-T23", "H2-D1", "AMIGO2", "SERPING1", "GGTA1P", "GGTA1", "UGT1A1",  "FKBP5", "SRGN", "IIGP1", "S100A10", "CLCF1", "TGM1", "PTX3", "SPHK1", "CD109", "PTGS2", "SLC10A6", "TM4SF1", "B3GNT5", "CD14")  
cyt_ma <- ma_plot_irfinder(data = ac_cyt_vcp_vs_ctrl, labels_list = astrocyte.reactivity.labels.ac_cyt.filt) + ylab(expression( Cytoplasm~Delta~IR~ratio )) + xlim(0.6,5.2) + ylim(-0.4,0.4) +
  geom_segment(aes(x = 5, xend = 5, y= 0.1, yend= 0.39), arrow=arrow(length=unit(0.4,"cm")), color = "dodgerblue") +
  geom_segment(aes(x = 5, xend = 5, y= -0.1, yend= -0.39),arrow=arrow(length=unit(0.4,"cm")), color = "firebrick2") +
  annotate("text", x = 4.2, y = 0.3, label = "VCP IR Up", colour = "dodgerblue", size = 3) +   annotate("text", x = 4.2, y = -0.4, label = "VCP IR Down", colour = "firebrick2", size = 3)
cyt_ma

# Venn overlap whole, nuclear, cyto
ctrl.filt <- ac_who_ctrl.IR %>% dplyr::select(Chr, Start, End, Name, Strand, gene_name, "reliable_ctrl" = reliable_retained)
mutant.filt <- ac_who_vcp.IR %>% dplyr::select(Chr, Start, End, Name, Strand, gene_name, "reliable_vcp" = reliable_retained)
ctrl_and_mutant.filt.who <- ctrl.filt %>% full_join(mutant.filt, by = c("Chr", "Start", "End", "Name", "Strand", "gene_name")) %>% 
  mutate(IR_who = case_when(reliable_ctrl == "reliable-retained" ~ "reliable-retained", reliable_vcp == "reliable-retained" ~ "reliable-retained", TRUE ~ "unreliable-spliced")) %>% 
  dplyr::select(-reliable_ctrl, -reliable_vcp)
ctrl.filt <- ac_nuc_ctrl.IR %>% dplyr::select(Chr, Start, End, Name, Strand, gene_name, "reliable_ctrl" = reliable_retained)
mutant.filt <- ac_nuc_vcp.IR %>% dplyr::select(Chr, Start, End, Name, Strand, gene_name,"reliable_vcp" = reliable_retained)
ctrl_and_mutant.filt.nuc <- ctrl.filt %>% full_join(mutant.filt, by = c("Chr", "Start", "End", "Name", "Strand", "gene_name"))  %>% 
  mutate(IR_nuc = case_when(reliable_ctrl == "reliable-retained" ~ "reliable-retained", reliable_vcp == "reliable-retained" ~ "reliable-retained", TRUE ~ "unreliable-spliced")) %>% 
  dplyr::select(-reliable_ctrl, -reliable_vcp)
ctrl.filt <- ac_cyt_ctrl.IR %>% dplyr::select(Chr, Start, End, Name, Strand, gene_name, "reliable_ctrl" = reliable_retained)
mutant.filt <- ac_cyt_vcp.IR %>% dplyr::select(Chr, Start, End, Name, Strand, gene_name, "reliable_vcp" = reliable_retained)
ctrl_and_mutant.filt.cyt <- ctrl.filt %>% full_join(mutant.filt, by = c("Chr", "Start", "End", "Name", "Strand", "gene_name"))  %>% 
  mutate(IR_cyt = case_when(reliable_ctrl == "reliable-retained" ~ "reliable-retained", reliable_vcp == "reliable-retained" ~ "reliable-retained", TRUE ~ "unreliable-spliced")) %>% 
  dplyr::select(-reliable_ctrl, -reliable_vcp)
who_nuc_IRoverlap <- ctrl_and_mutant.filt.who %>% full_join(ctrl_and_mutant.filt.nuc, c("Chr", "Start", "End", "Name", "Strand", "gene_name"))
who_nuc_cyt_IRoverlap <- who_nuc_IRoverlap %>% full_join(ctrl_and_mutant.filt.cyt, c("Chr", "Start", "End", "Name", "Strand", "gene_name"))
who_nuc_cyt_IRoverlap <- who_nuc_cyt_IRoverlap %>% mutate(overlap = case_when(IR_who == "reliable-retained" & IR_nuc == "reliable-retained" & IR_cyt == "reliable-retained" ~ "overlapping",
                                                                              IR_who == "reliable-retained" & IR_nuc == "reliable-retained" & IR_cyt != "reliable-retained" ~ "whole & nuclear",
                                                                              IR_who == "reliable-retained" & IR_nuc != "reliable-retained" & IR_cyt == "reliable-retained" ~ "whole & cytoplasmic",
                                                                              IR_who != "reliable-retained" & IR_nuc != "reliable-retained" & IR_cyt == "reliable-retained" ~ "cytoplasmic specific",
                                                                              IR_who != "reliable-retained" & IR_nuc == "reliable-retained" & IR_cyt != "reliable-retained" ~ "nuclear specific",
                                                                              IR_who == "reliable-retained" & IR_nuc != "reliable-retained" & IR_cyt != "reliable-retained" ~ "whole specific",
                                                                              TRUE ~ "none"))
who_nuc_cyt_IRoverlap.filt <- filter(who_nuc_cyt_IRoverlap, overlap != "none")

# Of the 16,317 retained introns identified in whole astrocytes, 92% (n = 14,952) were observed in nuclear isolates, 40% (n = 6,493) in cytoplasmic isolates, whilst 37% (n = 6,059) were observed in both 
table(who_nuc_cyt_IRoverlap.filt$overlap)
# cytoplasmic specific     nuclear specific          overlapping  whole & cytoplasmic      whole & nuclear       whole specific 
#                  347                25963                 6059                  434                 8893                  931 
8893+6059 # 14952
14952/16317 # 92%
434+6059 # 6493
6493/16317 # 40%

venn_who_nuc_cyt <- as.ggplot(
  ~draw.triple.venn(
    area1 = nrow( filter(who_nuc_cyt_IRoverlap.filt, overlap %in% c("whole specific", "whole & nuclear", "whole & cytoplasmic", "overlapping" ) ) ), # whole events
    area2 = nrow( filter(who_nuc_cyt_IRoverlap.filt, overlap %in% c("nuclear specific", "whole & nuclear", "cytoplasmic & nuclear", "overlapping" ) ) ), # nuclear events
    area3 = nrow( filter(who_nuc_cyt_IRoverlap.filt, overlap %in% c("cytoplasmic specific", "cytoplasmic & nuclear", "whole & cytoplasmic", "overlapping" ) ) ), # cytoplasmic events
    n12 = nrow( filter(who_nuc_cyt_IRoverlap.filt, overlap %in% c("whole & nuclear", "overlapping" ) ) ), # whole & nuclear
    n23 = nrow( filter(who_nuc_cyt_IRoverlap.filt, overlap %in% c("cytoplasmic & nuclear", "overlapping" ) ) ), # nuclear & cytoplasmic
    n13 = nrow( filter(who_nuc_cyt_IRoverlap.filt, overlap %in% c("whole & cytoplasmic", "overlapping" ) ) ), # whole & nuclear
    n123 = nrow(filter(who_nuc_cyt_IRoverlap.filt, overlap == "overlapping")),
    category = c("whole", "nuclear", "cytoplasm"), fill = c("#BC3C29", "#38618B", "#5A7A7C"), cat.col = c("#BC3C29", "#38618B", "#5A7A7C"),
    alpha = 0.5, cat.fontfamily = "sans",
    cat.pos = c(330,30,0), cat.dist = 0.07) ) # cat.cex = c(1,1,1),
venn_who_nuc_cyt


# Amongst 654 genes with decreased nuclear IR where cytoplasmic proteins were detected, there was a non-significant trend towards greater increases in protein than mRNA in VCP vs control (wilcoxon test p = 0.12
cyt_mass_spec_results <- readRDS("/camp/home/ziffo/home/projects/astrocyte-meta-analysis/expression/mass-spectrometry/cytoplasmic_mass_spec_results.RDS") %>% as_tibble()
ac_nuc_vcp_vs_ctrl.topIR.IRdown.filt <- ac_nuc_vcp_vs_ctrl.absoluteIR %>% group_by(gene_name) %>% slice_max(abs(IRratio.diff), n = 1, with_ties = FALSE)
nucIRdown_cytMS_cytGE.long <- cyt_mass_spec_results %>%  dplyr::select("gene_name" = name, "mass_spec" = vcp_vs_ctrl_ratio) %>% 
  left_join(dplyr::select(dge_res_ac_cyt_vcp_vs_ctrl, gene_name, "gene_expression" = log2FoldChange), by = "gene_name") %>%
  left_join(dplyr::select(ac_nuc_vcp_vs_ctrl.topIR.IRdown.filt, gene_name, IR.direction), by = "gene_name") %>% filter(IR.direction == "down") %>%
  mutate("mass_spec.scaled" = scale(mass_spec), "gene_expression.scaled" = scale(gene_expression)) %>% # # NB the variances are not equal between LFC in GE & LFC in MS > make the variance are equal
  dplyr::select(gene_name, protein = mass_spec.scaled, mRNA = gene_expression.scaled) %>%
 dplyr::select(gene_name, protein, mRNA) %>% pivot_longer(!gene_name, names_to = "type", values_to = "lfc")
cytMS_cytGE_LFC_IRdown_sina <- sinaplot(data = nucIRdown_cytMS_cytGE.long, groups = "type", continuous = "lfc", stat.y.position = 2.5, stats.test = "wilcoxon") + ylim(c(-2.5,2.5)) + ylab(label = expression(Cytoplasm~LFC~VCP:CTRL)) + xlab("IR down genes") +  
  geom_segment(aes(x = 2.5, xend = 2.5, y= 0.5, yend= 2.3), arrow=arrow(length=unit(0.3,"cm")), color = "black") +
  geom_segment(aes(x = 2.5, xend = 2.5, y= -0.5, yend= -2.3),arrow=arrow(length=unit(0.3,"cm")), color = "black") +
  annotate("text", x = 2.3, y = 1.8, label = "VCP Up", size = 2.8, colour = "black") +   annotate("text", x = 2.3, y = -1.8, label = "VCP Down", size = 2.9, colour = "black")
cytMS_cytGE_LFC_IRdown_sina

# [1] "Stats test used =  wilcoxon"
# # A tibble: 1 x 10
#   .y.   group1 group2     n1    n2 statistic     p p.adj p.adj.signif y.position
#   <chr> <chr>  <chr>   <int> <int>     <dbl> <dbl> <dbl> <chr>             <dbl>
# 1 y     mRNA   protein   654   654    201619 0.118 0.118 ns                  2.5


# GO 3 cats nuc IR & cyto DGE
dge_ir_genes <- filter(nuc_ir_cyt_dge, p0.05_reliable == "significant") %>%  split(.$direction) %>%  map("ensID") # create list of IR up/down & Expression up/down genes
dge_ir_genes <- gost(dge_ir_genes) # run gprofiler2 on all groups - runs on all KEGG, GO, REAC,TF, WP pathways simultaneously
dge_ir_gprofiles_filt <- dge_ir_genes$result %>% filter(str_detect(source, "KEGG|GO:BP|GO:MF|GO:CC|REAC|CORUM")) %>% arrange( -log10(p_value) ) %>% mutate( term_name = as.factor(term_name)) 
dge_ir_gprofiles_filt$query <- gsub("expression", "GE", dge_ir_gprofiles_filt$query)
irup_expdown_curated_list <- c("intracellular transport",                                                            
 "Rho GTPase binding",                                                  
 "dendrite cytoplasm",                                           
 # "cytoplasmic microtubule",                                           
 "mitotic cell cycle",                                                  
 # "Metabolism of lipids",                                       
 # "tubulin binding",                                                   
 "axonogenesis",                                                     
 "ribonucleotide binding",                                      
 "GTPase binding",                                               
"neuron development",                                                         
# "protein binding",                                                 
"cytoskeleton",                                                                                                         
"cellular component organization",                                     
"cytoplasm") 
irdown_expdown_curated_list <- c("RNA export from nucleus",
"RNA transport",
"RNA localization",
"transcription by RNA polymerase II",
"nucleocytoplasmic transport",
# "ribonucleoprotein granule",
"nuclear speck",
# "Processing of Capped Intron-Containing Pre-mRNA",
"RNA splicing", "RNA binding", "RNA processing",
"cytoplasm", "RNA binding",
"nucleus", "Nonsense-Mediated Decay (NMD)")
dge_ir_gprofiles_curated_irdown_expdown <- dge_ir_gprofiles_filt %>% filter(query == "IR down & GE down", term_name %in% irdown_expdown_curated_list)
dge_ir_gprofiles_curated_irup_expdown <- dge_ir_gprofiles_filt %>% filter(query == "IR up & GE down", term_name %in% irup_expdown_curated_list)
dge_ir_gprofiles_curated_irup_expup <- dge_ir_gprofiles_filt %>% filter(query == "IR up & GE up")
nucIR_cytGE.go_curated_3cats <- curated_profiles_colours(gprofiles = bind_rows(dge_ir_gprofiles_curated_irdown_expdown, dge_ir_gprofiles_curated_irup_expdown, dge_ir_gprofiles_curated_irup_expup)) 
nucIR_cytGE.go_curated_3cats

# GO 3 cats nuc IR & cyto MS
cyt_mass_spec_results <- readRDS("/camp/home/ziffo/home/projects/astrocyte-meta-analysis/expression/mass-spectrometry/cytoplasmic_mass_spec_results.RDS") %>% as_tibble()
ac_nucIR_cytDGE.absoluteIR <- cleanIRFinderQuantCompared(mutant = ac_nuc_vcp.IR, control = ac_nuc_ctrl.IR, differentialIR = ac_nuc_vcp_vs_ctrl.IRFinder, deseq_res = dge_res_ac_cyt_vcp_vs_ctrl, mass_spec = cyt_mass_spec_results) %>% as_tibble()
ms_ir_genes <- filter(ac_nucIR_cytDGE.absoluteIR, p0.05_reliable == "significant") %>%  split(.$massspec_ir_direction) %>%  map("ensID") # create list of IR up/down & Expression up/down genes
ms_ir_genes <- gost(ms_ir_genes) # run gprofiler2 on all groups - runs on all KEGG, GO, REAC,TF, WP pathways simultaneously
ms_ir_gprofiles_filt <- ms_ir_genes$result %>% filter(str_detect(source, "KEGG|GO:BP|GO:MF|GO:CC|REAC|CORUM")) %>% arrange( -log10(p_value) ) %>% mutate( term_name = as.factor(term_name)) 
ms_ir_gprofiles_filt$query <- gsub("protein", "MS", ms_ir_gprofiles_filt$query)
irdown_proteindown_curated_list <- c("rRNA processing",
"protein targeting to ER",
"Metabolism of proteins",
"Peptide chain elongation",
"actin cytoskeleton",
"Metabolism of RNA",
"Eukaryotic Translation Initiation",
"Nervous system development",
"Axon guidance",
"ribonucleoprotein complex",
"viral process",
"cytoskeletal protein binding",
"protein-containing complex",
"actin binding",
"RNA binding",
"focal adhesion",
"extracellular vesicle")
ms_ir_gprofiles_curated_irdown_proteindown <- ms_ir_gprofiles_filt %>% filter(query == "IR down & MS down", term_name %in% irdown_proteindown_curated_list)

irup_proteindown_curated_list <- c("filamin binding",
"cytoskeleton",
"cytoplasm",
"cytoskeletal protein binding",
"extracellular vesicle")
ms_ir_gprofiles_curated_irup_proteindown <- ms_ir_gprofiles_filt %>% filter(query == "IR up & MS down", term_name %in% irup_proteindown_curated_list)

irup_proteinup_curated_list <- c("RNA stem-loop binding",
"APC-IQGAP1-CLIP-170 complex",
"cytosol")
ms_ir_gprofiles_curated_irup_proteinup <- ms_ir_gprofiles_filt %>% filter(query == "IR up & MS up", term_name %in% irup_proteinup_curated_list)

nucIR_cytMS.go_curated_3cats <- curated_profiles_colours(gprofiles = bind_rows(ms_ir_gprofiles_curated_irdown_proteindown, ms_ir_gprofiles_curated_irup_proteindown, ms_ir_gprofiles_curated_irup_proteinup)) 
nucIR_cytMS.go_curated_3cats
```


## Merge

```{r}
figS7.merged <- ggarrange(
  ggarrange(nuc_ma, ac_nuc.reliability_histogram, ncol = 2, widths = c(1,1.4), labels = c("A", "C")),
  ggarrange(cyt_ma, ac_cyt.reliability_histogram, ncol = 2, widths = c(1,1.4), labels = c("B", "D")),
  ggarrange(venn_who_nuc_cyt, cytMS_cytGE_LFC_IRdown_sina, ncol = 2, labels = c("E", "F"), widths = c(1,1)),
  ggarrange(nucIR_cytGE.go_curated_3cats, nucIR_cytMS.go_curated_3cats, widths = c(1,1), labels = c("G", "H")),
  nrow = 4, labels = c("", "", "", ""), heights = c(1.5,1.4,1.5,2))
figS7.merged
ggsave(figS7.merged, filename = "/camp/home/ziffo/home/projects/astrocyte-meta-analysis/splicing/IRFinder/irlib/figures/figS7.merged.png", height = 12, width = 9)
```

# Figure 4: Nuclear IR & Cytoplasmic GE

Fractionation. Add schematic & western blot to top of figure in Illustrator.

## qPCR

```{r}
# qPCR
qpcr_ct <- readxl::read_xlsx("/camp/home/ziffo/home/projects/astrocyte-fractionation-bulk-rnaseq/sample-details/fractionation_qc_qpcr_raw_data_hamish.xlsx", sheet = "CT_values")
qpcr_ct.df <- qpcr_ct %>% dplyr::select("GAPDH Exon", "GAPDH Intron", "Malat1", "NEAT", "ref_gene") %>% as.data.frame()
group_var <- qpcr_ct %>% pull(fraction)
qpcr_ct.fractions <- qpcr_ct %>% filter(fraction != "Whole")
qpcr_ct.df <- qpcr_ct.fractions %>% dplyr::select("GAPDH Exon","ref_gene") %>% as.data.frame()
group_var <- qpcr_ct.fractions %>% pull(mutation_fraction)
res_cytoplasmic_markers <- pcr_analyze(qpcr_ct.df, group_var = group_var, reference_gene = "ref_gene", reference_group = "ctrl_nuclear") # nuclear reference for cytoplasmic markers
qpcr_ct.fractions <- qpcr_ct %>% filter(fraction != "Whole")
qpcr_ct.df <- qpcr_ct.fractions %>% dplyr::select("GAPDH Intron","Malat1", "NEAT","ref_gene") %>% as.data.frame()
group_var <- qpcr_ct.fractions %>% pull(mutation_fraction)
res_nuclear_markers <- pcr_analyze(qpcr_ct.df,group_var = group_var,reference_gene = "ref_gene", reference_group = "ctrl_cytoplasm") # cytoplasmic reference for nuclear markers

GAPDH_Exon <- filter(res_cytoplasmic_markers, gene == "GAPDH Exon") %>% mutate(mutation = str_split_fixed(group, "_", 2)[,1], fraction = str_split_fixed(group, "_", 2)[,2]) 
GAPDH_Exon_test <- GAPDH_Exon %>% dplyr::select(relative_expression, fraction) %>% t_test(relative_expression ~ fraction, var.equal = TRUE) %>% add_significance() %>% mutate(y.position = 4.2)
GAPDH_Exon_plot <- ggplot(data=GAPDH_Exon, aes(x=fraction, y=relative_expression, fill=fraction)) + geom_bar(stat="identity", width=0.8, color = "black", position=position_dodge()) + 
  geom_errorbar(aes(ymin=lower, ymax=upper), width=.2, position=position_dodge(.9)) + theme_classic() + theme(legend.position="none", axis.text=element_text(size=8),axis.text.x = element_text(angle = 45, hjust = 1)) +  facet_grid(. ~ mutation) + scale_fill_manual(values=c("white", "black")) +  xlab("") + ylab( expression(GAPDH~Exon)  )

GAPDH_Intron <- filter(res_nuclear_markers, gene == "GAPDH Intron") %>% mutate(mutation = str_split_fixed(group, "_", 2)[,1],fraction = str_split_fixed(group, "_", 2)[,2]) 
GAPDH_Intron_test <- GAPDH_Intron %>% dplyr::select(relative_expression, fraction) %>% t_test(relative_expression ~ fraction, var.equal = TRUE) %>% add_significance() %>% mutate(y.position = 4.2)
GAPDH_Intron_plot <- ggplot(data=GAPDH_Intron, aes(x=fraction, y=relative_expression, fill=fraction)) + geom_bar(stat="identity", width=0.8, color = "black", position=position_dodge()) + 
  geom_errorbar(aes(ymin=lower, ymax=upper), width=.2, position=position_dodge(.9)) + theme_classic() + theme(legend.position="none", axis.text=element_text(size=8), axis.text.x = element_text(angle = 45, hjust = 1), axis.title=element_text(size=10)) +  facet_grid(. ~ mutation) +  scale_fill_manual(values=c("white", "black")) +  xlab("") + ylab( expression(GAPDH~Intron)) 

Malat1 <- filter(res_nuclear_markers, gene == "Malat1") %>% mutate(mutation = str_split_fixed(group, "_", 2)[,1],fraction = str_split_fixed(group, "_", 2)[,2]) 
Malat1_test <- Malat1 %>% dplyr::select(relative_expression, fraction) %>% t_test(relative_expression ~ fraction, var.equal = TRUE) %>% add_significance() %>% mutate(y.position = 4.2)
Malat1_plot <- ggplot(data=Malat1, aes(x=fraction, y=relative_expression, fill=fraction)) + geom_bar(stat="identity", width=0.8, color = "black", position=position_dodge()) + 
  geom_errorbar(aes(ymin=lower, ymax=upper), width=.2, position=position_dodge(.9)) + theme_classic() + theme(legend.position="none", axis.text=element_text(size=8),axis.text.x = element_text(angle = 45, hjust = 1)) +  facet_grid(. ~ mutation) +  scale_fill_manual(values=c("white", "black")) +  xlab("") + ylab( expression(Malat1))

NEAT <- filter(res_nuclear_markers, gene == "NEAT") %>% mutate(mutation = str_split_fixed(group, "_", 2)[,1],fraction = str_split_fixed(group, "_", 2)[,2]) 
NEAT_test <- NEAT %>% dplyr::select(relative_expression, fraction) %>% t_test(relative_expression ~ fraction, var.equal = TRUE) %>% add_significance() %>% mutate(y.position = 4.2)
NEAT_plot <- ggplot(data=NEAT, aes(x=fraction, y=relative_expression, fill=fraction)) + geom_bar(stat="identity", width=0.8, color = "black", position=position_dodge()) + 
  geom_errorbar(aes(ymin=lower, ymax=upper), width=.2, position=position_dodge(.9)) + theme_classic() + theme(legend.position="none", axis.text=element_text(size=8), axis.text.x = element_text(angle = 45, hjust = 1)) +  facet_grid(. ~ mutation) +  scale_fill_manual(values=c("white", "black")) +  xlab("") + ylab( expression(NEAT))
```

## IR nuc & cyt

```{r}
# 5.8 fold more retained introns in nuclear than cytoplasmic fractions (41,138 vs 7,036 respectively
ac_nuc_vcp_vs_ctrl.absoluteIR %>% dplyr::filter(A.reliable_retained == "reliable-retained" | B.reliable_retained == "reliable-retained") %>% nrow # 41,138 reliable unique retained introns 
ac_cyt_vcp_vs_ctrl.absoluteIR %>% dplyr::filter(A.reliable_retained == "reliable-retained" | B.reliable_retained == "reliable-retained") %>% nrow # 7,063 reliable unique retained introns 
41138/7063 # 5.8 fold

# reduction in the absolute numbers of retained introns in VCP relative to control in both the nucleus (7% less, 34,464 vs 36,943 respectively) 
ac_nuc_vcp_vs_ctrl.absoluteIR %>% dplyr::filter(A.reliable_retained == "reliable-retained") %>% nrow  # 34,464 reliable unique retained introns in VCP
ac_nuc_vcp_vs_ctrl.absoluteIR %>% dplyr::filter(B.reliable_retained == "reliable-retained") %>% nrow  # 36,943 reliable unique retained introns in control
(36943-34464)/36943 * 100 # 6.7% fewer retained introns in VCP compared with control 

# and the cytoplasm (58% less, 2,791 vs 6,607)
ac_cyt_vcp_vs_ctrl.absoluteIR %>% dplyr::filter(A.reliable_retained == "reliable-retained") %>% nrow  # 2,791 reliable unique retained introns in VCP
ac_cyt_vcp_vs_ctrl.absoluteIR %>% dplyr::filter(B.reliable_retained == "reliable-retained") %>% nrow  # 6,607 reliable unique retained introns in control
(6607-2791)/6607 * 100 # 57.8% fewer retained introns in VCP compared with control 

# Comparing IR levels between VCP and control, revealed 1.8 fold more significantly differentially retained introns in the cytoplasm (5,138 in 3,065 genes) 
ac_cyt_vcp_vs_ctrl %>% dplyr::filter(p0.05_reliable == "significant") %>% nrow # 5,138 reliable introns retained at significantly different levels
ac_cyt_vcp_vs_ctrl %>% dplyr::filter(p0.05_reliable == "significant") %>% distinct(gene_name) %>% nrow # from 3,065 unique genes

# than the nucleus (2,915 in 2,079 genes)
ac_nuc_vcp_vs_ctrl %>% dplyr::filter(p0.05_reliable == "significant") %>% nrow # 2915 reliable introns retained at significantly different levels
ac_nuc_vcp_vs_ctrl %>% dplyr::filter(p0.05_reliable == "significant") %>% distinct(gene_name) %>% nrow # from 2079 unique genes
5138/2915 # 1.76 fold

# these were mostly decreased in VCP in both nuclear (1,882 / 2,915, 65%) and cytoplasmic fractions (5,089 / 5,138, 99%).
ac_nuc_vcp_vs_ctrl %>% dplyr::filter(p0.05_reliable == "significant", IRratio.diff < 0) %>% nrow # 1882 displaying decreased IR levels in VCP
1882/2915 * 100 # 64.6% decreased in VCP
ac_cyt_vcp_vs_ctrl %>% dplyr::filter(p0.05_reliable == "significant", IRratio.diff < 0) %>% nrow # 5089 displaying decreased IR levels in VCP
5089/5138 * 100 # 99.0% decreased in VCP


# VCP vs CTRL nuc & cyto sina
nuclear <- ac_nuc_vcp_vs_ctrl %>% dplyr::filter(p0.05_reliable == "significant") %>% dplyr::select(Intron.GeneName.GeneID, Chr, Start, End, gene_name, ensID, p.diff, IRratio.diff, IRratio.diff, p0.05_reliable, log2FoldChange, p0.05_reliable_IRdirection) %>% mutate(fraction = "nuclear")
cytoplasmic <- ac_cyt_vcp_vs_ctrl %>% dplyr::filter(p0.05_reliable == "significant") %>% dplyr::select(Intron.GeneName.GeneID, Chr, Start, End, gene_name, ensID, p.diff, IRratio.diff, IRratio.diff, p0.05_reliable, log2FoldChange, p0.05_reliable_IRdirection) %>% mutate(fraction = "cytoplasmic")
nuc_cyt <- bind_rows(nuclear, cytoplasmic)
# Sina violin plot: one sample stats nuc & cyto IR ratio LFC VCP vs CTRL
nuc_cyt_sina_plot <- sinaplot.one.sample(data =  filter(nuc_cyt, p0.05_reliable == "significant"), groups = "fraction", continuous = "IRratio.diff", labels = "gene_name", colours = 2,  label_number = 0, stat.y.position = 0.25, width = 0.1, flip = "no", stats.test = "wilcoxon") + ylab(label = expression( Delta~IR~ratio~VCP-CTRL )) + xlab("") + ylim(c(-0.3,0.3)) +
  geom_segment(aes(x = 2.5, xend = 2.5, y= 0.05, yend= 0.25), arrow=arrow(length=unit(0.4,"cm")), color = "black") +
  geom_segment(aes(x = 2.5, xend = 2.5, y= -0.05, yend= -0.25),arrow=arrow(length=unit(0.4,"cm")), color = "black") +
  annotate("text", x = 2.3, y = 0.27, label = "IR Up", colour = "black", size = 2.8) +   annotate("text", x = 2.3, y = -0.27, label = "IR Down", colour = "black", size = 2.8)+
  stat_compare_means(comparisons = list(c("nuclear", "cytoplasmic")), label = "p.label", method = "wilcox", label.y = 0.3, tip.length = 0.01, size = 2.8) 
nuc_cyt_sina_plot

# Comparing this decrease between the cytoplasm and nucleus confirmed that the cytoplasmic decrease was significantly greater (wilcoxon test, p < 2.22e-16)
# p < 2.22e-16

# [1] "Stats test used =  wilcoxon"
# # A tibble: 2 x 9
#   group   .y.   group1 group2      n statistic        p p.signif y.position
#   <chr>   <chr> <chr>  <chr>   <int>     <dbl>    <dbl> <chr>         <dbl>
# 1 cytopl? lfc   1      null m?  5138   169338  0.       ****            0.3
# 2 nuclear lfc   1      null m?  2915  1614252. 2.60e-29 ****            0.3


# noted 10% more differentially retained introns in VCP (31,358 events in 9,630 genes)
ac_vcp_cyt_vs_nuc %>% dplyr::filter(p0.05_reliable == "significant") %>% nrow # 31,358 reliable introns retained at significantly different levels
ac_vcp_cyt_vs_nuc %>% dplyr::filter(p0.05_reliable == "significant") %>% distinct(gene_name) %>% nrow # from 9,630 unique genes
# than control (28,447 events in 9,459 genes), 
ac_ctrl_cyt_vs_nuc %>% dplyr::filter(p0.05_reliable == "significant") %>% nrow # 28,447 reliable introns retained at significantly different levels
ac_ctrl_cyt_vs_nuc %>% dplyr::filter(p0.05_reliable == "significant") %>% distinct(gene_name) %>% nrow # from 9,459 unique genes
(31358-28447)/28447 # 10.2%

# however both conditions exhibited 99.9% being nuclear confined (control: 28,410 / 28,447; VCP: 31,330 / 31,358)
ac_ctrl_cyt_vs_nuc %>% dplyr::filter(p0.05_reliable == "significant", IRratio.diff < 0) %>% nrow # 28,410 displaying decreased IR levels in VCP
28410/28447 * 100 # 99.87% decreased in Cytoplasm
ac_vcp_cyt_vs_nuc %>% dplyr::filter(p0.05_reliable == "significant", IRratio.diff < 0) %>% nrow # 31,330 displaying decreased IR levels in VCP
31330/31358 * 100 # 99.91% decreased in Cytoplasm

ac_ctrl_cyt_vs_nuc.absoluteIR %>% dplyr::filter(A.reliable_retained == "reliable-retained" | B.reliable_retained == "reliable-retained") %>% nrow # 37,724 reliable unique retained introns 
ac_vcp_cyt_vs_nuc.absoluteIR %>% dplyr::filter(A.reliable_retained == "reliable-retained" | B.reliable_retained == "reliable-retained") %>% nrow # 35,093 reliable unique retained introns 
37724/35093 # 1.07 fold more retained introns in CTRL than VCP

# Cyto vs Nuc VCP & CTRL sina
ctrl <- ac_ctrl_cyt_vs_nuc %>% dplyr::filter(p0.05_reliable == "significant") %>% dplyr::select(Intron.GeneName.GeneID, Chr, Start, End, gene_name, ensID,p.diff, IRratio.diff, p0.05_reliable, log2FoldChange) %>% mutate(condition = "CTRL")
vcp <- ac_vcp_cyt_vs_nuc %>% dplyr::filter(p0.05_reliable == "significant") %>% dplyr::select(Intron.GeneName.GeneID, Chr, Start, End, gene_name, ensID,p.diff, IRratio.diff, p0.05_reliable, log2FoldChange) %>% mutate(condition = "VCP")
ctrl_vcp <- bind_rows(ctrl, vcp)
ctrl_vcp$condition <- factor(ctrl_vcp$condition, levels = c("VCP", "CTRL"))

# Sina violin plot: IR ratio LFC Cyt vs Nuc
vcp_ctrl_sina_plot <- sinaplot.one.sample(data =  filter(ctrl_vcp, p0.05_reliable == "significant"), groups = "condition", continuous = "IRratio.diff", labels = "gene_name", colours = 2,  label_number = 0, stat.y.position = 0.4, flip = "no", stats.test = "wilcoxon") + ylab(label = expression(Delta~IR~ratio~Cytoplasm-Nuclear) ) + xlab("") + ylim(c(-0.5,0.5)) +
  geom_segment(aes(x = 2.5, xend = 2.5, y= 0.1, yend= 0.4), arrow=arrow(length=unit(0.4,"cm")), color = "black") +
  geom_segment(aes(x = 2.5, xend = 2.5, y= -0.1, yend= -0.4),arrow=arrow(length=unit(0.4,"cm")), color = "black") +
  annotate("text", x = 2.2, y = 0.45, label = "cytoplasm", size = 2.8, colour = "black") +   annotate("text", x = 2.3, y = -0.45, label = "nuclear", size = 2.8, colour = "black")+
  stat_compare_means(comparisons = list(c("VCP", "CTRL")), label = "p.label", method = "wilcox", label.y = 0.5, tip.length = 0.01, size = 2.8)
vcp_ctrl_sina_plot

# Comparing this nuclear confinement between VCP and control showed no significant difference (wilcoxon test, p = 0.66).
# p = 0.66
# [1] "Stats test used =  wilcoxon"
# # A tibble: 2 x 9
#   group .y.   group1 group2         n statistic     p p.signif y.position
#   <fct> <chr> <chr>  <chr>      <int>     <dbl> <dbl> <chr>         <dbl>
# 1 VCP   lfc   1      null model 31358    477437     0 ****            0.4
# 2 CTRL  lfc   1      null model 28447    502672     0 ****            0.4

# Gene set enrichment analysis of the NMD gene set (n = 120) in the cytoplasmic fraction revealed significant upregulation in VCP (normalised enrichment score 1.15, enrichment p < 0.05)
# # # NMD GSEA
geneList <- dge_res_ac_cyt_vcp_vs_ctrl %>% mutate(log2FoldChange = case_when(log2FoldChange == "NA" ~ 0, TRUE ~ log2FoldChange)) %>% pull(log2FoldChange)
names(geneList) <- dge_res_ac_cyt_vcp_vs_ctrl %>% mutate(log2FoldChange = case_when(log2FoldChange == "NA" ~ 0, TRUE ~ log2FoldChange)) %>% pull(gene_name) %>% as.character()
geneList = sort(geneList, decreasing = TRUE)
NMD_genes.list <- list("NMD" = go.pathways.msigdb$GO_NUCLEAR_TRANSCRIBED_MRNA_CATABOLIC_PROCESS_NONSENSE_MEDIATED_DECAY[go.pathways.msigdb$GO_NUCLEAR_TRANSCRIBED_MRNA_CATABOLIC_PROCESS_NONSENSE_MEDIATED_DECAY %in% dge_res_ac_cyt_vcp_vs_ctrl$gene_name])
fgseaRes <- fgsea(pathways = NMD_genes.list, stats = geneList)
fgseaRes
#    pathway       pval       padj   log2err      ES      NES size
# 1:     NMD 0.04928524 0.04928524 0.3217759 0.28766 1.147124  120

# with 94 / 120 (78%) of the NMD genes being increased (one-sample wilcoxon test p = 2.36x10-7)
filter(dge_res_ac_cyt_vcp_vs_ctrl, gene_name %in% go.pathways.msigdb$GO_NUCLEAR_TRANSCRIBED_MRNA_CATABOLIC_PROCESS_NONSENSE_MEDIATED_DECAY, log2FoldChange > 0) %>% nrow # 94
# NMD sinaplot
dge_res_ac_cyt_vcp_vs_ctrl$fraction <- "Cytoplasm"
nmd_cyt_sina_plot <- sinaplot.one.sample(data =  filter(dge_res_ac_cyt_vcp_vs_ctrl, gene_name %in% go.pathways.msigdb$GO_NUCLEAR_TRANSCRIBED_MRNA_CATABOLIC_PROCESS_NONSENSE_MEDIATED_DECAY), groups = "fraction",
                                             continuous = "log2FoldChange", labels = "gene_name", colours = 2,  label_number = 0, stat.y.position = 1, width = 0.1, flip = "no", stats.test = "wilcoxon") + ylab(label = expression( NMD~GSEA~LFC~VCP:CTRL )) + xlab("") + ylim(c(-1,1)) +
  geom_segment(aes(x = 1.5, xend = 1.5, y= 0.4, yend= 0.8), arrow=arrow(length=unit(0.4,"cm")), color = "black") +
  geom_segment(aes(x = 1.5, xend = 1.5, y= -0.4, yend= -0.8),arrow=arrow(length=unit(0.4,"cm")), color = "black") +
  annotate("text", x = 1.1, y = 0.9, label = "NMD Up", colour = "black", size = 2.8) +   annotate("text", x = 1.1, y = -1, label = "NMD Down", colour = "black", size = 2.8)
nmd_cyt_sina_plot

# [1] "Stats test used =  wilcoxon"
# # A tibble: 1 x 9
#   group   .y.   group1 group2      n statistic        p p.signif y.position
#   <chr>   <chr> <chr>  <chr>   <int>     <dbl>    <dbl> <chr>         <dbl>
# 1 Cytopl? lfc   1      null m?   120      5604  2.36e-7 ****              1

```

## notNMD in nuc & cyt

Run in new R environment to avoid package clashes

```{r}
# ### Differential IR notNMD
# .libPaths("/camp/lab/luscomben/home/users/ziffo/.conda/envs/r4.0.3/lib/R/library")
# load("/camp/home/ziffo/home/projects/astrocyte-meta-analysis/scripts/figures_and_tables.RData")
# library(GenomicRanges)
# library(stringr)
# library(BSgenome.Hsapiens.UCSC.hg38)
# library(Gviz)
# library(rtracklayer)
# library(irlib)
# library(tidyverse)
# library(GeneStructureTools)
# nuclear <- ac_nuc_vcp_vs_ctrl %>% dplyr::select(Intron.GeneName.GeneID.Coords, coords, Chr, Start, End, Direction, intron_length, gc_content, reliable_retained, retained, p0.05_reliable, p0.05_reliable_IRdirection) %>% dplyr::mutate(fraction = "nuclear")
# cytoplasmic <- ac_cyt_vcp_vs_ctrl %>% dplyr::select(Intron.GeneName.GeneID.Coords, coords, Chr, Start, End, Direction, intron_length, gc_content, reliable_retained, retained,p0.05_reliable, p0.05_reliable_IRdirection) %>% dplyr::mutate(fraction = "cytoplasmic")
# nuc_cyt <- bind_rows(nuclear, cytoplasmic)
# gr_IRFinder <- as.GenomicRange(nuc_cyt$Intron.GeneName.GeneID.Coords) # GRanges of IRFinder 8,739 retained introns
# start(gr_IRFinder) = start(gr_IRFinder) +1
# seqlevelsStyle(gr_IRFinder) <- "UCSC"
# gr_IRFinder$id = paste0(seqnames(gr_IRFinder), ":", start(gr_IRFinder), "-", end(gr_IRFinder)) # make an id for each intron
# Homo_sapiens.GRCh38.99.exons <- readRDS("/camp/home/ziffo/home/genomes/ensembl/Homo_sapiens.GRCh38.99.exons.RDS")
# seqlevelsStyle(Homo_sapiens.GRCh38.99.exons) <- "UCSC"
# colnames(mcols(Homo_sapiens.GRCh38.99.exons)) = gsub("biotype","type", colnames(mcols(Homo_sapiens.GRCh38.99.exons))) # changes biotype to type in column names (required for GeneStrutureTools)
# exons.intronRetention <- findIntronContainingTranscripts(gr_IRFinder, Homo_sapiens.GRCh38.99.exons )
# IntronRetentionTranscripts <- addIntronInTranscript(exons.intronRetention, Homo_sapiens.GRCh38.99.exons)
# intron_ids = unique(IntronRetentionTranscripts$whippet_id)
# # get ORF details for each set of transcripts
# g <- BSgenome.Hsapiens.UCSC.hg38::BSgenome.Hsapiens.UCSC.hg38
# # reference transcripts
# normalTranscripts <- Homo_sapiens.GRCh38.99.exons[Homo_sapiens.GRCh38.99.exons$transcript_id %in% unlist(lapply(stringr::str_split(IntronRetentionTranscripts$transcript_id, "[+]"), "[[", 1))]
# orfs_normal <- getOrfs(normalTranscripts, BSgenome = g, returnLongestOnly = FALSE, allFrames = TRUE,uORFs = TRUE)
# orfs_skipped <- getOrfs(IntronRetentionTranscripts[IntronRetentionTranscripts$set == "spliced_intron"],BSgenome = g,returnLongestOnly = FALSE, allFrames = TRUE,uORFs = TRUE)
# orfs_retained <- getOrfs(IntronRetentionTranscripts[IntronRetentionTranscripts$set == "retained_intron"],BSgenome = g,returnLongestOnly = FALSE, allFrames = TRUE,uORFs = TRUE)
# # differences comparing ORFs
# orfChange <- orfDiff(orfsX = orfs_skipped,
#                      orfsY = orfs_retained,
#                      filterNMD = FALSE,
#                      compareBy="gene",
#                      geneSimilarity = TRUE,
#                      compareUTR=TRUE,
#                      allORFs = orfs_normal)
# # add NMD calculations
# orfs_normal$nmd_prob <- notNMD::predictNMD(orfs_normal, "prob")
# orfs_normal$nmd_class <- notNMD::predictNMD(orfs_normal)
# orfs_skipped$nmd_prob <- notNMD::predictNMD(orfs_skipped, "prob")
# orfs_skipped$nmd_class <- notNMD::predictNMD(orfs_skipped)
# orfs_retained$nmd_prob <- notNMD::predictNMD(orfs_retained, "prob")
# orfs_retained$nmd_class <- notNMD::predictNMD(orfs_retained)
# orfs_normal <- orfs_normal[which(!is.na(orfs_normal$orf_length)),]
# orfs_skipped <- orfs_skipped[which(!is.na(orfs_skipped$orf_length)),]
# orfs_retained <- orfs_retained[which(!is.na(orfs_retained$orf_length)),]
# # compare normal and retained isoforms
# # this time setting filterNMD to TRUE, which removes NMD targeted frames/isoforms where possible
# orfChange_filter <- orfDiff(orfsX = orfs_skipped, # normal transcripts
#                      orfsY = orfs_retained, # alternative transcripts
#                      filterNMD = TRUE,
#                      geneSimilarity = TRUE,
#                      compareUTR=TRUE,
#                      allORFs = orfs_normal)
# # test changes in NMD between retained and skipped introns.
# nmdChange <- attrChangeAltSpliced(orfs_skipped, orfs_retained,
#                                   attribute="nmd_prob",
#                                   compareBy="gene",
#                                   useMax=FALSE) # want useMax=F, as we want the minimum NMD probablity (if NMD prob is low, means that at least 1 isoform not targeted)
# m <- match(orfChange_filter$id, nmdChange$id)
# orfChange_filter <- cbind(orfChange_filter, nmdChange[m,-1])
# write.table(orfChange_filter, file="/camp/home/ziffo/home/projects/astrocyte-meta-analysis/splicing/IRFinder/notNMD/nuc_cyt_orf_nmd_changes.txt", sep='\t', row.names = F)
# save.image("/camp/home/ziffo/home/projects/astrocyte-meta-analysis/splicing/IRFinder/notNMD/nuc_cyt_notNMD_results.RData")
# load(file = "/camp/home/ziffo/home/projects/astrocyte-meta-analysis/splicing/IRFinder/notNMD/nuc_cyt_notNMD_results.RData")
# notNMDprob <- orfChange_filter %>% select("coords" = id, "nmd_prob_retained" = nmd_prob_bygroup_y, "nmd_prob_spliced" = nmd_prob_bygroup_x) #
# notNMDprob$coords <- gsub("chr", "", notNMDprob$coords)
# notNMDprob$Chr <- str_split_fixed(notNMDprob$coords, ":", 2)[,1]
# notNMDprob$End <- as.numeric(str_split_fixed(notNMDprob$coords, "-", 2)[,2])
# notNMDprob$Start <- str_split_fixed(notNMDprob$coords, "-", 2)[,1]
# notNMDprob$Start <- as.numeric(str_split_fixed(notNMDprob$Start, ":", 2)[,2])
# notNMDprob <- notNMDprob %>% mutate("Start" = Start - 1) %>% select(-coords) # remove 1 from Start to revert back to original IRFinder coords
# saveRDS(notNMDprob, "/camp/home/ziffo/home/projects/astrocyte-meta-analysis/splicing/IRFinder/notNMD/notNMDprob.nuc_cyt.RDS")
```

Above takes ~2 hours to run and needs high memory - run in an individual script with:

```{bash}
cd /camp/lab/luscomben/home/users/ziffo/projects/astrocyte-meta-analysis/splicing/IRFinder/notNMD

sbatch --cpus-per-task=10 --mem=1500G -N 1 --partition=hmem -t 48:00:00 --wrap="Rscript /camp/lab/luscomben/home/users/ziffo/projects/astrocyte-meta-analysis/scripts/notNMD_nuc_cyt_differentialIR.R" --mail-type=ALL,ARRAY_TASKS --mail-user=oliver.ziff@crick.ac.uk  --job-name=notNMD
```

```{r}
notNMDprob <- readRDS("/camp/home/ziffo/home/projects/astrocyte-meta-analysis/splicing/IRFinder/notNMD/notNMDprob.nuc_cyt.RDS")
nuc_cyt.IR <-  nuc_cyt %>% left_join(notNMDprob, by = c("Chr", "Start", "End")) # left join results to IRFinder
nuc_cyt.IR <-  nuc_cyt.IR %>% mutate(nmd_prob = nmd_prob_retained, #case_when(retained == "retained" ~ nmd_prob_retained, TRUE ~ nmd_prob_spliced),
                                 nmd_outcome = case_when(nmd_prob > 0.5 ~ "NMD", TRUE ~ "not NMD"))

# Differential sina plot
notNMD.IRup_IRdown.nuc_cyt.sina_plot <- ggplot(dplyr::filter(nuc_cyt.IR, p0.05_reliable == "significant"), aes(x = p0.05_reliable_IRdirection, y= nmd_prob, color = p0.05_reliable_IRdirection)) + 
      geom_violin() + geom_sina(size = 0.5) + geom_boxplot(data = dplyr::filter(nuc_cyt.IR, p0.05_reliable == "significant"), aes(fill = p0.05_reliable_IRdirection), colour = "black", width=0.1, outlier.shape = NA) +
      scale_colour_manual(values = c("firebrick2", "dodgerblue3")) + scale_fill_manual(values = c("firebrick2", "dodgerblue3")) +  
      theme_classic() + theme(legend.position = "none", axis.text.x=element_text(size=10), axis.text.y=element_text(size=8)) +
      geom_hline(yintercept = 0.5, linetype = 2) + facet_wrap(. ~ fraction) +
      stat_compare_means(comparisons = list(c("IR down", "IR up")), label = "p.signif", method = "wilcox", label.y = 0.95) + ylab(label = expression(NMD~probability) ) + xlab("")
notNMD.IRup_IRdown.nuc_cyt.sina_plot
```

## Nuc IR & Cyto GE

```{r}
# Nuc IR & Cyto mRNA violin
nuc_ir_cyt_dge$p0.05_reliable_IRdirection.nuc <- gsub("IR", "nuc IR", nuc_ir_cyt_dge$p0.05_reliable_IRdirection)
nucIR_cytDGE_sina_plot <- sinaplot(data = filter(nuc_ir_cyt_dge, p0.05_reliable == "significant"), groups = "p0.05_reliable_IRdirection.nuc", continuous = "log2FoldChange", stat.y.position = 1.5) + ylim(c(-1.5,1.5)) + ylab(label = expression(Cytoplasm~mRNA~VCP:CTRL)) + xlab("") +  
  geom_segment(aes(x = 1.5, xend = 1.5, y= 0.5, yend= 1.2), arrow=arrow(length=unit(0.4,"cm")), color = "black") +
  geom_segment(aes(x = 1.5, xend = 1.5, y= -0.5, yend= -1.3),arrow=arrow(length=unit(0.4,"cm")), color = "black") +
  annotate("text", x = 1.5, y = 1.3, label = "mRNA Up", size = 2.8, colour = "black") +   annotate("text", x = 1.5, y = -1.5, label = "mRNA down", size = 2.8, colour = "black")
nucIR_cytDGE_sina_plot

# [1] "Stats test used =  t_test"
# # A tibble: 1 x 11
#   .y.   group1 group2    n1    n2 statistic    df        p    p.adj
#   <chr> <chr>  <chr>  <int> <int>     <dbl> <dbl>    <dbl>    <dbl>
# 1 y     nuc I? nuc I?  1882  1033      12.6  2905 1.71e-35 1.71e-35


# Scatter plot: Nuclear Delta IR vs Cytoplasmic DGE LFC
labels_lenient <- filter(nuc_ir_cyt_dge, p0.05_reliable == "significant") %>% filter(gene_name %in% astrocyte.reactivity.labels.ac_nucIR_cytDGE) %>% group_by(gene_name) %>% slice_max(-IRratio.diff, n = 1, with_ties = FALSE) 
nucIR_cytDGE_scatter <- ggplot(filter(nuc_ir_cyt_dge, p0.05_reliable == "significant"), aes( x = IRratio.diff, y = log2FoldChange,  label = gene_name, colour = log10(baseMean))) + 
  geom_point(alpha = 0.5, size = 0.5) +  theme_bw() +  xlab( expression( Nuclear~Delta~IR~ratio~VCP-CTRL)  ) +  ylab( expression( Cytoplasm~mRNA~LFC~VCP:CTRL) ) +
  scale_colour_gradient(low = "dodgerblue3", high = "firebrick2") +
  geom_text_repel(data = labels_lenient, aes(x = IRratio.diff, y = log2FoldChange, label = gene_name), 
                  fontface = "italic", colour = "black", min.segment.length = 0, box.padding = 0.5, point.padding = 0.5, size = 2.5, max.overlaps = Inf) +
  theme(strip.background = element_rect(colour="black", fill="white"), panel.grid = element_blank(), legend.position = "none", axis.text=element_text(size=10)) + 
  geom_hline(yintercept = 0, linetype = 3, colour = "darkgrey") + geom_vline(xintercept = 0, linetype = 3, colour = "darkgrey") + 
  geom_smooth(data = nuc_ir_cyt_dge, aes(x = IRratio.diff, y = log2FoldChange), method = "lm", se = FALSE, show.legend = TRUE, colour = "black", linetype = 2) + 
  xlim(-0.36,0.36) +  ylim(-2,2.3) +
  annotate(geom="text", x=.2, y=-2, label="IR up & mRNA down", color="darkgrey")  + annotate(geom="text", x=.2, y=2.2, label="IR up & mRNA up", color="darkgrey")  + 
  annotate(geom="text", x=-.2, y=2.2, label="IR down & mRNA up", color="firebrick2") +   annotate(geom="text", x=-.2, y=-2, label="IR down & mRNA down", color="darkgrey")
nucIR_cytDGE_scatter

# Correlate IR LFC with gene expression LFC
nuc_ir_cyt_dge.filt <- nuc_ir_cyt_dge %>% drop_na(IRratio.diff, log2FoldChange) %>% filter(p0.05_reliable == "significant", IRratio.diff != Inf) %>% filter(IRratio.diff != -Inf) %>% filter(log2FoldChange != Inf) %>% filter(log2FoldChange != -Inf)
cor.test(x = nuc_ir_cyt_dge.filt$IRratio.diff, y = nuc_ir_cyt_dge.filt$log2FoldChange) # R = -0.21
summary( lm( IRratio.diff ~ log2FoldChange, data = nuc_ir_cyt_dge.filt)) 

# Nuc IR down & Cyto DGE up GO terms
dge_ir_genes <- filter(nuc_ir_cyt_dge, p0.05_reliable == "significant") %>%  split(.$direction) %>%  map("ensID") # create list of IR up/down & Expression up/down genes
dge_ir_genes <- gost(dge_ir_genes) # run gprofiler2 on all groups - runs on all KEGG, GO, REAC,TF, WP pathways simultaneously
dge_ir_gprofiles_filt <- dge_ir_genes$result %>% filter(str_detect(source, "KEGG|GO:BP|GO:MF|GO:CC|REAC|CORUM")) %>% arrange( -log10(p_value) ) %>% mutate( term_name = as.factor(term_name)) 
dge_ir_gprofiles_curated_irdown_expup <- dge_ir_gprofiles_filt %>% filter(query == "IR down & expression up")
irdown_expup_curated_list <- c("Interleukin- signaling",                                                                                             
"cellular response to cytokine stimulus",                                                                                  
"actin cytoskeleton",                                                        
"interferon-gamma-mediated signaling pathway",                                                                              
"cytokine-mediated signaling pathway",                                                                      
"cell death",                                                                                           
"response to cytokine",                                                                                  
"NIK-->noncanonical NF-kB signaling",
# "Regulation of Apoptosis",                                                                                 
"positive regulation of Wnt signaling pathway",                                                          
"Nonsense-Mediated Decay (NMD)",                                                                          
"Dectin- mediated noncanonical NF-kB signaling",                                                            
"Downstream signaling events of B Cell Receptor (BCR)",                                                                                                                                 
# "Infectious disease",                                                                                     
"gene expression",                                                                                         
"Antigen processing-Cross presentation",                                                                                                                                                      
"Cellular responses to stress",                                                                                                                                                              
# "peptide biosynthetic process",                                                                                                                                                       
# "extracellular exosome",                                                                             
"Metabolism of RNA",
# "cellular protein metabolic process",                                                                             
"Translation",                                                                                             
"viral process",
# "RNA binding",                                                                                             
# "nucleus",                                                                                                                                                                              
"focal adhesion")
dge_ir_gprofiles_curated_irdown_expup <- dge_ir_gprofiles_filt %>% filter(query == "IR down & expression up")  %>% filter(term_name %in% irdown_expup_curated_list) %>% mutate(query = "")
dge_ir_gprofiles_curated_irdown_expup$term_name <- strtrim(dge_ir_gprofiles_curated_irdown_expup$term_name, 38) # limit term_name string length
dge_ir_gprofiles_curated_irdown_expup$term_name <- gsub("interferon-gamma-mediated signaling pa", "interferon-gamma-mediated signaling", dge_ir_gprofiles_curated_irdown_expup$term_name)
dge_ir_gprofiles_curated_irdown_expup$term_name <- gsub("positive regulation of Wnt signaling p", "positive regulation of Wnt signaling", dge_ir_gprofiles_curated_irdown_expup$term_name)
dge_ir_gprofiles_curated_irdown_expup$term_name <- gsub("NIK-->", "", dge_ir_gprofiles_curated_irdown_expup$term_name)
nuc_IRdown_cytGEup_go_curated <- curated_profiles(gprofiles = dge_ir_gprofiles_curated_irdown_expup, colours = "firebrick2") 
nuc_IRdown_cytGEup_go_curated

 # Of the astrocyte reactivity genes identified in Liddelow et al (25) we observed decreased nuclear IR and increased cytoplasmic expression predominantly in A1 markers (enrichment p = 0.14)
ac_nucIR_cytDGE.IR_GE <- ac_nucIR_cytDGE.absoluteIR %>% group_by(gene_name) %>% dplyr::summarise(IR.mean = mean(IRratio.diff, na.rm = TRUE), GE.lfc = median(log2FoldChange))
ac_nucIR_cytDGE.IR_GE[is.na(ac_nucIR_cytDGE.IR_GE)] <- 0
ac_nucIR_cytDGE.IR_GE <- ac_nucIR_cytDGE.IR_GE %>% data.frame(row.names = "gene_name")
ac_nucIR_cytDGE.IR_GE <- scale(ac_nucIR_cytDGE.IR_GE) # rescale both IR & GE to ensure they have the same variance
ac_nucIR_cytDGE.IR_GE <- ac_nucIR_cytDGE.IR_GE %>% as_tibble(rownames = "gene_name") %>% mutate("IR_DGE" = (GE.lfc + (IR.mean*-1)/2))
geneList = ac_nucIR_cytDGE.IR_GE$IR_DGE
names(geneList) <- as.character(ac_nucIR_cytDGE.IR_GE$gene_name)
geneList = sort(geneList, decreasing = TRUE)
liddelow_genes.list <- list("Astrocyte_reactivity" = astrocyte.reactivity.markers$gene_name[astrocyte.reactivity.markers$gene_name %in% ac_nucIR_cytDGE.absoluteIR$gene_name]) # astrocyte reactivity gene set vector
ac_nucIR_cytDGE.fgseaRes <- fgsea(pathways = liddelow_genes.list, stats = geneList)
ac_nucIR_cytDGE.fgseaRes 
# pathway      pval      padj  log2err        ES      NES
# 1: Astrocyte_reactivity 0.1444759 0.1444759 0.197822 0.3931498 1.224119
```

## Proteomics

```{r}
cyt_mass_spec_results <- readRDS("/camp/home/ziffo/home/projects/astrocyte-meta-analysis/expression/mass-spectrometry/cytoplasmic_mass_spec_results.RDS") %>% as_tibble()
ac_nucIR_cytDGE.absoluteIR <- cleanIRFinderQuantCompared(mutant = ac_nuc_vcp.IR, control = ac_nuc_ctrl.IR, differentialIR = ac_nuc_vcp_vs_ctrl.IRFinder, deseq_res = dge_res_ac_cyt_vcp_vs_ctrl, mass_spec = cyt_mass_spec_results) %>% as_tibble()

# select top IR event per gene
ac_nucIR_cytDGE_cytMS.absoluteIR.topevent <- ac_nucIR_cytDGE.absoluteIR %>% filter(reliable == "reliable") %>% 
  group_by(gene_name) %>% slice_max(abs(IRratio.diff), n = 1, with_ties = FALSE) %>%
  dplyr::select(gene_name, "delta_IR_ratio" = IRratio.diff, "IR_p_value" = p.diff, "IR_significant" = p0.05_reliable_IR.direction,  "gene_expression.lfc" = log2FoldChange, "gene_expression.p_value" = padj, "massspec_p_value" = vcp_vs_ctrl_p.val, "mass_spec.lfc" = vcp_vs_ctrl_ratio, baseMean, IR.direction, IR.lfc)

# no significant differences in protein intensities between genes with decreased and increased nuclear IR in VCP versus control astrocytes (R = -0.02)

# # Nuc IR & Cyto protein violin
nucIR_cytMS <- cyt_mass_spec_results %>% left_join(ac_nuc_vcp_vs_ctrl, by = c("name"="gene_name"))
nucIR_cytMS$p0.05_reliable_IRdirection.nuc <- gsub("IR", "nuc IR", nucIR_cytMS$p0.05_reliable_IRdirection)
nucIR_cytMS_sina_plot <- sinaplot(data = filter(nucIR_cytMS, p0.05_reliable == "significant"), groups = "p0.05_reliable_IRdirection.nuc", continuous = "vcp_vs_ctrl_ratio", stat.y.position = 1.5) + ylim(c(-1.5,1.5)) + ylab(label = expression(Cytoplasm~protein~VCP:CTRL)) + xlab("") +
  geom_segment(aes(x = 1.5, xend = 1.5, y= 0.5, yend= 1.3), arrow=arrow(length=unit(0.4,"cm")), color = "black") +
  geom_segment(aes(x = 1.5, xend = 1.5, y= -0.5, yend= -1.3),arrow=arrow(length=unit(0.4,"cm")), color = "black") +
  annotate("text", x = 1.5, y = 1.4, label = "protein up", size = 2.8, colour = "black") +   annotate("text", x = 1.5, y = -1.5, label = "protein down", size = 2.8, colour = "black")
nucIR_cytMS_sina_plot

# [1] "Stats test used =  t_test"
# # A tibble: 1 x 11
#   .y.   group1      group2       n1    n2 statistic    df     p p.adj p.adj.signif y.position
#   <chr> <chr>       <chr>     <int> <int>     <dbl> <dbl> <dbl> <dbl> <chr>             <dbl>
# 1 y     nuc IR down nuc IR up   341    59     -1.49   398 0.136 0.136 ns                  1.5

# scatter plot: nuclear IR vs cyto protein
nucIR_cytMS_scatter.labels <- filter(ac_nucIR_cytDGE_cytMS.absoluteIR.topevent, delta_IR_ratio < 0 & mass_spec.lfc > 0, gene_name %in% c(astrocyte.reactivity.labels.ac_protein, astrocyte.reactivity.labels.ac_nucIR_cytDGE, astrocyte.reactivity.labels.ac_cyt, astrocyte.reactivity.labels.ac_nuc, astrocyte.reactivity.labels.ac_c9orf72, astrocyte.reactivity.labels.ac_sod1))
nucIR_cytMS_scatter <- ggplot(ac_nucIR_cytDGE_cytMS.absoluteIR.topevent, aes( x = delta_IR_ratio, y = mass_spec.lfc,  label = gene_name)) + 
  geom_point(colour = "dodgerblue3", alpha = 0.5, size = 0.5) +  theme_bw() +  xlab( expression( Nuclear~Delta~IR~ratio~VCP-CTRL)  ) +  ylab( expression( Cytoplasm~protein~LFC~VCP:CTRL) ) +
  geom_text_repel(data = nucIR_cytMS_scatter.labels, 
                  aes(x = delta_IR_ratio, y = mass_spec.lfc, label = gene_name), fontface = "italic", colour = "black", min.segment.length = 0, box.padding = 0.5, point.padding = 0.5, size = 2.5, max.overlaps = Inf) +  theme(strip.background = element_rect(colour="black", fill="white"), panel.grid = element_blank(), legend.position = "none", legend.box.margin=margin(-10,-10,-10,-10),legend.spacing.x = unit(1.0, 'cm'), axis.text=element_text(size=10)) + labs(colour = expression(log[10]~expression)) +
  geom_hline(yintercept = 0, linetype = 3, colour = "darkgrey") + geom_vline(xintercept = 0, linetype = 3, colour = "darkgrey") + 
  geom_smooth(data = ac_nucIR_cytDGE_cytMS.absoluteIR.topevent, aes(x = delta_IR_ratio, y = mass_spec.lfc), method = "lm", se = FALSE, show.legend = TRUE, colour = "black", linetype = 2) + 
  xlim(-0.23,0.23) +  ylim(-3.7,3.7) +
  annotate(geom="text", x=.12, y=-3.5, label="IR up & protein down", color="darkgrey")  + annotate(geom="text", x=.12, y=3.5, label="IR up & protein up", color="darkgrey")  + 
  annotate(geom="text", x=-.12, y=3.5, label="IR down & protein up", color="dodgerblue3") +   annotate(geom="text", x=-.12, y=-3.5, label="IR down & protein down", color="darkgrey")
nucIR_cytMS_scatter

# Correlate IR LFC with gene expression LFC
ac_nucIR_cytDGE.absoluteIR_filtered <- ac_nucIR_cytDGE_cytMS.absoluteIR.topevent %>% 
  drop_na(delta_IR_ratio, mass_spec.lfc) %>% filter(delta_IR_ratio != Inf) %>% filter(delta_IR_ratio != -Inf) %>% filter(mass_spec.lfc != Inf) %>% filter(mass_spec.lfc != -Inf)
cor.test(x = ac_nucIR_cytDGE.absoluteIR_filtered$delta_IR_ratio, y = ac_nucIR_cytDGE.absoluteIR_filtered$mass_spec.lfc) # R = -0.03
summary( lm( delta_IR_ratio ~ mass_spec.lfc, data = ac_nucIR_cytDGE.absoluteIR_filtered)) 

# Nuc IR down & Cyto MS up GO terms
ms_ir_genes <- filter(ac_nucIR_cytDGE.absoluteIR, p0.05_reliable == "significant") %>%  split(.$massspec_ir_direction) %>%  map("ensID") # create list of IR up/down & Expression up/down genes
ms_ir_genes <- gost(ms_ir_genes) # run gprofiler2 on all groups - runs on all KEGG, GO, REAC,TF, WP pathways simultaneously
ms_ir_gprofiles_filt <- ms_ir_genes$result %>% filter(str_detect(source, "KEGG|GO:BP|GO:MF|GO:CC|REAC|CORUM")) %>% arrange( -log10(p_value) ) %>% mutate( term_name = as.factor(term_name)) 
ms_ir_gprofiles_curated_irdown_proteinup <- ms_ir_gprofiles_filt %>% filter(query == "IR down & protein up")
paste('"',unique(ms_ir_gprofiles_curated_irdown_proteinup$term_name),'",', sep = "", collapse = '\n') %>% cat()
irdown_proteinup_curated_list <- c(#"SRP-dependent cotranslational protein targeting to membrane",
"cytoskeletal protein binding",
"Amyotrophic lateral sclerosis",
# "C complex spliceosome",
"cellular localization",
"ribosome",
"viral process",
"cell junction",
# "cytoskeleton",
"Processing of Capped Intron-Containing Pre-mRNA",
"actin cytoskeleton",
# "RNA splicing, via transesterification reactions",
"mRNA splicing, via spliceosome",
# "RNA splicing, via transesterification reactions with bulged adenosine as nucleophile",
"Metabolism of RNA",
"extracellular vesicle",
"spliceosomal complex",
"cadherin binding",
"anchoring junction",
"ribonucleoprotein complex",
"focal adhesion",
"RNA binding")
ms_ir_gprofiles_curated_irdown_proteinup <- ms_ir_gprofiles_filt %>% filter(query == "IR down & protein up")  %>% filter(term_name %in% irdown_proteinup_curated_list) %>% mutate(query = "")
# ms_ir_gprofiles_curated_irdown_proteinup$term_name <- strtrim(ms_ir_gprofiles_curated_irdown_proteinup$term_name, 38) # limit term_name string length
ms_ir_gprofiles_curated_irdown_proteinup$term_name <- gsub("Processing of Capped Intron-Containing Pre-mRNA", "Intron-containing mRNA processing", ms_ir_gprofiles_curated_irdown_proteinup$term_name)
# ms_ir_gprofiles_curated_irdown_proteinup$term_name <- gsub("positive regulation of Wnt signaling p", "positive regulation of Wnt signaling", ms_ir_gprofiles_curated_irdown_proteinup$term_name)
# ms_ir_gprofiles_curated_irdown_proteinup$term_name <- gsub("NIK-->", "", ms_ir_gprofiles_curated_irdown_proteinup$term_name)
nuc_IRdown_cytMSup_go_curated <- curated_profiles(gprofiles = ms_ir_gprofiles_curated_irdown_proteinup, colours = "dodgerblue3") 
nuc_IRdown_cytMSup_go_curated


# # Liddelow heatmap Nuc IR & Cyt DGE & Cyt Protein
annot <- data.frame(row.names = astrocyte.reactivity.markers$gene_name, group = factor(astrocyte.reactivity.markers$group))
annot_cols <- c("black", "firebrick2", "dodgerblue3")
names(annot_cols) <- unique(astrocyte.reactivity.markers$group)
annot_cols <- list(group = annot_cols)
breaksList = seq(-2, 2, by = 0.1)
nuc_ir_cyt_rna_protein.all <- cleanIRFinder(IRFinder = ac_nuc_vcp_vs_ctrl.IRFinder) %>% full_join(dge_res_ac_cyt_vcp_vs_ctrl, by = c("ensID", "gene_name")) %>% full_join(mutate(cyt_mass_spec_results, "gene_name" = name), by = "gene_name") 
dge_ir_filt <- nuc_ir_cyt_rna_protein.all %>% filter(gene_name %in% astrocyte.reactive.genes)  %>% 
  group_by(gene_name) %>% slice_max(-IR.lfc, n = 1, with_ties = FALSE) %>% 
  dplyr::select(gene_name, "Nuclear IR" = IR.lfc, "Cyto mRNA" = log2FoldChange, "Cyto Protein" = vcp_vs_ctrl_ratio)
dge_ir_filt$"Nuclear IR" <- replace_na(dge_ir_filt$"Nuclear IR", 0)
dge_ir_filt$"Cyto mRNA" <- replace_na(dge_ir_filt$"Cyto mRNA", 0)
dge_ir_filt$"Cyto Protein" <- replace_na(dge_ir_filt$"Cyto Protein", 0)
dge_ir_filt <- dge_ir_filt %>% arrange(factor(gene_name, levels = rownames(annot))) # order the matrix by the annotation dataframe
dge_ir_filt.mat <- dge_ir_filt %>% ungroup %>% dplyr::select(.,-gene_name) %>% as.matrix
rownames(dge_ir_filt.mat) = pull(dge_ir_filt,gene_name)
dge_ir_filt.mat[dge_ir_filt.mat == -Inf] <- -2.0
dge_ir_filt.mat[dge_ir_filt.mat == Inf] <- 2.0
dge_ir_filt.mat[dge_ir_filt.mat > 2.0] <- 2.0
dge_ir_filt.mat[dge_ir_filt.mat < -2.0] <- -2.0

dge_res_padj <- nuc_ir_cyt_rna_protein.all %>% filter(gene_name %in% astrocyte.reactive.genes) %>% arrange(IR.lfc) %>% distinct(gene_name, .keep_all = TRUE) %>%
  mutate(IR.sig = case_when(p0.05_reliable == "significant" ~ "\u2217", TRUE ~ ""), DGE.sig = case_when(padj <= 0.05 ~  "\u2217", TRUE ~ ""), prot.sig = case_when(vcp_vs_ctrl_p.val <= 0.05 ~  "\u2217", TRUE ~ "")) %>%  dplyr::select(gene_name, IR.sig, DGE.sig, prot.sig) %>%
  arrange(factor(gene_name, levels = rownames(annot))) %>% dplyr::select(IR.sig,DGE.sig, prot.sig) %>%  as.matrix  # order the matrix by the annotation dataframe
ac_nucIR_cytRNA_cytProt_liddelow_heatmap <- pheatmap(t(dge_ir_filt.mat), cluster_rows=F, show_rownames=T, show_colnames = T, cluster_cols=F, fontsize = 8,
                  gaps_col = c(13,29), gaps_row = c(1,2),
                  annotation_col = annot, annotation_names_col = F, annotation_legend = FALSE, annotation_colors = annot_cols, main = "", display_numbers = t(dge_res_padj), 
                  color = colorRampPalette(rev(greenred(75)))(length(breaksList)), breaks = breaksList,
                  cellheight = 10, cellwidth = 14, border_color=NA)
```

## Merge 

```{r}
fig4.merged <- ggarrange(
  ggarrange(GAPDH_Intron_plot, Malat1_plot, NEAT_plot, GAPDH_Exon_plot, nrow = 1, ncol = 4, labels = c("C")),
  ggarrange(nuc_cyt_sina_plot, vcp_ctrl_sina_plot, nmd_cyt_sina_plot, notNMD.IRup_IRdown.nuc_cyt.sina_plot, labels = c("D", "E", "F", "G"), ncol = 4, widths = c(1,1,0.7,1.6)),
  ggarrange(nucIR_cytDGE_sina_plot, nucIR_cytDGE_scatter, nuc_IRdown_cytGEup_go_curated, ncol = 3, labels = c("H", "I", "J"), widths = c(0.7,1.4,1.1)), # nuc IR & cyt GE
  ggarrange(nucIR_cytMS_sina_plot, nucIR_cytMS_scatter, nuc_IRdown_cytMSup_go_curated, ncol = 3, labels = c("K", "L", "M"), widths = c(0.7,1.4,1.1)), # mass spec
  NULL,
  ggarrange(annotate_figure(ac_nucIR_cytRNA_cytProt_liddelow_heatmap[[4]],  bottom = text_grob("Pan-reactive                                                 A1                                                           A2", color = "black", hjust = 0.6, vjust = -13.2, x = 0.5, face = "italic", size = 10)), labels = "N"), # fourth row
  nrow = 6, heights = c(0.8, 1.3, 1.4, 1.4, 0.02, 0.6))
fig4.merged
ggsave(fig4.merged, filename = "/camp/home/ziffo/home/projects/astrocyte-meta-analysis/splicing/IRFinder/irlib/figures/fig4.merged.png", height = 13.5, width = 10)
```


# Figure S8: Fractions IR characteristics

## GC, length, Phast

```{r}
# Differential IR characteristics dataframe
gr_introns <- as.GenomicRange(ac_nuc_vcp_vs_ctrl$Intron.GeneName.GeneID.Coords)
seqlevelsStyle(gr_introns) <- "UCSC"
ac_nuc_vcp_vs_ctrl$phast_cons <- gscores(phast, gr_introns)$default # takes ~1min - too slow to add to function
nuclear <- ac_nuc_vcp_vs_ctrl %>% dplyr::select(Intron.GeneName.GeneID.Coords, coords, Chr, Start, End, Direction, intron_length, gc_content, reliable_retained, retained, p0.05_reliable, p0.05_reliable_IRdirection, phast_cons) %>% dplyr::mutate(fraction = "nuclear")
gr_introns <- as.GenomicRange(ac_cyt_vcp_vs_ctrl$Intron.GeneName.GeneID.Coords)
seqlevelsStyle(gr_introns) <- "UCSC"
ac_cyt_vcp_vs_ctrl$phast_cons <- gscores(phast, gr_introns)$default # takes ~1min - too slow to add to function
cytoplasmic <- ac_cyt_vcp_vs_ctrl %>% dplyr::select(Intron.GeneName.GeneID.Coords, coords, Chr, Start, End, Direction, intron_length, gc_content, reliable_retained, retained,p0.05_reliable, p0.05_reliable_IRdirection, phast_cons) %>% dplyr::mutate(fraction = "cytoplasmic")
nuc_cyt <- bind_rows(nuclear, cytoplasmic)

# Absolute IR characteristics dataframe - bind_rows (compared retained vs spliced in each fraction)
VCP.nuc.ctrl.filt <- ac_nuc_ctrl.IR %>% dplyr::select(Name.Coords, coords, ensID, gene_name, Chr, Start, End, Strand, gc_content, intron_length, reliable, ctrl_retained = retained) %>% filter(reliable == "reliable") %>% dplyr::mutate(fraction = "nuclear")
VCP.nuc.mutant.filt <- ac_nuc_vcp.IR %>% dplyr::select(Name.Coords, coords, ensID, gene_name, Chr, Start, End, Strand, reliable, vcp_retained = retained, gc_content, intron_length) %>% filter(reliable == "reliable") %>% dplyr::mutate(fraction = "nuclear")
absolute.nuclear <- VCP.nuc.ctrl.filt %>% full_join(VCP.nuc.mutant.filt, by = c("Name.Coords", "coords", "ensID", "gene_name", "Chr", "Start", "End", "Strand", "gc_content", "intron_length", "fraction"))  %>% 
  mutate(retained = case_when(ctrl_retained == "retained" | vcp_retained == "retained" ~ "retained", TRUE ~ "spliced"))
VCP.cyt.ctrl.filt <- ac_cyt_ctrl.IR %>% dplyr::select(Name.Coords, coords, ensID, gene_name, Chr, Start, End, Strand, gc_content, intron_length, reliable, ctrl_retained = retained) %>% filter(reliable == "reliable") %>% dplyr::mutate(fraction = "cytoplasmic")
VCP.cyt.mutant.filt <- ac_cyt_vcp.IR %>% dplyr::select(Name.Coords, coords, ensID, gene_name, Chr, Start, End, Strand, reliable, vcp_retained = retained, gc_content, intron_length) %>% filter(reliable == "reliable") %>% dplyr::mutate(fraction = "cytoplasmic")
absolute.cytoplasmic <- VCP.cyt.ctrl.filt %>% full_join(VCP.cyt.mutant.filt, by = c("Name.Coords", "coords", "ensID", "gene_name", "Chr", "Start", "End", "Strand", "gc_content", "intron_length", "fraction"))  %>% mutate(retained = case_when(ctrl_retained == "retained" | vcp_retained == "retained" ~ "retained", TRUE ~ "spliced"))
absolute.nuc_cyt <- bind_rows(absolute.nuclear, absolute.cytoplasmic)
gr_introns <- as.GenomicRange(absolute.nuc_cyt$Name.Coords)
seqlevelsStyle(gr_introns) <- "UCSC"
absolute.nuc_cyt$phast_cons <- gscores(phast, gr_introns)$default # takes ~1min - too slow to add to function

# # # Absolute IR characteristics - full_join
# VCP.nuc.ctrl.filt <- ac_nuc_ctrl.IR %>% filter(reliable == "reliable") %>%
#   dplyr::select(Name.Coords, coords, ensID, gene_name, Chr, Start, End, Strand, gc_content, intron_length,VCP.nuc.ctrl.retained = retained)
# VCP.nuc.mutant.filt <- ac_nuc_vcp.IR %>% filter(reliable == "reliable") %>%
#   dplyr::select(Name.Coords, coords, ensID, gene_name, Chr, Start, End, Strand, VCP.nuc.mutant.retained = retained, gc_content, intron_length)
# VCP.cyt.ctrl.filt <- ac_cyt_ctrl.IR %>% filter(reliable == "reliable") %>%
#   dplyr::select(Name.Coords, coords, ensID, gene_name, Chr, Start, End, Strand, gc_content, intron_length, VCP.cyt.ctrl.retained = retained)
# VCP.cyt.mutant.filt <- ac_cyt_vcp.IR %>% filter(reliable == "reliable") %>%
#   dplyr::select(Name.Coords, coords, ensID, gene_name, Chr, Start, End, Strand, VCP.cyt.mutant.retained = retained, gc_content, intron_length)
# ctrl.vcp.nuc.cyt.IR <- VCP.nuc.ctrl.filt %>% full_join(VCP.nuc.mutant.filt, by = c("Name.Coords", "coords", "ensID", "gene_name", "Chr", "Start", "End", "Strand", "gc_content", "intron_length")) %>%
#   full_join(VCP.cyt.ctrl.filt, by = c("Name.Coords", "coords", "ensID", "gene_name", "Chr", "Start", "End", "Strand", "gc_content", "intron_length")) %>%
#   full_join(VCP.cyt.mutant.filt, by = c("Name.Coords", "coords", "ensID", "gene_name", "Chr", "Start", "End", "Strand", "gc_content", "intron_length")) %>%
#   drop_na %>%
#   mutate(retained = case_when(VCP.nuc.ctrl.retained == "retained" | VCP.nuc.mutant.retained == "retained" | VCP.cyt.ctrl.retained == "retained" | VCP.cyt.mutant.retained == "retained" ~ "retained",
#                               TRUE ~ "spliced"))

# GC Content in Fractions -------------------
# Absolute Cytoplasmic vs nuclear retained introns
gc.retained_spliced.nuc_cyt.sina_plot <- ggplot(absolute.nuc_cyt, aes(x = retained, y= gc_content, color = retained)) + 
      geom_violin() + geom_sina(size = 0.5) + geom_boxplot(data = absolute.nuc_cyt, aes(fill = retained), colour = "black", width=0.2, outlier.shape = NA) +
      scale_colour_manual(values = c("firebrick2", "dodgerblue3")) + scale_fill_manual(values = c("firebrick2", "dodgerblue3")) +  
      theme_classic() + theme(legend.position = "none", axis.text=element_text(size=10), axis.text.x = element_text(angle = 45, hjust = 1)) +
      geom_hline(yintercept = 0.5, linetype = 2) + facet_wrap(. ~ fraction) +
      stat_compare_means(comparisons = list(c("retained", "spliced")), label = "p.signif", label.y = 0.9) + ylab(label = expression(mean~GC~content) ) + xlab("")
gc.retained_spliced.nuc_cyt.sina_plot

# Differential IR up vs IR down in fractions
gc.IRup_IRdown.nuc_cyt.sina_plot <- ggplot(dplyr::filter(nuc_cyt, p0.05_reliable == "significant"), aes(x = p0.05_reliable_IRdirection, y= gc_content, color = p0.05_reliable_IRdirection)) + 
      geom_violin() + geom_sina(size = 0.5) + geom_boxplot(data = dplyr::filter(nuc_cyt, p0.05_reliable == "significant"), aes(fill = p0.05_reliable_IRdirection), colour = "black", width=0.2, outlier.shape = NA) +
      scale_colour_manual(values = c("firebrick2", "dodgerblue3")) + scale_fill_manual(values = c("firebrick2", "dodgerblue3")) +  
      theme_classic() + theme(legend.position = "none", axis.text=element_text(size=10), axis.text.x = element_text(angle = 45, hjust = 1)) +
      geom_hline(yintercept = 0.5, linetype = 2) + facet_wrap(. ~ fraction)+
      stat_compare_means(comparisons = list(c("IR down", "IR up")), label = "p.signif", label.y = 0.85) + ylab(label = expression(mean~GC~content) ) + xlab("")
gc.IRup_IRdown.nuc_cyt.sina_plot

 
# Intron lengths in fractions -------------------------------
# Cytoplasmic vs nuclear retained introns
intron_length.retained_spliced.nuc_cyt.sina_plot <- ggplot(absolute.nuc_cyt, aes(x = retained, y= intron_length, color = retained)) + 
      geom_violin() + geom_sina(size = 0.5) + geom_boxplot(data = absolute.nuc_cyt, aes(fill = retained), colour = "black", width=0.2, outlier.shape = NA) +
      scale_colour_manual(values = c("firebrick2", "dodgerblue3")) + scale_fill_manual(values = c("firebrick2", "dodgerblue3")) +  
      theme_classic() + theme(legend.position = "none", axis.text=element_text(size=10), axis.text.x = element_text(angle = 45, hjust = 1)) +
      facet_wrap(. ~ fraction) +
      stat_compare_means(comparisons = list(c("retained", "spliced")), label = "p.signif", label.y = 5.7) + ylab(label = expression(log[10]~intron~length) ) + xlab("") +  scale_y_log10() 
intron_length.retained_spliced.nuc_cyt.sina_plot

# IR up vs IR down in fractions
intron_length.IRup_IRdown.nuc_cyt.sina_plot <- ggplot(dplyr::filter(nuc_cyt, p0.05_reliable == "significant"), aes(x = p0.05_reliable_IRdirection, y= intron_length, color = p0.05_reliable_IRdirection)) + 
      geom_violin() + geom_sina(size = 0.5) + geom_boxplot(data = dplyr::filter(nuc_cyt, p0.05_reliable == "significant"), aes(fill = p0.05_reliable_IRdirection), colour = "black", width=0.2, outlier.shape = NA) +
      scale_colour_manual(values = c("firebrick2", "dodgerblue3")) + scale_fill_manual(values = c("firebrick2", "dodgerblue3")) +  
      theme_classic() + theme(legend.position = "none", axis.text=element_text(size=10), axis.text.x = element_text(angle = 45, hjust = 1)) +
      facet_wrap(. ~ fraction)+
      stat_compare_means(comparisons = list(c("IR down", "IR up")), label = "p.signif", label.y = 4.5) + ylab(label = expression(log[10]~intron~length) ) + xlab("") +  scale_y_log10() 
intron_length.IRup_IRdown.nuc_cyt.sina_plot


# PhastCons Score: conservation score -------------------------------
# Cytoplasmic vs nuclear retained introns
PhastCons.retained_spliced.nuc_cyt.sina_plot <- ggplot(absolute.nuc_cyt, aes(x = retained, y= phast_cons, color = retained)) + 
      geom_violin() + geom_sina(size = 0.5) + geom_boxplot(data = absolute.nuc_cyt, aes(fill = retained), colour = "black", width=0.2, outlier.shape = NA) +
      scale_colour_manual(values = c("firebrick2", "dodgerblue3")) + scale_fill_manual(values = c("firebrick2", "dodgerblue3")) +  
      theme_classic() + theme(legend.position = "none", axis.text=element_text(size=10), axis.text.x = element_text(angle = 45, hjust = 1)) +
      facet_wrap(. ~ fraction) +
      stat_compare_means(comparisons = list(c("retained", "spliced")), label = "p.signif", label.y = 0.1) + ylab(label = expression(log[10]~phast~conservation~score) ) + xlab("") +  scale_y_log10() 
PhastCons.retained_spliced.nuc_cyt.sina_plot

# IR up vs IR down in fractions
PhastCons.IRup_IRdown.nuc_cyt.sina_plot <- ggplot(dplyr::filter(nuc_cyt, p0.05_reliable == "significant"), aes(x = p0.05_reliable_IRdirection, y= phast_cons, color = p0.05_reliable_IRdirection)) + 
      geom_violin() + geom_sina(size = 0.5) + geom_boxplot(data = dplyr::filter(nuc_cyt, p0.05_reliable == "significant"), aes(fill = p0.05_reliable_IRdirection), colour = "black", width=0.2, outlier.shape = NA) +
      scale_colour_manual(values = c("firebrick2", "dodgerblue3")) + scale_fill_manual(values = c("firebrick2", "dodgerblue3")) +  
      theme_classic() + theme(legend.position = "none", axis.text=element_text(size=10), axis.text.x = element_text(angle = 45, hjust = 1)) +
      facet_wrap(. ~ fraction)+
      stat_compare_means(comparisons = list(c("IR down", "IR up")), label = "p.signif", label.y = 0.1) + ylab(label = expression(log[10]~phast~conservation~score) ) + xlab("") +  scale_y_log10() 
PhastCons.IRup_IRdown.nuc_cyt.sina_plot
```

## iCLIP

```{r}
# iCLIP xlinks score -------------------------------

### absolute IR iCLIP
# grange_introns <- as.GenomicRange(absolute.nuc_cyt$Name.Coords) # create GRanges for each intron (both retained and non retained)
# intron_coords <- tibble(intron = 1:length(absolute.nuc_cyt$coords), coords = absolute.nuc_cyt$coords) # intron coordinates for to enable matching back to original coords in IRFinder dataframe
# ## sum the iclip xlink scores for each RBP in HepG2 in a for loop
# clip_hepg2 <- list.files("/camp/home/ziffo/home/genomes/clip-bedgraphs/HepG2/")
# clip_hepg2 <- clip_hepg2[!grepl("3nt_peaks",clip_hepg2)] # only assess raw crosslink files without any peak calling for enrichment
# coverage_list_nuc <- list()
# for (i in seq_along(clip_hepg2)) {
#   print(clip_hepg2[[i]])
#   rbp <- str_remove(clip_hepg2[[i]], ".merged.bed.gz")
#   bed <- import(paste("/camp/home/ziffo/home/genomes/clip-bedgraphs/HepG2/", clip_hepg2[[i]],sep=""))
#   seqlevelsStyle(bed) <- "NCBI" # remove "chr"
#   grange_overlap <- findOverlaps(query = grange_introns, subject=bed, ignore.strand=FALSE) # find overlaps between RBP binding sites & retained introns
#   range <- bed[ queryHits(grange_overlap) ] %>% as.data.frame() # subset iCLIP xlinks that overlap with the introns - 50,287 rows each with a xlinks score. There are 8739 introns - many introns have multiple scores.
#   coverage_list_nuc[[i]] <- range %>%  mutate( intron = queryHits(grange_overlap)) %>%  group_by(intron) %>% summarise(iclip_score = sum(score) ) # sum the scores for each intron - 4266 introns with scores, remaining introns have 0 scores
#   coverage_list_nuc[[i]]$iclip_score[is.na(coverage_list_nuc[[i]]$iclip_score)]<-0 # set NA scores to 0
#   coverage_list_nuc[[i]] <- coverage_list_nuc[[i]] %>% left_join(intron_coords, by = "intron") # add intron coordinates to match back to IRFinder results
#   names(coverage_list_nuc)[i] <- rbp
# }
# saveRDS(coverage_list_nuc, "/camp/home/ziffo/home/projects/astrocyte-meta-analysis/splicing/IRFinder/ac_absolute.nuc_cyt.vcp_vs_ctrl.coverage_list_nuc.RDS")
# 
# coverage_list_df <- bind_rows(coverage_list_nuc, .id = "RBP") # collapse list of dataframes into a single df
# # iclip_scores <- coverage_list_df %>%  group_by(RBP) %>%  mutate(row = row_number()) %>%
# #   tidyr::pivot_wider(names_from = intron, values_from = iclip_score, values_fill = 0) %>% select(-row) # each column becomes an RBP. Each row is an intron - there are 7815 in IRFinder but only 7620 have xlink scores.
# iclip_scores <- pivot_wider(coverage_list_df, names_from = RBP, values_from = iclip_score, values_fill = 0) # 18,584 coords
# iclip_scores[is.na(iclip_scores)] <- 0  # set NA scores to 0
# absolute.nuc_cyt.unique.coords <- absolute.nuc_cyt %>% distinct(coords, .keep_all = TRUE)
# iclip_scores$intron_length <- absolute.nuc_cyt.unique.coords$intron_length[iclip_scores$coords %in% absolute.nuc_cyt.unique.coords$coords]
# iclip_scores_normalised <- iclip_scores %>% select(-intron) %>% mutate_if(is.numeric, ~ (./intron_length)) %>% dplyr::select(-intron_length) # normalise all scores for intron_length
# iclip_scores_normalised$sum <- iclip_scores_normalised %>% select_if(is.numeric) %>% rowSums # sum all normalised RBP scores for each intron
# saveRDS(iclip_scores_normalised, "/camp/home/ziffo/home/projects/astrocyte-meta-analysis/splicing/IRFinder/ac_absolute.nuc_cyt.vcp_vs_ctrl.CLIPscores")
iclip_scores_normalised <- readRDS("/camp/home/ziffo/home/projects/astrocyte-meta-analysis/splicing/IRFinder/ac_absolute.nuc_cyt.vcp_vs_ctrl.CLIPscores")
iclip_scores_normalised.filt <- iclip_scores_normalised %>% select(coords, "iclip_enrichment" = sum)
iclip_scores_normalised.filt <- aggregate(iclip_scores_normalised.filt$iclip_enrichment, list(coords = iclip_scores_normalised.filt$coords), FUN = mean) %>% drop_na
absolute.nuc_cyt$iclip_enrichment <- iclip_scores_normalised.filt$x[match(absolute.nuc_cyt$coords,iclip_scores_normalised.filt$coords)];
# set NA to 0
absolute.nuc_cyt$iclip_enrichment[is.na(absolute.nuc_cyt$iclip_enrichment)] <- 0  # set NA scores to 0

# Cytoplasmic vs nuclear retained introns
CLIP.retained_spliced.sina_plot <- ggplot(absolute.nuc_cyt, aes(x = retained, y= iclip_enrichment, color = retained)) + 
      geom_violin() + geom_sina(size = 0.5) + geom_boxplot(data = absolute.nuc_cyt, aes(fill = retained), colour = "black", width=0.2, outlier.shape = NA) +
      scale_colour_manual(values = c("firebrick2", "dodgerblue3")) + scale_fill_manual(values = c("firebrick2", "dodgerblue3")) +  
      theme_classic() + theme(legend.position = "none", axis.text=element_text(size=10), axis.text.x = element_text(angle = 45, hjust = 1)) +
      facet_wrap(. ~ fraction) +
      stat_compare_means(comparisons = list(c("retained", "spliced")), label = "p.signif", label.y = 3.1) + ylab(label = expression(log[10]~RBP~iCLIP~crosslinks~score) ) + xlab("") +  scale_y_log10() 
CLIP.retained_spliced.sina_plot


# differential IR iCLIP
# nuclear <- ac_nuc_vcp_vs_ctrl %>% dplyr::select(Intron.GeneName.GeneID.Coords, coords, intron_length, gc_content, reliable_retained, retained, p0.05_reliable, p0.05_reliable_IRdirection) %>% 
#   dplyr::mutate(fraction = "nuclear")
# cytoplasmic <- ac_cyt_vcp_vs_ctrl %>% dplyr::select(Intron.GeneName.GeneID.Coords, coords, intron_length, gc_content, reliable_retained, retained,p0.05_reliable, p0.05_reliable_IRdirection) %>% 
#   dplyr::mutate(fraction = "cytoplasmic")
# nuc_cyt <- bind_rows(nuclear, cytoplasmic)
# grange_introns <- as.GenomicRange(nuc_cyt$Intron.GeneName.GeneID.Coords) # create GRanges for each intron (both retained and non retained)
# intron_coords <- tibble(intron = 1:length(nuc_cyt$coords), coords = nuc_cyt$coords) # intron coordinates for to enable matching back to original coords in IRFinder dataframe
# ## sum the iclip xlink scores for each RBP in HepG2 in a for loop
# clip_hepg2 <- list.files("/camp/home/ziffo/home/genomes/clip-bedgraphs/HepG2/")
# clip_hepg2 <- clip_hepg2[!grepl("3nt_peaks",clip_hepg2)] # only assess raw crosslink files without any peak calling for enrichment
# coverage_list_nuc <- list()
# for (i in seq_along(clip_hepg2)) {
#   print(clip_hepg2[[i]])
#   rbp <- str_remove(clip_hepg2[[i]], ".merged.bed.gz")
#   bed <- import(paste("/camp/home/ziffo/home/genomes/clip-bedgraphs/HepG2/", clip_hepg2[[i]],sep=""))
#   seqlevelsStyle(bed) <- "NCBI" # remove "chr"
#   grange_overlap <- findOverlaps(query = grange_introns, subject=bed, ignore.strand=FALSE) # find overlaps between RBP binding sites & retained introns
#   range <- bed[ queryHits(grange_overlap) ] %>% as.data.frame() # subset iCLIP xlinks that overlap with the introns - 50,287 rows each with a xlinks score. There are 8739 introns - many introns have multiple scores.
#   coverage_list_nuc[[i]] <- range %>%  mutate( intron = queryHits(grange_overlap)) %>%  group_by(intron) %>% summarise(iclip_score = sum(score) ) # sum the scores for each intron - 4266 introns with scores, remaining introns have 0 scores
#   coverage_list_nuc[[i]]$iclip_score[is.na(coverage_list_nuc[[i]]$iclip_score)]<-0 # set NA scores to 0
#   coverage_list_nuc[[i]] <- coverage_list_nuc[[i]] %>% left_join(intron_coords, by = "intron") # add intron coordinates to match back to IRFinder results
#   names(coverage_list_nuc)[i] <- rbp
# }
# # saveRDS(coverage_list_nuc, "/camp/home/ziffo/home/projects/astrocyte-meta-analysis/splicing/IRFinder/ac_nuc_cyt.vcp_vs_ctrl.coverage_list_nuc.RDS")
# coverage_list_df <- bind_rows(coverage_list_nuc, .id = "RBP") # collapse list of dataframes into a single df
# # iclip_scores <- coverage_list_df %>%  group_by(RBP) %>%  mutate(row = row_number()) %>%
# #   tidyr::pivot_wider(names_from = intron, values_from = iclip_score, values_fill = 0) %>% select(-row) # each column becomes an RBP. Each row is an intron - there are 7815 in IRFinder but only 7620 have xlink scores.
# iclip_scores <- pivot_wider(coverage_list_df, names_from = RBP, values_from = iclip_score, values_fill = 0) # 18,584 coords
# iclip_scores[is.na(iclip_scores)] <- 0  # set NA scores to 0
# nuc_cyt.unique.coords <- nuc_cyt %>% distinct(coords, .keep_all = TRUE)
# iclip_scores$intron_length <- nuc_cyt.unique.coords$intron_length[iclip_scores$coords %in% nuc_cyt.unique.coords$coords]
# iclip_scores_normalised <- iclip_scores %>% select(-intron) %>% mutate_if(is.numeric, ~ (./intron_length)) %>% dplyr::select(-intron_length) # normalise all scores for intron_length
# iclip_scores_normalised$sum <- iclip_scores_normalised %>% select_if(is.numeric) %>% rowSums # sum all normalised RBP scores for each intron
# saveRDS(iclip_scores_normalised, "/camp/home/ziffo/home/projects/astrocyte-meta-analysis/splicing/IRFinder/ac_nuc_cyt.vcp_vs_ctrl.CLIPscores")
iclip_scores_normalised.IRdown.IRup <- readRDS("/camp/home/ziffo/home/projects/astrocyte-meta-analysis/splicing/IRFinder/ac_nuc_cyt.vcp_vs_ctrl.CLIPscores")
iclip_scores_normalised.IRdown.IRup.filt <- iclip_scores_normalised.IRdown.IRup %>% select(coords, "iclip_enrichment" = sum)
iclip_scores_normalised.IRdown.IRup.filt <- aggregate(iclip_scores_normalised.IRdown.IRup.filt$iclip_enrichment, list(coords = iclip_scores_normalised.IRdown.IRup.filt$coords), FUN = mean) %>% drop_na
nuc_cyt$iclip_enrichment <- iclip_scores_normalised.IRdown.IRup.filt$x[match(nuc_cyt$coords,iclip_scores_normalised.IRdown.IRup.filt$coords)];
# set NA to 0
nuc_cyt$iclip_enrichment[is.na(nuc_cyt$iclip_enrichment)] <- 0  # set NA scores to 0

# IR up vs IR down in fractions
CLIP.IRup_IRdown.sina_plot <- ggplot(dplyr::filter(nuc_cyt, p0.05_reliable == "significant"), aes(x = p0.05_reliable_IRdirection, y= iclip_enrichment, color = p0.05_reliable_IRdirection)) + 
      geom_violin() + geom_sina(size = 0.5) + geom_boxplot(data = dplyr::filter(nuc_cyt, p0.05_reliable == "significant"), aes(fill = p0.05_reliable_IRdirection), colour = "black", width=0.2, outlier.shape = NA) +
      scale_colour_manual(values = c("firebrick2", "dodgerblue3")) + scale_fill_manual(values = c("firebrick2", "dodgerblue3")) +  
      theme_classic() + theme(legend.position = "none", axis.text=element_text(size=10), axis.text.x = element_text(angle = 45, hjust = 1)) +
      facet_wrap(. ~ fraction)+
      stat_compare_means(comparisons = list(c("IR down", "IR up")), label = "p.signif", label.y = 3) + ylab(label = expression(log[10]~RBP~iCLIP~crosslinks~score) ) + xlab("") +  scale_y_log10() 
CLIP.IRup_IRdown.sina_plot
```

## MaxEntScan

```{r}
## MaxEntScan https://github.com/monahton/GencoDymo https://rdrr.io/github/monahton/GencoDymo/f/vignettes/Vignette.Rmd https://bioconductor.org/packages/devel/bioc/manuals/VarCon/man/VarCon.pdf
### MaxEntScan Absolute IR
ctrl.vcp.nuc.cyt.IR.coords <- ctrl.vcp.nuc.cyt.IR %>% dplyr::select(coords)
hg38 <- BSgenome.Hsapiens.UCSC.hg38::BSgenome.Hsapiens.UCSC.hg38
# download.file(url="ftp://ftp.ebi.ac.uk/pub/databases/gencode/Gencode_human/release_36/gencode.v36.basic.annotation.gtf.gz",
#               destfile = "/camp/home/ziffo/home/genomes/gencode/GRCh38.p13.v36.basic.annotation.gtf.gz", method = "wget")
gen36 <- load_gtf("/camp/home/ziffo/home/genomes/gencode/GRCh38.p13.v36.basic.annotation.gtf.gz")
gen36_introns <- extract_introns(gen36) # Extract introns from the gtf file
gen36_introns_unique <- eliminate_redundant_introns(gen36_introns) # Remove redundant introns
# filter gen36_introns gtf to IRFinder differentially retained introns
ctrl.vcp.nuc.cyt.IR.coords <- ctrl.vcp.nuc.cyt.IR %>% dplyr::select("seqnames" = Chr, "intron_start" = Start, "intron_end" = End, "strand" = Strand) %>%
  mutate(seqnames = paste("chr", seqnames, sep = ""),  # add chr to match gencode annotation style
         intron_start = 1 + intron_start) # add 1 to start position of intron to avoid overlapping the exon
gen36_IRFinder_introns <- ctrl.vcp.nuc.cyt.IR.coords %>% left_join(gen36_introns_unique, by = c("seqnames", "intron_start", "intron_end", "strand"))
setwd("/camp/home/ziffo/home/projects/astrocyte-meta-analysis/splicing/IRFinder/MaxEntScan") # set director to save splice site fasta files for MaxEntScan
ss5_motif_df <- extract_5ss_motif(gen36_IRFinder_introns, BSgenome.Hsapiens.UCSC.hg38) # Produce a fasta file with 5'ss 9-mers:
ss3_motif_df <- extract_3ss_motif(gen36_IRFinder_introns, BSgenome.Hsapiens.UCSC.hg38) # Produce a fasta file with 3'ss 23-mers:
ss_motif <- ss5_motif_df %>% left_join(ss3_motif_df, by = c("seqnames", "strand", "transcript_id", "intron_number")) %>%
  dplyr::rename("motif_start.5" = motif_start.x, "motif_start.3" = motif_start.y, "motif_end.5" = motif_end.x, "motif_end.3" = motif_end.y) %>%
  mutate(MaxEntScore_5ss = calculateMaxEntScanScore(donor_ss_motif,5), MaxEntScore_3ss = calculateMaxEntScanScore(acceptor_ss_motif,3)) %>% # Calculate MaxEntScores at both splice sites
  left_join(dplyr::select(gen36_introns_unique, seqnames, intron_start, intron_end, strand, transcript_id, intron_number), by = c("seqnames", "strand", "transcript_id", "intron_number")) %>% # intron start end
  mutate(seqnames = gsub("chr", "", seqnames), intron_start = intron_start - 1) %>% rename("Chr" = seqnames, "Strand" = strand, "Start" = intron_start, "End" = intron_end)
saveRDS(ss_motif, "/camp/lab/luscomben/home/users/ziffo/projects/astrocyte-meta-analysis/splicing/IRFinder/MaxEntScan/MaxEntScores.absolute_ctrl.vcp.nuc.cyt.IR.RDS")
```

Run above in separate script with:

```{bash}
cd /camp/lab/luscomben/home/users/ziffo/projects/astrocyte-meta-analysis/splicing/IRFinder/MaxEntScan

sbatch -N 1 -c 8 --mem=20G -t 48:00:00 --wrap="Rscript /camp/lab/luscomben/home/users/ziffo/projects/astrocyte-meta-analysis/scripts/MaxEntScan_nuc_cyt_absolute_retained_vs_spliced.R" --mail-type=ALL,ARRAY_TASKS --mail-user=oliver.ziff@crick.ac.uk --job-name=MaxEntScan
```

```{r}
ss_motif <- readRDS("/camp/lab/luscomben/home/users/ziffo/projects/astrocyte-meta-analysis/splicing/IRFinder/MaxEntScan/MaxEntScores.absolute_ctrl.vcp.nuc.cyt.IR.RDS")
absolute.nuc_cyt.IR <- absolute.nuc_cyt %>% left_join(dplyr::select(ss_motif, Chr, Start, End, Strand, MaxEntScore_5ss, MaxEntScore_3ss), by = c("Chr", "Start", "End", "Strand")) %>%
  mutate(MaxEntScore_5ss = as.numeric(MaxEntScore_5ss), MaxEntScore_3ss = as.numeric(MaxEntScore_3ss)) # add MaxEntScan scores back to IRFinder results

maxent5.retained_spliced.sina_plot <- ggplot(absolute.nuc_cyt.IR, aes(x = retained, y= MaxEntScore_5ss, color = retained)) + 
      geom_violin() + geom_sina(size = 0.5) + geom_boxplot(data = absolute.nuc_cyt.IR, aes(fill = retained), colour = "black", width=0.2, outlier.shape = NA) +
      scale_colour_manual(values = c("firebrick2", "dodgerblue3")) + scale_fill_manual(values = c("firebrick2", "dodgerblue3")) +  
      theme_classic() + theme(legend.position = "none", axis.text=element_text(size=10), axis.text.x = element_text(angle = 45, hjust = 1)) +
      facet_wrap(. ~ fraction) + ylim(c(0,12)) + 
      stat_compare_means(comparisons = list(c("retained", "spliced")), label = "p.signif", label.y = 11.5) + ylab(label = "5' MaxEntScore" ) + xlab("")
maxent5.retained_spliced.sina_plot

maxent3.retained_spliced.sina_plot <- ggplot(absolute.nuc_cyt.IR, aes(x = retained, y= MaxEntScore_3ss, color = retained)) + 
      geom_violin() + geom_sina(size = 0.5) + geom_boxplot(data = absolute.nuc_cyt.IR, aes(fill = retained), colour = "black", width=0.2, outlier.shape = NA) +
      scale_colour_manual(values = c("firebrick2", "dodgerblue3")) + scale_fill_manual(values = c("firebrick2", "dodgerblue3")) +  
      theme_classic() + theme(legend.position = "none", axis.text=element_text(size=10), axis.text.x = element_text(angle = 45, hjust = 1)) +
      facet_wrap(. ~ fraction) + ylim(c(0,15)) + 
      stat_compare_means(comparisons = list(c("retained", "spliced")), label = "p.signif", label.y = 14.5) + ylab(label = "3' MaxEntScore" ) + xlab("")
maxent3.retained_spliced.sina_plot


# ### MaxEntScan Differential IR
# nuc_cyt.IR.coords <- nuc_cyt %>% dplyr::select(coords)
# hg38 <- BSgenome.Hsapiens.UCSC.hg38::BSgenome.Hsapiens.UCSC.hg38
# # download.file(url="ftp://ftp.ebi.ac.uk/pub/databases/gencode/Gencode_human/release_36/gencode.v36.basic.annotation.gtf.gz",
# #               destfile = "/camp/home/ziffo/home/genomes/gencode/GRCh38.p13.v36.basic.annotation.gtf.gz", method = "wget")
# gen36 <- load_gtf("/camp/home/ziffo/home/genomes/gencode/GRCh38.p13.v36.basic.annotation.gtf.gz")
# gen36_introns <- extract_introns(gen36) # Extract introns from the gtf file
# gen36_introns_unique <- eliminate_redundant_introns(gen36_introns) # Remove redundant introns
# # filter gen36_introns gtf to IRFinder differentially retained introns
# nuc_cyt.IR.coords <- nuc_cyt %>% dplyr::select("seqnames" = Chr, "intron_start" = Start, "intron_end" = End, "strand" = Direction) %>%
#   mutate(seqnames = paste("chr", seqnames, sep = ""),  # add chr to match gencode annotation style
#          intron_start = 1 + intron_start) # add 1 to start position of intron to avoid overlapping the exon
# gen36_IRFinder_introns <- nuc_cyt.IR.coords %>% left_join(gen36_introns_unique, by = c("seqnames", "intron_start", "intron_end", "strand"))
# setwd("/camp/home/ziffo/home/projects/astrocyte-meta-analysis/splicing/IRFinder/MaxEntScan") # set director to save splice site fasta files for MaxEntScan
# ss5_motif_df <- extract_5ss_motif(gen36_IRFinder_introns, BSgenome.Hsapiens.UCSC.hg38) # Produce a fasta file with 5'ss 9-mers:
# ss3_motif_df <- extract_3ss_motif(gen36_IRFinder_introns, BSgenome.Hsapiens.UCSC.hg38) # Produce a fasta file with 3'ss 23-mers:
# ss_motif <- ss5_motif_df %>% left_join(ss3_motif_df, by = c("seqnames", "strand", "transcript_id", "intron_number")) %>%
#   dplyr::rename("motif_start.5" = motif_start.x, "motif_start.3" = motif_start.y, "motif_end.5" = motif_end.x, "motif_end.3" = motif_end.y) %>%
#   mutate(MaxEntScore_5ss = calculateMaxEntScanScore(donor_ss_motif,5), MaxEntScore_3ss = calculateMaxEntScanScore(acceptor_ss_motif,3)) %>% # Calculate MaxEntScores at both splice sites
#   left_join(dplyr::select(gen36_introns_unique, seqnames, intron_start, intron_end, strand, transcript_id, intron_number), by = c("seqnames", "strand", "transcript_id", "intron_number")) %>% # intron start end
#   mutate(seqnames = gsub("chr", "", seqnames), intron_start = intron_start - 1) %>% rename("Chr" = seqnames, "Direction" = strand, "Start" = intron_start, "End" = intron_end)
# saveRDS(ss_motif, "/camp/lab/luscomben/home/users/ziffo/projects/astrocyte-meta-analysis/splicing/IRFinder/MaxEntScan/MaxEntScores.differential_nuc_cyt.IR.RDS")

ss_motif.IRdown.IRup <- readRDS("/camp/lab/luscomben/home/users/ziffo/projects/astrocyte-meta-analysis/splicing/IRFinder/MaxEntScan/MaxEntScores.differential_nuc_cyt.IR.RDS")
nuc_cyt.IR <- nuc_cyt %>% left_join(dplyr::select(ss_motif.IRdown.IRup, Chr, Start, End, Direction, MaxEntScore_5ss, MaxEntScore_3ss), by = c("Chr", "Start", "End", "Direction")) %>%
  mutate(MaxEntScore_5ss = as.numeric(MaxEntScore_5ss), MaxEntScore_3ss = as.numeric(MaxEntScore_3ss)) # add MaxEntScan scores back to IRFinder results

maxent5.nuc_cyt.differential.sina_plot <- ggplot(dplyr::filter(nuc_cyt.IR, p0.05_reliable == "significant"), aes(x = p0.05_reliable_IRdirection, y= MaxEntScore_5ss, color = p0.05_reliable_IRdirection)) + 
      geom_violin() + geom_sina(size = 0.5) + 
  geom_boxplot(data = dplyr::filter(nuc_cyt.IR, p0.05_reliable == "significant"), aes(fill = p0.05_reliable_IRdirection), colour = "black", width=0.2, outlier.shape = NA) + 
  scale_colour_manual(values = c("firebrick2", "dodgerblue3")) + scale_fill_manual(values = c("firebrick2", "dodgerblue3")) +  
      theme_classic() + theme(legend.position = "none", axis.text=element_text(size=10), axis.text.x = element_text(angle = 45, hjust = 1)) +
      facet_wrap(. ~ fraction) +  ylim(c(0,12)) +
      stat_compare_means(comparisons = list(c("IR down", "IR up")), label = "p.signif", label.y = 11) + ylab(label = "5' MaxEntScore" ) + xlab("") 
maxent5.nuc_cyt.differential.sina_plot

maxent3.nuc_cyt.differential.sina_plot <- ggplot(dplyr::filter(nuc_cyt.IR, p0.05_reliable == "significant"), aes(x = p0.05_reliable_IRdirection, y= MaxEntScore_3ss, color = p0.05_reliable_IRdirection)) + 
      geom_violin() + geom_sina(size = 0.5) + 
  geom_boxplot(data = dplyr::filter(nuc_cyt.IR, p0.05_reliable == "significant"), aes(fill = p0.05_reliable_IRdirection), colour = "black", width=0.2, outlier.shape = NA) + 
  scale_colour_manual(values = c("firebrick2", "dodgerblue3")) + scale_fill_manual(values = c("firebrick2", "dodgerblue3")) +  
      theme_classic() + theme(legend.position = "none", axis.text=element_text(size=10), axis.text.x = element_text(angle = 45, hjust = 1)) +
      facet_wrap(. ~ fraction) +  ylim(c(0,15)) +
      stat_compare_means(comparisons = list(c("IR down", "IR up")), label = "p.signif", label.y = 14) + ylab(label = "3' MaxEntScore" ) + xlab("") 
maxent3.nuc_cyt.differential.sina_plot
```

## notNMD

```{r}
### notNMD ----------------------
# GeneStructureTools functions clash with other packages so need a fresh R environment (dont source irfinder_functions.R)

### Absolute IR
.libPaths("/camp/lab/luscomben/home/users/ziffo/.conda/envs/r4.0.3/lib/R/library")
load("/camp/home/ziffo/home/projects/astrocyte-meta-analysis/scripts/figures_and_tables.RData")
library(GenomicRanges)
library(stringr)
library(BSgenome.Hsapiens.UCSC.hg38)
library(Gviz)
library(rtracklayer)
library(irlib)
library(tidyverse)
library(GeneStructureTools)
# Absolute IR characteristics dataframe - bind_rows (compared retained vs spliced in each fraction)
VCP.nuc.ctrl.filt <- ac_nuc_ctrl.IR %>% dplyr::select(Name.Coords, coords, ensID, gene_name, Chr, Start, End, Strand, gc_content, intron_length, reliable, ctrl_retained = retained) %>% filter(reliable == "reliable") %>% dplyr::mutate(fraction = "nuclear")
VCP.nuc.mutant.filt <- ac_nuc_vcp.IR %>% dplyr::select(Name.Coords, coords, ensID, gene_name, Chr, Start, End, Strand, reliable, vcp_retained = retained, gc_content, intron_length) %>% filter(reliable == "reliable") %>% dplyr::mutate(fraction = "nuclear")
absolute.nuclear <- VCP.nuc.ctrl.filt %>% full_join(VCP.nuc.mutant.filt, by = c("Name.Coords", "coords", "ensID", "gene_name", "Chr", "Start", "End", "Strand", "gc_content", "intron_length", "fraction"))  %>% 
  mutate(retained = case_when(ctrl_retained == "retained" | vcp_retained == "retained" ~ "retained", TRUE ~ "spliced"))
VCP.cyt.ctrl.filt <- ac_cyt_ctrl.IR %>% dplyr::select(Name.Coords, coords, ensID, gene_name, Chr, Start, End, Strand, gc_content, intron_length, reliable, ctrl_retained = retained) %>% filter(reliable == "reliable") %>% dplyr::mutate(fraction = "cytoplasmic")
VCP.cyt.mutant.filt <- ac_cyt_vcp.IR %>% dplyr::select(Name.Coords, coords, ensID, gene_name, Chr, Start, End, Strand, reliable, vcp_retained = retained, gc_content, intron_length) %>% filter(reliable == "reliable") %>% dplyr::mutate(fraction = "cytoplasmic")
absolute.cytoplasmic <- VCP.cyt.ctrl.filt %>% full_join(VCP.cyt.mutant.filt, by = c("Name.Coords", "coords", "ensID", "gene_name", "Chr", "Start", "End", "Strand", "gc_content", "intron_length", "fraction"))  %>% mutate(retained = case_when(ctrl_retained == "retained" | vcp_retained == "retained" ~ "retained", TRUE ~ "spliced"))
absolute.nuc_cyt <- bind_rows(absolute.nuclear, absolute.cytoplasmic)
gr_introns <- as.GenomicRange(absolute.nuc_cyt$Name.Coords)
seqlevelsStyle(gr_introns) <- "UCSC"
absolute.nuc_cyt$phast_cons <- gscores(phast, gr_introns)$default # takes ~1min - too slow to add to function

gr_IRFinder <- as.GenomicRange(absolute.nuc_cyt$Name.Coords) # GRanges of IRFinder 8,739 retained introns
start(gr_IRFinder) = start(gr_IRFinder) +1
seqlevelsStyle(gr_IRFinder) <- "UCSC"
gr_IRFinder$id = paste0(seqnames(gr_IRFinder), ":", start(gr_IRFinder), "-", end(gr_IRFinder)) # make an id for each intron
Homo_sapiens.GRCh38.99.exons <- readRDS("/camp/home/ziffo/home/genomes/ensembl/Homo_sapiens.GRCh38.99.exons.RDS")
seqlevelsStyle(Homo_sapiens.GRCh38.99.exons) <- "UCSC"
colnames(mcols(Homo_sapiens.GRCh38.99.exons)) = gsub("biotype","type", colnames(mcols(Homo_sapiens.GRCh38.99.exons))) # changes biotype to type in column names (required for GeneStrutureTools)
exons.intronRetention <- findIntronContainingTranscripts(gr_IRFinder, Homo_sapiens.GRCh38.99.exons )
IntronRetentionTranscripts <- addIntronInTranscript(exons.intronRetention, Homo_sapiens.GRCh38.99.exons)
intron_ids = unique(IntronRetentionTranscripts$whippet_id)
# get ORF details for each set of transcripts
g <- BSgenome.Hsapiens.UCSC.hg38::BSgenome.Hsapiens.UCSC.hg38

# reference transcripts
normalTranscripts <- Homo_sapiens.GRCh38.99.exons[Homo_sapiens.GRCh38.99.exons$transcript_id %in% unlist(lapply(stringr::str_split(IntronRetentionTranscripts$transcript_id, "[+]"), "[[", 1))]
orfs_normal <- getOrfs(normalTranscripts, BSgenome = g, returnLongestOnly = FALSE, allFrames = TRUE,uORFs = TRUE)
orfs_skipped <- getOrfs(IntronRetentionTranscripts[IntronRetentionTranscripts$set == "spliced_intron"],BSgenome = g,returnLongestOnly = FALSE, allFrames = TRUE,uORFs = TRUE)
orfs_retained <- getOrfs(IntronRetentionTranscripts[IntronRetentionTranscripts$set == "retained_intron"],BSgenome = g,returnLongestOnly = FALSE, allFrames = TRUE,uORFs = TRUE)
# differences comparing ORFs
orfChange <- orfDiff(orfsX = orfs_skipped,
                     orfsY = orfs_retained,
                     filterNMD = FALSE,
                     compareBy="gene",
                     geneSimilarity = TRUE,
                     compareUTR=TRUE,
                     allORFs = orfs_normal)
# add NMD calculations
orfs_normal$nmd_prob <- notNMD::predictNMD(orfs_normal, "prob")
orfs_normal$nmd_class <- notNMD::predictNMD(orfs_normal)
orfs_skipped$nmd_prob <- notNMD::predictNMD(orfs_skipped, "prob")
orfs_skipped$nmd_class <- notNMD::predictNMD(orfs_skipped)
orfs_retained$nmd_prob <- notNMD::predictNMD(orfs_retained, "prob")
orfs_retained$nmd_class <- notNMD::predictNMD(orfs_retained)
orfs_normal <- orfs_normal[which(!is.na(orfs_normal$orf_length)),]
orfs_skipped <- orfs_skipped[which(!is.na(orfs_skipped$orf_length)),]
orfs_retained <- orfs_retained[which(!is.na(orfs_retained$orf_length)),]
# compare normal and retained isoforms
# this time setting filterNMD to TRUE, which removes NMD targeted frames/isoforms where possible
orfChange_filter <- orfDiff(orfsX = orfs_skipped, # normal transcripts
                     orfsY = orfs_retained, # alternative transcripts
                     filterNMD = TRUE,
                     geneSimilarity = TRUE,
                     compareUTR=TRUE,
                     allORFs = orfs_normal)
# test changes in NMD between retained and skipped introns.
nmdChange <- attrChangeAltSpliced(orfs_skipped, orfs_retained,
                                  attribute="nmd_prob",
                                  compareBy="gene",
                                  useMax=FALSE) # want useMax=F, as we want the minimum NMD probablity (if NMD prob is low, means that at least 1 isoform not targeted)
m <- match(orfChange_filter$id, nmdChange$id)
orfChange_filter <- cbind(orfChange_filter, nmdChange[m,-1])
write.table(orfChange_filter, file="/camp/home/ziffo/home/projects/astrocyte-meta-analysis/splicing/IRFinder/notNMD/absolute.nuc_cyt_orf_nmd_changes.txt", sep='\t', row.names = F)
save.image("/camp/home/ziffo/home/projects/astrocyte-meta-analysis/splicing/IRFinder/notNMD/absolute.nuc_cyt_notNMD_results.RData")

# load(file = "/camp/home/ziffo/home/projects/astrocyte-meta-analysis/splicing/IRFinder/notNMD/absolute.nuc_cyt_notNMD_results.RData")
notNMDprob <- orfChange_filter %>% select("coords" = id, "nmd_prob_retained" = nmd_prob_bygroup_y, "nmd_prob_spliced" = nmd_prob_bygroup_x) #
notNMDprob$coords <- gsub("chr", "", notNMDprob$coords)
notNMDprob$Chr <- str_split_fixed(notNMDprob$coords, ":", 2)[,1]
notNMDprob$End <- as.numeric(str_split_fixed(notNMDprob$coords, "-", 2)[,2])
notNMDprob$Start <- str_split_fixed(notNMDprob$coords, "-", 2)[,1]
notNMDprob$Start <- as.numeric(str_split_fixed(notNMDprob$Start, ":", 2)[,2])
notNMDprob <- notNMDprob %>% mutate("Start" = Start - 1) %>% select(-coords) # remove 1 from Start to revert back to original IRFinder coords
saveRDS(notNMDprob, "/camp/home/ziffo/home/projects/astrocyte-meta-analysis/splicing/IRFinder/notNMD/notNMDprob.absolute.nuc_cyt.RDS")
```

Run above in a separate script with:

```{bash}
cd /camp/lab/luscomben/home/users/ziffo/projects/astrocyte-meta-analysis/splicing/IRFinder/notNMD

sbatch --cpus-per-task=10 --mem=1500G -N 1 --partition=hmem -t 48:00:00 --wrap="Rscript /camp/lab/luscomben/home/users/ziffo/projects/astrocyte-meta-analysis/scripts/notNMD_nuc_cyt_absolute_retained_vs_spliced.R" --mail-type=ALL,ARRAY_TASKS --mail-user=oliver.ziff@crick.ac.uk  --job-name=notNMD
```


```{r}
notNMDprob <- readRDS("/camp/home/ziffo/home/projects/astrocyte-meta-analysis/splicing/IRFinder/notNMD/notNMDprob.absolute.nuc_cyt.RDS")
absolute.nuc_cyt <-  absolute.nuc_cyt %>% left_join(notNMDprob, by = c("Chr", "Start", "End")) # left join results to IRFinder
absolute.nuc_cyt <-  absolute.nuc_cyt %>% mutate(nmd_prob = nmd_prob_retained, #case_when(retained == "retained" ~ nmd_prob_retained, TRUE ~ nmd_prob_spliced),
                                 nmd_outcome = case_when(nmd_prob > 0.5 ~ "NMD", TRUE ~ "not NMD"))

# Cytoplasmic vs nuclear retained introns
notNMD.nuc_cyt.retained_spliced.sina_plot <- ggplot(absolute.nuc_cyt, aes(x = retained, y= nmd_prob, color = retained)) + 
      geom_violin() + geom_sina(size = 0.5) + geom_boxplot(data = absolute.nuc_cyt, aes(fill = retained), colour = "black", width=0.2, outlier.shape = NA) +
      scale_colour_manual(values = c("firebrick2", "dodgerblue3")) + scale_fill_manual(values = c("firebrick2", "dodgerblue3")) +  
      theme_classic() + theme(legend.position = "none", axis.text=element_text(size=10), axis.text.x = element_text(angle = 45, hjust = 1)) +
      facet_wrap(. ~ fraction) +
      stat_compare_means(comparisons = list(c("retained", "spliced")), label = "p.signif") + ylab(label = expression(NMD~probability) ) + xlab("") 
notNMD.nuc_cyt.retained_spliced.sina_plot




notNMDprob <- readRDS("/camp/home/ziffo/home/projects/astrocyte-meta-analysis/splicing/IRFinder/notNMD/notNMDprob.nuc_cyt.RDS")
nuc_cyt.IR <-  nuc_cyt %>% left_join(notNMDprob, by = c("Chr", "Start", "End")) # left join results to IRFinder
nuc_cyt.IR <-  nuc_cyt.IR %>% mutate(nmd_prob = nmd_prob_retained, #case_when(retained == "retained" ~ nmd_prob_retained, TRUE ~ nmd_prob_spliced),
                                 nmd_outcome = case_when(nmd_prob > 0.5 ~ "NMD", TRUE ~ "not NMD"))

# Differential sina plot
notNMD.IRup_IRdown.nuc_cyt.sina_plot <- ggplot(dplyr::filter(nuc_cyt.IR, p0.05_reliable == "significant"), aes(x = p0.05_reliable_IRdirection, y= nmd_prob, color = p0.05_reliable_IRdirection)) + 
      geom_violin() + geom_sina(size = 0.5) + geom_boxplot(data = dplyr::filter(nuc_cyt.IR, p0.05_reliable == "significant"), aes(fill = p0.05_reliable_IRdirection), colour = "black", width=0.1, outlier.shape = NA) +
      scale_colour_manual(values = c("firebrick2", "dodgerblue3")) + scale_fill_manual(values = c("firebrick2", "dodgerblue3")) +  
      theme_classic() + theme(legend.position = "none", axis.text=element_text(size=10), axis.text.x = element_text(angle = 45, hjust = 1)) +
      geom_hline(yintercept = 0.5, linetype = 2) + facet_wrap(. ~ fraction)+
      stat_compare_means(comparisons = list(c("IR down", "IR up")), label = "p.signif", method = "wilcox") + ylab(label = expression(NMD~probability) ) + xlab("")
notNMD.IRup_IRdown.nuc_cyt.sina_plot

```


## Merge

```{r}
figS8.merged <- ggarrange(
  ggarrange(gc.retained_spliced.nuc_cyt.sina_plot, intron_length.retained_spliced.nuc_cyt.sina_plot, PhastCons.retained_spliced.nuc_cyt.sina_plot, CLIP.retained_spliced.sina_plot, 
            ncol = 4, labels = c("A", "B", "C", "D"), widths = c(1,1,1,1)),
  ggarrange(maxent5.retained_spliced.sina_plot, maxent3.retained_spliced.sina_plot, gc.IRup_IRdown.nuc_cyt.sina_plot, intron_length.IRup_IRdown.nuc_cyt.sina_plot, 
            ncol = 4, labels = c("E", "F", "G", "H"), widths = c(1,1,1,1)),
  ggarrange(PhastCons.IRup_IRdown.nuc_cyt.sina_plot, CLIP.IRup_IRdown.sina_plot, maxent5.nuc_cyt.differential.sina_plot, maxent3.nuc_cyt.differential.sina_plot, 
            ncol = 4, labels = c("I", "J", "K", "L"), widths = c(1,1,1,1)), 
  nrow = 3, heights = c(1,1,1))
figS8.merged
ggsave(figS8.merged, filename = "/camp/home/ziffo/home/projects/astrocyte-meta-analysis/splicing/IRFinder/irlib/figures/figS8.merged.png", height = 12, width = 10)
```


# Table S6: Mass spectrometry

```{r}
cyt_mass_spec_results <- readRDS("/camp/home/ziffo/home/projects/astrocyte-meta-analysis/expression/mass-spectrometry/cytoplasmic_mass_spec_results.RDS") %>% as_tibble()
# cyt_mass_spec_results <- readRDS("/camp/home/ziffo/home/projects/astrocyte-meta-analysis/expression/mass-spectrometry/cytoplasmic_mass_spec_results_filtered.RDS") %>% as_tibble()
ac_nucIR_cytDGE.absoluteIR <- cleanIRFinderQuantCompared(mutant = ac_nuc_vcp.IR, control = ac_nuc_ctrl.IR, differentialIR = ac_nuc_vcp_vs_ctrl.IRFinder, deseq_res = dge_res_ac_cyt_vcp_vs_ctrl, mass_spec = cyt_mass_spec_results) %>% as_tibble()

# select top IR event per gene
ac_nucIR_cytDGE_cytMS.absoluteIR.topevent <- ac_nucIR_cytDGE.absoluteIR %>% filter(reliable == "reliable") %>% group_by(gene_name) %>% slice_max(abs(IRratio.diff), n = 1, with_ties = FALSE) %>%
  dplyr::select(gene_name, "Delta.IRratio" = IRratio.diff, "IR.pval" = p.diff, "mRNA.lfc" = log2FoldChange, "mRNA.FDR" = padj)

cyt_mass_spec_results %>%
  dplyr::select("gene_name" = name, "protein.pval" = vcp_vs_ctrl_p.val, "protein.lfc" = vcp_vs_ctrl_ratio) %>%
  left_join(ac_nucIR_cytDGE_cytMS.absoluteIR.topevent, by = "gene_name") %>%
  mutate(protein.direction = case_when(protein.lfc > 0 ~ "protein up", protein.lfc < 0 ~ "protein down", TRUE ~ "NA"),
         mRNA.direction = case_when(mRNA.lfc > 0 ~ "mRNA up", mRNA.lfc < 0 ~ "mRNA down", TRUE ~ "NA"),
         IR.direction = case_when(Delta.IRratio > 0 ~ "IR up", Delta.IRratio < 0 ~ "IR down", TRUE ~ "NA"),
         protein.mRNA.IR.direction = case_when(protein.lfc > 0 & mRNA.lfc > 0 & Delta.IRratio > 0 ~ "protein up, mRNA up, IR up",
                                                     protein.lfc < 0 & mRNA.lfc < 0 & Delta.IRratio < 0 ~ "protein down, mRNA down, IR down",
                                                     protein.lfc > 0 & mRNA.lfc < 0 & Delta.IRratio < 0 ~ "protein up, mRNA down, IR down",
                                                     protein.lfc < 0 & mRNA.lfc > 0 & Delta.IRratio < 0 ~ "protein down, mRNA up, IR down",
                                                     protein.lfc < 0 & mRNA.lfc < 0 & Delta.IRratio > 0 ~ "protein down, mRNA down, IR up",
                                                     protein.lfc > 0 & mRNA.lfc > 0 & Delta.IRratio < 0 ~ "protein up, mRNA up, IR down",
                                                     protein.lfc > 0 & mRNA.lfc < 0 & Delta.IRratio > 0 ~ "protein up, mRNA down, IR up",
                                                     protein.lfc < 0 & mRNA.lfc > 0 & Delta.IRratio > 0 ~ "protein down, mRNA up, IR up",
                                                     TRUE ~ "NA")) %>% 
  dplyr::select(gene_name, protein.lfc, protein.pval, protein.direction, Delta.IRratio, IR.pval, IR.direction, mRNA.lfc, mRNA.FDR, mRNA.direction, protein.mRNA.IR.direction) %>%
  write.csv(., "/camp/home/ziffo/home/projects/astrocyte-meta-analysis/splicing/IRFinder/irlib/results/TableS6.absolute_ac_cyt_vcp_vs_ctrl.mass_spec_results.csv")

#we detected 1160 proteins in the cytoplasmic fractions of which 580 were increased in protein levels in VCP versus control
cyt_mass_spec_results %>% nrow # # how many proteins detected? 1160
cyt_mass_spec_results %>% filter(vcp_vs_ctrl_ratio > 0) %>% nrow # 580

# Of these 580, 312 (54%) were derived from genes with decreased nuclear IR, of which 52% (163/312) are involved with reactive transformation
cyt_mass_spec_results %>% dplyr::select("gene_name" = name, "massspec_p_value" = vcp_vs_ctrl_p.val, "protein.lfc" = vcp_vs_ctrl_ratio) %>%
  left_join(ac_nucIR_cytDGE_cytMS.absoluteIR.topevent, by = "gene_name") %>%  filter(protein.lfc > 0, Delta.IRratio < 0) %>% nrow() # 312 genes                                                                
312/578 # 53.9%
cyt_mass_spec_results %>%  dplyr::select("gene_name" = name, "massspec_p_value" = vcp_vs_ctrl_p.val, "protein.lfc" = vcp_vs_ctrl_ratio) %>%
  left_join(ac_nucIR_cytDGE_cytMS.absoluteIR.topevent, by = "gene_name") %>%filter(protein.lfc > 0, Delta.IRratio < 0, gene_name %in% all.reactivity.go) %>% nrow # 163
163/312 # 52.2%
```

# Figure S9: Fractionated Coverage Tracks

```{r}
txdb <- makeTxDbFromGFF("/camp/home/ziffo/home/genomes/ensembl/Homo_sapiens.GRCh38.99.chr_patch_hapl_scaff.gtf", format="gtf") # make txdb from GTF - turn on only when needed
seqlevels(txdb, pruning.mode="coarse") <- c("1", "2", "3", "4", "5", "6", "7", "8", "9", "10", "11", "12", "13", "14", "15", "16", "17", "18", "19", "20", "21", "22", "X", "Y") # removes all the other chromsome 
# Create Alignment Track from BAM files (rather than BedGraph) - merged biological repeats into a single bam
ac_nuc_ctrl.bam <- "/camp/home/ziffo/home/projects/astrocyte-meta-analysis/alignment/STAR/merged/ac_nuc_ctrl.bam"
ac_nuc_vcp.bam <- "/camp/home/ziffo/home/projects/astrocyte-meta-analysis/alignment/STAR/merged/ac_nuc_vcp.bam"
ac_cyt_ctrl.bam <- "/camp/home/ziffo/home/projects/astrocyte-meta-analysis/alignment/STAR/merged/ac_cyt_ctrl.bam"
ac_cyt_vcp.bam <- "/camp/home/ziffo/home/projects/astrocyte-meta-analysis/alignment/STAR/merged/ac_cyt_vcp.bam"

# Plot coverage tracks in nuclear & cytoplasmic fractions
track_viewer_fractions(gene_test = "IRF7", txt = ac_vcp_vs_ctrl.filt, mut.nuc.bam = ac_nuc_vcp.bam, ctrl.nuc.bam = ac_nuc_ctrl.bam, mut.cyt.bam = ac_cyt_vcp.bam, ctrl.cyt.bam = ac_cyt_ctrl.bam)
track_viewer_fractions(gene_test = "RND3", txt = ac_vcp_vs_ctrl.filt, mut.nuc.bam = ac_nuc_vcp.bam, ctrl.nuc.bam = ac_nuc_ctrl.bam, mut.cyt.bam = ac_cyt_vcp.bam, ctrl.cyt.bam = ac_cyt_ctrl.bam)
track_viewer_fractions(gene_test = "MCAM", txt = ac_vcp_vs_ctrl.filt, mut.nuc.bam = ac_nuc_vcp.bam, ctrl.nuc.bam = ac_nuc_ctrl.bam, mut.cyt.bam = ac_cyt_vcp.bam, ctrl.cyt.bam = ac_cyt_ctrl.bam)
track_viewer_fractions(gene_test = "ADAM19", txt = ac_vcp_vs_ctrl.filt, mut.nuc.bam = ac_nuc_vcp.bam, ctrl.nuc.bam = ac_nuc_ctrl.bam, mut.cyt.bam = ac_cyt_vcp.bam, ctrl.cyt.bam = ac_cyt_ctrl.bam)
track_viewer_fractions(gene_test = "MAMDC4", txt = ac_vcp_vs_ctrl.filt, mut.nuc.bam = ac_nuc_vcp.bam, ctrl.nuc.bam = ac_nuc_ctrl.bam, mut.cyt.bam = ac_cyt_vcp.bam, ctrl.cyt.bam = ac_cyt_ctrl.bam)
track_viewer_fractions(gene_test = "ILK", txt = ac_vcp_vs_ctrl.filt, mut.nuc.bam = ac_nuc_vcp.bam, ctrl.nuc.bam = ac_nuc_ctrl.bam, mut.cyt.bam = ac_cyt_vcp.bam, ctrl.cyt.bam = ac_cyt_ctrl.bam)
track_viewer_fractions(gene_test = "TEP1", txt = ac_vcp_vs_ctrl.filt, mut.nuc.bam = ac_nuc_vcp.bam, ctrl.nuc.bam = ac_nuc_ctrl.bam, mut.cyt.bam = ac_cyt_vcp.bam, ctrl.cyt.bam = ac_cyt_ctrl.bam)
track_viewer_fractions(gene_test = "ASIC3", txt = ac_vcp_vs_ctrl.filt, mut.nuc.bam = ac_nuc_vcp.bam, ctrl.nuc.bam = ac_nuc_ctrl.bam, mut.cyt.bam = ac_cyt_vcp.bam, ctrl.cyt.bam = ac_cyt_ctrl.bam)
```

# Response letter figures

## Scatterplots IR vs GE

```{r}
scatterplots.merged <- ggarrange(ac_who_dge_ir_scatter, ac_bir_c9orf_scatter, ac_tyz_sod1_scatter,ac_bar_a1_scatter,ac_tdp_ko_scatter, nucIR_cytDGE_scatter, nrow = 3, ncol = 2, 
                                 labels = c("2B", "2C", "2D", "3F", "3G", "4I"))
scatterplots.merged
ggsave(scatterplots.merged, filename = "/camp/home/ziffo/home/projects/astrocyte-meta-analysis/splicing/IRFinder/irlib/figures/scatterplots.merged.png", height = 11.25, width = 8.75)

```

## TDP43 iCLIP

Consider including after TDP43 deletion

Does TDP43 bind more to IR up and IR down?

TDP43 binds IR up > IR down --> TDP43 sequestered by IR transcripts in VCP --> Depletes free TDP43 --> TDP43 unable to block splice sites --> excessive splicing in VCP

```{r}
clip_k562 <- list.files("/camp/home/ziffo/home/genomes/clip-bedgraphs/K562/")
clip_k562 <- clip_k562[!grepl("3nt_peaks",clip_k562)] # only assess raw crosslink files without any peak calling for enrichment
VCP <- ac_who_vcp_vs_ctrl %>% dplyr::select(Intron.GeneName.GeneID.Coords, coords, ensID, gene_name, Chr, Start, End, Direction, gc_content, intron_length, p0.05_reliable, VCP.p0.05_reliable_IRdirection = p0.05_reliable_IRdirection) %>% filter(p0.05_reliable == "significant")
SOD1 <- ac_tyz_sod1_vs_ctrl %>% dplyr::select(Intron.GeneName.GeneID.Coords, coords, ensID, gene_name, Chr, Start, End, Direction,p0.05_reliable, SOD1.p0.05_reliable_IRdirection = p0.05_reliable_IRdirection, gc_content, intron_length)  %>% filter(p0.05_reliable == "significant")
C9ORF72 <- ac_bir_c9orf_vs_ctrl %>% dplyr::select(Intron.GeneName.GeneID.Coords, coords, ensID, gene_name, Chr, Start, End, Direction,p0.05_reliable, C9ORF72.p0.05_reliable_IRdirection = p0.05_reliable_IRdirection, gc_content, intron_length) %>% filter(p0.05_reliable == "significant")
differential.3datasets.IR <- VCP %>% full_join(SOD1, by = c("Intron.GeneName.GeneID.Coords", "coords", "ensID", "gene_name", "Chr", "Start", "End", "Direction", "gc_content", "intron_length")) %>%
  full_join(C9ORF72, by = c("Intron.GeneName.GeneID.Coords", "coords", "ensID", "gene_name", "Chr", "Start", "End", "Direction", "gc_content", "intron_length")) %>%
  mutate(p0.05_reliable_IRdirection = case_when(VCP.p0.05_reliable_IRdirection == "IR down" | SOD1.p0.05_reliable_IRdirection == "IR down" | C9ORF72.p0.05_reliable_IRdirection == "IR down" ~ "IR down", TRUE ~ "IR up"))
grange_introns <- as.GenomicRange(differential.3datasets.IR$Intron.GeneName.GeneID.Coords) # create GRanges for each intron (both retained and non retained)
intron_coords <- tibble(intron = 1:length(differential.3datasets.IR$coords), coords = differential.3datasets.IR$coords) # intron coordinates for to enable matching back to original coords in IRFinder dataframe
rbp <- "TARDBP"
bed <- import("/camp/home/ziffo/home/genomes/clip-bedgraphs/K562/TARDBP.merged.bed.gz")
seqlevelsStyle(bed) <- "NCBI" # remove "chr"
grange_overlap <- findOverlaps(query = grange_introns, subject=bed, ignore.strand=FALSE) # find overlaps between RBP binding sites & retained introns
range <- bed[ queryHits(grange_overlap) ] %>% as.data.frame() # subset TDP43 xlinks that overlap with the introns.
TARDBP_ac_who_k562 <- range %>%  mutate( intron = queryHits(grange_overlap)) %>%  group_by(intron) %>% summarise(iclip_score = sum(score) ) %>% # sum the scores for each intron - 1834 introns with scores
  left_join(intron_coords, by = "intron") %>% dplyr::select(-intron) # add intron coordinates to match back to IRFinder results
TARDBP_iclip_scores <- differential.3datasets.IR %>% dplyr::select(coords, intron_length) %>% full_join(TARDBP_ac_who_k562, by = "coords") %>% # add back the introns without xlink scores
  drop_na() # NAs indicate no xlink score for the intron. Can drop NAs or set NA scores to 0
# TARDBP_iclip_scores[is.na(TARDBP_iclip_scores)] <- 0  # set NA scores to 0
TARDBP_iclip_scores <- TARDBP_iclip_scores %>% mutate(iclip_score.normalised = iclip_score/intron_length) # normalise all scores for intron_length
differential.3datasets.IR <- differential.3datasets.IR %>% left_join(select(TARDBP_iclip_scores, iclip_score.normalised, coords), by = "coords") #%>% # add normalised scores to IRFinder results
TARDBP.clip.differential.sina_plot <- sinaplot(data = differential.3datasets.IR, groups = "p0.05_reliable_IRdirection", continuous = "iclip_score.normalised", dashed.line = "NA", stat.y.position = 0.11, stats.test = "t_test") +  ylab(label = expression(log[10]~TARDBP~iCLIP~binding) ) + xlab("") + scale_y_log10() + theme(axis.text.y = element_text(size=6, angle = 45), axis.text.x = element_text(angle = 45, hjust = 1))
TARDBP.clip.differential.sina_plot

clip.differential.sina_plot <- sinaplot(data = differential.3datasets.IR, groups = "p0.05_reliable_IRdirection", continuous = "sum", dashed.line = "NA", stat.y.position = 2, stats.test = "t_test") + 
  ylab(label = expression(log[10]~all~RBPs~iCLIP~binding) ) + xlab("") + scale_y_log10() + theme(axis.text.y = element_text(size=6, angle = 45), axis.text.x = element_text(angle = 45, hjust = 1))
clip.differential.sina_plot

allRBPs.TARDBP.iCLIP.binding.sinaplot <- ggarrange(clip.differential.sina_plot, TARDBP.clip.differential.sina_plot, nrow = 1, ncol = 2)
allRBPs.TARDBP.iCLIP.binding.sinaplot
ggsave(allRBPs.TARDBP.iCLIP.binding.sinaplot, filename = "/camp/home/ziffo/home/projects/astrocyte-meta-analysis/splicing/IRFinder/irlib/figures/allRBPs.TARDBP.iCLIP.binding.sinaplot.png", height = 3, width = 8)


# Nucleus & cytoplasm VCP vs CTRL
nuclear <- ac_nuc_vcp_vs_ctrl %>% dplyr::select(Intron.GeneName.GeneID.Coords, coords, intron_length, gc_content, reliable_retained, retained, p0.05_reliable, p0.05_reliable_IRdirection) %>% 
  dplyr::mutate(fraction = "nuclear")
cytoplasmic <- ac_cyt_vcp_vs_ctrl %>% dplyr::select(Intron.GeneName.GeneID.Coords, coords, intron_length, gc_content, reliable_retained, retained,p0.05_reliable, p0.05_reliable_IRdirection) %>% 
  dplyr::mutate(fraction = "cytoplasmic")
nuc_cyt <- bind_rows(nuclear, cytoplasmic)
grange_introns <- as.GenomicRange(nuc_cyt$Intron.GeneName.GeneID.Coords) # create GRanges for each intron (both retained and non retained)
intron_coords <- tibble(intron = 1:length(nuc_cyt$coords), coords = nuc_cyt$coords) # intron coordinates for to enable matching back to original coords in IRFinder dataframe
rbp <- "TARDBP"
bed <- import("/camp/home/ziffo/home/genomes/clip-bedgraphs/K562/TARDBP.merged.bed.gz")
seqlevelsStyle(bed) <- "NCBI" # remove "chr"
grange_overlap <- findOverlaps(query = grange_introns, subject=bed, ignore.strand=FALSE) # find overlaps between RBP binding sites & retained introns
range <- bed[ queryHits(grange_overlap) ] %>% as.data.frame() # subset TDP43 xlinks that overlap with the introns.
TARDBP_ac_who_k562 <- range %>%  mutate( intron = queryHits(grange_overlap)) %>%  group_by(intron) %>% summarise(iclip_score = sum(score) ) %>% # sum the scores for each intron - 1834 introns with scores
  left_join(intron_coords, by = "intron") %>% dplyr::select(-intron) # add intron coordinates to match back to IRFinder results
TARDBP_iclip_scores <- differential.3datasets.IR %>% dplyr::select(coords, intron_length) %>% full_join(TARDBP_ac_who_k562, by = "coords") #%>% # add back the introns without xlink scores
  # drop_na() # NAs indicate no xlink score for the intron. Can drop NAs or set NA scores to 0
TARDBP_iclip_scores[is.na(TARDBP_iclip_scores)] <- 0  # set NA scores to 0
TARDBP_iclip_scores <- TARDBP_iclip_scores %>% mutate(iclip_score.normalised = iclip_score/intron_length) # normalise all scores for intron_length
nuc_cyt <- nuc_cyt %>% left_join(select(TARDBP_iclip_scores, iclip_score.normalised, coords), by = "coords") #%>% # add normalised scores to IRFinder results

# IR up vs IR down in fractions
CLIP.IRup_IRdown.sina_plot <- ggplot(dplyr::filter(nuc_cyt, p0.05_reliable == "significant"), aes(x = p0.05_reliable_IRdirection, y= iclip_score.normalised, color = p0.05_reliable_IRdirection)) + 
      geom_violin() + geom_sina(size = 0.5) + geom_boxplot(data = dplyr::filter(nuc_cyt, p0.05_reliable == "significant"), aes(fill = p0.05_reliable_IRdirection), colour = "black", width=0.2, outlier.shape = NA) +
      scale_colour_manual(values = c("firebrick2", "dodgerblue3")) + scale_fill_manual(values = c("firebrick2", "dodgerblue3")) +  
      theme_classic() + theme(legend.position = "none", axis.text=element_text(size=12)) +
      facet_wrap(. ~ fraction)+
      stat_compare_means(comparisons = list(c("IR down", "IR up")), label = "p.signif") + ylab(label = expression(log[10]~normalised~TARDBP~xlinks~score~sum) ) + xlab("") +  scale_y_log10() 
CLIP.IRup_IRdown.sina_plot
```

## Fraction differential IR charachteristics

```{r}
fraction.differentialIRcharacterstics <- ggarrange(gc.IRup_IRdown.nuc_cyt.sina_plot, intron_length.IRup_IRdown.nuc_cyt.sina_plot, PhastCons.IRup_IRdown.nuc_cyt.sina_plot, CLIP.IRup_IRdown.sina_plot, nrow = 1, ncol = 4)
fraction.differentialIRcharacterstics
ggsave(fraction.differentialIRcharacterstics, filename = "/camp/home/ziffo/home/projects/astrocyte-meta-analysis/splicing/IRFinder/irlib/figures/fraction.differentialIRcharacterstics.png", height = 4, width = 10)
```



